<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:37 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>MultipleAlignmentTools (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="MultipleAlignmentTools (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/MultipleAlignmentTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html" title="class in org.biojava.nbio.structure.align.multiple.util"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html" title="class in org.biojava.nbio.structure.align.multiple.util"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html" target="_top">Frames</a></li>
<li><a href="MultipleAlignmentTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.structure.align.multiple.util</div>
<h2 title="Class MultipleAlignmentTools" class="title">Class MultipleAlignmentTools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.structure.align.multiple.util.MultipleAlignmentTools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">MultipleAlignmentTools</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Utility functions for working with <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.
 <p>
 Supported functions:
 <ul>
 <li>Obtain the alignment as sequence strings
 <li>Map from sequence alignment position to structure Atom
 <li>Map from sequence alignment position to Block number
 <li>Transform the aligned Atoms of a MultipleAlignment
 <li>Get all the core alignment positions of the alignment
 <li>Calculate the average residue distance of all aligned positions
 <li>Sort Blocks in a MultipleAlignment by a specified row
 <li>Convert a MultipleAlignment to a MultipleSequenceAlignment
 </ul></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>4.1.0</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Spencer Bliven, Aleix Lafita</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#MultipleAlignmentTools()">MultipleAlignmentTools</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20int,%20int)">getAtomForSequencePosition</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                          <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                          int&nbsp;str,
                          int&nbsp;sequencePos)</code>
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAverageResidueDistances(java.util.List)">getAverageResidueDistances</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed)</code>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAverageResidueDistances</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20int)">getBlockForSequencePosition</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                           int&nbsp;sequencePos)</code>
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getBlockSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</code>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List)">getBlockSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                         <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</code>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getCorePositions(org.biojava.nbio.structure.align.multiple.Block)">getCorePositions</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&nbsp;block)</code>
<div class="block">Calculate a List of alignment indicies that correspond to the core of a
 Block, which means that all structures have a residue in that positon.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getHSDMTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getKimuraTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRMSDMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List)">getSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                    <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</code>
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getStructuralTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#sortBlocks(java.util.List,%20int)">sortBlocks</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&gt;&nbsp;blocks,
          int&nbsp;sortedIndex)</code>
<div class="block">Sort blocks so that the specified row is in sequential order.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toProteinMSA</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">transformAtoms</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Transforms atoms according to the superposition stored in the alignment.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="MultipleAlignmentTools()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>MultipleAlignmentTools</h4>
<pre>public&nbsp;MultipleAlignmentTools()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAlignment</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getSequenceAlignment(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                                <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</pre>
<div class="block">Calculate the sequence alignment Strings for the whole alignment. This
 method creates a sequence alignment where aligned residues are in
 uppercase and unaligned residues are in lowercase, thus providing a more
 compact way to represent the alignment.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps are represented by '-'. Separation between different
 Blocks is indicated by a gap in all positions, meaning that there is a
 possible discontinuity.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - input MultipleAlignment</dd><dd><code>mapSeqToStruct</code> - provides a link from the sequence alignment position to the
            structure alignment position. Specially designed for the GUI.
            Has to be initialized previously and will be overwritten.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a string for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd></dl>
</li>
</ul>
<a name="getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAlignment</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getSequenceAlignment(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</pre>
<div class="block">Calculate the sequence alignment Strings for the whole alignment. This
 method creates a sequence alignment where aligned residues are in
 uppercase and unaligned residues are in lowercase, thus providing a more
 compact way to represent the alignment.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps are represented by '-'. Separation between different
 Blocks is indicated by a gap in all positions, meaning that there is a
 possible discontinuity.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - input MultipleAlignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>String for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd></dl>
</li>
</ul>
<a name="getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBlockSequenceAlignment</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getBlockSequenceAlignment(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                                     <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</pre>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment. This method creates a sequence alignment where all residues
 are in uppercase and a residue alone with gaps in all the other
 structures represents unaligned residues. Because of this latter
 constraint only the residues within the Blocks are represented, for a
 more compact alignment. For a sequence alignment of the full protein use
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>getSequenceAlignment(MultipleAlignment)</code></a>.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps between blocks are omitted, while gaps within blocks are
 represented by '-'. Separation between different Blocks is indicated by a
 gap in all positions, meaning that there is something unaligned
 inbetween.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - input MultipleAlignment</dd><dd><code>mapSeqToStruct</code> - provides a link from the sequence alignment position to the
            structure alignment position. Specially designed for the GUI.
            Has to be initialized previously and will be overwritten.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a string for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd></dl>
</li>
</ul>
<a name="getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBlockSequenceAlignment</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getBlockSequenceAlignment(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</pre>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment. This method creates a sequence alignment where all residues
 are in uppercase and a residue alone with gaps in all the other
 structures represents unaligned residues. Because of this latter
 constraint only the residues within the Blocks are represented, for a
 more compact alignment. For a sequence alignment of the full protein use
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>getSequenceAlignment(MultipleAlignment)</code></a>.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps between blocks are omitted, while gaps within blocks are
 represented by '-'. Separation between different Blocks is indicated by a
 gap in all positions, meaning that there is something unaligned
 inbetween.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - input MultipleAlignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>String for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd></dl>
</li>
</ul>
<a name="getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment, java.util.List, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomForSequencePosition</h4>
<pre>public static&nbsp;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;getAtomForSequencePosition(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                              <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                              int&nbsp;str,
                              int&nbsp;sequencePos)</pre>
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>multAln</code> - the MultipleAlignment object from where the sequence alignment
            has been generated</dd><dd><code>mapSeqToStruct</code> - the mapping between sequence and structure generated with the
            sequence alignment</dd><dd><code>str</code> - the structure index of the alignment (row)</dd><dd><code>sequencePos</code> - the sequence alignment position (column)</dd>
<dt><span class="strong">Returns:</span></dt><dd>Atom the atom in that position or null if there is a gap</dd></dl>
</li>
</ul>
<a name="getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment, java.util.List, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBlockForSequencePosition</h4>
<pre>public static&nbsp;int&nbsp;getBlockForSequencePosition(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                              <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                              int&nbsp;sequencePos)</pre>
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>multAln</code> - the MultipleAlignment object from where the sequence alignment
            has been generated.</dd><dd><code>mapSeqToStruct</code> - the mapping between sequence and structure generated with the
            sequence alignment</dd><dd><code>sequencePos</code> - the position in the sequence alignment (column)</dd>
<dt><span class="strong">Returns:</span></dt><dd>int the block index, or -1 if the position is not aligned</dd></dl>
</li>
</ul>
<a name="getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAverageResidueDistances</h4>
<pre>public static&nbsp;<a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;getAverageResidueDistances(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</pre>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.
 <p>
 Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment
 length.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - MultipleAlignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>Matrix containing all average residue distances</dd></dl>
</li>
</ul>
<a name="getAverageResidueDistances(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAverageResidueDistances</h4>
<pre>public static&nbsp;<a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;getAverageResidueDistances(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed)</pre>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.
 <p>
 Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment
 length.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>transformed</code> - List of Atom arrays containing only the aligned atoms of each
            structure, or null if there is a gap.</dd>
<dt><span class="strong">Returns:</span></dt><dd>Matrix containing all average residue distances. Entry -1 means
         there is a gap in the position.</dd></dl>
</li>
</ul>
<a name="transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>transformAtoms</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformAtoms(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</pre>
<div class="block">Transforms atoms according to the superposition stored in the alignment.
 <p>
 For each structure in the alignment, returns an atom for each
 representative atom in the aligned columns, omitting unaligned residues
 (i.e. an array of length <tt>alignment.length()</tt> ).
 <p>
 All blocks are concatenated together, so Atoms may not appear in the same
 order as in their parent structure. If the alignment blocks contain null
 residues (gaps), then the returned array will also contain null Atoms in
 the same positions.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alignment</code> - MultipleAlignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>List of Atom arrays of only the aligned atoms of every structure
         (null Atom if a gap position)</dd></dl>
</li>
</ul>
<a name="getCorePositions(org.biojava.nbio.structure.align.multiple.Block)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCorePositions</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;getCorePositions(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&nbsp;block)</pre>
<div class="block">Calculate a List of alignment indicies that correspond to the core of a
 Block, which means that all structures have a residue in that positon.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>block</code> - alignment Block</dd>
<dt><span class="strong">Returns:</span></dt><dd>List of positions in the core of the alignment</dd></dl>
</li>
</ul>
<a name="sortBlocks(java.util.List, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>sortBlocks</h4>
<pre>public static&nbsp;void&nbsp;sortBlocks(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&gt;&nbsp;blocks,
              int&nbsp;sortedIndex)</pre>
<div class="block">Sort blocks so that the specified row is in sequential order. The sort
 happens in place.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>blocks</code> - List of blocks</dd><dd><code>sortedIndex</code> - Index of the row to be sorted</dd></dl>
</li>
</ul>
<a name="toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toProteinMSA</h4>
<pre>public static&nbsp;<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;toProteinMSA(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)
                                                                                 throws <a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues. This method is only valid for protein structure alignments.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>msta</code> - Multiple Structure Alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>MultipleSequenceAlignment of protein sequences</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRMSDMatrix</h4>
<pre>public static&nbsp;<a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;getRMSDMatrix(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</pre>
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>msa</code> - Multiple Structure Alignment</dd>
<dt><span class="strong">Returns:</span></dt><dd>Matrix of RMSD with size the number of structures squared</dd></dl>
</li>
</ul>
<a name="getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getKimuraTree</h4>
<pre>public static&nbsp;org.forester.phylogeny.Phylogeny&nbsp;getKimuraTree(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)
                                                      throws <a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a>,
                                                             <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt><span class="strong">Returns:</span></dt><dd>Phylogeny phylogenetic tree</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getHSDMTree</h4>
<pre>public static&nbsp;org.forester.phylogeny.Phylogeny&nbsp;getHSDMTree(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)
                                                    throws <a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a>,
                                                           <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt><span class="strong">Returns:</span></dt><dd>Phylogeny phylogenetic tree</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getStructuralTree</h4>
<pre>public static&nbsp;org.forester.phylogeny.Phylogeny&nbsp;getStructuralTree(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</pre>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt><span class="strong">Returns:</span></dt><dd>Phylogeny phylogenetic tree</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/MultipleAlignmentTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html" title="class in org.biojava.nbio.structure.align.multiple.util"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html" title="class in org.biojava.nbio.structure.align.multiple.util"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html" target="_top">Frames</a></li>
<li><a href="MultipleAlignmentTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
