<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:18 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>SeqIOTools (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SeqIOTools (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SeqIOTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/bio/seq/io/SequenceBuilder.html" title="interface in org.biojava.bio.seq.io"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/io/SeqIOTools.html" target="_top">Frames</a></li>
<li><a href="SeqIOTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.bio.seq.io</div>
<h2 title="Class SeqIOTools" class="title">Class SeqIOTools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.bio.seq.io.SeqIOTools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use org.biojavax.bio.seq.RichSequence.IOTools</i></div>
</div>
<br>
<pre>public final class <a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.64">SeqIOTools</a>
extends <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">A set of convenience methods for handling common file formats.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>1.1</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Thomas Down, Mark Schreiber, Nimesh Singh, Matthew Pocock, Keith James</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#biojavaToFile(int,%20java.io.OutputStream,%20java.lang.Object)">biojavaToFile</a></strong>(int&nbsp;fileType,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Converts a Biojava object to the given filetype.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#biojavaToFile(java.lang.String,%20java.lang.String,%20java.io.OutputStream,%20java.lang.Object)">biojavaToFile</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a Biojava <code>SequenceIterator</code>,
 <code>SequenceDB</code>, <code>Sequence</code> or <code>Aligment</code>
 to an <code>OutputStream</code></div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#fileToBiojava(int,%20java.io.BufferedReader)">fileToBiojava</a></strong>(int&nbsp;fileType,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Reads a file and returns the corresponding Biojava object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#fileToBiojava(java.lang.String,%20java.lang.String,%20java.io.BufferedReader)">fileToBiojava</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Reads a file with the specified format and alphabet</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#formatToFactory(org.biojava.bio.seq.io.SequenceFormat,%20org.biojava.bio.symbol.Alphabet)">formatToFactory</a></strong>(<a href="../../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;format,
                              <a href="../../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>as this essentially duplicates the operation
 available in the method <code>identifyBuilderFactory</code>.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getAlphabet(int)">getAlphabet</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getAlphabet</code> accepts a value which represents a
 sequence format and returns the relevant
 <code>FiniteAlphabet</code> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getBuilderFactory(int)">getBuilderFactory</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getBuilderFactory</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceBuilderFactory</code> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getEmblBuilderFactory()">getEmblBuilderFactory</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling EMBL
 files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getFastaBuilderFactory()">getFastaBuilderFactory</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling FASTA
 files.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getGenbankBuilderFactory()">getGenbankBuilderFactory</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling GenBank
 files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getGenpeptBuilderFactory()">getGenpeptBuilderFactory</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling Genpept
 files.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getSequenceFormat(int)">getSequenceFormat</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getSequenceFormat</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceFormat</code> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#getSwissprotBuilderFactory()">getSwissprotBuilderFactory</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling Swissprot
 files.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#guessFileType(java.io.File)">guessFileType</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;seqFile)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>because there is no standard file naming convention
 and guessing by file name is inherantly error prone and bad.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#identifyFormat(java.lang.String,%20java.lang.String)">identifyFormat</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>identifyFormat</code> performs a case-insensitive mapping
 of a pair of common sequence format name (such as 'embl',
 'genbank' or 'fasta') and alphabet name (such as 'dna', 'rna',
 'protein', 'aa') to an integer.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readEmbl(java.io.BufferedReader)">readEmbl</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readEmblNucleotide(java.io.BufferedReader)">readEmblNucleotide</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readEmblRNA(java.io.BufferedReader)">readEmblRNA</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream, but for RNA.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFasta(java.io.BufferedReader,%20org.biojava.bio.seq.io.SymbolTokenization)">readFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br,
                  <a href="../../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;sTok)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Read a fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFasta(java.io.BufferedReader,%20org.biojava.bio.seq.io.SymbolTokenization,%20org.biojava.bio.seq.io.SequenceBuilderFactory)">readFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br,
                  <a href="../../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;sTok,
                  <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;seqFactory)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Read a fasta file using a custom type of SymbolList.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFasta(java.io.InputStream,%20org.biojava.bio.symbol.Alphabet)">readFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;seqFile,
                  <a href="../../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Create a sequence database from a fasta file provided as an
 input stream.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFastaDNA(java.io.BufferedReader)">readFastaDNA</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of DNA sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFastaProtein(java.io.BufferedReader)">readFastaProtein</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of Protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFastaRNA(java.io.BufferedReader)">readFastaRNA</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of RNA sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readGenbank(java.io.BufferedReader)">readGenbank</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an Genbank-format stream.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readGenbankXml(java.io.BufferedReader)">readGenbankXml</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an GenbankXML-format stream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readGenpept(java.io.BufferedReader)">readGenpept</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an Genpept-format stream.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readSwissprot(java.io.BufferedReader)">readSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Iterate over the sequences in an Swissprot-format stream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeEmbl(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeEmbl</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                  <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in EMBL format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeEmbl(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeEmbl</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                  <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in EMBL format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeFasta(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                    <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Fasta format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeFasta(java.io.OutputStream,%20org.biojava.bio.seq.db.SequenceDB)">writeFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                    <a href="../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Write a sequenceDB to an output stream in fasta format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeFasta(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                    <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes sequences from a SequenceIterator to an OutputStream in
 Fasta Format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenbank(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeGenbank</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Genbank format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenbank(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeGenbank</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in Genbank
 format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenpept(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeGenpept</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Genpept format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenpept(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeGenpept</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in Genpept
 format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeSwissprot(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                            <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in SwissProt format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeSwissprot(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                            <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in SwissProt
 format.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getEmblBuilderFactory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getEmblBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.82">getEmblBuilderFactory</a>()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling EMBL
 files.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <CODE>SmartSequenceBuilder.FACTORY</CODE></dd></dl>
</li>
</ul>
<a name="readEmbl(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readEmbl</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.96">readEmbl</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the EMBL source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readEmblRNA(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readEmblRNA</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.109">readEmblRNA</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream, but for RNA.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the EMBL source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readEmblNucleotide(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readEmblNucleotide</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.122">readEmblNucleotide</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an EMBL-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the EMBL source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="getGenbankBuilderFactory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenbankBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.134">getGenbankBuilderFactory</a>()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling GenBank
 files.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>SmartSequenceBuilder.FACTORY</code></dd></dl>
</li>
</ul>
<a name="readGenbank(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readGenbank</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.148">readGenbank</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an Genbank-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the Genbank source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readGenbankXml(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readGenbankXml</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.161">readGenbankXml</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an GenbankXML-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the GenbanXML source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="getGenpeptBuilderFactory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenpeptBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.174">getGenpeptBuilderFactory</a>()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling Genpept
 files.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>SmartSequenceBuilder.FACTORY</code></dd></dl>
</li>
</ul>
<a name="readGenpept(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readGenpept</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.188">readGenpept</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an Genpept-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the Genpept source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="getSwissprotBuilderFactory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSwissprotBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.200">getSwissprotBuilderFactory</a>()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling Swissprot
 files.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>SmartSequenceBuilder.FACTORY</code></dd></dl>
</li>
</ul>
<a name="readSwissprot(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readSwissprot</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.214">readSwissprot</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an Swissprot-format stream.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - A reader for the Swissprot source or file</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="getFastaBuilderFactory()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFastaBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.226">getFastaBuilderFactory</a>()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Get a default SequenceBuilderFactory for handling FASTA
 files.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>SmartSequenceBuilder.FACTORY</code></dd></dl>
</li>
</ul>
<a name="readFasta(java.io.BufferedReader, org.biojava.bio.seq.io.SymbolTokenization)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFasta</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.241">readFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br,
                         <a href="../../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;sTok)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Read a fasta file.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - the BufferedReader to read data from</dd><dd><code>sTok</code> - a SymbolTokenization that understands the sequences</dd>
<dt><span class="strong">Returns:</span></dt><dd>a SequenceIterator over each sequence in the fasta file</dd></dl>
</li>
</ul>
<a name="readFasta(java.io.BufferedReader, org.biojava.bio.seq.io.SymbolTokenization, org.biojava.bio.seq.io.SequenceBuilderFactory)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFasta</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.261">readFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br,
                         <a href="../../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;sTok,
                         <a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;seqFactory)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Read a fasta file using a custom type of SymbolList. For example,
 use SmartSequenceBuilder.FACTORY to emulate readFasta(BufferedReader,
 SymbolTokenization) and SmartSequenceBuilder.BIT_PACKED to force all
 symbols to be encoded using bit-packing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - the BufferedReader to read data from</dd><dd><code>sTok</code> - a SymbolTokenization that understands the sequences</dd><dd><code>seqFactory</code> - a factory used to build a SymbolList</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readFastaDNA(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFastaDNA</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.279">readFastaDNA</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of DNA sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - the BufferedReader to read data from</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readFastaRNA(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFastaRNA</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.292">readFastaRNA</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of RNA sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - the BufferedReader to read data from</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readFastaProtein(java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFastaProtein</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.305">readFastaProtein</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Iterate over the sequences in an FASTA-format stream of Protein sequences.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>br</code> - the BufferedReader to read data from</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <CODE>SequenceIterator</CODE> that iterates over each
 <CODE>Sequence</CODE> in the file</dd></dl>
</li>
</ul>
<a name="readFasta(java.io.InputStream, org.biojava.bio.symbol.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readFasta</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.327">readFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;seqFile,
                   <a href="../../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)
                            throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Create a sequence database from a fasta file provided as an
 input stream.  Note this somewhat duplicates functionality in
 the readFastaDNA and readFastaProtein methods but uses a stream
 rather than a reader and returns a SequenceDB rather than a
 SequenceIterator. If the returned DB is likely to be large then
 the above mentioned methods should be used.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqFile</code> - The file containg the fasta formatted sequences</dd><dd><code>alpha</code> - The <code>Alphabet</code> of the sequence, ie DNA, RNA etc</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <code>SequenceDB</code> containing all the <code>Sequences</code>
 in the file.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if problems occur during reading of the
 stream.</dd><dt><span class="strong">Since:</span></dt>
  <dd>1.2</dd></dl>
</li>
</ul>
<a name="writeFasta(java.io.OutputStream, org.biojava.bio.seq.db.SequenceDB)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeFasta</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.358">writeFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
              <a href="../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)
                       throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Write a sequenceDB to an output stream in fasta format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the stream to write the fasta formatted data to.</dd><dd><code>db</code> - the database of <code>Sequence</code>s to write</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd><dt><span class="strong">Since:</span></dt>
  <dd>1.2</dd></dl>
</li>
</ul>
<a name="writeFasta(java.io.OutputStream, org.biojava.bio.seq.SequenceIterator)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeFasta</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.374">writeFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
              <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)
                       throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes sequences from a SequenceIterator to an OutputStream in
 Fasta Format.  This makes for a useful format filter where a
 StreamReader can be sent to the StreamWriter after formatting.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - The stream to write fasta formatted data to</dd><dd><code>in</code> - The source of input <code>Sequences</code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd><dt><span class="strong">Since:</span></dt>
  <dd>1.2</dd></dl>
</li>
</ul>
<a name="writeFasta(java.io.OutputStream, org.biojava.bio.seq.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeFasta</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.387">writeFasta</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
              <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)
                       throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Fasta format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>seq</code> - the Sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeEmbl(java.io.OutputStream, org.biojava.bio.seq.SequenceIterator)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeEmbl</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.399">writeEmbl</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
             <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)
                      throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in EMBL format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>in</code> - a SequenceIterator.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeEmbl(java.io.OutputStream, org.biojava.bio.seq.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeEmbl</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.412">writeEmbl</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
             <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)
                      throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in EMBL format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>seq</code> - the Sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeSwissprot(java.io.OutputStream, org.biojava.bio.seq.SequenceIterator)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeSwissprot</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.425">writeSwissprot</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                  <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)
                           throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                  <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in SwissProt
 format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>in</code> - a SequenceIterator.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the <CODE>Sequence</CODE> cannot be converted to SwissProt
 format</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeSwissprot(java.io.OutputStream, org.biojava.bio.seq.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeSwissprot</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.441">writeSwissprot</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                  <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)
                           throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                  <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in SwissProt format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>seq</code> - the Sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the <CODE>Sequence</CODE> cannot be written to SwissProt format</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeGenpept(java.io.OutputStream, org.biojava.bio.seq.SequenceIterator)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeGenpept</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.454">writeGenpept</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)
                         throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in Genpept
 format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>in</code> - a SequenceIterator.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the <CODE>Sequence</CODE> cannot be written to Genpept format</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeGenpept(java.io.OutputStream, org.biojava.bio.seq.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeGenpept</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.470">writeGenpept</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)
                         throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Genpept format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>seq</code> - the Sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the <CODE>Sequence</CODE> cannot be written to Genpept format</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeGenbank(java.io.OutputStream, org.biojava.bio.seq.SequenceIterator)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeGenbank</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.483">writeGenbank</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)
                         throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in Genbank
 format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>in</code> - a SequenceIterator.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="writeGenbank(java.io.OutputStream, org.biojava.bio.seq.Sequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>writeGenbank</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.496">writeGenbank</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)
                         throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Genbank format.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>os</code> - the OutputStream.</dd><dd><code>seq</code> - the Sequence.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there was an error while writing.</dd></dl>
</li>
</ul>
<a name="identifyFormat(java.lang.String, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>identifyFormat</h4>
<pre>public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.519">identifyFormat</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block"><code>identifyFormat</code> performs a case-insensitive mapping
 of a pair of common sequence format name (such as 'embl',
 'genbank' or 'fasta') and alphabet name (such as 'dna', 'rna',
 'protein', 'aa') to an integer. The value returned will be one
 of the public static final fields in
 <code>SeqIOConstants</code>, or a bitwise-or combination of
 them. The method will reject known illegal combinations of
 format and alphabet (such as swissprot + dna) by throwing an
 <code>IllegalArgumentException</code>. It will return the
 <code>SeqIOConstants.UNKNOWN</code> value when either format or
 alphabet are unknown.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>formatName</code> - a <code>String</code>.</dd><dd><code>alphabetName</code> - a <code>String</code>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>an <code>int</code>.</dd></dl>
</li>
</ul>
<a name="getSequenceFormat(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceFormat</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.625">getSequenceFormat</a>(int&nbsp;identifier)
                                        throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block"><code>getSequenceFormat</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceFormat</code> object.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>identifier</code> - an <code>int</code> which represents a binary
 value with bits set according to the scheme described in
 <code>SeqIOConstants</code>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <code>SequenceFormat</code>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if an error occurs.</dd></dl>
</li>
</ul>
<a name="getBuilderFactory(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBuilderFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.672">getBuilderFactory</a>(int&nbsp;identifier)
                                                throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block"><code>getBuilderFactory</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceBuilderFactory</code> object.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>identifier</code> - an <code>int</code> which represents a binary
 value with bits set according to the scheme described in
 <code>SeqIOConstants</code>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <code>SequenceBuilderFactory</code>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if an error occurs.</dd></dl>
</li>
</ul>
<a name="getAlphabet(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlphabet</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.719">getAlphabet</a>(int&nbsp;identifier)
                                  throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block"><code>getAlphabet</code> accepts a value which represents a
 sequence format and returns the relevant
 <code>FiniteAlphabet</code> object.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>identifier</code> - an <code>int</code> which represents a binary
 value with bits set according to the scheme described in
 <code>SeqIOConstants</code>.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a <code>FiniteAlphabet</code>.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if an error occurs.</dd></dl>
</li>
</ul>
<a name="guessFileType(java.io.File)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>guessFileType</h4>
<pre>public static&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.760">guessFileType</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;seqFile)
                         throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>because there is no standard file naming convention
 and guessing by file name is inherantly error prone and bad.</i></div>
<div class="block">Attempts to guess the filetype of a file given the name.  For
 use with the functions below that take an int fileType as a
 parameter. EMBL and Genbank files are assumed to contain DNA
 sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqFile</code> - the <CODE>File</CODE> to read from.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a value that describes the file type.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if <CODE>seqFile</CODE> cannot be read</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a></code> - if <CODE>seqFile</CODE> cannot be found</dd></dl>
</li>
</ul>
<a name="formatToFactory(org.biojava.bio.seq.io.SequenceFormat, org.biojava.bio.symbol.Alphabet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>formatToFactory</h4>
<pre>public static&nbsp;<a href="../../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.853">formatToFactory</a>(<a href="../../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;format,
                                     <a href="../../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)
                                              throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>as this essentially duplicates the operation
 available in the method <code>identifyBuilderFactory</code>.</i></div>
<div class="block">Attempts to retrieve the most appropriate
 <code>SequenceBuilder</code> object for some combination of
 <code>Alphabet</code> and <code>SequenceFormat</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>format</code> - currently supports <code>FastaFormat</code>,
 <code>GenbankFormat</code>, <code>EmblLikeFormat</code></dd><dd><code>alpha</code> - currently only supports the DNA and Protein
 alphabets</dd>
<dt><span class="strong">Returns:</span></dt><dd>the <code>SequenceBuilderFactory</code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the combination of alpha and format is
 unrecognized.</dd></dl>
</li>
</ul>
<a name="fileToBiojava(java.lang.String, java.lang.String, java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fileToBiojava</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.901">fileToBiojava</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                   <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                   <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)
                            throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Reads a file with the specified format and alphabet</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>formatName</code> - the name of the format eg genbank or
 swissprot (case insensitive)</dd><dd><code>alphabetName</code> - the name of the alphabet eg dna or rna or
 protein (case insensitive)</dd><dd><code>br</code> - a BufferedReader for the input</dd>
<dt><span class="strong">Returns:</span></dt><dd>either an Alignment object or a SequenceIterator
 (depending on the format read)</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if an error occurs while reading or a
 unrecognized format, alphabet combination is used (eg swissprot
 and DNA).</dd><dt><span class="strong">Since:</span></dt>
  <dd>1.3</dd></dl>
</li>
</ul>
<a name="fileToBiojava(int, java.io.BufferedReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fileToBiojava</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.922">fileToBiojava</a>(int&nbsp;fileType,
                   <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)
                            throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Reads a file and returns the corresponding Biojava object. You
 need to cast it as an Alignment or a SequenceIterator as
 appropriate.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fileType</code> - a value that describes the file type</dd><dd><code>br</code> - the reader for the input</dd>
<dt><span class="strong">Returns:</span></dt><dd>either a <code>SequenceIterator</code> if the file type is a 
 sequence file, or a <code>Alignment</code> if the file is a sequence
 alignment.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if the file cannot be parsed</dd></dl>
</li>
</ul>
<a name="biojavaToFile(java.lang.String, java.lang.String, java.io.OutputStream, java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>biojavaToFile</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.968">biojavaToFile</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)
                          throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a>,
                                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                 <a href="../../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Writes a Biojava <code>SequenceIterator</code>,
 <code>SequenceDB</code>, <code>Sequence</code> or <code>Aligment</code>
 to an <code>OutputStream</code></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>formatName</code> - eg fasta, GenBank (case insensitive)</dd><dd><code>alphabetName</code> - eg DNA, RNA (case insensititve)</dd><dd><code>os</code> - where to write to</dd><dd><code>biojava</code> - the object to write</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - problems getting data from the biojava object.</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if there are IO problems</dd>
<dd><code><a href="../../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - a Symbol cannot be parsed</dd></dl>
</li>
</ul>
<a name="biojavaToFile(int, java.io.OutputStream, java.lang.Object)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>biojavaToFile</h4>
<pre>public static&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/io/SeqIOTools.html#line.989">biojavaToFile</a>(int&nbsp;fileType,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)
                          throws <a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a>,
                                 <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                 <a href="../../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;</div>
<div class="block">Converts a Biojava object to the given filetype.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fileType</code> - a value that describes the type of sequence file</dd><dd><code>os</code> - the stream to write the formatted results to</dd><dd><code>biojava</code> - a <code>SequenceIterator</code>, <code>SequenceDB</code>, 
 <code>Sequence</code>, or <code>Alignment</code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></code> - if <code>biojava</code> cannot be 
 converted to that format.</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - if the output cannot be written to 
 <code>os</code></dd>
<dd><code><a href="../../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - if <code>biojava
 </code> contains a <code>Symbol</code> that cannot be understood by the
 parser.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SeqIOTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/bio/seq/io/SequenceBuilder.html" title="interface in org.biojava.bio.seq.io"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/io/SeqIOTools.html" target="_top">Frames</a></li>
<li><a href="SeqIOTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
