<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:23 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>Chromatogram (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Chromatogram (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/biojava/bio/chromatogram/ChromatogramFactory.html" title="class in org.biojava.bio.chromatogram"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/chromatogram/Chromatogram.html" target="_top">Frames</a></li>
<li><a href="Chromatogram.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.bio.chromatogram</div>
<h2 title="Interface Chromatogram" class="title">Interface Chromatogram</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../org/biojava/bio/program/abi/ABIFChromatogram.html" title="class in org.biojava.bio.program.abi">ABIFChromatogram</a>, <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram">AbstractChromatogram</a>, <a href="../../../../org/biojava/bio/program/scf/SCF.html" title="class in org.biojava.bio.program.scf">SCF</a>, <a href="../../../../org/biojava/bio/chromatogram/SimpleChromatogram.html" title="class in org.biojava.bio.chromatogram">SimpleChromatogram</a></dd>
</dl>
<hr>
<br>
<pre>public interface <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.36">Chromatogram</a></pre>
<div class="block">Encapsulates the basic information you would want from a chromatogram.
 Read-only.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>1.3</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Rhett Sutphin (<a href="http://genome.uiowa.edu/">UI CBCB</a>), Matthew Pocock</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#DNA">DNA</a></strong></code>
<div class="block">The sequence label for the list of called bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#OFFSETS">OFFSETS</a></strong></code>
<div class="block">The sequence label for the trace offsets of the called bases.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getBaseCalls()">getBaseCalls</a></strong>()</code>
<div class="block">Returns an alignment that describes the base calls for this chromatogram.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax()">getMax</a></strong>()</code>
<div class="block">Gets the max intensity from all the traces.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)">getMax</a></strong>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code>
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getSequenceLength()">getSequenceLength</a></strong>()</code>
<div class="block">Returns the number of bases called by whatever base-calling software
 analyzed the chromatogram as loaded.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getSignificantBits()">getSignificantBits</a></strong>()</code>
<div class="block">Returns the number of bits of the traces which are significant.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)">getTrace</a></strong>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code>
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTraceLength()">getTraceLength</a></strong>()</code>
<div class="block">Returns the length of the trace of the Chromatogram.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#reverseComplement()">reverseComplement</a></strong>()</code>
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.</div>
</td>
</tr>
</table>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="DNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNA</h4>
<pre>static final&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.40">DNA</a></pre>
<div class="block">The sequence label for the list of called bases.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#org.biojava.bio.chromatogram.Chromatogram.DNA">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="OFFSETS">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>OFFSETS</h4>
<pre>static final&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.45">OFFSETS</a></pre>
<div class="block">The sequence label for the trace offsets of the called bases.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#org.biojava.bio.chromatogram.Chromatogram.OFFSETS">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre>int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.52">getMax</a>()</pre>
<div class="block">Gets the max intensity from all the traces.  Must be equivalent
  to the max of calling <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)"><code>getMax(AtomicSymbol)</code></a> on each
  of the four non-ambiguous DNA nucleotides.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the max intensity</dd></dl>
</li>
</ul>
<a name="getMax(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre>int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.60">getMax</a>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
           throws <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotide</code> - the trace to examine.  Must be a concrete 
         (non-ambiguous) nucleotide from the DNA alphabet</dd>
<dt><span class="strong">Returns:</span></dt><dd>the max intensity</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - when the nucleotide isn't from the DNA
          alphabet</dd></dl>
</li>
</ul>
<a name="getTraceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTraceLength</h4>
<pre>int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.66">getTraceLength</a>()</pre>
<div class="block">Returns the length of the trace of the Chromatogram.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of samples in the trace</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)"><code>getTrace(AtomicSymbol)</code></a></dd></dl>
</li>
</ul>
<a name="getTrace(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrace</h4>
<pre>int[]&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.81">getTrace</a>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
               throws <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.  
  This may be a reference the actual internal representation of the 
  samples, so callers <b>must not modify it</b>.
  <p>
  The resulting array for each nucleotide must be <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTraceLength()"><code>getTraceLength()</code></a>
  <code>int</code>s long.
  </p></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nucleotide</code> - the trace to examine.  Must be the symbol for A, C, G, or T
         as provided by <a href="../../../../org/biojava/bio/seq/DNATools.html" title="class in org.biojava.bio.seq"><code>DNATools</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>an array of integers representing the values of a particular 
          chromatogram trace.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - if <code>nucleotide</code> isn't in the DNA alphabet</dd></dl>
</li>
</ul>
<a name="getSignificantBits()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignificantBits</h4>
<pre>int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.89">getSignificantBits</a>()</pre>
<div class="block">Returns the number of bits of the traces which are significant.  For 
 instance, if the chromatogram were originally encoded with a single byte 
 per trace sample, this method must return 8.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of significant bits</dd></dl>
</li>
</ul>
<a name="getBaseCalls()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBaseCalls</h4>
<pre><a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.115">getBaseCalls</a>()</pre>
<div class="block">Returns an alignment that describes the base calls for this chromatogram.
 All of the <code>SymbolList</code>s in this alignment must be the same
 length and that length must equal <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getSequenceLength()"><code>getSequenceLength()</code></a>.
 <p>
 The alignment must contain, at the least, two sequences:
 </p>
 <ol>
   <li>A sequence containing the called bases.  The alphabet of this list
       must be <a href="../../../../org/biojava/bio/seq/DNATools.html#getDNA()"><code>DNATools.getDNA()</code></a>.  
       The label for this list in the alignment must be 
       <code>Chromatogram.DNA</code></li>
   <li>A sequence containing the trace offsets at which the called bases 
       were called.  The alphabet of this list must be an 
       <a href="../../../../org/biojava/bio/symbol/IntegerAlphabet.html" title="class in org.biojava.bio.symbol"><code>IntegerAlphabet</code></a> or a
       <a href="../../../../org/biojava/bio/symbol/IntegerAlphabet.SubIntegerAlphabet.html" title="class in org.biojava.bio.symbol"><code>IntegerAlphabet.SubIntegerAlphabet</code></a>.
       The label for this list in the alignment must be 
       <code>Chromatogram.OFFSETS</code>.</li>
 </ol>
 <p>
 Implementors may provide other sequences as they see fit.
 </p></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>an alignment of at least two sequences, as described above.</dd></dl>
</li>
</ul>
<a name="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre>int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.122">getSequenceLength</a>()</pre>
<div class="block">Returns the number of bases called by whatever base-calling software
 analyzed the chromatogram as loaded.  Must equal 
 <code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getBaseCalls()"><code>getBaseCalls()</code></a>.length()</code>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the number of bases</dd></dl>
</li>
</ul>
<a name="reverseComplement()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>reverseComplement</h4>
<pre><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.133">reverseComplement</a>()</pre>
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.
 <p>
 Implementors should copy the metadata about the chromatogram (i.e., base 
 calls) as is appropriate to their formats.
 </p></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new chromatogram that is the reverse complement of this one</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/biojava/bio/chromatogram/ChromatogramFactory.html" title="class in org.biojava.bio.chromatogram"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/chromatogram/Chromatogram.html" target="_top">Frames</a></li>
<li><a href="Chromatogram.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
