<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:29 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>Uses of Class org.biojava.bio.BioException (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.biojava.bio.BioException (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/class-use/BioException.html" target="_top">Frames</a></li>
<li><a href="BioException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.bio.BioException" class="title">Uses of Class<br>org.biojava.bio.BioException</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.alignment">org.biojava.bio.alignment</a></td>
<td class="colLast">
<div class="block">
Classes to generate and describe sequence alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.annodb">org.biojava.bio.annodb</a></td>
<td class="colLast">
<div class="block">
Databases of generic structured data (annotation) objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.chromatogram">org.biojava.bio.chromatogram</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for chromatogram data, as produced by DNA sequencing equipment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.dist">org.biojava.bio.dist</a></td>
<td class="colLast">
<div class="block">
Probability distributions over Alphabets.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.dp">org.biojava.bio.dp</a></td>
<td class="colLast">
<div class="block">HMM and Dynamic Programming Algorithms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.dp.onehead">org.biojava.bio.dp.onehead</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.dp.twohead">org.biojava.bio.dp.twohead</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.gui">org.biojava.bio.gui</a></td>
<td class="colLast">
<div class="block">Graphical interfaces for biojava objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.molbio">org.biojava.bio.molbio</a></td>
<td class="colLast">
<div class="block">The classes and interfaces in this package cover common molecular
biological techniques such as restriction digests and PCR.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.formats">org.biojava.bio.program.formats</a></td>
<td class="colLast">
<div class="block">
Experimental parsers using the tagvalue framework.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff">org.biojava.bio.program.gff</a></td>
<td class="colLast">
<div class="block">GFF manipulation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff3">org.biojava.bio.program.gff3</a></td>
<td class="colLast">
<div class="block">
Support for reading and writing GFF3.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.homologene">org.biojava.bio.program.homologene</a></td>
<td class="colLast">
<div class="block">
Support classes for Homologene data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.indexdb">org.biojava.bio.program.indexdb</a></td>
<td class="colLast">
<div class="block">A flat-file ascii index of ascii flat files as per the OBDA specification.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.phred">org.biojava.bio.program.phred</a></td>
<td class="colLast">
<div class="block">Parser for Phred output</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.sax.blastxml">org.biojava.bio.program.sax.blastxml</a></td>
<td class="colLast">
<div class="block">
Parsers for the XML output from NCBI blast.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssaha">org.biojava.bio.program.ssaha</a></td>
<td class="colLast">
<div class="block">SSAHA sequence searching API.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssbind">org.biojava.bio.program.ssbind</a></td>
<td class="colLast">
<div class="block">
Creation of objects from SAX events using the BioJava BlastLikeDataSetCollection DTD.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.unigene">org.biojava.bio.program.unigene</a></td>
<td class="colLast">
<div class="block">Objects for representing Unigene clusters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.xff">org.biojava.bio.program.xff</a></td>
<td class="colLast">
<div class="block">Event-driven parsing system for the Extensible Feature Format (XFF).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.proteomics">org.biojava.bio.proteomics</a></td>
<td class="colLast">
<div class="block">Utilities to aid in performing various physical analysis of proteins.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.proteomics.aaindex">org.biojava.bio.proteomics.aaindex</a></td>
<td class="colLast">
<div class="block">Classes and interfaces to load Amino Acid Index database files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.search">org.biojava.bio.search</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for representing sequence similarity search results.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq">org.biojava.bio.seq</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for defining biological sequences and informatics
objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db">org.biojava.bio.seq.db</a></td>
<td class="colLast">
<div class="block">
Collections of biological sequence data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biofetch">org.biojava.bio.seq.db.biofetch</a></td>
<td class="colLast">
<div class="block">Client for the OBDA BioFetch protocol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biosql">org.biojava.bio.seq.db.biosql</a></td>
<td class="colLast">
<div class="block">General purpose Sequence storage in a relational database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.flat">org.biojava.bio.seq.db.flat</a></td>
<td class="colLast">
<div class="block">
Support for OBDA flatfile databases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.distributed">org.biojava.bio.seq.distributed</a></td>
<td class="colLast">
<div class="block">Sequences and SequenceDBs which are composed from data taken
from a number of data sources.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.homol">org.biojava.bio.seq.homol</a></td>
<td class="colLast">
<div class="block">The classes and interfaces for defining sequence similarity and
honology.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.impl">org.biojava.bio.seq.impl</a></td>
<td class="colLast">
<div class="block">Standard in-memory implementations of <code>Sequence</code> and
<code>Feature</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io">org.biojava.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for processing and producing flat-file representations
of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.projection">org.biojava.bio.seq.projection</a></td>
<td class="colLast">
<div class="block">Code for projecting Feature objects and systematically altering their
properties.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.symbol">org.biojava.bio.symbol</a></td>
<td class="colLast">
<div class="block">Representation of the Symbols that make up a sequence, and locations within
them.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.taxa">org.biojava.bio.taxa</a></td>
<td class="colLast">
<div class="block">Taxonomy object for representing species information.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.directory">org.biojava.directory</a></td>
<td class="colLast">
<div class="block">Open Bio Sequence Database Access (OBDA) registry support.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.utils.automata">org.biojava.utils.automata</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.utils.math">org.biojava.utils.math</a></td>
<td class="colLast">
<div class="block">
Mathematical utility classes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.utils.regex">org.biojava.utils.regex</a></td>
<td class="colLast">
<div class="block">This package is used to perform regular expression searches of
SymbolLists defined in arbitrary Alphabets.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.utils.walker">org.biojava.utils.walker</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio">org.biojavax.bio</a></td>
<td class="colLast">
<div class="block">Classes to represent biological entities and their relationships.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.alignment">org.biojavax.bio.alignment</a></td>
<td class="colLast">
<div class="block">This package contains the interfaces that need to be implemented by discrete alignment services.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.alignment.blast">org.biojavax.bio.alignment.blast</a></td>
<td class="colLast">
<div class="block">This package holds implementations of the org.biojava3.ws.alignment interfaces for the NCBI remote QBlast services
  through the WWW.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db">org.biojavax.bio.db</a></td>
<td class="colLast">
<div class="block">Interactions between biojavax objects and a DB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.biosql">org.biojavax.bio.db.biosql</a></td>
<td class="colLast">
<div class="block">Interface between biojava and biosql databases</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.ncbi">org.biojavax.bio.db.ncbi</a></td>
<td class="colLast">
<div class="block">Interfaces to NCBI data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.seq">org.biojavax.bio.seq</a></td>
<td class="colLast">
<div class="block">Rich implementations of Sequences, Locations and Features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.seq.io">org.biojavax.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes to support the I/O of RichSequence and 
     Bioentry objects.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.bio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FlexibleAlignment.</span><code><strong><a href="../../../../org/biojava/bio/alignment/FlexibleAlignment.html#addSequence(org.biojava.bio.alignment.AlignmentElement)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/alignment/AlignmentElement.html" title="interface in org.biojava.bio.alignment">AlignmentElement</a>&nbsp;ae)</code>
<div class="block">add a new a alignment usings a location to the reference sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ARAlignment.</span><code><strong><a href="../../../../org/biojava/bio/alignment/ARAlignment.html#addSequence(org.biojava.bio.alignment.AlignmentElement)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/alignment/AlignmentElement.html" title="interface in org.biojava.bio.alignment">AlignmentElement</a>&nbsp;ae)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AlignmentPair.</span><code><strong><a href="../../../../org/biojava/bio/alignment/AlignmentPair.html#formatOutput()">formatOutput</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AlignmentPair.</span><code><strong><a href="../../../../org/biojava/bio/alignment/AlignmentPair.html#formatOutput(int)">formatOutput</a></strong>(int&nbsp;width)</code>
<div class="block">This method provides a BLAST-like formated alignment from the given
 <code>String</code>s, in which the sequence coordinates and the
 information "Query" or "Sbjct", respectively is added to each line.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a></code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#getSubstitutionMatrix(java.io.BufferedReader)">getSubstitutionMatrix</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;reader)</code>
<div class="block">This constructor can be used to guess the alphabet of this substitution
 matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a></code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#getSubstitutionMatrix(org.biojava.bio.symbol.FiniteAlphabet,%20java.io.BufferedReader)">getSubstitutionMatrix</a></strong>(<a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;alphabet,
                                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;reader)</code>
<div class="block">Return a new substitution matrix with the specified alphabet.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a></code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#getSubstitutionMatrix(org.biojava.bio.symbol.FiniteAlphabet,%20java.io.BufferedReader,%20java.lang.String)">getSubstitutionMatrix</a></strong>(<a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;alphabet,
                                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;reader,
                                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Return a new substitution matrix with the specified alphabet and name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>short</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#getValueAt(org.biojava.bio.symbol.Symbol,%20org.biojava.bio.symbol.Symbol)">getValueAt</a></strong>(<a href="../../../../org/biojava/bio/symbol/Symbol.html" title="interface in org.biojava.bio.symbol">Symbol</a>&nbsp;row,
                    <a href="../../../../org/biojava/bio/symbol/Symbol.html" title="interface in org.biojava.bio.symbol">Symbol</a>&nbsp;col)</code>
<div class="block">There are some substitution matrices containing more columns than lines.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a></code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#normalizeMatrix()">normalizeMatrix</a></strong>()</code>
<div class="block">With this method you can get a &ldquo;normalized&rdquo;
 <code>SubstitutionMatrix</code> object; however, since this
 implementation uses an short matrix, the normalized matrix will be scaled
 by ten.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/alignment/AlignmentPair.html" title="class in org.biojava.bio.alignment">AlignmentPair</a></code></td>
<td class="colLast"><span class="strong">NeedlemanWunsch.</span><code><strong><a href="../../../../org/biojava/bio/alignment/NeedlemanWunsch.html#pairwiseAlignment(org.biojava.bio.symbol.SymbolList,%20org.biojava.bio.symbol.SymbolList)">pairwiseAlignment</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;query,
                                  <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;subject)</code>
<div class="block">Global pairwise sequence alignment of two BioJava-Sequence objects
 according to the Needleman-Wunsch-algorithm.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/AlignmentPair.html#AlignmentPair(org.biojava.bio.seq.Sequence,%20org.biojava.bio.seq.Sequence,%20int,%20int,%20int,%20int,%20org.biojava.bio.alignment.SubstitutionMatrix)">AlignmentPair</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;query,
                          <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;subject,
                          int&nbsp;queryStart,
                          int&nbsp;queryEnd,
                          int&nbsp;subjectStart,
                          int&nbsp;subjectEnd,
                          <a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a>&nbsp;subMatrix)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/AlignmentPair.html#AlignmentPair(org.biojava.bio.seq.Sequence,%20org.biojava.bio.seq.Sequence,%20org.biojava.bio.alignment.SubstitutionMatrix)">AlignmentPair</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;query,
                          <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;subject,
                          <a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html" title="class in org.biojava.bio.alignment">SubstitutionMatrix</a>&nbsp;subMatrix)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/FlexibleAlignment.html#FlexibleAlignment(java.util.List)">FlexibleAlignment</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/bio/alignment/AlignmentElement.html" title="interface in org.biojava.bio.alignment">AlignmentElement</a>&gt;&nbsp;seqList)</code>
<div class="block">construct this object with the reference sequence which can either be a
 gappedSymbolList or not label in all cases refers to an object that holds
 the display name (generally just a String). since more than one sequence
 in an alignment could have the same name this works as long as the labels
 are different objects even though they may hold the same name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/SimpleAlignmentElement.html#SimpleAlignmentElement(java.lang.String,%20org.biojava.bio.symbol.SymbolList,%20org.biojava.bio.symbol.Location)">SimpleAlignmentElement</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;label,
                                            <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;seq,
                                            <a href="../../../../org/biojava/bio/symbol/Location.html" title="interface in org.biojava.bio.symbol">Location</a>&nbsp;loc)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#SubstitutionMatrix(java.io.File)">SubstitutionMatrix</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>
<div class="block">This constructor can be used to guess the alphabet of this substitution
 matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#SubstitutionMatrix(org.biojava.bio.symbol.FiniteAlphabet,%20java.io.File)">SubstitutionMatrix</a></strong>(<a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;alpha,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;matrixFile)</code>
<div class="block">This constructs a <code>SubstitutionMatrix</code> object that contains
 two <code>Map</code> data structures having BioJava symbols as keys and
 the value being the index of the matrix containing the substitution
 score.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/alignment/SubstitutionMatrix.html#SubstitutionMatrix(org.biojava.bio.symbol.FiniteAlphabet,%20java.lang.String,%20java.lang.String)">SubstitutionMatrix</a></strong>(<a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;alpha,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matrixString,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">With this constructor it is possible to construct a SubstitutionMatrix
 object from a substitution matrix file.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.annodb">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/annodb/package-summary.html">org.biojava.bio.annodb</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/annodb/package-summary.html">org.biojava.bio.annodb</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/annodb/IndexedAnnotationDB.html#IndexedAnnotationDB(java.lang.String,%20java.io.File,%20org.biojava.bio.program.tagvalue.Index2Model,%20java.util.List,%20int,%20org.biojava.bio.AnnotationType,%20org.biojava.bio.annodb.IndexedAnnotationDB.ParserListenerFactory)">IndexedAnnotationDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbName,
                                      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeLoc,
                                      <a href="../../../../org/biojava/bio/program/tagvalue/Index2Model.html" title="class in org.biojava.bio.program.tagvalue">Index2Model</a>&nbsp;model,
                                      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;toIndex,
                                      int&nbsp;maxKeyLen,
                                      <a href="../../../../org/biojava/bio/AnnotationType.html" title="interface in org.biojava.bio">AnnotationType</a>&nbsp;schema,
                                      <a href="../../../../org/biojava/bio/annodb/IndexedAnnotationDB.ParserListenerFactory.html" title="interface in org.biojava.bio.annodb">IndexedAnnotationDB.ParserListenerFactory</a>&nbsp;plFactory)</code>
<div class="block">Create a new IndexedAnnotationDB.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.chromatogram">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/chromatogram/package-summary.html">org.biojava.bio.chromatogram</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/chromatogram/package-summary.html">org.biojava.bio.chromatogram</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/UnsupportedChromatogramFormatException.html" title="class in org.biojava.bio.chromatogram">UnsupportedChromatogramFormatException</a></strong></code>
<div class="block">Exception indicating that some provided data does not represent
 a supported chromatogram format.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dist">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/dist/package-summary.html">org.biojava.bio.dist</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/dist/package-summary.html">org.biojava.bio.dist</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">DistributionTools.</span><code><strong><a href="../../../../org/biojava/bio/dist/DistributionTools.html#areEmissionSpectraEqual(org.biojava.bio.dist.Distribution[],%20org.biojava.bio.dist.Distribution[])">areEmissionSpectraEqual</a></strong>(<a href="../../../../org/biojava/bio/dist/Distribution.html" title="interface in org.biojava.bio.dist">Distribution</a>[]&nbsp;a,
                                              <a href="../../../../org/biojava/bio/dist/Distribution.html" title="interface in org.biojava.bio.dist">Distribution</a>[]&nbsp;b)</code>
<div class="block">Compares the emission spectra of two distribution arrays.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">DistributionTools.</span><code><strong><a href="../../../../org/biojava/bio/dist/DistributionTools.html#areEmissionSpectraEqual(org.biojava.bio.dist.Distribution,%20org.biojava.bio.dist.Distribution)">areEmissionSpectraEqual</a></strong>(<a href="../../../../org/biojava/bio/dist/Distribution.html" title="interface in org.biojava.bio.dist">Distribution</a>&nbsp;a,
                                              <a href="../../../../org/biojava/bio/dist/Distribution.html" title="interface in org.biojava.bio.dist">Distribution</a>&nbsp;b)</code>
<div class="block">Compares the emission spectra of two distributions.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/dp/IllegalTransitionException.html" title="class in org.biojava.bio.dp">IllegalTransitionException</a></strong></code>
<div class="block">This exception indicates that there is no transition between two states.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">WeightMatrixAnnotator.</span><code><strong><a href="../../../../org/biojava/bio/dp/WeightMatrixAnnotator.html#annotate(org.biojava.bio.seq.Sequence)">annotate</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleHMMTrainer.</span><code><strong><a href="../../../../org/biojava/bio/dp/SimpleHMMTrainer.html#completeCycle()">completeCycle</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HMMTrainer.</span><code><strong><a href="../../../../org/biojava/bio/dp/HMMTrainer.html#completeCycle()">completeCycle</a></strong>()</code>
<div class="block">indicate that a cycle of training is completed and the
 emission/transition matrices should be updated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/DP.html" title="class in org.biojava.bio.dp">DP</a></code></td>
<td class="colLast"><span class="strong">DPFactory.</span><code><strong><a href="../../../../org/biojava/bio/dp/DPFactory.html#createDP(org.biojava.bio.dp.MarkovModel)">createDP</a></strong>(<a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a>&nbsp;model)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/DP.html" title="class in org.biojava.bio.dp">DP</a></code></td>
<td class="colLast"><span class="strong">DPFactory.DefaultFactory.</span><code><strong><a href="../../../../org/biojava/bio/dp/DPFactory.DefaultFactory.html#createDP(org.biojava.bio.dp.MarkovModel)">createDP</a></strong>(<a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a>&nbsp;model)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/DPMatrix.html" title="interface in org.biojava.bio.dp">DPMatrix</a></code></td>
<td class="colLast"><span class="strong">DP.</span><code><strong><a href="../../../../org/biojava/bio/dp/DP.html#forwardsBackwards(org.biojava.bio.symbol.SymbolList[],%20org.biojava.bio.dp.ScoreType)">forwardsBackwards</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>[]&nbsp;symList,
                                  <a href="../../../../org/biojava/bio/dp/ScoreType.html" title="interface in org.biojava.bio.dp">ScoreType</a>&nbsp;scoreType)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/StatePath.html" title="interface in org.biojava.bio.dp">StatePath</a></code></td>
<td class="colLast"><span class="strong">DP.</span><code><strong><a href="../../../../org/biojava/bio/dp/DP.html#generate(int)">generate</a></strong>(int&nbsp;length)</code>
<div class="block">
 Generates an alignment from a model.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/dp/WeightMatrix.html" title="interface in org.biojava.bio.dp">WeightMatrix</a></code></td>
<td class="colLast"><span class="strong">XmlMarkovModel.</span><code><strong><a href="../../../../org/biojava/bio/dp/XmlMarkovModel.html#readMatrix(org.w3c.dom.Element)">readMatrix</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/org/w3c/dom/Element.html?is-external=true" title="class or interface in org.w3c.dom">Element</a>&nbsp;root)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a></code></td>
<td class="colLast"><span class="strong">XmlMarkovModel.</span><code><strong><a href="../../../../org/biojava/bio/dp/XmlMarkovModel.html#readModel(org.w3c.dom.Element)">readModel</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/org/w3c/dom/Element.html?is-external=true" title="class or interface in org.w3c.dom">Element</a>&nbsp;root)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">WMAsMM.</span><code><strong><a href="../../../../org/biojava/bio/dp/WMAsMM.html#registerWithTrainer(org.biojava.bio.dp.ModelTrainer)">registerWithTrainer</a></strong>(<a href="../../../../org/biojava/bio/dp/ModelTrainer.html" title="interface in org.biojava.bio.dp">ModelTrainer</a>&nbsp;modelTrainer)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Trainable.</span><code><strong><a href="../../../../org/biojava/bio/dp/Trainable.html#registerWithTrainer(org.biojava.bio.dp.ModelTrainer)">registerWithTrainer</a></strong>(<a href="../../../../org/biojava/bio/dp/ModelTrainer.html" title="interface in org.biojava.bio.dp">ModelTrainer</a>&nbsp;mt)</code>
<div class="block">Perform any registration that is necessary with mt.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/State.html" title="interface in org.biojava.bio.dp">State</a>[]</code></td>
<td class="colLast"><span class="strong">DP.</span><code><strong><a href="../../../../org/biojava/bio/dp/DP.html#stateList(org.biojava.bio.dp.MarkovModel)">stateList</a></strong>(<a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a>&nbsp;mm)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">TrainingAlgorithm.</span><code><strong><a href="../../../../org/biojava/bio/dp/TrainingAlgorithm.html#train(org.biojava.bio.seq.db.SequenceDB,%20double,%20org.biojava.bio.dp.StoppingCriteria)">train</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
          double&nbsp;nullWeight,
          <a href="../../../../org/biojava/bio/dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code>
<div class="block">Trains the sequences in db untill stopper says to finnish.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractTrainer.</span><code><strong><a href="../../../../org/biojava/bio/dp/AbstractTrainer.html#train(org.biojava.bio.seq.db.SequenceDB,%20double,%20org.biojava.bio.dp.StoppingCriteria)">train</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
          double&nbsp;nullModelWeight,
          <a href="../../../../org/biojava/bio/dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code>
<div class="block">Trains the sequences in db until stopper says to finnish.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp.onehead">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/dp/onehead/package-summary.html">org.biojava.bio.dp.onehead</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/dp/onehead/package-summary.html">org.biojava.bio.dp.onehead</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/dp/onehead/SingleDP.html#SingleDP(org.biojava.bio.dp.MarkovModel)">SingleDP</a></strong>(<a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a>&nbsp;model)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp.twohead">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/dp/twohead/package-summary.html">org.biojava.bio.dp.twohead</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/dp/twohead/package-summary.html">org.biojava.bio.dp.twohead</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/dp/twohead/CellCalculatorFactory.html" title="interface in org.biojava.bio.dp.twohead">CellCalculatorFactory</a></code></td>
<td class="colLast"><span class="strong">CellCalculatorFactoryMaker.</span><code><strong><a href="../../../../org/biojava/bio/dp/twohead/CellCalculatorFactoryMaker.html#make(org.biojava.bio.dp.DP)">make</a></strong>(<a href="../../../../org/biojava/bio/dp/DP.html" title="class in org.biojava.bio.dp">DP</a>&nbsp;dp)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/dp/twohead/package-summary.html">org.biojava.bio.dp.twohead</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/dp/twohead/PairwiseDP.html#PairwiseDP(org.biojava.bio.dp.MarkovModel,%20org.biojava.bio.dp.twohead.CellCalculatorFactoryMaker)">PairwiseDP</a></strong>(<a href="../../../../org/biojava/bio/dp/MarkovModel.html" title="interface in org.biojava.bio.dp">MarkovModel</a>&nbsp;mm,
                    <a href="../../../../org/biojava/bio/dp/twohead/CellCalculatorFactoryMaker.html" title="interface in org.biojava.bio.dp.twohead">CellCalculatorFactoryMaker</a>&nbsp;ccfm)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/gui/package-summary.html">org.biojava.bio.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/gui/package-summary.html">org.biojava.bio.gui</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FeatureTree.</span><code><strong><a href="../../../../org/biojava/bio/gui/FeatureTree.html#setSequenceDB(org.biojava.bio.seq.db.SequenceDB)">setSequenceDB</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block">Use this method to provide the sequences for the tree to work with.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.molbio">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/molbio/package-summary.html">org.biojava.bio.molbio</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/molbio/package-summary.html">org.biojava.bio.molbio</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><span class="strong">RestrictionEnzyme.</span><code><strong><a href="../../../../org/biojava/bio/molbio/RestrictionEnzyme.html#getUpstreamCut()">getUpstreamCut</a></strong>()</code>
<div class="block"><code>getUpstreamCut</code> returns the cut site upstream of
 the recognition site.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">RestrictionEnzyme.</span><code><strong><a href="../../../../org/biojava/bio/molbio/RestrictionEnzyme.html#getUpstreamEndType()">getUpstreamEndType</a></strong>()</code>
<div class="block"><code>getUpstreamEndType</code> returns the double-stranded end
 type produced by the secondary (upstream) cut.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.formats">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/formats/package-summary.html">org.biojava.bio.program.formats</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/formats/package-summary.html">org.biojava.bio.program.formats</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/formats/Format.html" title="interface in org.biojava.bio.program.formats">Format</a></code></td>
<td class="colLast"><span class="strong">FormatTools.</span><code><strong><a href="../../../../org/biojava/bio/program/formats/FormatTools.html#getFormat(java.lang.String)">getFormat</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName)</code>
<div class="block">Attempt to find aformat for a format identifer string.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#annotateSequences(org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.program.gff.GFFEntrySet)">annotateSequences</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqs,
                                  <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a>&nbsp;ents)</code>
<div class="block">Annotates all sequences in a sequence DB with features from a GFF entry set.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/program/gff/SimpleGFFRecord.html" title="class in org.biojava.bio.program.gff">SimpleGFFRecord</a></code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#createGFFRecord(org.biojava.bio.seq.Feature,%20java.lang.String)">createGFFRecord</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feature,
                              <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Internal method to create a <span class="type">GFFRecord</span>
 from an individual <span class="type">Feature</span>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/program/gff/GFFRecord.html" title="interface in org.biojava.bio.program.gff">GFFRecord</a></code></td>
<td class="colLast"><span class="strong">GFFParser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFParser.html#createRecord(org.biojava.bio.program.gff.GFFDocumentHandler,%20java.util.List,%20java.lang.String,%20java.lang.String)">createRecord</a></strong>(<a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;aList,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;rest,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;comment)</code>
<div class="block">Actually turns a list of tokens, some value string and a comment into a
 <span class="type">GFFRecord</span> and informs
 <span class="arg">handler</span>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#doPreProcessSequence(org.biojava.bio.seq.Sequence,%20org.biojava.bio.program.gff.GFFDocumentHandler,%20java.lang.String)">doPreProcessSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                                        <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler,
                                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Emit any per-sequence header information.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#doProcessFeature(org.biojava.bio.seq.Feature,%20org.biojava.bio.program.gff.GFFDocumentHandler,%20java.lang.String)">doProcessFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feature,
                                <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Internal method to process an individual <span class="type">Feature</span>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#doProcessSequence(org.biojava.bio.seq.Sequence,%20org.biojava.bio.program.gff.GFFDocumentHandler,%20java.lang.String)">doProcessSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                                  <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler,
                                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Internal method to process an individual <span class="type">Sequence</span>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#gffFromSeqDB(org.biojava.bio.seq.db.SequenceDB)">gffFromSeqDB</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code>
<div class="block">Creates a GFFEntrySet containing one entry for each feature on each
 sequence of a SequenceDB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#gffFromSequence(org.biojava.bio.seq.Sequence)">gffFromSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block">Creates a GFFEntrySet containing one entry for each feature on a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFWriter.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFWriter.html#invalidEnd(java.lang.String,%20java.lang.NumberFormatException)">invalidEnd</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;token,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a>&nbsp;nfe)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFWriter.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFWriter.html#invalidFrame(java.lang.String,%20java.lang.NumberFormatException)">invalidFrame</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;token,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a>&nbsp;nfe)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFWriter.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFWriter.html#invalidScore(java.lang.String,%20java.lang.NumberFormatException)">invalidScore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;token,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a>&nbsp;nfe)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFWriter.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFWriter.html#invalidStart(java.lang.String,%20java.lang.NumberFormatException)">invalidStart</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;token,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/NumberFormatException.html?is-external=true" title="class or interface in java.lang">NumberFormatException</a>&nbsp;nfe)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFWriter.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFWriter.html#invalidStrand(java.lang.String)">invalidStrand</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;token)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFParser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFParser.html#parse(java.io.BufferedReader,%20org.biojava.bio.program.gff.GFFDocumentHandler)">parse</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;bReader,
          <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler)</code>
<div class="block">Informs <span class="arg">handler</span> of each line of
 gff read from <span class="arg">bReader</span>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFFParser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFParser.html#parse(java.io.BufferedReader,%20org.biojava.bio.program.gff.GFFDocumentHandler,%20java.lang.String)">parse</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;bReader,
          <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler,
          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;locator)</code>
<div class="block">Informs <span class="arg">handler</span> of each line of
 GFF read from <span class="arg">bReader</span></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#processDB(org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.program.gff.GFFDocumentHandler)">processDB</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
                  <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler)</code>
<div class="block">Process all <span class="type">Sequence</span>s within a
 <span class="type">SequenceDB</span>, informing
 <span class="arg">handler</span> of any suitable features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequencesAsGFF.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#processSequence(org.biojava.bio.seq.Sequence,%20org.biojava.bio.program.gff.GFFDocumentHandler)">processSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                              <a href="../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler)</code>
<div class="block">Process an individual <span class="type">Sequence</span>, informing
 <span class="arg">handler</span> of any suitable features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.io.BufferedReader)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;gffIn)</code>
<div class="block">Read all GFF entries from a buffered reader.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.io.BufferedReader,%20org.biojava.bio.program.gff.GFFRecordFilter)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;gffIn,
              <a href="../../../../org/biojava/bio/program/gff/GFFRecordFilter.html" title="interface in org.biojava.bio.program.gff">GFFRecordFilter</a>&nbsp;recFilt)</code>
<div class="block">Read all GFF entries matching a filter from a buffered reader.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.io.File)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;inFile)</code>
<div class="block">Reads a <code>GFFEntrySet</code> from a file with no filtering.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.io.File,%20org.biojava.bio.program.gff.GFFRecordFilter)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;inFile,
              <a href="../../../../org/biojava/bio/program/gff/GFFRecordFilter.html" title="interface in org.biojava.bio.program.gff">GFFRecordFilter</a>&nbsp;recFilt)</code>
<div class="block">Reads a GFFEntrySet from a file with the specified filter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.lang.String)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fileName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use: readGff(File)</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="strong">GFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/gff/GFFTools.html#readGFF(java.lang.String,%20org.biojava.bio.program.gff.GFFRecordFilter)">readGFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fileName,
              <a href="../../../../org/biojava/bio/program/gff/GFFRecordFilter.html" title="interface in org.biojava.bio.program.gff">GFFRecordFilter</a>&nbsp;recFilt)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use: readGff(File,GFFRecordFilter)</i></div>
</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff3">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/gff3/package-summary.html">org.biojava.bio.program.gff3</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/gff3/package-summary.html">org.biojava.bio.program.gff3</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/program/gff3/GFF3Record.html" title="interface in org.biojava.bio.program.gff3">GFF3Record</a></code></td>
<td class="colLast"><span class="strong">GFF3Parser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff3/GFF3Parser.html#createRecord(org.biojava.bio.program.gff3.GFF3DocumentHandler,%20java.util.List,%20java.lang.String,%20java.lang.String,%20org.biojava.ontology.Ontology,%20org.biojava.ontology.Ontology)">createRecord</a></strong>(<a href="../../../../org/biojava/bio/program/gff3/GFF3DocumentHandler.html" title="interface in org.biojava.bio.program.gff3">GFF3DocumentHandler</a>&nbsp;handler,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;aList,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;rest,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;comment,
                        <a href="../../../../org/biojava/ontology/Ontology.html" title="interface in org.biojava.ontology">Ontology</a>&nbsp;ontology,
                        <a href="../../../../org/biojava/ontology/Ontology.html" title="interface in org.biojava.ontology">Ontology</a>&nbsp;fallBack)</code>
<div class="block">Actually turns a list of tokens, some value string and a comment into a
 <span class="type">GFF3Record</span> and informs
 <span class="arg">handler</span>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFF3Parser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff3/GFF3Parser.html#parse(java.io.BufferedReader,%20org.biojava.bio.program.gff3.GFF3DocumentHandler,%20org.biojava.ontology.Ontology)">parse</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;bReader,
          <a href="../../../../org/biojava/bio/program/gff3/GFF3DocumentHandler.html" title="interface in org.biojava.bio.program.gff3">GFF3DocumentHandler</a>&nbsp;handler,
          <a href="../../../../org/biojava/ontology/Ontology.html" title="interface in org.biojava.ontology">Ontology</a>&nbsp;ontology)</code>
<div class="block">Informs <span class="arg">handler</span> of each line of
 gff read from <span class="arg">bReader</span>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">GFF3Parser.</span><code><strong><a href="../../../../org/biojava/bio/program/gff3/GFF3Parser.html#parse(java.io.BufferedReader,%20org.biojava.bio.program.gff3.GFF3DocumentHandler,%20org.biojava.ontology.Ontology,%20java.lang.String)">parse</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;bReader,
          <a href="../../../../org/biojava/bio/program/gff3/GFF3DocumentHandler.html" title="interface in org.biojava.bio.program.gff3">GFF3DocumentHandler</a>&nbsp;handler,
          <a href="../../../../org/biojava/ontology/Ontology.html" title="interface in org.biojava.ontology">Ontology</a>&nbsp;ontology,
          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;locator)</code>
<div class="block">Informs <span class="arg">handler</span> of each line of
 GFF read from <span class="arg">bReader</span></div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.homologene">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/homologene/package-summary.html">org.biojava.bio.program.homologene</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/homologene/package-summary.html">org.biojava.bio.program.homologene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/program/homologene/DuplicateTaxonException.html" title="class in org.biojava.bio.program.homologene">DuplicateTaxonException</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.indexdb">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/indexdb/package-summary.html">org.biojava.bio.program.indexdb</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/indexdb/package-summary.html">org.biojava.bio.program.indexdb</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/indexdb/BioStore.html" title="class in org.biojava.bio.program.indexdb">BioStore</a></code></td>
<td class="colLast"><span class="strong">BioStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/BioStoreFactory.html#createBioStore()">createBioStore</a></strong>()</code>
<div class="block"><code>createBioStore</code> creates a <code>BioStore</code>
 reflecting the current state of the factory and returns a
 reference to it.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/indexdb/Record.html" title="interface in org.biojava.bio.program.indexdb">Record</a></code></td>
<td class="colLast"><span class="strong">IndexStore.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexStore.html#get(java.lang.String)">get</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block"><code>get</code> returns a record specified by a primary
 identifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a></code></td>
<td class="colLast"><span class="strong">IndexStore.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexStore.html#get(java.lang.String,%20java.lang.String)">get</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;namespace)</code>
<div class="block"><code>get</code> returns a list of <code>Record</code>s by
 searching against the primary identifiers if the namespace
 argument is equal to the primary namespace or otherwise by
 searching the secondary namespaces.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a></code></td>
<td class="colLast"><span class="strong">BioStore.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/BioStore.html#get(java.lang.String,%20java.lang.String)">get</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;namespace)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">IndexTools.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexTools.html#indexEmbl(java.lang.String,%20java.io.File,%20java.io.File[],%20int)">indexEmbl</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;location,
                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>[]&nbsp;seqFiles,
                  int&nbsp;alphabetIdentifier)</code>
<div class="block"><code>indexEmbl</code> indexes DNA, RNA or protein EMBL format
 sequence files on ID as primary identifier and AC as secondary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">IndexTools.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexTools.html#indexFasta(java.lang.String,%20java.io.File,%20java.io.File[],%20int)">indexFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;location,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>[]&nbsp;seqFiles,
                    int&nbsp;alphabetIdentifier)</code>
<div class="block"><code>indexFasta</code> indexes DNA, RNA or protein Fasta
 format sequence files on primary identifier.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">IndexTools.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexTools.html#indexGenbank(java.lang.String,%20java.io.File,%20java.io.File[],%20int)">indexGenbank</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;location,
                        <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>[]&nbsp;seqFiles,
                        int&nbsp;alphabetIdentifier)</code>
<div class="block"><code>indexGenbank</code> indexes DNA, RNA or protein Genbank
 format sequence files on LOCUS as primary identifier and
 ACCESSION as secondary.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">IndexTools.</span><code><strong><a href="../../../../org/biojava/bio/program/indexdb/IndexTools.html#indexSwissprot(java.lang.String,%20java.io.File,%20java.io.File[])">indexSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;location,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>[]&nbsp;seqFiles)</code>
<div class="block"><code>indexSwissprot</code> indexes Swissprot format protein
 sequence files on ID as primary identifier.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/program/indexdb/package-summary.html">org.biojava.bio.program.indexdb</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/program/indexdb/BioStore.html#BioStore(java.io.File,%20boolean)">BioStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;location,
                boolean&nbsp;cache)</code>
<div class="block">Creates a new <code>BioStore</code> flatfile index at the
 specified location with the specified caching behaviour.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.phred">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">PhredTools.</span><code><strong><a href="../../../../org/biojava/bio/program/phred/PhredTools.html#writePhredQuality(java.io.OutputStream,%20java.io.OutputStream,%20org.biojava.bio.seq.db.SequenceDB)">writePhredQuality</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;qual,
                                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;seq,
                                  <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block">Writes Phred quality data in a Fasta type format.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.sax.blastxml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/sax/blastxml/package-summary.html">org.biojava.bio.program.sax.blastxml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/program/sax/blastxml/package-summary.html">org.biojava.bio.program.sax.blastxml</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/program/sax/blastxml/BlastXMLParserFacade.html#BlastXMLParserFacade()">BlastXMLParserFacade</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssaha">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="strong">NIODataStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/NIODataStoreFactory.html#buildDataStore(java.io.File,%20org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.symbol.Packing,%20int,%20int)">buildDataStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                            <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
                            <a href="../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
                            int&nbsp;wordLength,
                            int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="strong">MappedDataStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/MappedDataStoreFactory.html#buildDataStore(java.io.File,%20org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.symbol.Packing,%20int,%20int)">buildDataStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                            <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
                            <a href="../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
                            int&nbsp;wordLength,
                            int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="strong">DataStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/DataStoreFactory.html#buildDataStore(java.io.File,%20org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.symbol.Packing,%20int,%20int)">buildDataStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                            <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
                            <a href="../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
                            int&nbsp;wordLength,
                            int&nbsp;threshold)</code>
<div class="block">Build a new DataStore.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="strong">CompactedDataStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/CompactedDataStoreFactory.html#buildDataStore(java.io.File,%20org.biojava.bio.seq.db.SequenceDB,%20org.biojava.bio.symbol.Packing,%20int,%20int)">buildDataStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                            <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
                            <a href="../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
                            int&nbsp;wordLength,
                            int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="strong">CompactedDataStoreFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/CompactedDataStoreFactory.html#buildDataStore(java.io.File,%20org.biojava.bio.program.ssaha.SequenceStreamer,%20org.biojava.bio.symbol.Packing,%20int,%20int,%20int)">buildDataStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                            <a href="../../../../org/biojava/bio/program/ssaha/SequenceStreamer.html" title="interface in org.biojava.bio.program.ssaha">SequenceStreamer</a>&nbsp;streamer,
                            <a href="../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
                            int&nbsp;wordLength,
                            int&nbsp;stepSize,
                            int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceStreamer.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/SequenceStreamer.html#reset()">reset</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceStreamer.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/SequenceStreamer.html#streamNext(org.biojava.bio.seq.io.SeqIOListener)">streamNext</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io">SeqIOListener</a>&nbsp;listener)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceStreamer.SequenceDBStreamer.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/SequenceStreamer.SequenceDBStreamer.html#streamNext(org.biojava.bio.seq.io.SeqIOListener)">streamNext</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io">SeqIOListener</a>&nbsp;listener)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceStreamer.FileStreamer.</span><code><strong><a href="../../../../org/biojava/bio/program/ssaha/SequenceStreamer.FileStreamer.html#streamNext(org.biojava.bio.seq.io.SeqIOListener)">streamNext</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io">SeqIOListener</a>&nbsp;listener)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssbind">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">ViewSequenceFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssbind/ViewSequenceFactory.html#makeQueryViewSequence(java.lang.String)">makeQueryViewSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;queryID)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/search/SeqSimilaritySearchResult.html" title="interface in org.biojava.bio.search">SeqSimilaritySearchResult</a></code></td>
<td class="colLast"><span class="strong">BlastLikeSearchBuilder.</span><code><strong><a href="../../../../org/biojava/bio/program/ssbind/BlastLikeSearchBuilder.html#makeSearchResult()">makeSearchResult</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">ViewSequenceFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/ssbind/ViewSequenceFactory.html#makeSubjectViewSequence(java.lang.String)">makeSubjectViewSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;subjectID)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a></code></td>
<td class="colLast"><span class="strong">AlphabetResolver.</span><code><strong><a href="../../../../org/biojava/bio/program/ssbind/AlphabetResolver.html#resolveAlphabet(java.lang.String)">resolveAlphabet</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;identifier)</code>
<div class="block"><code>resolveAlphabet</code> returns an appropriate
 <code>Alphabet</code> for an arbitrary identifier.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.unigene">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneCluster.html" title="interface in org.biojava.bio.program.unigene">UnigeneCluster</a></code></td>
<td class="colLast"><span class="strong">UnigeneDB.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html#addCluster(org.biojava.bio.program.unigene.UnigeneCluster)">addCluster</a></strong>(<a href="../../../../org/biojava/bio/program/unigene/UnigeneCluster.html" title="interface in org.biojava.bio.program.unigene">UnigeneCluster</a>&nbsp;cluster)</code>
<div class="block">Add a cluster to a database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">UnigeneTools.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneTools.html#createUnigene(java.net.URL)">createUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;dbURL)</code>
<div class="block">Create a new UnigeneDB instance referred to by a URL.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">UnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneFactory.html#createUnigene(java.net.URL)">createUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;unigeneURL)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">SQLUnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/SQLUnigeneFactory.html#createUnigene(java.net.URL)">createUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;dbURL)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">FlatFileUnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/FlatFileUnigeneFactory.html#createUnigene(java.net.URL)">createUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;unigeneLoc)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/unigene/UnigeneFactory.html" title="interface in org.biojava.bio.program.unigene">UnigeneFactory</a></code></td>
<td class="colLast"><span class="strong">UnigeneTools.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneTools.html#findFactory(java.net.URL)">findFactory</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;dbURL)</code>
<div class="block">Find the UnigeneFactory that can accept a URL.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneCluster.html" title="interface in org.biojava.bio.program.unigene">UnigeneCluster</a></code></td>
<td class="colLast"><span class="strong">UnigeneDB.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html#getCluster(java.lang.String)">getCluster</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;clusterID)</code>
<div class="block">Fetch a cluster by its cluster id.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">UnigeneTools.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneTools.html#loadUnigene(java.net.URL)">loadUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;dbURL)</code>
<div class="block">Load a UnigeneDB instance referred to by a URL.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">UnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/UnigeneFactory.html#loadUnigene(java.net.URL)">loadUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;unigeneURL)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">SQLUnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/SQLUnigeneFactory.html#loadUnigene(java.net.URL)">loadUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;dbURL)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/program/unigene/UnigeneDB.html" title="interface in org.biojava.bio.program.unigene">UnigeneDB</a></code></td>
<td class="colLast"><span class="strong">FlatFileUnigeneFactory.</span><code><strong><a href="../../../../org/biojava/bio/program/unigene/FlatFileUnigeneFactory.html#loadUnigene(java.net.URL)">loadUnigene</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/net/URL.html?is-external=true" title="class or interface in java.net">URL</a>&nbsp;unigeneLoc)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.xff">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/program/xff/package-summary.html">org.biojava.bio.program.xff</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/program/xff/package-summary.html">org.biojava.bio.program.xff</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">XFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/xff/XFFTools.html#annotateXFF(java.io.File,%20org.biojava.bio.seq.Sequence)">annotateXFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;xffFile,
                      <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">XFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/xff/XFFTools.html#annotateXFF(java.io.File,%20org.biojava.bio.seq.Sequence,%20org.biojava.bio.Annotation)">annotateXFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;xffFile,
                      <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;sequence,
                      <a href="../../../../org/biojava/bio/Annotation.html" title="interface in org.biojava.bio">Annotation</a>&nbsp;ann)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">XFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/xff/XFFTools.html#readXFF(java.io.File,%20java.lang.String)">readXFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;xffFile,
              <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqID)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">XFFTools.</span><code><strong><a href="../../../../org/biojava/bio/program/xff/XFFTools.html#readXFF(java.io.File,%20java.lang.String,%20org.biojava.bio.symbol.FiniteAlphabet)">readXFF</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;xffFile,
              <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqID,
              <a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a>&nbsp;alpha)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.proteomics">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/proteomics/package-summary.html">org.biojava.bio.proteomics</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/proteomics/package-summary.html">org.biojava.bio.proteomics</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Digest.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/Digest.html#addDigestFeatures()">addDigestFeatures</a></strong>()</code>
<div class="block">Adds peptides as features to the Sequence in this class.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#createProtease(java.lang.String,%20boolean,%20java.lang.String)">createProtease</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;cleaveRes,
                            boolean&nbsp;endoProtease,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#createProtease(java.lang.String,%20boolean,%20java.lang.String,%20java.lang.String)">createProtease</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;cleaveRes,
                            boolean&nbsp;endoProtease,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;notCleaveRes,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#createProtease(org.biojava.bio.symbol.SymbolList,%20boolean,%20java.lang.String)">createProtease</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;cleaveRes,
                            boolean&nbsp;endoProtease,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#createProtease(org.biojava.bio.symbol.SymbolList,%20boolean,%20org.biojava.bio.symbol.SymbolList,%20java.lang.String)">createProtease</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;cleaveRes,
                            boolean&nbsp;endoProtease,
                            <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;notCleaveRes,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Creates and registers a new Protease.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">IsoelectricPointCalc.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/IsoelectricPointCalc.html#getIsoelectricPoint(org.biojava.bio.symbol.SymbolList)">getIsoelectricPoint</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;peptide)</code>
<div class="block">Static public method to compute the pI for a polypeptide in
 denaturating and reduced conditions with both free ends.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">IsoelectricPointCalc.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/IsoelectricPointCalc.html#getPI(org.biojava.bio.symbol.SymbolList,%20boolean,%20boolean)">getPI</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;peptide,
          boolean&nbsp;hasFreeNTerm,
          boolean&nbsp;hasFreeCTerm)</code>
<div class="block">Computes isoelectric point of specified peptide.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#getProteaseByName(java.lang.String)">getProteaseByName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteaseName)</code>
<div class="block">Gets a Protease instance by name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a></code></td>
<td class="colLast"><span class="strong">Protease.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/Protease.html#getProteaseByName(java.lang.String)">getProteaseByName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;proteaseName)</code>
<div class="block">Retrieves a reference to the named Protease.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">ProteaseManager.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/ProteaseManager.html#registerProtease(org.biojava.bio.proteomics.Protease)">registerProtease</a></strong>(<a href="../../../../org/biojava/bio/proteomics/Protease.html" title="class in org.biojava.bio.proteomics">Protease</a>&nbsp;prot)</code>
<div class="block">Registers a protease and ensures its flyweight status</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/proteomics/package-summary.html">org.biojava.bio.proteomics</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/proteomics/Protease.html#Protease(java.lang.String,%20boolean)">Protease</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;cleavageRes,
                boolean&nbsp;endoProtease)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Creating a Protease with this constructor will not register it
 with the ProteaseManager (use ProteaseManager.createProtease())</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/proteomics/Protease.html#Protease(java.lang.String,%20boolean,%20java.lang.String)">Protease</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;cleaveRes,
                boolean&nbsp;endoProtease,
                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;notCleaveRes)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Creating a Protease with this constructor will not register it
 with the ProteaseManager (use ProteaseManager.createProtease())</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/proteomics/Protease.html#Protease(org.biojava.bio.symbol.SymbolList,%20boolean,%20org.biojava.bio.symbol.SymbolList)">Protease</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;cleaveRes,
                boolean&nbsp;endoProtease,
                <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;notCleaveRes)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Creating a Protease with this constructor will not register it
 with the ProteaseManager (use ProteaseManager.createProtease())</i></div>
</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.proteomics.aaindex">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/proteomics/aaindex/package-summary.html">org.biojava.bio.proteomics.aaindex</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/proteomics/aaindex/package-summary.html">org.biojava.bio.proteomics.aaindex</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/symbol/SymbolPropertyTable.html" title="interface in org.biojava.bio.symbol">SymbolPropertyTable</a></code></td>
<td class="colLast"><span class="strong">SymbolPropertyTableIterator.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/aaindex/SymbolPropertyTableIterator.html#nextTable()">nextTable</a></strong>()</code>
<div class="block">Returns the next <a href="../../../../org/biojava/bio/symbol/SymbolPropertyTable.html" title="interface in org.biojava.bio.symbol"><code>SymbolPropertyTable</code></a> 
 object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/symbol/SymbolPropertyTable.html" title="interface in org.biojava.bio.symbol">SymbolPropertyTable</a></code></td>
<td class="colLast"><span class="strong">AAindexStreamReader.</span><code><strong><a href="../../../../org/biojava/bio/proteomics/aaindex/AAindexStreamReader.html#nextTable()">nextTable</a></strong>()</code>
<div class="block">Returns the next <a href="../../../../org/biojava/bio/symbol/SymbolPropertyTable.html" title="interface in org.biojava.bio.symbol"><code>SymbolPropertyTable</code></a> 
 object.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/proteomics/aaindex/package-summary.html">org.biojava.bio.proteomics.aaindex</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/proteomics/aaindex/SimpleSymbolPropertyTableDB.html#SimpleSymbolPropertyTableDB(org.biojava.bio.proteomics.aaindex.SymbolPropertyTableIterator)">SimpleSymbolPropertyTableDB</a></strong>(<a href="../../../../org/biojava/bio/proteomics/aaindex/SymbolPropertyTableIterator.html" title="interface in org.biojava.bio.proteomics.aaindex">SymbolPropertyTableIterator</a>&nbsp;tableIterator)</code>
<div class="block">Initializes the database by copying all symbol property tables from 
 a given iterator into the database.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.search">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/search/SeqSimilaritySearchResult.html" title="interface in org.biojava.bio.search">SeqSimilaritySearchResult</a></code></td>
<td class="colLast"><span class="strong">SearchBuilder.</span><code><strong><a href="../../../../org/biojava/bio/search/SearchBuilder.html#makeSearchResult()">makeSearchResult</a></strong>()</code>
<div class="block">The <code>makeSearchResult</code> method returns a
 <code>SeqSimilaritySearchResult</code> instance created from
 accumulated data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/search/SeqSimilaritySearchResult.html" title="interface in org.biojava.bio.search">SeqSimilaritySearchResult</a></code></td>
<td class="colLast"><span class="strong">SeqSimilaritySearcher.</span><code><strong><a href="../../../../org/biojava/bio/search/SeqSimilaritySearcher.html#search(org.biojava.bio.symbol.SymbolList,%20org.biojava.bio.seq.db.SequenceDB,%20java.util.Map)">search</a></strong>(<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;querySeq,
            <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;searchParameters)</code>
<div class="block">Using this sequence similarity searcher, search with the given
 sequence against the given sequence database.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/search/FilteringContentHandler.html#FilteringContentHandler(org.biojava.bio.search.BlastLikeSearchFilter)">FilteringContentHandler</a></strong>(<a href="../../../../org/biojava/bio/search/BlastLikeSearchFilter.html" title="interface in org.biojava.bio.search">BlastLikeSearchFilter</a>&nbsp;filter)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/search/FilteringContentHandler.html#FilteringContentHandler(org.biojava.bio.search.BlastLikeSearchFilter,%20org.biojava.bio.search.SearchContentHandler)">FilteringContentHandler</a></strong>(<a href="../../../../org/biojava/bio/search/BlastLikeSearchFilter.html" title="interface in org.biojava.bio.search">BlastLikeSearchFilter</a>&nbsp;filter,
                                              <a href="../../../../org/biojava/bio/search/SearchContentHandler.html" title="interface in org.biojava.bio.search">SearchContentHandler</a>&nbsp;delegate)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SequenceTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/SequenceTools.html#addAllFeatures(org.biojava.bio.seq.Sequence,%20org.biojava.bio.seq.FeatureHolder)">addAllFeatures</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                            <a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;fh)</code>
<div class="block">Add features to a sequence that contain the same information as all
 those in a feature holder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleFeatureRealizer.</span><code><strong><a href="../../../../org/biojava/bio/seq/SimpleFeatureRealizer.html#addImplementation(java.lang.Class,%20java.lang.Class)">addImplementation</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>&nbsp;template,
                                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Class.html?is-external=true" title="class or interface in java.lang">Class</a>&nbsp;impl)</code>
<div class="block">Install a new mapping from a class of Feature.Template to
 a class of Feature implementations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceAnnotator.</span><code><strong><a href="../../../../org/biojava/bio/seq/SequenceAnnotator.html#annotate(org.biojava.bio.seq.Sequence)">annotate</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block">Return an annotated version of a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleAssembly.</span><code><strong><a href="../../../../org/biojava/bio/seq/SimpleAssembly.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">NewSimpleAssembly.</span><code><strong><a href="../../../../org/biojava/bio/seq/NewSimpleAssembly.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">LazyFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/LazyFeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">FeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/FeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;ft)</code>
<div class="block">Create a new Feature, and add it to this FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">CircularView.</span><code><strong><a href="../../../../org/biojava/bio/seq/CircularView.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>
<div class="block">Over rides ViewSequence to allow the use of locations that have
 coordinates outside of the sequence length (which are needed to
 describe locations that overlap the origin of a circular sequence).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">AbstractFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/AbstractFeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">RemoteFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/RemoteFeature.html#getRemoteFeature()">getRemoteFeature</a></strong>()</code>
<div class="block">Retrieve the Feature on some assembly Sequence that can represent this
 RemoteFeature properly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceIterator.</span><code><strong><a href="../../../../org/biojava/bio/seq/SequenceIterator.html#nextSequence()">nextSequence</a></strong>()</code>
<div class="block">Returns the next sequence in the iterator.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleAssembly.</span><code><strong><a href="../../../../org/biojava/bio/seq/SimpleAssembly.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;fh,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">RealizingFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/RealizingFeatureHolder.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;parent,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>
<div class="block">Realize a feature template.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">NewSimpleAssembly.</span><code><strong><a href="../../../../org/biojava/bio/seq/NewSimpleAssembly.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;fh,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleFeatureRealizer.</span><code><strong><a href="../../../../org/biojava/bio/seq/SimpleFeatureRealizer.html#realizeFeature(org.biojava.bio.seq.Sequence,%20org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                            <a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;parent,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">FeatureRealizer.</span><code><strong><a href="../../../../org/biojava/bio/seq/FeatureRealizer.html#realizeFeature(org.biojava.bio.seq.Sequence,%20org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq,
                            <a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;parent,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>
<div class="block">Install a feature on the specified sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">LazyFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/LazyFeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/FeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>
<div class="block">Remove a feature from this FeatureHolder.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/AbstractFeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">RemoteFeature.Resolver.</span><code><strong><a href="../../../../org/biojava/bio/seq/RemoteFeature.Resolver.html#resolve(org.biojava.bio.seq.RemoteFeature)">resolve</a></strong>(<a href="../../../../org/biojava/bio/seq/RemoteFeature.html" title="interface in org.biojava.bio.seq">RemoteFeature</a>&nbsp;rFeat)</code>
<div class="block">Resolve rFeat.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">FilterUtils.</span><code><strong><a href="../../../../org/biojava/bio/seq/FilterUtils.html#visitFilter(org.biojava.bio.seq.FeatureFilter,%20org.biojava.utils.walker.Visitor)">visitFilter</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;filter,
                      <a href="../../../../org/biojava/utils/walker/Visitor.html" title="interface in org.biojava.utils.walker">Visitor</a>&nbsp;visitor)</code>
<div class="block">Applies a visitor to a filter, and returns the visitor's result or null.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/IllegalIDException.html" title="class in org.biojava.bio.seq.db">IllegalIDException</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">IndexedSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/IndexedSequenceDB.html#addFile(java.io.File)">addFile</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;seqFile)</code>
<div class="block">Add sequences from a file to the sequence database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceDBLite.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html#addSequence(org.biojava.bio.seq.Sequence)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block">Adds a sequence to the database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DummySequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/DummySequenceDB.html#addSequence(org.biojava.bio.seq.Sequence)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/AbstractSequenceDB.html#addSequence(org.biojava.bio.seq.Sequence)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">TabIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/TabIndexStore.html#commit()">commit</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">IndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/IndexStore.html#commit()">commit</a></strong>()</code>
<div class="block">Commit the stored indices to permanent storage.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">EmblCDROMIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/EmblCDROMIndexStore.html#commit()">commit</a></strong>()</code>
<div class="block"><code>commit</code> commits changes.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioIndex.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/BioIndex.html#commit()">commit</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">AnnotatedSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/AnnotatedSequenceDB.html#doAnnotation(org.biojava.bio.seq.Sequence)">doAnnotation</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block">Apply the annotation to a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a></code></td>
<td class="colLast"><span class="strong">TabIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/TabIndexStore.html#fetch(java.lang.String)">fetch</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a></code></td>
<td class="colLast"><span class="strong">IndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/IndexStore.html#fetch(java.lang.String)">fetch</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Fetch an Index based upon an ID.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a></code></td>
<td class="colLast"><span class="strong">EmblCDROMIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/EmblCDROMIndexStore.html#fetch(java.lang.String)">fetch</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a></code></td>
<td class="colLast"><span class="strong">BioIndex.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/BioIndex.html#fetch(java.lang.String)">fetch</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">WebSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/WebSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Gets a sequence using its unique ID (eg for GenBank this would be the GI number)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">ViewingSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/ViewingSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SwissprotSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/SwissprotSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SubSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/SubSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceDBLite.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Retrieve a single sequence by its id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">IndexedSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/IndexedSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">GenpeptSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/GenpeptSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">DummySequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/DummySequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">CachingSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/CachingSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">AnnotatedSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/AnnotatedSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenbankSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/GenbankSequenceDB.html#getSequences(java.util.Set)">getSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list)</code>
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenbankSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/GenbankSequenceDB.html#getSequences(java.util.Set,%20org.biojava.bio.seq.db.SequenceDB)">getSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list,
                        <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;database)</code>
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceDBLite.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Remove the sequence associated with an ID from the database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HashSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/HashSequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DummySequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/DummySequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/AbstractSequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NCBISequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/NCBISequenceDB.html#setDatabase(java.lang.String)">setDatabase</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dataBase)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">NCBISequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/NCBISequenceDB.html#setSequenceFormat(org.biojava.bio.seq.io.SequenceFormat)">setSequenceFormat</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;format)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">TabIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/TabIndexStore.html#store(org.biojava.bio.seq.db.Index)">store</a></strong>(<a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a>&nbsp;indx)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">IndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/IndexStore.html#store(org.biojava.bio.seq.db.Index)">store</a></strong>(<a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a>&nbsp;indx)</code>
<div class="block">Add the Index to the store.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">EmblCDROMIndexStore.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/EmblCDROMIndexStore.html#store(org.biojava.bio.seq.db.Index)">store</a></strong>(<a href="../../../../org/biojava/bio/seq/db/Index.html" title="interface in org.biojava.bio.seq.db">Index</a>&nbsp;index)</code>
<div class="block"><code>store</code> adds an <code>Index</code> to the store.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/BioIndex.html#BioIndex(java.io.File)">BioIndex</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;indexDirectory)</code>
<div class="block">Load an existing index file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/BioIndex.html#BioIndex(java.io.File,%20java.lang.String,%20int)">BioIndex</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;indexDirectory,
                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;namespace,
                int&nbsp;idLength)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/SubSequenceDB.html#SubSequenceDB(org.biojava.bio.seq.db.SequenceDB,%20java.util.Set)">SubSequenceDB</a></strong>(<a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/TabIndexStore.html#TabIndexStore(java.io.File,%20java.io.File,%20java.lang.String,%20org.biojava.bio.seq.io.SequenceFormat,%20org.biojava.bio.seq.io.SequenceBuilderFactory,%20org.biojava.bio.seq.io.SymbolTokenization)">TabIndexStore</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;indexFile,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                          <a href="../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;format,
                          <a href="../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a>&nbsp;sbFactory,
                          <a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;symbolParser)</code>
<div class="block">Create a new TabIndexStore.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biofetch">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/db/biofetch/package-summary.html">org.biojava.bio.seq.db.biofetch</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/db/biofetch/package-summary.html">org.biojava.bio.seq.db.biofetch</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">BioFetchSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biofetch/BioFetchSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html" title="interface in org.biojava.bio.seq.db">SequenceDBLite</a></code></td>
<td class="colLast"><span class="strong">BioFetchSequenceDBProvider.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biofetch/BioFetchSequenceDBProvider.html#getSequenceDB(java.util.Map)">getSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;config)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#addName(java.sql.Connection,%20org.biojava.bio.taxa.Taxon,%20java.lang.String,%20java.lang.String)">addName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
              <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a>&nbsp;taxon,
              <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameClass,
              <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Adds a new name of the given <code>nameClass</code> to the taxon.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#addSequence(org.biojava.bio.seq.Sequence)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#createDummySequence(java.lang.String,%20org.biojava.bio.symbol.Alphabet,%20int)">createDummySequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                                      <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alphabet,
                                      int&nbsp;length)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Stack.html?is-external=true" title="class or interface in java.util">Stack</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#getChildrenOf(java.sql.Connection,%20java.lang.String)">getChildrenOf</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;scientificName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Returns all children of the specified taxon.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Stack.html?is-external=true" title="class or interface in java.util">Stack</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#getChildrenOf(java.sql.Connection,%20org.biojava.bio.taxa.Taxon)">getChildrenOf</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                          <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a>&nbsp;t)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Returns the children as a <code>Stack</code> of this given taxon.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#getSequence(int)">getSequence</a></strong>(int&nbsp;bioentry_id)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html" title="interface in org.biojava.bio.seq.db">SequenceDBLite</a></code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDBProvider.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDBProvider.html#getSequenceDB(java.util.Map)">getSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;config)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#removeName(java.sql.Connection,%20org.biojava.bio.seq.db.biosql.DBHelper,%20org.biojava.bio.taxa.Taxon,%20java.lang.String,%20java.lang.String)">removeName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                    <a href="../../../../org/biojava/bio/seq/db/biosql/DBHelper.html" title="class in org.biojava.bio.seq.db.biosql">DBHelper</a>&nbsp;helper,
                    <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a>&nbsp;taxon,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameClass,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;oldName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Deletes the specified name from of the taxon from the database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#removeTaxon(java.sql.Connection,%20org.biojava.bio.seq.db.biosql.DBHelper,%20java.lang.String)">removeTaxon</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                      <a href="../../../../org/biojava/bio/seq/db/biosql/DBHelper.html" title="class in org.biojava.bio.seq.db.biosql">DBHelper</a>&nbsp;helper,
                      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Deletes a taxon specified by one of it's names with all it's different
 names, annotations and sequences from the database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#removeTaxon(java.sql.Connection,%20int,%20org.biojava.bio.seq.db.biosql.DBHelper)">removeTaxon</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                      int&nbsp;ncbi_id,
                      <a href="../../../../org/biojava/bio/seq/db/biosql/DBHelper.html" title="class in org.biojava.bio.seq.db.biosql">DBHelper</a>&nbsp;helper)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Deletes the taxon given by it's NCBI-Taxon-ID from the database and returns the
 removed taxon.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a></code></td>
<td class="colLast"><span class="strong">TaxonSQL.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/TaxonSQL.html#setCommonName(java.sql.Connection,%20org.biojava.bio.taxa.Taxon,%20java.lang.String)">setCommonName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/sql/Connection.html?is-external=true" title="class or interface in java.sql">Connection</a>&nbsp;conn,
                          <a href="../../../../org/biojava/bio/taxa/Taxon.html" title="interface in org.biojava.bio.taxa">Taxon</a>&nbsp;taxon,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;newName)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">With this method the common name of the given taxon can be changed or created,
 if there was none before.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#BioSQLSequenceDB(javax.sql.DataSource,%20java.lang.String,%20boolean)">BioSQLSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/javax/sql/DataSource.html?is-external=true" title="class or interface in javax.sql">DataSource</a>&nbsp;ds,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;biodatabase,
                                boolean&nbsp;create)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#BioSQLSequenceDB(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20boolean)">BioSQLSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbURL,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbUser,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbPass,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;biodatabase,
                                boolean&nbsp;create)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Connect to a BioSQL database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html#BioSQLSequenceDB(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20boolean)">BioSQLSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbDriver,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbURL,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbUser,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbPass,
                                <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;biodatabase,
                                boolean&nbsp;create)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Connect to a BioSQL database.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.flat">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/db/flat/package-summary.html">org.biojava.bio.seq.db.flat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/db/flat/package-summary.html">org.biojava.bio.seq.db.flat</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">FlatSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/flat/FlatSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html" title="interface in org.biojava.bio.seq.db">SequenceDBLite</a></code></td>
<td class="colLast"><span class="strong">FlatSequenceDBProvider.</span><code><strong><a href="../../../../org/biojava/bio/seq/db/flat/FlatSequenceDBProvider.html#getSequenceDB(java.util.Map)">getSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;config)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/seq/db/flat/package-summary.html">org.biojava.bio.seq.db.flat</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/db/flat/FlatSequenceDB.html#FlatSequenceDB(java.lang.String,%20java.lang.String)">FlatSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;location,
                            <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbName)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.distributed">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#getFeatures(org.biojava.bio.seq.FeatureFilter)">getFeatures</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#getFeatures(org.biojava.bio.seq.FeatureFilter)">getFeatures</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#getFeatures(org.biojava.bio.seq.FeatureFilter)">getFeatures</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff)</code>
<div class="block">Get all features matching a FeatureFilter provided by this DistDataSource.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#getFeatures(java.lang.String,%20org.biojava.bio.seq.FeatureFilter,%20boolean)">getFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                      <a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff,
                      boolean&nbsp;recurse)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#getFeatures(java.lang.String,%20org.biojava.bio.seq.FeatureFilter,%20boolean)">getFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                      <a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff,
                      boolean&nbsp;recurse)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#getFeatures(java.lang.String,%20org.biojava.bio.seq.FeatureFilter,%20boolean)">getFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                      <a href="../../../../org/biojava/bio/seq/FeatureFilter.html" title="interface in org.biojava.bio.seq">FeatureFilter</a>&nbsp;ff,
                      boolean&nbsp;recurse)</code>
<div class="block">Get all features matching a FeatureFilter on a Sequence with an ID and recurse flats.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">DistributedSequenceDB.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistributedSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Get a Sequence object for an ID.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#hasFeatures(java.lang.String)">hasFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#hasFeatures(java.lang.String)">hasFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#hasFeatures(java.lang.String)">hasFeatures</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Find out if this DistDataSource can provide features on a sequence with a particular ID.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#hasSequence(java.lang.String)">hasSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#hasSequence(java.lang.String)">hasSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#hasSequence(java.lang.String)">hasSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Find out if this DistDataSource provides the sequence information for a sequence ID.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a></code></td>
<td class="colLast"><span class="strong">SequenceDBDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#ids(boolean)">ids</a></strong>(boolean&nbsp;topLevel)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a></code></td>
<td class="colLast"><span class="strong">GFFDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/GFFDataSource.html#ids(boolean)">ids</a></strong>(boolean&nbsp;topLevel)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a></code></td>
<td class="colLast"><span class="strong">DistDataSource.</span><code><strong><a href="../../../../org/biojava/bio/seq/distributed/DistDataSource.html#ids(boolean)">ids</a></strong>(boolean&nbsp;topLevel)</code>
<div class="block">Get the complete set of sequence IDs provided by this DistDataSource.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.homol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/homol/package-summary.html">org.biojava.bio.seq.homol</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/homol/package-summary.html">org.biojava.bio.seq.homol</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleHomology.</span><code><strong><a href="../../../../org/biojava/bio/seq/homol/SimpleHomology.html#setAlignment(org.biojava.bio.alignment.Alignment)">setAlignment</a></strong>(<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;alignment)</code>
<div class="block"><code>setAlignment</code> sets the alignment which describes
 the homology.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.impl">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ViewSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/ViewSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SubSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SubSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;templ)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleGappedSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleGappedSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;templ)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleFeature.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">RevCompSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/RevCompSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;ft)</code>
<div class="block">createFeature() will call createFeature() on the underlying Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">LazyFilterFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/LazyFilterFeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">DummySequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/DummySequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleSequence.html#createFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;fh,
                          <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Please use new 1-arg createFeature instead.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleRemoteFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleRemoteFeature.html#getRemoteFeature()">getRemoteFeature</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a></code></td>
<td class="colLast"><span class="strong">TemplateUtils.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/TemplateUtils.html#instantiateTemplate(org.biojava.bio.seq.Feature)">instantiateTemplate</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feat)</code>
<div class="block">This attempts to divine the 'best' template class for a feature and return
 a new instance readly for pupulating.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a></code></td>
<td class="colLast"><span class="strong">TemplateUtils.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/TemplateUtils.html#makeTemplate(org.biojava.bio.seq.Feature)">makeTemplate</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feat)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">TemplateUtils.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/TemplateUtils.html#populate(org.biojava.bio.seq.Feature.Template,%20org.biojava.bio.seq.Feature)">populate</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;templ,
                <a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feat)</code>
<div class="block">This attempts to populate the fields of a template using
 the publically accessible information in a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ViewSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/ViewSequence.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;parent,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleSequence.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;parent,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;template)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleFeature.html#realizeFeature(org.biojava.bio.seq.FeatureHolder,%20org.biojava.bio.seq.Feature.Template)">realizeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>&nbsp;fh,
                            <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;templ)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SubSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SubSequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleSequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>
<div class="block">Remove a feature attached to this sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleGappedSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleGappedSequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RevCompSequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/RevCompSequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">LazyFilterFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/LazyFilterFeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">DummySequence.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/DummySequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;feature)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleRemoteFeature.DBResolver.</span><code><strong><a href="../../../../org/biojava/bio/seq/impl/SimpleRemoteFeature.DBResolver.html#resolve(org.biojava.bio.seq.RemoteFeature)">resolve</a></strong>(<a href="../../../../org/biojava/bio/seq/RemoteFeature.html" title="interface in org.biojava.bio.seq">RemoteFeature</a>&nbsp;rFeat)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/io/ParseException.html" title="class in org.biojava.bio.seq.io">ParseException</a></strong></code>
<div class="block">ParseException should be thrown to indicate that there was a problem with
 parsing sequence information.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/ComponentFeature.html" title="interface in org.biojava.bio.seq">ComponentFeature</a></code></td>
<td class="colLast"><span class="strong">SimpleAssemblyBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SimpleAssemblyBuilder.html#addComponentSequence(org.biojava.bio.seq.ComponentFeature.Template)">addComponentSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/ComponentFeature.Template.html" title="class in org.biojava.bio.seq">ComponentFeature.Template</a>&nbsp;cft)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#biojavaToFile(int,%20java.io.OutputStream,%20java.lang.Object)">biojavaToFile</a></strong>(int&nbsp;fileType,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Converts a Biojava object to the given filetype.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#biojavaToFile(java.lang.String,%20java.lang.String,%20java.io.OutputStream,%20java.lang.Object)">biojavaToFile</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;biojava)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a Biojava <code>SequenceIterator</code>,
 <code>SequenceDB</code>, <code>Sequence</code> or <code>Aligment</code>
 to an <code>OutputStream</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FeatureTableParser.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FeatureTableParser.html#endFeature()">endFeature</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FeatureTableParser.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FeatureTableParser.html#featureData(java.lang.String)">featureData</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;line)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#fileToBiojava(int,%20java.io.BufferedReader)">fileToBiojava</a></strong>(int&nbsp;fileType,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Reads a file and returns the corresponding Biojava object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#fileToBiojava(java.lang.String,%20java.lang.String,%20java.io.BufferedReader)">fileToBiojava</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;formatName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphabetName,
                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;br)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Reads a file with the specified format and alphabet</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#formatToFactory(org.biojava.bio.seq.io.SequenceFormat,%20org.biojava.bio.symbol.Alphabet)">formatToFactory</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a>&nbsp;format,
                              <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>as this essentially duplicates the operation
 available in the method <code>identifyBuilderFactory</code>.</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/FiniteAlphabet.html" title="interface in org.biojava.bio.symbol">FiniteAlphabet</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#getAlphabet(int)">getAlphabet</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getAlphabet</code> accepts a value which represents a
 sequence format and returns the relevant
 <code>FiniteAlphabet</code> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/io/SequenceBuilderFactory.html" title="interface in org.biojava.bio.seq.io">SequenceBuilderFactory</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#getBuilderFactory(int)">getBuilderFactory</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getBuilderFactory</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceBuilderFactory</code> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/io/SequenceFormat.html" title="interface in org.biojava.bio.seq.io">SequenceFormat</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#getSequenceFormat(int)">getSequenceFormat</a></strong>(int&nbsp;identifier)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>getSequenceFormat</code> accepts a value which represents
 a sequence format and returns the relevant
 <code>SequenceFormat</code> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a></code></td>
<td class="colLast"><span class="strong">ChunkedSymbolListFactory.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/ChunkedSymbolListFactory.html#make(org.biojava.bio.seq.io.SymbolReader)">make</a></strong>(<a href="../../../../org/biojava/bio/seq/io/SymbolReader.html" title="interface in org.biojava.bio.seq.io">SymbolReader</a>&nbsp;sr)</code>
<div class="block">Method to create a Sequence with a SymbolReader.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceDBSequenceBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SequenceDBSequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>
<div class="block">create the sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SmartSequenceBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SmartSequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SimpleSequenceBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SimpleSequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SimpleAssemblyBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SimpleAssemblyBuilder.html#makeSequence()">makeSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceBuilderFilter.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SequenceBuilderFilter.html#makeSequence()">makeSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceBuilderBase.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SequenceBuilderBase.html#makeSequence()">makeSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SequenceBuilder.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>
<div class="block">Return the Sequence object which has been constructed
 by this builder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">StreamReader.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/StreamReader.html#nextSequence()">nextSequence</a></strong>()</code>
<div class="block">Pull the next sequence out of the stream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a></code></td>
<td class="colLast"><span class="strong">EmblLikeLocationParser.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/EmblLikeLocationParser.html#parseLocation(java.lang.String,%20org.biojava.bio.seq.Feature.Template)">parseLocation</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;location,
                          <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;theTemplate)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block"><code>parseLocation</code> creates a <code>Location</code> from
 the String and returns a stranded location.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFasta(java.io.InputStream,%20org.biojava.bio.symbol.Alphabet)">readFasta</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;seqFile,
                  <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Create a sequence database from a fasta file provided as an
 input stream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">SequenceFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SequenceFormat.html#readSequence(java.io.BufferedReader,%20org.biojava.bio.seq.io.SymbolTokenization,%20org.biojava.bio.seq.io.SeqIOListener)">readSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;reader,
                        <a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;symParser,
                        <a href="../../../../org/biojava/bio/seq/io/SeqIOListener.html" title="interface in org.biojava.bio.seq.io">SeqIOListener</a>&nbsp;listener)</code>
<div class="block">Read a sequence and pass data on to a SeqIOListener.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FeatureTableParser.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FeatureTableParser.html#startFeature(java.lang.String)">startFeature</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MSFAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/MSFAlignmentFormat.html#write(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment,%20int)">write</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
          <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align,
          int&nbsp;fileType)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FastaAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FastaAlignmentFormat.html#write(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment,%20int)">write</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
          <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align,
          int&nbsp;fileType)</code>
<div class="block">Writes out the alignment to an FASTA file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MSFAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/MSFAlignmentFormat.html#writeDna(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment)">writeDna</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FastaAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FastaAlignmentFormat.html#writeDna(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment)">writeDna</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenpept(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeGenpept</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in Genpept format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeGenpept(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeGenpept</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in Genpept
 format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MSFAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/MSFAlignmentFormat.html#writeProtein(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment)">writeProtein</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FastaAlignmentFormat.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/FastaAlignmentFormat.html#writeProtein(java.io.OutputStream,%20org.biojava.bio.alignment.Alignment)">writeProtein</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                        <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeSwissprot(java.io.OutputStream,%20org.biojava.bio.seq.Sequence)">writeSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                            <a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a single Sequence to an OutputStream in SwissProt format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SeqIOTools.</span><code><strong><a href="../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeSwissprot(java.io.OutputStream,%20org.biojava.bio.seq.SequenceIterator)">writeSwissprot</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
                            <a href="../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;in)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
<div class="block">Writes a stream of Sequences to an OutputStream in SwissProt
 format.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/seq/io/CrossProductTokenization.html#CrossProductTokenization(org.biojava.bio.symbol.Alphabet)">CrossProductTokenization</a></strong>(<a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.projection">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/seq/projection/package-summary.html">org.biojava.bio.seq.projection</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/seq/projection/package-summary.html">org.biojava.bio.seq.projection</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ReparentContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ReparentContext.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;projTempl)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ProjectionContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectionContext.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;projTempl)</code>
<div class="block">Create a projected feature with properties matching the template.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ProjectedFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectedFeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;templ)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ProjectedFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectedFeature.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;temp)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ReparentContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ReparentContext.html#createFeature(org.biojava.bio.seq.Feature,%20org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f,
                          <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;projTempl)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ProjectionContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectionContext.html#createFeature(org.biojava.bio.seq.Feature,%20org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;projParent,
                          <a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;projTempl)</code>
<div class="block">Create a new projected feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ReparentContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ReparentContext.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;dyingChild)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ProjectionContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectionContext.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;dyingChild)</code>
<div class="block">Remove the dying child.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ProjectedFeatureHolder.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectedFeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;dyingChild)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ProjectedFeature.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectedFeature.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ReparentContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ReparentContext.html#removeFeature(org.biojava.bio.seq.Feature,%20org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f,
                          <a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ProjectionContext.</span><code><strong><a href="../../../../org/biojava/bio/seq/projection/ProjectionContext.html#removeFeature(org.biojava.bio.seq.Feature,%20org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;projParent,
                          <a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;dyingChild)</code>
<div class="block">Remove the dying child.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.symbol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/symbol/package-summary.html">org.biojava.bio.symbol</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/symbol/package-summary.html">org.biojava.bio.symbol</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/symbol/IllegalAlphabetException.html" title="class in org.biojava.bio.symbol">IllegalAlphabetException</a></strong></code>
<div class="block">
 The exception to indicate that an invalid alphabet has been used.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></strong></code>
<div class="block">The exception to indicate that a symbol is not valid within a context.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/bio/symbol/package-summary.html">org.biojava.bio.symbol</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/AlphabetIndex.html" title="interface in org.biojava.bio.symbol">AlphabetIndex</a></code></td>
<td class="colLast"><span class="strong">AlphabetManager.</span><code><strong><a href="../../../../org/biojava/bio/symbol/AlphabetManager.html#getAlphabetIndex(org.biojava.bio.symbol.Symbol[])">getAlphabetIndex</a></strong>(<a href="../../../../org/biojava/bio/symbol/Symbol.html" title="interface in org.biojava.bio.symbol">Symbol</a>[]&nbsp;syms)</code>
<div class="block">Get an indexer for an array of symbols.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a></code></td>
<td class="colLast"><span class="strong">SoftMaskedAlphabet.</span><code><strong><a href="../../../../org/biojava/bio/symbol/SoftMaskedAlphabet.html#getTokenization(java.lang.String)">getTokenization</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a></code></td>
<td class="colLast"><span class="strong">DoubleAlphabet.SubDoubleAlphabet.</span><code><strong><a href="../../../../org/biojava/bio/symbol/DoubleAlphabet.SubDoubleAlphabet.html#getTokenization(java.lang.String)">getTokenization</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a></code></td>
<td class="colLast"><span class="strong">Alphabet.</span><code><strong><a href="../../../../org/biojava/bio/symbol/Alphabet.html#getTokenization(java.lang.String)">getTokenization</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">
 Get a SymbolTokenization by name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a></code></td>
<td class="colLast"><span class="strong">AbstractAlphabet.</span><code><strong><a href="../../../../org/biojava/bio/symbol/AbstractAlphabet.html#getTokenization(java.lang.String)">getTokenization</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlphabetManager.</span><code><strong><a href="../../../../org/biojava/bio/symbol/AlphabetManager.html#loadAlphabets(org.xml.sax.InputSource)">loadAlphabets</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/org/xml/sax/InputSource.html?is-external=true" title="class or interface in org.xml.sax">InputSource</a>&nbsp;is)</code>
<div class="block">Load additional Alphabets, defined in XML format, into the AlphabetManager's registry.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/MergeLocation.html" title="class in org.biojava.bio.symbol">MergeLocation</a></code></td>
<td class="colLast"><span class="strong">MergeLocation.</span><code><strong><a href="../../../../org/biojava/bio/symbol/MergeLocation.html#mergeLocations(java.util.List)">mergeLocations</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;componentLocations)</code>
<div class="block">Static Factory method for getting an instance of a <code>MergeLocation</code></div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/MergeLocation.html" title="class in org.biojava.bio.symbol">MergeLocation</a></code></td>
<td class="colLast"><span class="strong">MergeLocation.</span><code><strong><a href="../../../../org/biojava/bio/symbol/MergeLocation.html#mergeLocations(org.biojava.bio.symbol.Location,%20org.biojava.bio.symbol.Location)">mergeLocations</a></strong>(<a href="../../../../org/biojava/bio/symbol/Location.html" title="interface in org.biojava.bio.symbol">Location</a>&nbsp;locA,
                            <a href="../../../../org/biojava/bio/symbol/Location.html" title="interface in org.biojava.bio.symbol">Location</a>&nbsp;locB)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CodonPrefFilter.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefFilter.html#put(org.biojava.bio.symbol.CodonPref)">put</a></strong>(<a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a>&nbsp;codonPref)</code>
<div class="block">handles storage of a CodonPref object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CodonPrefFilter.ByName.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefFilter.ByName.html#put(org.biojava.bio.symbol.CodonPref)">put</a></strong>(<a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a>&nbsp;codonPref)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CodonPrefFilter.EverythingToXML.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefFilter.EverythingToXML.html#put(org.biojava.bio.symbol.CodonPref)">put</a></strong>(<a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a>&nbsp;codonPref)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a>[]</code></td>
<td class="colLast"><span class="strong">CodonPrefTools.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefTools.html#readFromXML(java.io.InputStream)">readFromXML</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;prefStream)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">CodonPrefTools.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefTools.html#readFromXML(java.io.InputStream,%20org.biojava.bio.symbol.CodonPrefFilter)">readFromXML</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;prefStream,
                      <a href="../../../../org/biojava/bio/symbol/CodonPrefFilter.html" title="interface in org.biojava.bio.symbol">CodonPrefFilter</a>&nbsp;filter)</code>
<div class="block">read an CodonPref XML stream and handle it with a CodonPrefFilter object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a></code></td>
<td class="colLast"><span class="strong">CodonPrefTools.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefTools.html#readFromXML(java.io.InputStream,%20java.lang.String)">readFromXML</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;prefStream,
                      <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">reads a specified CodonPref from an file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">CodonPrefTools.</span><code><strong><a href="../../../../org/biojava/bio/symbol/CodonPrefTools.html#writeToXML(org.biojava.bio.symbol.CodonPref,%20java.io.PrintWriter)">writeToXML</a></strong>(<a href="../../../../org/biojava/bio/symbol/CodonPref.html" title="interface in org.biojava.bio.symbol">CodonPref</a>&nbsp;codonPref,
                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/PrintWriter.html?is-external=true" title="class or interface in java.io">PrintWriter</a>&nbsp;writer)</code>
<div class="block">write out a specified CodonPref object in XML format.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.taxa">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/taxa/package-summary.html">org.biojava.bio.taxa</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/bio/taxa/package-summary.html">org.biojava.bio.taxa</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/taxa/CircularReferenceException.html" title="class in org.biojava.bio.taxa">CircularReferenceException</a></strong></code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>replaced by classes in <a href="../../../../org/biojavax/bio/taxa/package-summary.html"><code>org.biojavax.bio.taxa</code></a></i></div>
</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.directory">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/directory/package-summary.html">org.biojava.directory</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/directory/package-summary.html">org.biojava.directory</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html" title="interface in org.biojava.bio.seq.db">SequenceDBLite</a></code></td>
<td class="colLast"><span class="strong">Registry.</span><code><strong><a href="../../../../org/biojava/directory/Registry.html#getDatabase(java.lang.String)">getDatabase</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbName)</code>
<div class="block"><code>getDatabase</code> retrieves a database instance known by
 a name <code>String</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/db/SequenceDBLite.html" title="interface in org.biojava.bio.seq.db">SequenceDBLite</a></code></td>
<td class="colLast"><span class="strong">SequenceDBProvider.</span><code><strong><a href="../../../../org/biojava/directory/SequenceDBProvider.html#getSequenceDB(java.util.Map)">getSequenceDB</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;config)</code>
<div class="block">Get a sequence database.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.utils.automata">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/automata/package-summary.html">org.biojava.utils.automata</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/automata/package-summary.html">org.biojava.utils.automata</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/utils/automata/AutomatonException.html" title="class in org.biojava.utils.automata">AutomatonException</a></strong></code>
<div class="block">An exception thrown by classes of this package.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.utils.math">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/math/package-summary.html">org.biojava.utils.math</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/utils/math/package-summary.html">org.biojava.utils.math</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">BinarySearch.</span><code><strong><a href="../../../../org/biojava/utils/math/BinarySearch.html#solve(double,%20double,%20double,%20org.biojava.utils.math.ComputeObject)">solve</a></strong>(double&nbsp;min,
          double&nbsp;max,
          double&nbsp;tolerance,
          <a href="../../../../org/biojava/utils/math/ComputeObject.html" title="interface in org.biojava.utils.math">ComputeObject</a>&nbsp;obj)</code>
<div class="block">method that will attempt solving the equation.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.utils.regex">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/regex/package-summary.html">org.biojava.utils.regex</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/regex/package-summary.html">org.biojava.utils.regex</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/utils/regex/RegexException.html" title="class in org.biojava.utils.regex">RegexException</a></strong></code>
<div class="block">An exception thrown by classes of this package.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.utils.walker">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojava/utils/walker/package-summary.html">org.biojava.utils.walker</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojava/utils/walker/package-summary.html">org.biojava.utils.walker</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/utils/walker/Walker.html" title="interface in org.biojava.utils.walker">Walker</a></code></td>
<td class="colLast"><span class="strong">WalkerFactory.</span><code><strong><a href="../../../../org/biojava/utils/walker/WalkerFactory.html#getWalker(org.biojava.utils.walker.Visitor)">getWalker</a></strong>(<a href="../../../../org/biojava/utils/walker/Visitor.html" title="interface in org.biojava.utils.walker">Visitor</a>&nbsp;visitor)</code>
<div class="block">Get a Walker that is customosed to a particular visitor.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/package-summary.html">org.biojavax.bio</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/package-summary.html">org.biojavax.bio</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">BioEntryIterator.</span><code><strong><a href="../../../../org/biojavax/bio/BioEntryIterator.html#nextBioEntry()">nextBioEntry</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/alignment/package-summary.html">org.biojavax.bio.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/alignment/package-summary.html">org.biojavax.bio.alignment</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html#getAlignmentOption(java.lang.String)">getAlignmentOption</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key)</code>
<div class="block">Method that returns the value associated with the key given in parameter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentService.html#getAlignmentResults(java.lang.String,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentOutputProperties)">getAlignmentResults</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;rid,
                                      <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentOutputProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentOutputProperties</a>&nbsp;out)</code>
<div class="block">Getting the actual alignment results from this instantiated service for a given ID with specific
 formatting parameters held in a RemotePairwiseAlignmentOutputProperties-implemented object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentOutputProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentOutputProperties.html#getOutputOption(java.lang.String)">getOutputOption</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key)</code>
<div class="block">Method that returns the value associated with the key given in parameter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentService.html#isReady(java.lang.String,%20long)">isReady</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
              long&nbsp;present)</code>
<div class="block">Simple method to check if the specified request has been completed by the service used.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentService.html#sendAlignmentRequest(org.biojavax.bio.seq.RichSequence,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;rs,
                                        <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">Doing the actual analysis on the instantiated service using specified parameters and the RichSequence object</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentService.html#sendAlignmentRequest(java.lang.String,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;str,
                                        <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">Doing the actual analysis on the instantiated service using specified parameters on the string representation
 of the Sequence object</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.alignment.blast">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/alignment/blast/package-summary.html">org.biojavax.bio.alignment.blast</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/alignment/blast/package-summary.html">org.biojavax.bio.alignment.blast</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastAlignmentProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastAlignmentProperties.html#getAlignmentOption(java.lang.String)">getAlignmentOption</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#getAlignmentResults(java.lang.String,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentOutputProperties)">getAlignmentResults</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                                      <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentOutputProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentOutputProperties</a>&nbsp;rb)</code>
<div class="block">
 This method extracts the actual Blast report for this request ID.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastOutputProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastOutputProperties.html#getOutputOption(java.lang.String)">getOutputOption</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;o)</code>
<div class="block">Method that returns any value associated to any key for this RemoteQBlastOutputProperties object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#isReady(java.lang.String,%20long)">isReady</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
              long&nbsp;present)</code>
<div class="block">
 This method is used only for the executeBlastSearch method to check for
 completion of request using the NCBI specified RTOE variable
 </div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#printRemoteBlastInfo()">printRemoteBlastInfo</a></strong>()</code>
<div class="block">A simple method to check the availability of the QBlast service</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#sendAlignmentRequest(int,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(int&nbsp;gid,
                                        <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">This method is a wrapper that executes the Blast request via the Put
 command of the CGI-BIN interface with the specified parameters and a
 GenBank GID.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#sendAlignmentRequest(org.biojavax.bio.seq.RichSequence,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;rs,
                                        <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">This method is a wrapper that executes the Blast request via the Put
 command of the CGI-BIN interface with the specified parameters and a
 RichSequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemoteQBlastService.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#sendAlignmentRequest(java.lang.String,%20org.biojavax.bio.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;str,
                                        <a href="../../../../org/biojavax/bio/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojavax.bio.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">This method is a wrapper that executes the Blast request via the Put
 command of the CGI-BIN interface with the specified parameters and a
 string representing the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RemoteQBlastOutputProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastOutputProperties.html#setAlignmentOutputFormat(org.biojavax.bio.alignment.blast.RemoteQBlastOutputFormat)">setAlignmentOutputFormat</a></strong>(<a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastOutputFormat.html" title="enum in org.biojavax.bio.alignment.blast">RemoteQBlastOutputFormat</a>&nbsp;rf)</code>
<div class="block">This method is use to set the alignment output format to get from the QBlast service</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RemoteQBlastAlignmentProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastAlignmentProperties.html#setBlastProgram(java.lang.String)">setBlastProgram</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;program)</code>
<div class="block">This method set the program to be use with blastall.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RemoteQBlastOutputProperties.</span><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastOutputProperties.html#setOutputFormat(org.biojavax.bio.alignment.blast.RemoteQBlastOutputFormat)">setOutputFormat</a></strong>(<a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastOutputFormat.html" title="enum in org.biojavax.bio.alignment.blast">RemoteQBlastOutputFormat</a>&nbsp;rf)</code>
<div class="block">This method is use to set the stream output format to get from the QBlast service</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojavax/bio/alignment/blast/package-summary.html">org.biojavax.bio.alignment.blast</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojavax/bio/alignment/blast/RemoteQBlastService.html#RemoteQBlastService()">RemoteQBlastService</a></strong>()</code>
<div class="block">The constructor for a QBlast service request.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#addBioEntry(org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>
<div class="block">Add a BioEntry, the name of the BioEntry will be used as the ID</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioEntryDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/BioEntryDBLite.html#addBioEntry(org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>
<div class="block">Adds a sequence to the database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#addBioEntry(org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractBioEntryDB.html#addBioEntry(org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#addBioEntry(java.lang.String,%20org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                      <a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RichSequenceDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/RichSequenceDBLite.html#addRichSequence(org.biojavax.bio.seq.RichSequence)">addRichSequence</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;seq)</code>
<div class="block">Adds a sequence to the database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#addRichSequence(org.biojavax.bio.seq.RichSequence)">addRichSequence</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;seq)</code>
<div class="block">Add a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#addRichSequence(org.biojavax.bio.seq.RichSequence)">addRichSequence</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#addRichSequence(java.lang.String,%20org.biojavax.bio.seq.RichSequence)">addRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                              <a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#addSequence(org.biojava.bio.seq.Sequence)">addSequence</a></strong>(<a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#getBioEntry(java.lang.String)">getBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">BioEntryDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/BioEntryDBLite.html#getBioEntry(java.lang.String)">getBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Retrieve a single BioEntry by its id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#getBioEntry(java.lang.String)">getBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#getBioEntrys(java.util.Set)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">BioEntryDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/BioEntryDBLite.html#getBioEntrys(java.util.Set)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>
<div class="block">Retrieve multiple BioEntry by their ids.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#getBioEntrys(java.util.Set)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#getBioEntrys(java.util.Set,%20org.biojavax.bio.db.BioEntryDB)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                        <a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a>&nbsp;db)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">BioEntryDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/BioEntryDBLite.html#getBioEntrys(java.util.Set,%20org.biojavax.bio.db.BioEntryDB)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                        <a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a>&nbsp;db)</code>
<div class="block">Retrieve multiple BioEntry into a specific sequence database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#getBioEntrys(java.util.Set,%20org.biojavax.bio.db.BioEntryDB)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                        <a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a>&nbsp;db)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequenceDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/RichSequenceDBLite.html#getRichSequence(java.lang.String)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Retrieve a single RichSequence by its id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#getRichSequence(java.lang.String)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">RichSequenceDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/RichSequenceDBLite.html#getRichSequences(java.util.Set)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>
<div class="block">Retrieve multiple RichSequence by its id.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#getRichSequences(java.util.Set)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">RichSequenceDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/RichSequenceDBLite.html#getRichSequences(java.util.Set,%20org.biojavax.bio.db.RichSequenceDB)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                                <a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a>&nbsp;db)</code>
<div class="block">Retrieve multiple RichSequence into a specific sequence database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#getRichSequences(java.util.Set,%20org.biojavax.bio.db.RichSequenceDB)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                                <a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a>&nbsp;db)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#getSequence(java.lang.String)">getSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HashBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashBioEntryDB.html#removeBioEntry(java.lang.String)">removeBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioEntryDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/BioEntryDBLite.html#removeBioEntry(java.lang.String)">removeBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Remove the BioEntry associated with an ID from the database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#removeBioEntry(java.lang.String)">removeBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractBioEntryDB.html#removeBioEntry(java.lang.String)">removeBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RichSequenceDBLite.</span><code><strong><a href="../../../../org/biojavax/bio/db/RichSequenceDBLite.html#removeRichSequence(java.lang.String)">removeRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Remove the RichSequence associated with an ID from the database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#removeRichSequence(java.lang.String)">removeRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HashRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/HashRichSequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html#removeSequence(java.lang.String)">removeSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#_addBioEntry(org.biojavax.bio.BioEntry)">_addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#addBioEntry(org.biojavax.bio.BioEntry)">addBioEntry</a></strong>(<a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#addRichSequence(org.biojavax.bio.seq.RichSequence)">addRichSequence</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#fullyLoadRichSequence(org.biojavax.bio.seq.RichSequence)">fullyLoadRichSequence</a></strong>(<a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#getBioEntry(java.lang.String)">getBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#getBioEntrys(java.util.Set)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a></code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#getBioEntrys(java.util.Set,%20org.biojavax.bio.db.BioEntryDB)">getBioEntrys</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                        <a href="../../../../org/biojavax/bio/db/BioEntryDB.html" title="interface in org.biojavax.bio.db">BioEntryDB</a>&nbsp;db)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#getRichSequence(java.lang.String)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#getRichSequences(java.util.Set)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#getRichSequences(java.util.Set,%20org.biojavax.bio.db.RichSequenceDB)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids,
                                <a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a>&nbsp;db)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLBioEntryDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLBioEntryDB.html#removeBioEntry(java.lang.String)">removeBioEntry</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BioSQLRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html#removeRichSequence(java.lang.String)">removeRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.ncbi">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">GenpeptRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html#getRichSequence(java.lang.String)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Given the appropriate Genbank ID, return the matching RichSequence object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">GenbankRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenbankRichSequenceDB.html#getRichSequence(java.lang.String)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>
<div class="block">Given the appropriate Genbank ID, return the matching RichSequence object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">GenpeptRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html#getRichSequence(java.lang.String,%20org.biojavax.Namespace)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                              <a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;nsp)</code>
<div class="block">Given the appropriate Genbank ID, return the matching RichSequence object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">GenbankRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenbankRichSequenceDB.html#getRichSequence(java.lang.String,%20org.biojavax.Namespace)">getRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id,
                              <a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;nsp)</code>
<div class="block">Given the appropriate Genbank ID, return the matching RichSequence object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenpeptRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html#getRichSequences(java.util.Set)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list)</code>
<div class="block">Retrieve rich sequences from a Genbank</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenbankRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenbankRichSequenceDB.html#getRichSequences(java.util.Set)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list)</code>
<div class="block">Retrieve rich sequences from a Genbank</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenpeptRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html#getRichSequences(java.util.Set,%20org.biojavax.bio.db.RichSequenceDB)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list,
                                <a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a>&nbsp;database)</code>
<div class="block">Retrieve rich sequences from a Genbank</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></code></td>
<td class="colLast"><span class="strong">GenbankRichSequenceDB.</span><code><strong><a href="../../../../org/biojavax/bio/db/ncbi/GenbankRichSequenceDB.html#getRichSequences(java.util.Set,%20org.biojavax.bio.db.RichSequenceDB)">getRichSequences</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list,
                                <a href="../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a>&nbsp;database)</code>
<div class="block">Retrieve rich sequences from a Genbank</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.seq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/seq/package-summary.html">org.biojavax.bio.seq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/seq/package-summary.html">org.biojavax.bio.seq</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">ThinRichSequence.</span><code><strong><a href="../../../../org/biojavax/bio/seq/ThinRichSequence.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;ft)</code>
<div class="block">Create a new Feature, and add it to this FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></code></td>
<td class="colLast"><span class="strong">SimpleRichFeature.</span><code><strong><a href="../../../../org/biojavax/bio/seq/SimpleRichFeature.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.Template.html" title="class in org.biojava.bio.seq">Feature.Template</a>&nbsp;ft)</code>
<div class="block">Create a new Feature, and add it to this FeatureHolder.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequence.Tools.</span><code><strong><a href="../../../../org/biojavax/bio/seq/RichSequence.Tools.html#createRichSequence(org.biojavax.Namespace,%20java.lang.String,%20java.lang.String,%20org.biojava.bio.symbol.Alphabet)">createRichSequence</a></strong>(<a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;ns,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                                    <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block">Create a new RichSequence in the specified namespace.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequence.Tools.</span><code><strong><a href="../../../../org/biojavax/bio/seq/RichSequence.Tools.html#createRichSequence(java.lang.String,%20java.lang.String,%20org.biojava.bio.symbol.Alphabet)">createRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                                    <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block">Create a new RichSequence in the default namespace.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequence.Tools.</span><code><strong><a href="../../../../org/biojavax/bio/seq/RichSequence.Tools.html#createRichSequence(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.bio.symbol.Alphabet)">createRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;namespace,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
                                    <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                                    <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block">Create a new RichSequence in the specified namespace.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequenceIterator.</span><code><strong><a href="../../../../org/biojavax/bio/seq/RichSequenceIterator.html#nextRichSequence()">nextRichSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojavax/bio/seq/RichSequenceIterator.html" title="interface in org.biojavax.bio.seq">RichSequenceIterator</a></code></td>
<td class="colLast"><span class="strong">RichSequence.IOTools.</span><code><strong><a href="../../../../org/biojavax/bio/seq/RichSequence.IOTools.html#readHashedFastaDNA(java.io.BufferedInputStream,%20org.biojavax.Namespace)">readHashedFastaDNA</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedInputStream.html?is-external=true" title="class or interface in java.io">BufferedInputStream</a>&nbsp;is,
                                    <a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;ns)</code>
<div class="block">Iterate over the sequences in an FASTA-format stream of DNA
 sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ThinRichSequence.</span><code><strong><a href="../../../../org/biojavax/bio/seq/ThinRichSequence.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>
<div class="block">Remove a feature from this FeatureHolder.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SimpleRichFeature.</span><code><strong><a href="../../../../org/biojavax/bio/seq/SimpleRichFeature.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></strong>(<a href="../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>&nbsp;f)</code>
<div class="block">Remove a feature from this FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">ThinRichSequence.</span><code><strong><a href="../../../../org/biojavax/bio/seq/ThinRichSequence.html#setAlphabetName(java.lang.String)">setAlphabetName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphaname)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SimpleRichSequence.</span><code><strong><a href="../../../../org/biojavax/bio/seq/SimpleRichSequence.html#setAlphabetName(java.lang.String)">setAlphabetName</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alphaname)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SimpleRichSequence.</span><code><strong><a href="../../../../org/biojavax/bio/seq/SimpleRichSequence.html#setStringSequence(java.lang.String)">setStringSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seq)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.seq.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a> in <a href="../../../../org/biojavax/bio/seq/io/package-summary.html">org.biojavax.bio.seq.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../org/biojavax/bio/seq/io/package-summary.html">org.biojavax.bio.seq.io</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">SimpleRichSequenceBuilder.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/SimpleRichSequenceBuilder.html#makeRichSequence()">makeRichSequence</a></strong>()</code>
<div class="block">Build a RichSequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichSequenceBuilder.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichSequenceBuilder.html#makeRichSequence()">makeRichSequence</a></strong>()</code>
<div class="block">Build a RichSequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">SimpleRichSequenceBuilder.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/SimpleRichSequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>
<div class="block">Return the Sequence object which has been constructed
 by this builder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">RichSequenceBuilder.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichSequenceBuilder.html#makeSequence()">makeSequence</a></strong>()</code>
<div class="block">Return the Sequence object which has been constructed
 by this builder.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">RichStreamReader.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichStreamReader.html#nextBioEntry()">nextBioEntry</a></strong>()</code></td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/BioEntry.html" title="interface in org.biojavax.bio">BioEntry</a></code></td>
<td class="colLast"><span class="strong">HashedFastaIterator.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/HashedFastaIterator.html#nextBioEntry()">nextBioEntry</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">RichStreamReader.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichStreamReader.html#nextRichSequence()">nextRichSequence</a></strong>()</code></td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojavax/bio/seq/RichSequence.html" title="interface in org.biojavax.bio.seq">RichSequence</a></code></td>
<td class="colLast"><span class="strong">HashedFastaIterator.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/HashedFastaIterator.html#nextRichSequence()">nextRichSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">RichStreamReader.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichStreamReader.html#nextSequence()">nextSequence</a></strong>()</code>
<div class="block">Returns the next sequence in the iterator.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a></code></td>
<td class="colLast"><span class="strong">HashedFastaIterator.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/HashedFastaIterator.html#nextSequence()">nextSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">RichSequenceFormat.</span><code><strong><a href="../../../../org/biojavax/bio/seq/io/RichSequenceFormat.html#readRichSequence(java.io.BufferedReader,%20org.biojava.bio.seq.io.SymbolTokenization,%20org.biojavax.bio.seq.io.RichSeqIOListener,%20org.biojavax.Namespace)">readRichSequence</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedReader.html?is-external=true" title="class or interface in java.io">BufferedReader</a>&nbsp;reader,
                                <a href="../../../../org/biojava/bio/seq/io/SymbolTokenization.html" title="interface in org.biojava.bio.seq.io">SymbolTokenization</a>&nbsp;symParser,
                                <a href="../../../../org/biojavax/bio/seq/io/RichSeqIOListener.html" title="interface in org.biojavax.bio.seq.io">RichSeqIOListener</a>&nbsp;listener,
                                <a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;ns)</code>
<div class="block">Reads a sequence from the given buffered reader using the given tokenizer to parse
 sequence symbols.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../org/biojavax/bio/seq/io/package-summary.html">org.biojavax.bio.seq.io</a> that throw <a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">BioException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../org/biojavax/bio/seq/io/HashedFastaIterator.html#HashedFastaIterator(java.io.BufferedInputStream,%20org.biojava.bio.symbol.Alphabet,%20org.biojavax.Namespace)">HashedFastaIterator</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/io/BufferedInputStream.html?is-external=true" title="class or interface in java.io">BufferedInputStream</a>&nbsp;is,
                                      <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha,
                                      <a href="../../../../org/biojavax/Namespace.html" title="interface in org.biojavax">Namespace</a>&nbsp;ns)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../org/biojava/bio/BioException.html" title="class in org.biojava.bio">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/class-use/BioException.html" target="_top">Frames</a></li>
<li><a href="BioException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
