<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:51 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/multiple/class-use/MultipleAlignment.html" target="_top">Frames</a></li>
<li><a href="MultipleAlignment.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment" class="title">Uses of Interface<br>org.biojava.nbio.structure.align.multiple.MultipleAlignment</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></td>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></td>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.mc">org.biojava.nbio.structure.align.multiple.mc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></td>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleStructureAligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/MultipleStructureAligner.html#align(java.util.List)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleStructureAligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/MultipleStructureAligner.html#align(java.util.List,%20java.lang.Object)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">Run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#display(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">display</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code>
<div class="block">Display a MultipleAlignment with a JmolPanel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getAlignmentPanelMenu(javax.swing.JFrame,%20java.awt.event.ActionListener,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentPanelMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event">ActionListener</a>&nbsp;actionListener,
                     <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Create the menu for the Alignment Panel representation of
 Structural Alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getAlignmentTextMenu(javax.swing.JFrame,%20java.awt.event.ActionListener,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentTextMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
                    <a href="http://docs.oracle.com/javase/7/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event">ActionListener</a>&nbsp;actionListener,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Create the menu for the Text representations of Structural Alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.Atom[])">getPDBresnum</a></strong>(int&nbsp;structNum,
            <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Utility method used in the <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuItem.html?is-external=true" title="class or interface in javax.swing">JMenuItem</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getSaveAlignmentMenuItem(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSaveAlignmentMenuItem</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#initJmolMenu(javax.swing.JFrame,%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">initJmolMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
            <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;parent,
            <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Provide a JMenuBar that can be added to a JFrame containing
 a JmolPanel.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#showAlignmentImage(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.lang.String)">showAlignmentImage</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;result)</code>
<div class="block">Creates a new Frame with the String output representation of the
 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#showMultipleAligmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showMultipleAligmentPanel</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                         <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Creates a new Frame with the MultipleAlignment Sequence Panel.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MySaveFileListener.html#MySaveFileListener(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">MySaveFileListener</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Constructor to avoid checking which of the two is null before
 instantiating this class.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MySaveFileListener.html#MySaveFileListener(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">MySaveFileListener</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAligPanel.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#getMultipleAlignment()">getMultipleAlignment</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#MultipleAligPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                 <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jm)</code>
<div class="block">Constructor using a MultipleAlignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmol.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html#getMultipleAlignment()">getMultipleAlignment</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmol.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20org.jcolorbrewer.ColorBrewer,%20boolean)">getJmolString</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
             org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
             boolean&nbsp;colorByBlocks)</code>
<div class="block">Generate a Jmol command String that colors the aligned residues of every
 structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmol.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html#getMultiBlockJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20org.jcolorbrewer.ColorBrewer,%20boolean)">getMultiBlockJmolString</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
                       org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
                       boolean&nbsp;colorByBlocks)</code>
<div class="block">Colors every Block of the structures with a different color, following
 the palette.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html#MultipleAlignmentJmol(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List)">MultipleAlignmentJmol</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;rotatedAtoms)</code>
<div class="block">The constructor displays the Mutltiple Alignment in a new JmolPanel
 Frame.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> that implement <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentImpl</a></strong></code>
<div class="block">A general implementation of a <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html#clone()">clone</a></strong>()</code>
<div class="block">Creates and returns an identical copy of this alignment, including a
 deep copy of all constituent BlockSets.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">BlockSetImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSetImpl.html#getMultipleAlignment()">getMultipleAlignment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">BlockSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSet.html#getMultipleAlignment()">getMultipleAlignment</a></strong>()</code>
<div class="block">Returns the parent MultipleAlignment of the BlockSet.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsembleImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#getMultipleAlignment(int)">getMultipleAlignment</a></strong>(int&nbsp;index)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsemble.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html#getMultipleAlignment(int)">getMultipleAlignment</a></strong>(int&nbsp;index)</code>
<div class="block">Returns the MultipleAlignments at the specified index
 in the ensemble.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsembleImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#getMultipleAlignments()">getMultipleAlignments</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsemble.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html#getMultipleAlignments()">getMultipleAlignments</a></strong>()</code>
<div class="block">Returns the List of MultipleAlignments in the ensemble.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsembleImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">addMultipleAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsemble.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">addMultipleAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Add a new MultipleAlignment to the end of the ensemble and set its
 parent ensemble to this.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlockSetImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSetImpl.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlockSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSet.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code>
<div class="block">Set the back-reference to its parent MultipleAlignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsembleImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#setMultipleAlignments(java.util.List)">setMultipleAlignments</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsemble.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html#setMultipleAlignments(java.util.List)">setMultipleAlignments</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code>
<div class="block">Set the List of MultipleAlignments in the ensemble.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSetImpl.html#BlockSetImpl(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">BlockSetImpl</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Constructor.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.mc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html#align(java.util.List)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html#align(java.util.List,%20java.lang.Object)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html#call()">call</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html#optimize()">optimize</a></strong>()</code>
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html#MultipleMcOptimizer(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.align.multiple.mc.MultipleMcParameters,%20int)">MultipleMcOptimizer</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;seedAln,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcParameters.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcParameters</a>&nbsp;params,
                   int&nbsp;reference)</code>
<div class="block">Constructor.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#calculateScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">calculateScores</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Calculates and puts the RMSD and the average TM-Score of the
 MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20int,%20int)">getAtomForSequencePosition</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                          <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                          int&nbsp;str,
                          int&nbsp;sequencePos)</code>
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAverageResidueDistances</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getAvgTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAvgTMScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Calculates the average TMScore of all the possible pairwise structure
 comparisons of the given alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20int)">getBlockForSequencePosition</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                           int&nbsp;sequencePos)</code>
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getBlockSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</code>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List)">getBlockSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                         <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</code>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getHSDMTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getKimuraTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getMCScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20double,%20double,%20double)">getMCScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          double&nbsp;gapOpen,
          double&nbsp;gapExtension,
          double&nbsp;dCutoff)</code>
<div class="block">Calculates the MC score, specific for the MultipleAlignment algorithm.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getRefRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20int)">getRefRMSD</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;ref)</code>
<div class="block">/** Calculates the average RMSD from all structures to a reference s
 tructure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getRefTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20int)">getRefTMScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
             int&nbsp;ref)</code>
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRMSD</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Calculates the RMSD of all-to-all structure comparisons (distances) of
 the given MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRMSDMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentDisplay.html#getRotatedAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRotatedAtoms</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code>
<div class="block">New structures are downloaded if they were not cached in the alignment
 and they are entirely transformed here with the superposition information
 in the Multiple Alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List)">getSequenceAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
                    <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct)</code>
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static org.forester.phylogeny.Phylogeny</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getStructuralTree</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CoreSuperimposer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/CoreSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleSuperimposer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Superimpose all structures from a <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ReferenceSuperimposer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/ReferenceSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20int,%20int)">to3DFormat</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;queryIndex,
          int&nbsp;templateIndex)</code>
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#toAlignedResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toAlignedResidues</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code>
<div class="block">Converts the alignment to its simplest form: a list of groups of aligned
 residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#toFASTA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toFASTA</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Converts the <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a multiple sequence alignment
 String in FASTA format.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#toFatCat(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toFatCat</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Converts the <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a FatCat String format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toProteinMSA</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code>
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#toTransformMatrices(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">toTransformMatrices</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Converts the transformation Matrices of the alignment into a String
 output.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">transformAtoms</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Transforms atoms according to the superposition stored in the alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLParser.html#parseMultipleAlignment(org.w3c.dom.Node,%20org.biojava.nbio.structure.align.multiple.MultipleAlignmentEnsemble)">parseMultipleAlignment</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/org/w3c/dom/Node.html?is-external=true" title="class or interface in org.w3c.dom">Node</a>&nbsp;root,
                      <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsemble</a>&nbsp;ensemble)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/BlockSet.html" title="interface in org.biojava.nbio.structure.align.multiple">BlockSet</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLParser.html#parseBlockSet(org.w3c.dom.Node,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">parseBlockSet</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/org/w3c/dom/Node.html?is-external=true" title="class or interface in org.w3c.dom">Node</a>&nbsp;root,
             <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLConverter.html#printXMLalignment(org.biojava.nbio.core.util.PrettyXMLWriter,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">printXMLalignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                 <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#getMultipleAlignment()">getMultipleAlignment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SymmOptimizer.html#optimize()">optimize</a></strong>()</code>
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">GraphComponentRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">setMultipleAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multipleAlignment)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">fromAFP</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#toFullAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toFullAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code>
<div class="block">Method that converts a repeats symmetric alignment into an alignment of
 whole structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#toRepeatsAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toRepeatsAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code>
<div class="block">Method that converts a symmetry alignment into an alignment of the
 repeats only, as new independent structures.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#isRefined(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">isRefined</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code>
<div class="block">Returns true a symmetry multiple alignment has been refined, false
 otherwise.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#isSignificant(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20double)">isSignificant</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
             double&nbsp;symmetryThreshold)</code>
<div class="block">Returns true if the symmetry alignment is significant, false otherwise.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#updateSymmetryScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryScores</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code>
<div class="block">Update the scores (TM-score and RMSD) of a symmetry multiple alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.Atom[])">updateSymmetryTransformation</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
                            <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/multiple/class-use/MultipleAlignment.html" target="_top">Frames</a></li>
<li><a href="MultipleAlignment.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
