<p><strong>Note</strong>: The classes mentioned in this article are not integrated in
the BioJava 1.4 library, yet. However, they are avaiable via
<a href="http://cvs.biojava.org/cgi-bin/viewcvs/viewcvs.cgi/biojava-live/src/org/biojava/bio/proteomics/aaindex/?cvsroot=biojava">CVS</a>.</p>

<h2 id="how-do-i-analyze-the-symbol-properties-of-an-amino-acid-sequence-using-the-amino-acid-index-database">How do I analyze the symbol properties of an amino acid sequence using the Amino Acid Index database?</h2>

<p>To analyze the symbol properties of an amino acid sequence, e.g. the
average hydrophobicity of the protein, one can use the interface
<code class="highlighter-rouge">[http://www.biojava.org/docs/api14/org/biojava/bio/symbol/SymbolPropertyTable.html SymbolPropertyTable]</code>.
Its
<code class="highlighter-rouge">[http://www.biojava.org/docs/api14/org/biojava/bio/symbol/SymbolPropertyTable.html#getDoubleValue(org.biojava.bio.symbol.Symbol) getDoubleValue]</code>
method returns a numeric value for a given amino acid symbol, e.g. a
negative or positive value that indicates the hydrophobicity of the
amino acid relative to the other amino acids. The <a href="http://www.genome.ad.jp/dbget/aaindex.html">Amino Acid
Index</a> database contains
over 500 different amino acid property tables in a simple text file
called
<em><a href="ftp://ftp.genome.ad.jp/pub/db/genomenet/aaindex/aaindex1">aaindex1</a></em>.
<a href="http://www.genome.jp/dbget-bin/show_man?aaindex">AAindex1</a> is also the
format name.</p>

<p>This file can be loaded through the <code class="highlighter-rouge">AAindexStreamReader</code> class.
Afterwards, the property tables can be enumerated by calling the
<code class="highlighter-rouge">nextTable</code> method, which returns for each property table an <code class="highlighter-rouge">AAindex</code>
object that implements the <code class="highlighter-rouge">SymbolPropertyTable</code> interface. If the file
must be hold in memory and random access to the property tables (via the
table name) is needed, one can use a <code class="highlighter-rouge">SimpleSymbolPropertyTableDB</code>
object and initialize it with a <code class="highlighter-rouge">AAindexStreamReader</code> object.</p>

<p>The following example shows how to calculate the average hydrophobicity
for a given amino acid sequence (in this example the sequence only
contains the twenty amino acids) on the basis of the <em>CIDH920105</em> table
from the AAindex1 file <em>aaindex1</em>:</p>

<java>SimpleSymbolPropertyTableDB db = new
SimpleSymbolPropertyTableDB(new AAindexStreamReader(new
FileReader("aaindex1"))); AAindex hydrophobicity = (AAindex)
db.table("CIDH920105"); SymbolList symbols =
ProteinTools.createProtein("ARNDCEQGHILKMFPSTWYV"); double hp = 0.0; for
(int i = 1; i \&lt;= symbols.length(); i++) {

`   hp += hydrophobicity.getDoubleValue(symbols.symbolAt(i));`

} System.out.println("Average hydrophobicity: " + (hp /
symbols.length()));</java>
