<h2 id="welcome-to-the-biojava-forum-page">Welcome to the BioJava Forum Page!!</h2>

<p>This page is to harbour discussions on BioJava topics, especially topics
related to development of new features or extension of already present
features.</p>

<h3 id="dbsnp-objects-for-biojava">dbSNP objects for BioJava</h3>

<p>I started working with a research team focusing in genomics and observed
that there is nothing in BioJava to deal with dbSNP infos. I setup a
(very) simple NCBIdbSNP class that can fetch single dbSNP entries and
represent them as RichSequences. Anybody would have insights on the
subject where I could go with this?</p>

<p>–<a href="User:Foisys" title="wikilink">Foisys</a> 21:25, 23 January 2007 (EST)</p>

<p>Sounds interesting. How do you represent the SNP on the RichSequence? Is
it a Feature? Does dbSNP contain frequency information for the SNPs? If
it does it would be possible to represent each SNP as a Distribution.
You could make a simple extension of RichFeature that stores a
Distribution or possibly embed it into the RichAnnotation for the
feature (best to do it as a string rather than as a Distribution in this
case).</p>

<p>–<a href="User:Mark" title="wikilink">Mark</a> 23:33, 27 January 2007 (EST)</p>

<p>Right now, I have a simple parser that reads the XML file for a given
dbSNP and extracts some infos out of it. I want to make each dbSNP entry
into a RichSequence (content still in flux). I have not yet settled into
the finality of the object: possibly extracting frequencied in
populations would be an nice thing to have?</p>

<p>–<a href="User:Foisys" title="wikilink">Foisys</a> 11:13, 29 January 2007 (EST)</p>

<p>For me this class would be particularly interesting. In STRAP users can
manually give a list of SNPs
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&amp;cmd=Retrieve&amp;dopt=AbstractPlus&amp;list_uids=16322575 abstract" title="wikilink">http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=pubmed&amp;cmd=Retrieve&amp;dopt=AbstractPlus&amp;list_uids=16322575
abstract</a>
which are then projected on the AA-Sequence, NT-Sequence or 3D-structure
<a href="http://www.proteinscience.org/cgi/content/abstract/15/1/208 publication" title="wikilink">http://www.proteinscience.org/cgi/content/abstract/15/1/208
publication</a>
Your class could help to get it from the databases automatically.</p>

<p>–<a href="User:Christo" title="wikilink">Christo</a> 05:07, 10 March 2007 (EST)</p>

<h3 id="data-retrival-tools-for-biojava">Data retrival tools for BioJava</h3>

<p>It would be very useful to include tools to retrieve data from different
databases, in particular from NCBI. Some tools programmed in Java
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/query/static/esoap_java_help.html">1</a>
already exist, but it would be great to embed them in BioJava, along
with the tools required to parse the data retrieved.</p>

<p>–<a href="User:Gbonamy" title="wikilink">Ghislain</a> 22:19, 6 July 2008 (UTC)</p>

<p>Hi Ghislain,</p>

<p>I had this <a href="BioJava:CookBook:ExternalSources:NCBIFetch" title="wikilink"> little
example</a> ready
for a while because I had this same exact problem so many years ago ;-)
There is also a package to get info from Ensembl (biojava-ensembl) from
Thomas Down but I do not know what is its status right now. Thomas: can
you give us an update on this?</p>

<p>–<a href="User:Foisys" title="wikilink">Foisys</a> 12:35, 7 July 2008 (UTC)</p>
