<h2 id="seqres-and-atom-information-from-pdb-files">SEQRES and ATOM information from PDB files</h2>

<p>The SEQRES records in a PDB file contain the amino acid or nucleic acid
sequence of residues in each chain of the macromolecule that was
studied. The ATOM records provide the coordinates for those residues
that were possible to observe. In order to join this information,
Biojava aligns the SEQRES and the ATOM records for each chain. This is
optional and can be enabled or disabled by using the
<a href="http://www.biojava.org/docs/api/org/biojava/nbio/structure/io/PDBFileReader.html">PDBFileReader.setAlignSeqRes()</a>
method. By default this alignment will be calculated.</p>

<p>Access to the ATOM groups is possible via
<a href="http://www.biojava.org/docs/api/org/biojava/nbio/structure/Chain.html">Chain.getAtomGroups()</a>.
Access to the SEQRES groups is possible via
<a href="http://www.biojava.org/docs/api/org/biojava/nbio/structure/Chain.html">Chain.getSeqResGroups()</a>.
Groups derived from SEQRES groups will be empty (i.e. they will not
contain Atoms), unless they could be mapped to the ATOM records, in
which case the full ATOM information can be accessed.</p>

<p>Example: <java> import java.util.List; import
org.biojava.nbio.structure.Chain; import
org.biojava.nbio.structure.Group; import
org.biojava.nbio.structure.GroupType; import
org.biojava.nbio.structure.Structure; import
org.biojava.nbio.structure.io.FileParsingParameters; import
org.biojava.nbio.structure.io.PDBFileReader; import
org.biojava.nbio.structure.io.LocalPDBDirectory.FetchBehavior;</java></p>

<p>public class SeqResCookBook {</p>

<p><code class="highlighter-rouge">   public static void main(String[] args) throws Exception {</code><br />
<code class="highlighter-rouge">       </code><br />
<code class="highlighter-rouge">       String code =  "1aoi";</code></p>

<p><code class="highlighter-rouge">       PDBFileReader pdbreader = new PDBFileReader();</code><br />
<code class="highlighter-rouge">       FileParsingParameters params = new FileParsingParameters();</code><br />
<code class="highlighter-rouge">       params.setParseSecStruc(true);// parse the secondary structure information from PDB file</code><br />
<code class="highlighter-rouge">       params.setAlignSeqRes(true);  // align SEQRES and ATOM records</code><br />
<code class="highlighter-rouge">       pdbreader.setFileParsingParameters(params);</code><br />
<code class="highlighter-rouge">       </code><br />
<code class="highlighter-rouge">       pdbreader.setFetchBehavior(FetchBehavior.FETCH_FILES);   // fetch PDB files from web if they can't be found locally</code></p>

<p><code class="highlighter-rouge">       Structure struc = pdbreader.getStructureById(code);</code></p>

<p><code class="highlighter-rouge">       System.out.println("The SEQRES and ATOM information is available via the chains:");</code></p>

<p><code class="highlighter-rouge">       int modelnr = 0 ; // also is 0 if structure is an XRAY structure.</code></p>

<p><code class="highlighter-rouge">       List</code><chain>` chains = struc.getChains(modelnr);`  
`       for (Chain cha:chains){`  
`           List`<group>` agr = cha.getAtomGroups(GroupType.AMINOACID);`  
`           List`<group>` hgr = cha.getAtomGroups(GroupType.HETATM);`  
`           List`<group>` ngr = cha.getAtomGroups(GroupType.NUCLEOTIDE);`</group></group></group></chain></p>

<p><code class="highlighter-rouge">           System.out.print("chain: &gt;"+cha.getChainID()+"&lt;");</code><br />
<code class="highlighter-rouge">           System.out.print(" length SEQRES: " +cha.getSeqResLength());</code><br />
<code class="highlighter-rouge">           System.out.print(" length ATOM: " +cha.getAtomLength());</code><br />
<code class="highlighter-rouge">           System.out.print(" aminos: " +agr.size());</code><br />
<code class="highlighter-rouge">           System.out.print(" hetatms: "+hgr.size());</code><br />
<code class="highlighter-rouge">           System.out.println(" nucleotides: "+ngr.size());  </code><br />
<code class="highlighter-rouge">       }</code></p>

<p><code class="highlighter-rouge">   }</code></p>

<p>}</p>

<p>&lt;/java&gt;</p>

<p>gives this output:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>The SEQRES and ATOM information is available via the chains:
chain: &gt;I&lt; length SEQRES: 146 length ATOM: 153 aminos: 0 hetatms: 7 nucleotides: 146
chain: &gt;J&lt; length SEQRES: 146 length ATOM: 154 aminos: 0 hetatms: 8 nucleotides: 146
chain: &gt;A&lt; length SEQRES: 116 length ATOM: 98 aminos: 98 hetatms: 0 nucleotides: 0
chain: &gt;B&lt; length SEQRES: 87 length ATOM: 84 aminos: 83 hetatms: 1 nucleotides: 0
chain: &gt;C&lt; length SEQRES: 116 length ATOM: 115 aminos: 115 hetatms: 0 nucleotides: 0
chain: &gt;D&lt; length SEQRES: 99 length ATOM: 99 aminos: 99 hetatms: 0 nucleotides: 0
chain: &gt;E&lt; length SEQRES: 116 length ATOM: 117 aminos: 116 hetatms: 1 nucleotides: 0
chain: &gt;F&lt; length SEQRES: 87 length ATOM: 89 aminos: 87 hetatms: 2 nucleotides: 0
chain: &gt;G&lt; length SEQRES: 116 length ATOM: 108 aminos: 108 hetatms: 0 nucleotides: 0
chain: &gt;H&lt; length SEQRES: 99 length ATOM: 99 aminos: 99 hetatms: 0 nucleotides: 0
</code></pre>
</div>

<p>Next: <biojava:cookbook:pdb:align> - How to align two protein structures</biojava:cookbook:pdb:align></p>
