<p><strong>If you use BioJava in an application or publication please cite:</strong></p>

<h2 id="projects">Projects</h2>

<p>The following projects make use of BioJava. If you know of other
projects please add them to the list.</p>

<ul>
  <li><a href="http://www.genostar.com/">Metabolic Pathway Builder</a>: Software
suite dedicated to the exploration of connections among genes,
proteins, reactions and metabolic pathways</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.dengueinfo.org/">DengueInfo</a>: a Dengue genome
information portal that uses BioJava in the middleware and talks to
a biosql database.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.derkholm.net/thomas/dazzle">Dazzle</a>: A BioJava based DAS
server.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.idbs.com/InforSenseSuite/BioSense">BioSense</a>: A plugin
for the InforSense Suite, an analytics software platform by
<a href="http://www.idbs.com/">IDBS</a> that unitizes BioJava.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.bioclipse.net">Bioclipse</a>: A free, open source,
workbench for chemo- and bioinformatics with powerful editing and
visualization capabilities for molecules, sequences, proteins,
spectra etc.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://webclu.bio.wzw.tum.de/prompt">PROMPT</a>: A free, open source
framework and application for the comparison and mapping of protein
sets. Uses BioJava for handling most input data formats.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.cytoscape.org">Cytoscape</a>: An open source bioinformatics
software platform for visualizing molecular interaction networks.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.bioweka.org">BioWeka</a>: An open source biological data
mining application.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.biomatters.com">Geneious</a>: A molecular biology toolkit.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.ncbi.nlm.nih.gov/staff/slottad/MassSieve/">MassSieve</a>:
An open source application to analyze mass spec proteomics data.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.charite.de/bioinf/strap/">Strap</a>: A tool for multiple
sequence alignment and sequence based structure alignment.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.jstacs.de">Jstacs</a>: A Java framework for statistical
analysis and classification of biological sequences</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.bioinf.jku.at/software/LSTM_protein/">jLSTM</a> “Long
Short-Term Memory” for protein classification</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://lajolla.sourceforge.net">LaJolla</a> Structural alignment of
RNA and proteins using an index structure for fast alignment of
thousands of structures. Including an easy to use command line
interface. Open source at Sourceforge.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://www.geneinfinity.org/genbeans/index.html">GenBeans</a>: A rich
client platform for bioinformatics primarily focused on molecular
biology and sequence analysis.</li>
</ul>

<!-- -->

<ul>
  <li><a href="http://bioservices.hs-mittweida.de/equant/">eQuant</a>: A model
quality assessment server to state the reliability of protein
structures.</li>
</ul>

<h2 id="publications">Publications</h2>

<p>In 2008 we published our first Application note. As of Nov. 2014 Google
Scholar <a href="http://scholar.google.com/scholar?cites=3048631375755320177&amp;as_sdt=2005&amp;sciodt=0,5&amp;hl=en">counts more than 170
citations</a>.</p>

<pubmed name="BioJava2008"> 18689808 </pubmed>

<p>Below a list of publications in which BioJava has been used. If you know
of other publications please add them.</p>

<pubmed name="hidalgo1998"> 9564045 </pubmed>
<pubmed name="jacobs2000">10592251 </pubmed>
<pubmed name="xie2000">12761070</pubmed>
<pubmed> 12016048</pubmed>
<pubmed> 12493080 </pubmed>

<h3 id="section">2003</h3>

<pubmed> 12626717</pubmed>
<pubmed> 12967955 </pubmed>
<pubmed>
15130816 </pubmed>
<pubmed> 14594704 </pubmed>
<pubmed> 14592990
</pubmed>
<pubmed> 14668218</pubmed>
<pubmed> 12824432 </pubmed>
<pubmed> 14583100</pubmed>

<h3 id="section-1">2004</h3>

<pubmed> 15044242 </pubmed>
<pubmed> 15215471 </pubmed>
<pubmed>
15369604 </pubmed>
<pubmed> 15247332 </pubmed>
<pubmed> 14681429
</pubmed>
<pubmed> 15610565 </pubmed>

<h3 id="section-2">2005</h3>

<pubmed> 15537809</pubmed>
<pubmed> 15760844</pubmed>
<pubmed>
15572471 </pubmed>
<pubmed> 15984937 </pubmed>
<pubmed> 16083500
</pubmed>
<pubmed> 15608201 </pubmed>
<pubmed> 15640145 </pubmed>
<pubmed> 16204122 </pubmed>
<pubmed> 15905283 </pubmed>
<pubmed>
16288651 </pubmed>

<h3 id="section-3">2006</h3>

<pubmed> 16845037 </pubmed>
<pubmed> 17002805 </pubmed>
<pubmed>
16925840 </pubmed>
<pubmed> 16469097</pubmed>
<pubmed>
16789813</pubmed>
<pubmed> 16990246</pubmed>
<pubmed>
16402215</pubmed>
<pubmed> 17054788 </pubmed>
<pubmed> 16260186
</pubmed>
<pubmed> 17000643</pubmed>
<pubmed> 16423288</pubmed>
<pubmed> 16845480 </pubmed>
<pubmed> 16817977 </pubmed>
<pubmed>
16845480</pubmed>
<pubmed> 16817977</pubmed>
<pubmed>
16837528</pubmed>
<pubmed> 16872539</pubmed>

<h3 id="section-4">2007</h3>

<pubmed> 17068077</pubmed>
<pubmed> 17288609 </pubmed>
<pubmed>
17238282</pubmed>
<pubmed> 17237069 </pubmed>
<pubmed>
17332025</pubmed>
<pubmed> 17400476</pubmed>
<pubmed>
17537825</pubmed>
<pubmed> 17316423</pubmed>

<h3 id="section-5">2008</h3>

<pubmed> 18061398</pubmed>
<pubmed name="zajac2008">18061398</pubmed>
<pubmed name="vernikos2008">18071028</pubmed>
<pubmed name="liang2008">17054788</pubmed>
<pubmed name="chalk2008">18397893</pubmed>
<pubmed name="gront2008">18227118</pubmed>

<p>See above for a link to all recent citations on Google Scholar.</p>

<h3 id="section-6">2009</h3>

<p>Bauer, R.; Rother, K.; Moor, P.; Reinert, K.; Steinke, T.; Bujnicki, J.
M.; Preissner, R. Fast Structural Alignment of Biomolecules Using a Hash
Table, N-Grams and String Descriptors. <em>Algorithms</em> <strong>2009</strong>, 2,
692-709. <a href="http://www.mdpi.com/1999-4893/2/2/692">open access full text</a></p>

<p>More biojava publications can be found at <a href="http://scholar.google.com/scholar?q=biojava">Google
Scholar</a>.</p>
