<h2 id="how-do-i-work-with-nextgen-sequencing-reads-in-fastq-format">How do I work with nextgen sequencing reads in FASTQ format?</h2>

<p>The org.biojava3.sequencing.io.fastq package provides support for
reading and writing nextgen sequencing reads in FASTQ format
(FastqReader and FastqWriter, respectively) and for converting FASTQ
sequences into proper biojava3 DNASequences for analysis purposes
(FastqTools).</p>

<p>The following code snippets demonstrate how to use the APIs for common
use cases.</p>

<h3 id="convert-between-fastq-variants">Convert between FASTQ variants</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader(); FastqWriter
fastqWriter = new SangerFastqWriter(); fastqWriter.write(new
File("sanger.fastq"), fastqReader.read(new File("illumina.fastq")));
</java>

<h3 id="convert-only-long-sequences">Convert only long sequences</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader(); FastqWriter
fastqWriter = new SangerFastqWriter(); FileWriter fileWriter = new
FileWriter(new File("sanger.fastq"))));

for (Fastq fastq : fastqReader.read(new File("illumina.fastq"))) {

` if (fastq.getSequence().length() &gt; 16)`  
` {`  
`   fastqWriter.append(fileWriter, fastq);`  
` }`

} </java>

<h3 id="convert-between-fastq-variants-using-streaming-api">Convert between FASTQ variants using streaming API</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader();
InputSupplier inputSupplier = Files.newReaderSupplier(new
File("illumina.fastq")); final FastqWriter fastqWriter = new
SangerFastqWriter(); final FileWriter fileWriter = new FileWriter(new
File("sanger.fastq"))));

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     fastqWriter.append(fileWriter, fastq);`  
`   }`  
` });`

</java>

<h3 id="convert-only-long-sequences-using-streaming-api">Convert only long sequences using streaming API</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader();
InputSupplier inputSupplier = Files.newReaderSupplier(new
File("illumina.fastq")); FastqWriter fastqWriter = new
SangerFastqWriter(); FileWriter fileWriter = new FileWriter(new
File("sanger.fastq"))));

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     if (fastq.getSequence().length() &gt; 16)`  
`     {`  
`       fastqWriter.append(fileWriter, fastq);`  
`     }`  
`   }`  
` });`

</java>

<h3 id="count-sequences">Count sequences</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader();

int count = 0; for (Fastq fastq : fastqReader.read(new
File("illumina.fastq"))) {

` count++:`

} System.out.println(count); </java>

<h3 id="count-sequences-using-streaming-api">Count sequences using streaming API</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader();
InputSupplier inputSupplier = Files.newReaderSupplier(new
File("illumina.fastq"));

final AtomicInteger count = new AtomicInteger();
fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     count.incrementAndGet();`  
`   }`  
` });`

System.out.println(count.get()); </java>

<h3 id="count-sequences-using-low-level-api">Count sequences using low-level API</h3>

<java> FastqReader fastqReader = new IlluminaFastqReader();
InputSupplier inputSupplier = Files.newReaderSupplier(new
File("illumina.fastq"));

final AtomicInteger count = new AtomicInteger();
fastqReader.parse(inputSupplier, new ParseAdapter()

` {`  
`   @Override`  
`   public void complete() throws IOException`  
`   {`  
`     count.incrementAndGet();`  
`   }`  
` });`

System.out.println(count.get()); </java>

<h3 id="pattern-match-description-lines-using-low-level-api">Pattern match description lines using low-level API</h3>

<java> final Pattern pattern = Pattern.compile("^HWUSI-EAS100R:.\*$");
FastqReader fastqReader = new IlluminaFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("illumina.fastq"));

fastqReader.parse(inputSupplier, new ParseAdapter()

` {`  
`   @Override`  
`   public void description(final String description) throws IOException`  
`   {`  
`     if (pattern.matches(description))`  
`     {`  
`       System.out.println(description);`  
`     }`  
`   }`  
` });`

</java>

<h3 id="create-dnasequences-from-fastq-sequences">Create DNASequences from FASTQ sequences</h3>

<java> FastqReader fastqReader = new SangerFastqReader();
List<dnasequence> sequences = new LinkedList<dnasequence>();

for (Fastq fastq : fastqReader.read(new File("sanger.fastq"))) {

` sequences.add(FastqTools.createDNASequence(fastq));`

} 

### Create DNASequences with quality scores in QualityFeature from FASTQ sequences

<java> FastqReader fastqReader = new SangerFastqReader();
List<dnasequence> sequences = new LinkedList<dnasequence>();

for (Fastq fastq : fastqReader.read(new File("sanger.fastq"))) {

` sequences.add(FastqTools.createDNASequenceWithQualityScores(fastq));`

} 

### Create DNASequences with quality scores in QualityFeature and error probabilities in QuantityFeature from FASTQ sequences

<java> FastqReader fastqReader = new SangerFastqReader();
List<dnasequence> sequences = new LinkedList<dnasequence>();

for (Fastq fastq : fastqReader.read(new File("sanger.fastq"))) {

` sequences.add(FastqTools.createDNASequenceWithQualityScoresAndErrorProbabilities(fastq));`

} 

### Create DNASequences with streaming API

<java> FastqReader fastqReader = new SangerFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("sanger.fastq"));
List<dnasequence> sequences = new LinkedList<dnasequence>();

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     sequences.add(FastqTools.createDNASequence(fastq));`  
`   }`  
` });`



### Create DNASequences with quality scores in QualityFeature from FASTQ sequences with streaming API

<java> FastqReader fastqReader = new SangerFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("sanger.fastq"));
List<dnasequence> sequences = new LinkedList<dnasequence>();

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     sequences.add(FastqTools.createDNASequenceWithQualityScores(fastq));`  
`   }`  
` });`



### Create DNASequences with quality scores in QualityFeature and error probabilities in QuantityFeature from FASTQ sequences with streaming API

<java> FastqReader fastqReader = new SangerFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("sanger.fastq"));
List<dnasequence> sequences = new LinkedList<dnasequence>();

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     sequences.add(FastqTools.createDNASequenceWithQualityScoresAndErrorProbabilities(fastq));`  
`   }`  
` });`



### Calculate mean p scores using streaming API

<java> FastqReader fastqReader = new SangerFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("sanger.fastq"));
SummaryStatistics stats = new SummaryStatistics(); StringBuilder sb =
new StringBuilder(512);

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     stats.clear();`  
`     for (Double errorProbability : FastqTools.errorProbabilities(fastq))`  
`     {`  
`       stats.addValue(errorProbability);`  
`     }`  
`     sb.delete(0, sb.length());`  
`     sb.append(fastq.getDescription());`  
`     sb.append("\t");`  
`     sb.append(stats.getMean());`  
`     sb.append("\t");`  
`     sb.append(stats.getStandardDeviation());`  
`     System.out.println(sb.toString());`  
`   }`  
` });`

</java>

### Calculate mean p scores using streaming API and double array

<java> FastqReader fastqReader = new SangerFastqReader(); InputSupplier
inputSupplier = Files.newReaderSupplier(new File("sanger.fastq"));
SummaryStatistics stats = new SummaryStatistics(); StringBuilder sb =
new StringBuilder(512);

fastqReader.stream(inputSupplier, new StreamListener()

` {`  
`   @Override`  
`   public void fastq(final Fastq fastq)`  
`   {`  
`     stats.clear();`  
`     int size = fastq.getSequence().length();`  
`     double[] errorProbabilities = FastqTools.errorProbabilties(fastq, new double[size]);`  
`     for (int i = 0; i &lt; size; i++)`  
`     {`  
`       stats.addValue(errorProbabilities[i]);`  
`     }`  
`     sb.delete(0, sb.length());`  
`     sb.append(fastq.getDescription());`  
`     sb.append("\t");`  
`     sb.append(stats.getMean());`  
`     sb.append("\t");`  
`     sb.append(stats.getStandardDeviation());`  
`     System.out.println(sb.toString());`  
`   }`  
` });`

</java>
</dnasequence></dnasequence></java></dnasequence></dnasequence></java></dnasequence></dnasequence></java></dnasequence></dnasequence></java></dnasequence></dnasequence></java></dnasequence></dnasequence></java>
