<h3 id="task-1">Task 1</h3>

<p>Write a FASTA parser</p>

<p>Your solution should:</p>

<ul>
  <li>be a Java code of high quality (maintability, reusability, OO
design, etc)</li>
  <li>be efficient</li>
  <li>be capable of reading badly formatted FASTA files</li>
  <li>be capable of reading large files</li>
  <li>has convenient API</li>
  <li>be possible to extend to reading FASTQ files</li>
</ul>

<p>Please refrain from using any libraries that are not part of a standard
Java 6 development kit in your production code. For testing code fill
free to use any library you feel comfortable with.</p>

<h3 id="task-2">Task 2</h3>

<p>Write a FASTA writer</p>

<ul>
  <li>Use your parser to read a FASTA file which contains sequences with
ambiguous characters (you choose whether this is going to be
ambiguous DNA or protein sequence)</li>
  <li>Write two FASTA output files one with sequences which contains
ambiguous characters and another one without.</li>
</ul>

<h3 id="submission">Submission</h3>

<p>Please submit your completed exercise to <strong>gsocexercise at gmail dot
com</strong> by Friday the 6 of April inclusive. Your submission should be a
ZIP archive that contains an executable JAR file with your FASTA parser
and writer as well as</p>

<ul>
  <li>your source files in the <em>src</em> directory</li>
  <li>your documentation files in the <em>docs</em> directory</li>
  <li>the test data file named data.fasta up to 10Kb in size</li>
  <li>The executable JAR containing the program. This should be called
<em>runme.jar</em>.</li>
  <li>a pure ASCII text file called <em>choices.txt</em> describing the
significant design choices you made, uncertainties you had regarding
the project, and the decisions you made when resolving them.</li>
</ul>

