<h1 id="how-to-write-your-own-dazzle-plugin">How to write your own Dazzle plugin</h1>

<p>Each plugin for <a href="Dazzle" title="wikilink">Dazzle</a> has to implement certain
interfaces. Here we will show how to implement a Dazzle plugin that
supports the DAS - features command, using the GFFFeatureSource
interface. There are also other plugin mechanisms in Dazzle, but for the
moment let’s only consider this one.</p>

<h2 id="required-knowledge">Required knowledge</h2>

<p>For this turorial you should already know how to <a href="Dazzle:deployment" title="wikilink">deploy
Dazzle</a></p>

<h2 id="the-gfffeaturesource-interface">The GFFFeatureSource interface</h2>

<p>The Interface that needs to be implemented is the
<a href="http://www.derkholm.net/svn/repos/dazzle/trunk/src/org/biojava/servlets/dazzle/datasource/GFFFeatureSource.java">GFFFeatureSource</a>
interface.</p>

<p>What this means is that the DAS source provides a method called</p>

<p><code class="highlighter-rouge">GFFFeature getFeatures(String reference);</code></p>

<p>This method accepts a String as an argument, that represents either the
chromosomal region or the accession code that is requested. It returns
an array of
<a href="http://www.derkholm.net/svn/repos/dazzle/trunk/src/org/biojava/servlets/dazzle/datasource/GFFFeature.java">GFFFeature</a>
objects that contain the data that should be transported.</p>

<p>When Dazzle gets a DAS - Features request for your DAS source, it will
call this getFeatures method in order to obtain the data and then return
it as DAS-XML.</p>

<h2 id="the-abstractgfffeaturesource-class">The AbstractGFFFeatureSource class</h2>

<p>For full DAS-specification support a couple of more methods are
required, but they do not need to worry us right now, since there is a
utility class available that contains most of the required code already.
Your plugin simply needs to extend
<a href="http://www.derkholm.net/svn/repos/dazzle/trunk/src/org/biojava/servlets/dazzle/datasource/AbstractGFFFeatureSource.java">AbstractGFFFeatureSource</a>.</p>

<h2 id="a-minimal-plugin">A minimal plugin</h2>

<p>A minimal plugin for Dazzle looks like this: let’s call the file below
MyPlugin.java <java> package org.dazzle;</java></p>

<p>import org.biojava.servlets.dazzle.datasource.AbstractGFFFeatureSource;
import org.biojava.servlets.dazzle.datasource.DataSourceException;
import org.biojava.servlets.dazzle.datasource.GFFFeature;</p>

<p>public class MyPlugin extends AbstractGFFFeatureSource {</p>

<p><code class="highlighter-rouge">   public GFFFeature[] getFeatures(String reference) </code><br />
<code class="highlighter-rouge">   throws DataSourceException{</code><br />
<code class="highlighter-rouge">       System.out.println("got a features request for " + reference);</code><br />
<code class="highlighter-rouge">       return new GFFFeature[0];</code><br />
<code class="highlighter-rouge">   }</code></p>

<p>} &lt;/java&gt;</p>

<p>and to enable this in Dazzle we add the following lines to
<strong>dazzlecfg.xml</strong> :</p>

<div class="highlighter-rouge"><pre class="highlight"><code> &lt;datasource id="myplugin" jclass="org.dazzle.MyPlugin"&gt;
    &lt;string name="name" value="My 1st Plugin" /&gt;
    &lt;string name="description" value="a demo for how to write a Dazzle plugin" /&gt;
    &lt;string name="version" value="1.0" /&gt;
  &lt;/datasource&gt;
</code></pre>
</div>

<h3 id="das---dsn-request">DAS - DSN request</h3>

<p>Start your Dazzle instance. If you don’t know how you should do that,
please see <dazzle:deployment>. Once Dazzle is running you can do the
DAS - dsn (data source names) command, which lists all available
datasources.</dazzle:deployment></p>

<p><a href="http://localhost:8080/dazzleDemo/dsn"><code class="highlighter-rouge">http://localhost:8080/dazzleDemo/dsn</code></a></p>

<p>note: Dazzle provides XSL stylesheets for a nice display of the XML
response in your browser. To view the raw XML source code in the Firefox
browser, add view-source: in front of the URL.</p>

<p><a href="view-source:http://localhost:8080/dazzleDemo/dsn"><code class="highlighter-rouge">view-source:http://localhost:8080/dazzleDemo/dsn</code></a></p>

<p>You should get this response if you called your servlet dazzleDemo and
your <a href="Dazzle:deployment" title="wikilink">basic Dazzle installation</a> is
correct:</p>

<xml>

<?xml version='1.0' standalone='no' ?>
<?xml-stylesheet type="text/xsl" href="das.xsl"?>
&lt;!DOCTYPE DASDSN SYSTEM 'dasdsn.dtd' &gt;
<dasdsn>

` `<dsn>  
`   `

    My 1st Plugin

`   `<mapmaster>[`http://localhost:8080/dazzleDemo/myplugin/`](http://localhost:8080/dazzleDemo/myplugin/)</mapmaster>  
`   `<description>`a demo for how to write a Dazzle plugin`</description>  
` `</dsn>

` `<dsn>  
`   `

    Test seqs

`   `<mapmaster>[`http://localhost:8080/dazzleDemo/test/`](http://localhost:8080/dazzleDemo/test/)</mapmaster>  
`   `<description>`Test set for promoter-finding software`</description>  
` `</dsn>

` `<dsn>  
`   `

    TSS

`   `<mapmaster>[`http://localhost:8080/das/test/`](http://localhost:8080/das/test/)</mapmaster>  
`   `<description>`Transcription start sites`</description>  
` `</dsn>

` `<dsn>  
`   `

    uniprot_snps

`   `<mapmaster>[`http://localhost:8080/dazzleDemo/uniprot_snps/`](http://localhost:8080/dazzleDemo/uniprot_snps/)</mapmaster>  
`   `<description>`some snps on a uniprot sequence`</description>  
` `</dsn>

</dasdsn> </xml>

<h3 id="the-das-features-command">The DAS features command</h3>

<p>Now you can also do a first DAS - features command:</p>

<p><a href="http://localhost:8080/dazzleDemo/myplugin/features?segment=123"><code class="highlighter-rouge">http://localhost:8080/dazzleDemo/myplugin/features?segment=123</code></a></p>

<p>should give you now a very simple response, which will not contain
features. (we did not return any, did we?)</p>

<p>Check your server logs it should say something like</p>

<p><code class="highlighter-rouge">got a features request for 123</code></p>

<p>If you see that, you mastered the first step!</p>

<h2 id="adding-features">Adding Features</h2>

<p>So fare our response does not contain features. Let’s add one:</p>

<java> package org.dazzle;

import java.util.ArrayList; import java.util.List; import
org.biojava.servlets.dazzle.datasource.AbstractGFFFeatureSource; import
org.biojava.servlets.dazzle.datasource.DataSourceException; import
org.biojava.servlets.dazzle.datasource.GFFFeature;

public class MyPlugin extends AbstractGFFFeatureSource {

`   public GFFFeature[] getFeatures(String reference) `  
`   throws DataSourceException{`  
`       System.out.println("got a features request for " + reference);`  
`       `  
`       List`<gfffeature>` features = new ArrayList`<gfffeature>`();`  
`       `  
`       // This is up to YOU:`  
`       // get your data from somewhere, e.g. a database, parse a flat file`  
`       // whatever you like.`  
`       // then with your data we fill the GFFFeature objects`  
`       `  
`       // GFFFeature is a simple Java-bean`  
`       GFFFeature gff = new GFFFeature();`  
`       `  
`       gff.setType("annotation type");`  
`       gff.setLabel("the annotation label");`  
`       // start and end are strings to support e.g. PDB -file residue `  
`       // numbering, which can contain insertion codes`  
`       gff.setStart("123"); `  
`       gff.setEnd("234");`  
`       `  
`       gff.setName("the name of my feature");`  
`       gff.setMethod("the dazzle plugin tutorial");`  
`       gff.setLink("`[`http://www.biojava.org/wiki/Dazzle:writeplugin`](http://www.biojava.org/wiki/Dazzle:writeplugin)`");`  
`       gff.setNote("the note field contains the actual annotation!");`  
`       `  
`       // see the documentation for GFFFeature for all possible fields`  
`               `  
`       features.add(gff);`  
`           `  
`       // and we return our features `  
`       return (GFFFeature[]) features.toArray(new GFFFeature[features.size()]);`  
`   }`

} 

Now will give this response:

<xml>

<?xml version='1.0' standalone='no' ?>
<?xml-stylesheet type="text/xsl" href="das.xsl"?>
&lt;!DOCTYPE DASGFF SYSTEM 'dasgff.dtd' &gt;
<dasgff>

` `<gff version="1.0" href="http://localhost:8088/dazzleDemo/myplugin/features?segment=123">  
`   `<segment id="123" version="" start="1" stop="-1">  
`     `<feature id="__dazzle__annotation type_the name of my feature_123_234" label="the annotation label">  
`       `<type id="annotation type">`annotation type`</type>  
`       `<method id="the dazzle plugin tutorial">`the dazzle plugin tutorial`</method>  
`       `<start>`123`</start>  
`       `<end>`234`</end>  
`       `<score>`-`</score>  
`       `

<note>
the note field contains the actual annotation!

</note>
`       `<link href="http://www.biojava.org/wiki/Dazzle:writeplugin" />[`http://www.biojava.org/wiki/Dazzle:writeplugin`](http://www.biojava.org/wiki/Dazzle:writeplugin)  
`     `</feature>  
`   `</segment>  
` `</gff>

</dasgff> </xml>

Adding more DAS commands
------------------------

No we can already expose our annotations via the DAS - features command.
Our next step is to make this DAS source a reference source for sequence
annotations. For this we need to implement the interface
[DazzleReferenceSource](http://www.derkholm.net/svn/repos/dazzle/trunk/src/org/biojava/servlets/dazzle/datasource/DazzleReferenceSource.java),
which adds support for 2 new DAS commands - entry\_points and sequence.

<java> package org.dazzle;

import java.util.ArrayList; import java.util.List; import
java.util.NoSuchElementException; import java.util.Set; import
java.util.TreeSet; import org.biojava.bio.seq.ProteinTools; import
org.biojava.bio.seq.Sequence; import
org.biojava.bio.symbol.IllegalSymbolException; import
org.biojava.servlets.dazzle.datasource.AbstractGFFFeatureSource; import
org.biojava.servlets.dazzle.datasource.DataSourceException; import
org.biojava.servlets.dazzle.datasource.DazzleReferenceSource; import
org.biojava.servlets.dazzle.datasource.GFFFeature;

public class MyPlugin extends AbstractGFFFeatureSource implements
DazzleReferenceSource{

`   public GFFFeature[] getFeatures(String reference) `  
`   throws DataSourceException{`  
`       System.out.println("got a features request for " + reference);`  
`       `  
`       List`<gfffeature>` features = new ArrayList`<gfffeature>`();`  
`       `  
`       // This is up to YOU:`  
`       // get your data from somewhere, e.g. a database, parse a flat file`  
`       // whatever you like.`  
`       // then with your data we fill the GFFFeature objects`  
`       `  
`       // GFFFeature is a simple Java-bean`  
`       GFFFeature gff = new GFFFeature();`  
`       `  
`       gff.setType("annotation type");`  
`       gff.setLabel("the annotation label");`  
`       // start and end are strings to support e.g. PDB -file residue `  
`       // numbering, which can contain insertion codes`  
`       gff.setStart("123"); `  
`       gff.setEnd("234");`  
`       `  
`       gff.setName("the name of my feature");`  
`       gff.setMethod("the dazzle plugin tutorial");`  
`       gff.setLink("`[`http://www.biojava.org/wiki/Dazzle:writeplugin`](http://www.biojava.org/wiki/Dazzle:writeplugin)`");`  
`       gff.setNote("the note field contains the actual annotation!");`  
`       `  
`       // see the documentation for GFFFeature for all possible fields`  
`               `  
`       features.add(gff);`  
`           `  
`       // and we return our features `  
`       return (GFFFeature[]) features.toArray(new GFFFeature[features.size()]);`  
`   }`

`   /** This method deals with the DAS -entry points command.`  
`    * @return a set containing the references to the entry points`  
`    */ `  
`   public Set getEntryPoints() {`  
`       Set`<string>` s = new TreeSet`<string>` ();`  
`       // this example has only one feature.`  
`       // for your real data you might want to add a SQL query here.`  
`       s.add("123");`  
`       return s;`  
`   }`

`   /** This method deals with the DAS - sequence command.`  
`    * `  
`    * @return a biojava Sequence object`  
`    * `  
`    */`  
`   public Sequence getSequence(String ref) throws NoSuchElementException, DataSourceException {`  
`       String seq =  "ECNEUQESECNEUQESECNEUQESECNEUQESECNEUQES";`  
`       `  
`       try {`  
`           Sequence prot = ProteinTools.createProteinSequence(seq, ref);`  
`           return prot;`  
`       } catch ( IllegalSymbolException e){`  
`           throw new DataSourceException(e.getMessage());`  
`       }       `  
`   }`

} 

### The DAS entry\_points command

If we forgot which reference points we annotated, we can do a DAS -
entry\_points request:

[`http://localhost:8080/dazzleDemo/myplugin/entry_points`](http://localhost:8080/dazzleDemo/myplugin/entry_points)

now returns: <xml>

<?xml version='1.0' standalone='no' ?>
<?xml-stylesheet type="text/xsl" href="das.xsl"?>
&lt;!DOCTYPE DASEP SYSTEM 'dasep.dtd' &gt;
<dasep>

` `<entry_points href="http://localhost:8088/dazzleDemo/myplugin/entry_points" version="1.0">  
`   `<segment id="123" size="-1" subparts="no" />  
` `</entry_points>

</dasep> </xml>

### The DAS SEQUENCE command

The entry points command showed us that we could use "123" as a
reference (a chromosomal region, or a database accession code) for a
request.

[`http://localhost:8080/dazzleDemo/myplugin/sequence?segment=123`](http://localhost:8080/dazzleDemo/myplugin/sequence?segment=123)

gives the response: <xml>

<?xml version='1.0' standalone='no' ?>
<?xml-stylesheet type="text/xsl" href="das.xsl"?>
&lt;!DOCTYPE DASSEQUENCE SYSTEM 'dassequence.dtd' &gt;
<dassequence>

` `<sequence id="123" version="" start="1" stop="40" moltype="Protein">

ECNEUQESECNEUQESECNEUQESECNEUQESECNEUQES

` `</sequence>

</dassequence> </xml>

Congratulations! at this point you have set up our first DAS - reference
server!
</string></string></gfffeature></gfffeature></java></gfffeature></gfffeature></java>
