<h1 id="deploying-an-ensembl-das-reference-server">Deploying an ensembl-das reference server</h1>

<p>This document describes the steps required to install a DAS reference
server serving human genome data from an Ensembl-format SQL database.
Ensembl-DAS is implemented as a plugin for the Dazzle server framework.
If you are not familiar with Dazzle, it is recommended that you read the
<a href="Dazzle:deployment" title="wikilink">Dazzle deployment</a> guide first.</p>

<h2 id="prerequisites">Prerequisites</h2>

<p>To run an Ensembl-DAS server, you will need to have the full Ensembl
core database installed. There are some instructions for doing this
here. Note that unless you actually want a local copy of the Ensembl
website (or to use the Perl APIs directly) you don’t need to install any
of the Perl code. Ensembl-DAS uses pure Java APIs for accessing the
database. You only need the core databases for the species you are
interested in – currently, no other databases are used.</p>

<h3 id="you-will-also-need-the-following">You will also need the following:</h3>

<p><code class="highlighter-rouge">   * A Java 2 runtime environment, version 1.4 or later</code><br />
<code class="highlighter-rouge">   * A Java servlet container. We recommend Tomcat 5.0.</code><br />
<code class="highlighter-rouge">   * A recent snapshot of BioJava.</code><br />
<code class="highlighter-rouge">   * Dazzle 1.00 or later</code><br />
<code class="highlighter-rouge">   * The biojava-ensembl bridge code</code><br />
<code class="highlighter-rouge">   * A Java database driver for MySQL, available from MySQL AB</code></p>

<p>The easiest way to get a server up and running is to download the latest
ensembl-das webapp skeleton. This is based on the standard Dazzle
skeleton, except that it contains biojava-ensembl code, including the
ensembl-das plugins. You therefore just need to configure and deploy the
application.</p>

<p>If you choose to build everything from source, note that the order of
compilation is important: you must first compile dazzle, the ensure that
dazzle.jar is available in your working directory (or on your CLASSPATH)
when you compile biojava-ensembl, otherwise the ensembl-das plugins will
not be compiled. When the biojava-ensembl build script starts up, it
displays a message to tell you whether or not Dazzle has been detected.</p>

<h2 id="basic-configuration">Basic configuration</h2>

<p>Like all Dazzle plugins, ensembl-das is configured by editing the
dazzlecfg.xml file. A typical minimal configuration might look like:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>&lt;dazzle xmlns="http://www.biojava.org/2000/dazzle"&gt;
  &lt;resource id="hsa2134" jclass="org.ensembl.das.DatabaseHolder"&gt;
    &lt;string name="dbURL" value="jdbc:mysql://noranti.derkholm.net/homo_sapiens_core_23_34e" /&gt;
    &lt;string name="dbUser" value="ensembl" /&gt;
    &lt;string name="dbPass" value="xxx" /&gt;
  &lt;/resource&gt;


  &lt;datasource id="hsa2134" jclass="org.ensembl.das.EnsemblCoreReference"&gt;
    &lt;string name="name" value="Human" /&gt;
    &lt;string name="description" value="The human genome assembly from Ensembl" /&gt;
    &lt;string name="version" value="23.34" /&gt;

    &lt;string name="coreHolder" value="connection_hsa_core_23_34" /&gt;
    &lt;int name="schemaVersion" value="23" /&gt;

    &lt;string name="ensemblWebURL" value="http://www.ensembl.org/Homo_sapiens/" /&gt;
  &lt;/datasource&gt;
&lt;/dazzle&gt;
</code></pre>
</div>

<p>Note that the database connection is not configured in the main
datasource element, but in a separate resource element. This is to allow
a single database connection to be shared between multiple DAS
datasources.</p>

<p>The org.ensembl.das.DatabaseHolder resource type reflects a database
connection. The dbURL property specifies the following information:</p>

<p><code class="highlighter-rouge">   * The type of database driver (MySQL)</code><br />
<code class="highlighter-rouge">   * The host name of the database server machine (e.g. noranti.derkholm.net)</code><br />
<code class="highlighter-rouge">   * The name of the Ensembl database (e.g. homo_sapiens_core_23_34e)</code></p>

<p>Having defined a database resource, configuring an EnsemblCoreReference
datasource is fairly standard. Note the coreHolder property, which
should be used to point to the appropriate database connection.</p>

<p>Once you are happy with your configuration, you should package
everything as a .WAR file and deploy it as normal for your servlet
container. If in doubt, consult the Dazzle deployment guide. As a first
test, use a web browser to view the Dazzle status page, which will
typically be:</p>

<p><code class="highlighter-rouge">   </code><a href="http://your-server:8080/das/"><code class="highlighter-rouge">http://your-server:8080/das/</code></a><code class="highlighter-rouge"> </code></p>

<p>You could then try viewing the data source in a DAS client.</p>

<h2 id="annotation-servers">Annotation servers</h2>

<h2 id="protein-das">Protein DAS</h2>

<h2 id="snps">SNPs</h2>

<h2 id="the-generic-seqfeature-plugin">The Generic SeqFeature plugin</h2>

<p>The GenericSeqFeatureSource plugin is a general-purpose Dazzle plugin
which allows features to be served up from an SQL database. The current
version does not depend directly on any Ensembl code or databases, but
it is distributed alongside the ensembl-das plugins described above, and
has historically been popular with Ensembl users</p>

<p>To serve a new dataset, create a MySQL database and add one or more
tables matching the following schema:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>CREATE TABLE my_feature ( 
  contig_id    varchar(40) NOT NULL default '', 
  start        int(10) NOT NULL default '0',
  end          int(10) NOT NULL default '0', 
  strand       int(2) NOT NULL default '0', 
  id           varchar(40) NOT NULL default '', 
  score        double(16,4) NOT NULL default '0.0000', 
  gff_feature  varchar(40) default NULL, 
  gff_source   varchar(40) default NULL,
  name         varchar(40) default NULL, 
  hstart       int(11) NOT NULL default '0',
  hend         int(11) NOT NULL default '0',
  hid          varchar(40) NOT NULL default'',
  evalue       varchar(40) default NULL,
  perc_id      int(10) default NULL, 
  phase        int(11) NOT NULL default '0', 
  end_phase    int(11) NOT NULL default '0',
  
  KEY id_contig(contig_id), 
  KEY id_pos(id,start,end) 
); 
</code></pre>
</div>

<p>A single database can contain many datasets, each in its own table. Each
dataset is served by a separate instance of the GenericSeqFeatureSource
plugin, but they can share a single pool of database connections,
therefore reducing the load on your database server if you want to serve
up a large number of datasets.</p>

<p>The most important columns are:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>contig_id    The name of the sequence to which a feature is attached (may actually be a contig, clone, or chromosome name).
start    The minimum sequence position covered by the feature
end    The maximum position covered by the feature
strand    The strand of the feature (should be -1, 0, or 1).
id    A unique ID for each feature
gff_feature    The "type" of the feature
gff_source    The "source" of the feature (e.g. the name of the program which performed the analysis)
</code></pre>
</div>

<p>For many purposes, the remaining fields can be left with their default
values.</p>

<p>A typical configuration looks like:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>&lt;dazzle xmlns="http://www.biojava.org/2000/dazzle"&gt;
  &lt;resource id="generic_db" jclass="org.ensembl.das.DatabaseHolder"&gt;
    &lt;string name="dbURL" value="jdbc:mysql://noranti.derkholm.net/generic_features" /&gt;
    &lt;string name="dbUser" value="ensembl" /&gt;
    &lt;string name="dbPass" value="xxx" /&gt;
  &lt;/resource&gt;


  &lt;datasource id="somefeatures" jclass="org.ensembl.das.GenericSeqFeatureSource"&gt;
    &lt;string name="name" value="Some features" /&gt;
    &lt;string name="description" value="Some features which I think are really interesting" /&gt;
    &lt;string name="version" value="1" /&gt;
    
    &lt;string name="mapMaster" value="http://noranti.derkholm.net/das/hsa2334/"&gt;

    &lt;string name="dbHolder" value="generic_db" /&gt;
    &lt;string name="tableName" value="my_feature" /&gt;
  &lt;/datasource&gt;
&lt;/dazzle&gt;
</code></pre>
</div>

<p>If the features in your database have unique IDs, it is easy to add
links to other web pages. For example:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>  &lt;datasource id="somefeatures" jclass="org.ensembl.das.GenericSeqFeatureSource"&gt;
    &lt;string name="name" value="Some features" /&gt;
    &lt;string name="description" value="Some features which I think are really interesting" /&gt;
    &lt;string name="version" value="1" /&gt;
    
    &lt;string name="mapMaster" value="http://noranti.derkholm.net/das/hsa2334/"&gt;

    &lt;string name="dbHolder" value="generic_db" /&gt;
    &lt;string name="tableName" value="my_feature" /&gt;
    
    &lt;map name="uriPatterns"&gt;
      &lt;string name="test_link" value="http://www.example.org/exciting_features.jsp?id=####;format=table"&gt;
    &lt;/map&gt;
  &lt;/datasource&gt;
</code></pre>
</div>

<p>It is possible to provide several links for each feature, so long as you
give them unique names. For each feature, the #### string in the
pattern is replaced by the feature ID from the database.</p>
