<h2 id="biojava-at-gsoc-introduction">BioJava at GSoC Introduction</h2>

<p>BioJava is applying to take part in this year’s Google Summer of Code
again as part of the OBF - the Open Bioinformatics Foundation. See <a href="http://www.open-bio.org/wiki/Google_Summer_of_Code">this
year’s GSoC page of the
OBF</a>.</p>

<p>If you want to propose a project, have a look at the <biojava:modules>
page, for areas which are currently under development. Also take a look
at the [Feature Requests](BioJava3_Feature_Requests "wikilink") page.
There are also some ideas from last year at
[Google\_Summer\_of\_Code\_2012](Google_Summer_of_Code_2012 "wikilink")</biojava:modules></p>

<p>Please read the <a href="http://www.open-bio.org/wiki/Google_Summer_of_Code">GSoC page at the Open Bioinformatics
Foundation</a> and the
<a href="http://code.google.com/soc">main Google Summer of Code page</a> for more
details about the program.</p>

<h3 id="mentor-list">Mentor List</h3>

<p>The following developers are possible Mentors for 2013:</p>

<ul>
  <li>Peter Troshin</li>
  <li>Spencer Bliven</li>
  <li>Michael Heuer</li>
  <li>Andreas Prlic</li>
</ul>

<h3 id="project-proposals">Project Proposals</h3>

<hr />

<h4 id="port-the-biojava-1-or-2-functionality-to-biojava-3">Port the BioJava 1 or 2 functionality to BioJava 3</h4>

<p>As you might have noticed some functionality present in BioJava 1/2 is
missing from the BioJava 3. This is not because this functionality is
obsolete or not needed; this is because nobody had time to refactor it
to work in the BioJava 3.</p>

<p>So your challenge is to identify such functionality and write a proposal
where you specify</p>

<ul>
  <li>What functionality you are going to port</li>
  <li>How you are going to do that (e.g. what needs changing)</li>
</ul>

<p>This project is be suitable to a confident Java developer.</p>

<p>Please send your proposals to the BioJava dev mailing list early, so we
can discuss them in details.</p>

<p>P.S. Actually you are not limited to the old versions of BioJava, there
are plenty of small little known Java projects in the Bioinformatics
field which can be of interest to a wider user community. You can
integrate them as a whole or cannibalize them and extract something
useful on its own. You are more than welcome to optimize and improve the
code while porting it to BioJava as you see fit.</p>

<p><strong>Possible Mentors</strong></p>

<p>Peter Troshin 2nd mentor : not yet assigned</p>

<hr />

<h4 id="improve-structural-alignment-datastructures-to-support-topology-independent-alignments">Improve structural alignment datastructures to support topology-independent alignments</h4>

<p>Rationale</p>

<p>BioJava contains a number of algorithms for aligning protein structures.
In the most general case, an alignment consists of a mapping between
residues of two (or more) proteins. However, for historical and
performance reasons alignments are stored as linear, sorted arrays. This
makes it difficult to express cases where the order of aligned residues
differs between the two proteins. For instance, storing the following
alignment requires some creative work-arounds:</p>

<p><code class="highlighter-rouge"> 123456</code><br />
<code class="highlighter-rouge"> 456123</code></p>

<p>Additionally, the class to store structural alignments (AFPChain)
contains a number of unneccessary, poorly documented, or
algorithm-specific parameters which should be removed or refactored.</p>

<p>Approach &amp; Goals</p>

<p>Your challenge is to propose and implement a data structure for storing
structure alignments which</p>

<ul>
  <li>Is flexible enough to store topology-independent alignments</li>
  <li>Efficiently utilizes memory</li>
  <li>Has good performance for common tasks</li>
</ul>

<p>Difficulty and needed skills</p>

<p>Moderate technical difficulty, but requires strong planning and abstract
thinking.</p>

<p>This project requires an understanding of basic data structures and
performance considerations. A successful proposal should consider not
only the new data structure, but also suggest a plan for integrating it
into existing methods, particularly in the biojava3-structure and
biojava3-structure-gui modules.</p>

<p>Possible Mentors</p>

<p>Spencer Bliven 2nd mentor: not yet assigned</p>

<hr />

<h4 id="topology-diagrams-of-protein-structures">Topology Diagrams of Protein Structures</h4>

<p><strong>Rationale</strong></p>

<p>Topology diagrams are useful for visualising the arrangement and
connectivity of secondary structure elements of proteins. We are
currently not aware of an easy to use Java implementation of software
for drawing such diagrams.</p>

<p><strong>Approach &amp; Goals</strong></p>

<p>The goal of this project would be to use the available tools from
biojava (load structures, define secondary structure) and implement a
layout algorithm that would arrange a representation of secondary
structure elements in a way so it can be easily used for various
visualisation libraries. Depending on the speed of progress a
visualisation layer could be added on top of this (e.g HTML5 vector
graphics, JPanel, etc.).</p>

<p><strong>Difficulty and needed skills</strong></p>

<p>This project requires some algorithmic knowledge for developing the
layout-algorithm for the secondary structure elements.</p>

<p><strong>Possible Mentors</strong></p>

<p>Andreas Prlic 2nd mentor: not yet assigned</p>

<hr />

<h4 id="sequence-variation">Sequence Variation</h4>

<p><strong>Rationale</strong></p>

<p>Several similar file specifications exist for dealing with sequence
variation, including:</p>

<p>VCF (Variant Call Format) is a text file format used by the 1000 Genomes
project and others for representing variation against a reference
sequence.</p>

<p><a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41</a></p>

<p>The Genome Variation Format (GVF) is a text file format for describing
sequence variants at nucleotide resolution relative to a reference
genome. GVF is a type of GFF3 file with additional pragmas and
attributes specified.</p>

<p><a href="http://www.sequenceontology.org/resources/gvf_1.02.html">http://www.sequenceontology.org/resources/gvf_1.02.html</a>
<a href="http://www.sequenceontology.org/resources/gff3.html">http://www.sequenceontology.org/resources/gff3.html</a></p>

<p>Some support for these file specifications is already present in various
bioinformatics libraries (and in fact biojava3 already provides GFF3
support); it would be desireable to pull these together behind a set of
common APIs in biojava3.</p>

<p><strong>Approach &amp; Goals</strong></p>

<ul>
  <li>Consider existing open source VCF and GVF implementations (<a href="http://www.broadinstitute.org/gatk/">Genotype
Analysis Toolkit, GATK</a>,
<a href="http://vcftools.sourceforge.net/">VCFTools</a>,
<a href="http://picard.sourceforge.net/">Picard</a>,
<a href="https://github.com/srynobio/GVF-Parser">GVF-Parser</a>, etc.)</li>
  <li>Design APIs for common entities (Allele, Genotype, Haplotype, etc.)</li>
  <li>Create adaptors to third party implementations or implement support
directly in Biojava3</li>
</ul>

<p><strong>Difficulty and needed skills</strong></p>

<p>Moderate difficulty.</p>

<p>Strength in API design, the ability to learn from existing codebases,
and experience with Java and other languages (i.e. Perl and Python) will
be necessary.</p>

<p><strong>Possible Mentors</strong></p>

<p>Michael Heuer 2nd mentor: not yet assigned</p>

<h2 id="previous-years">Previous Years</h2>

<p><a href="Google Summer of Code 2012" title="wikilink">Google Summer of Code 2012</a></p>

<p><a href="Google Summer of Code 2011" title="wikilink">Google Summer of Code 2011</a></p>

<p><a href="Google Summer of Code 2010" title="wikilink">Google Summer of Code 2010</a></p>
