<p><strong>Amino Acids Physicochemical Properties Calculation</strong></p>

<p><em><a href="Google Summer of Code" title="wikilink">Google Summer of Code</a> Project by
<a href="Chuan Hock Koh" title="wikilink">Chuan Hock Koh</a></em></p>

<p><em>Mentored by <a href="http://biojava.org/wiki/User:Ptroshin">Peter Troshin</a></em></p>

<p><em>Co-mentored by <a href="Andreas Prlic" title="wikilink">Andreas Prlic</a></em></p>

<p>The calculation of physicochemical properties for biopolymers is an
important tool in the arsenal of molecular biologist. Theoretically
calculated quantities like extinction coefficients, isoelectric points,
hydrophobicities and instability indices are useful guides as to how a
molecule behaves in an experiment. Many tools for calculating these
properties exist, including widely used open-source implementations in
EMBOSS and BioPerl, but only some are currently available in BioJava3.
The aim of this project is to port or produce new implementations of
standard algorithms for a range of calculations within BioJava3 and make
them available in various levels (APIs, executable and web service).</p>

<h2 id="properties">Properties</h2>

<p>Listing the properties that could be considered for implementing. Mainly
based on the original proposal, PROFEAT and Sirius PSB. We welcome more
ideas for additional/removal of properties. Note: The semi-circle
bracket () after the name of a property indicates the number of values
this property will generate.</p>

<h3 id="originally-proposed-properties">Originally proposed properties</h3>

<ul>
  <li>Molecular weight (1)</li>
  <li>Extinction coefficient (2 – Assumes Cys are reduced and assumes Cys
form cystines)</li>
  <li>Instability index (1)</li>
  <li>Aliphatic index (1)</li>
  <li>Grand Average of Hydropathy (1)</li>
  <li>Isoelectric point (1)</li>
  <li>Number of amino acids in the protein (1)</li>
</ul>

<h3 id="other-obvious-properties-to-consider">Other obvious properties to consider</h3>

<ul>
  <li>Amino acid composition (20)</li>
  <li>Dipeptide acid compostion (400)</li>
  <li>Net Charge (1)</li>
</ul>

<h3 id="profeat-properties">PROFEAT properties</h3>

<p>Sequences are first transformed into representation of different
attribute and each attribute has 3 groups. In PROFEAT, there are 7
different attributes.</p>

<ul>
  <li>Hydrophobicity (Polar, Neutral, Hydrophobicity)</li>
  <li>Normalized van der Waals volume (Range 0 – 2.78, 2.95 – 4.0, 4.03 –
8.08)</li>
  <li>Polarity (Value 4.9 – 6.2, 8.0 – 9.2, 10.4 – 13.0)</li>
  <li>Polarizability (Value 0 – 1.08, 0.128 – 0.186, 0.219 – 0.409)</li>
  <li>Charge (Positive, Neutral, Negative)</li>
  <li>Secondary structure (Helix, Strand, Coil)</li>
  <li>Solvent accessibility (Buried, Exposed, Intermediate)</li>
</ul>

<p>After transformation, PROFEAT computes three type of properties</p>

<ul>
  <li>Composition (3 * 7 = 21)</li>
  <li>Transition (3 * 7 = 21) – [1&lt;-&gt;2, 1&lt;-&gt;3, 2&lt;-&gt;3]
    <ul>
      <li>A transition from class 1 to 2 is the percent frequency with
which 1 is followed by 2 or 2 is followed by 1 in the
transformed sequence</li>
    </ul>
  </li>
  <li>Distribution (3 * 5 * 7 = 105)
    <ul>
      <li>It measures the position percent in the whole sequence of
encoded residue for first residue, 25%, 50%, 75%, 100%.</li>
    </ul>
  </li>
</ul>

<h3 id="other-profeat-properties">Other PROFEAT properties</h3>

<p>PROFEAT also computes a series of properties based on autocorrelations
and sequence-order which are harder to comprehend. Please see PROFEAT
<a href="http://jing.cz3.nus.edu.sg/prof/prof_manual.pdf">manual</a> for details.</p>

<ul>
  <li>Normalized Moreau-Broto autocorrelation (240)</li>
  <li>Moran autocorrelation (240)</li>
  <li>Geary autocorrelation (240)</li>
  <li>Sequence-order-coupling number (60)</li>
  <li>Quasi-sequence-order descriptors (100)</li>
</ul>

<h3 id="sirius-psb-properties">Sirius PSB properties</h3>

<p>Likewise, Sirius PSB transforms sequences accordingly to their
respective grouping based on the different attributes. However, it
computes different properties from them.</p>

<ul>
  <li>Number of AA in (or Size) the maximum region (1 * 7 = 7)</li>
  <li>Value of the maximum region (1 * 7 = 7)</li>
  <li>Location of the maximum region (2 * 7 = 14) – start and end
    <ul>
      <li>Note: Maximum region is the region that has the highest value of
a particular grouping. E.g. the most hydrophobic region in the
sequence</li>
    </ul>
  </li>
  <li>Number of regions with at least size X (1 * 7 = 7)</li>
  <li>Number of regions with at least value Y (1 * 7 = 7)</li>
</ul>

<h2 id="timeline">Timeline</h2>

<h3 id="phase-1-7-weeks">Phase 1 (7 Weeks)</h3>

<p><em>April 25 to June 12</em> - Deliverables: APIs</p>

<ul>
  <li>Research and finalize with mentors the properties to be included</li>
  <li>Start coding of APIs while writing the documentation concurrently</li>
</ul>

<h3 id="phase-2-3-weeks">Phase 2 (3 Weeks)</h3>

<p><em>June 13 to July 3</em> - Deliverables: Tested APIs and Documentations</p>

<ul>
  <li>Write test cases to ensure accuracy of the APIs</li>
  <li>Clean up codes and documentation for readability</li>
  <li>Update BioJava wiki page and write tutorial for using APIs</li>
  <li>Check with mentor on areas of improvement</li>
</ul>

<h3 id="phase-3-2-weeks">Phase 3 (2 Weeks)</h3>

<p><em>July 4 - July 17</em> - Deliverables: Executable</p>

<ul>
  <li>Develop, test and document command line executable for generating
properties</li>
  <li>Update BioJava wiki page and write tutorial for using Executables</li>
</ul>

<h3 id="phase-4-2-weeks">Phase 4 (2 Weeks)</h3>

<p><em>July 18 - July 31</em> - Deliverables: SOAP web services</p>

<ul>
  <li>Expose APIs as SOAP web services within the JABAWS framework</li>
  <li>Test and document the services</li>
</ul>

<h3 id="phase-5-3-weeks">Phase 5 (3 Weeks)</h3>

<p><em>August 1 to August 21</em></p>

<ul>
  <li>Final touch-up based on mentors’ feedback</li>
</ul>

<h2 id="progress-log">Progress Log</h2>

<h3 id="april-26---may-4-week-1">April 26 - May 4 (Week 1)</h3>

<ul>
  <li>Set up project page in BioJava Wiki</li>
  <li>Arranged to have weekly skype meeting every Tuesday 1600 London time
(GMT +1) with both Mentors, Peter and Andreas.</li>
  <li>Suggests properties to consider for this project.</li>
  <li>Decision has been made to start working on the originally proposed
APIs with only the addition of amino acid composition. More
properties will be considered if there is time left after the
completion of these properties.</li>
  <li>Created a BioJava SVN account and successfully checkout biojava3 as
Maven Projects.</li>
</ul>

<h3 id="may-5---may-12-week-2">May 5 - May 12 (Week 2)</h3>

<ul>
  <li>Completed initialization. Ready to start coding.</li>
  <li>Created a user page for myself on BioJava Wiki.</li>
  <li>Introduced to the biojava developer mailing list.</li>
  <li>Request for input and suggestion from biojava developer mailing list
on this project.</li>
  <li>Committed the first draft of the interface class for the basic
physicochemical properties.</li>
  <li>Updated interface class for basic physicochemical properties based
on Mentors’ feedback.</li>
  <li>Learned and better understand how to use Maven on Eclipse.</li>
</ul>

<h3 id="may-13---may-20-week-3">May 13 - May 20 (Week 3)</h3>

<ul>
  <li>Gathered websites/publications needed for constraints.</li>
  <li>Refactoring packages and classes for more readability.</li>
  <li>Learning to properly document work according to set standard.</li>
  <li>Finalizing the approach for implementation.</li>
  <li>Initial implementation of a few properties.</li>
</ul>

<h3 id="may-21---may-27-week-4">May 21 - May 27 (Week 4)</h3>

<ul>
  <li>Implemented all basic properties.</li>
  <li>Verified them manually based on three web tools.
    <ul>
      <li><a href="http://web.expasy.org/protparam/">http://web.expasy.org/protparam/</a></li>
      <li><a href="http://web.expasy.org/compute_pi/">http://web.expasy.org/compute_pi/</a></li>
      <li><a href="http://www.innovagen.se/custom-peptide-synthesis/peptide-property-calculator/peptide-property-calculator.asp">http://www.innovagen.se/custom-peptide-synthesis/peptide-property-calculator/peptide-property-calculator.asp</a></li>
    </ul>
  </li>
  <li>Implemented adaptor methods for all basic properties.</li>
  <li>Basic properties includes the following
    <ul>
      <li>Molecular Weight</li>
      <li>Extinction Coefficient</li>
      <li>Instability Index</li>
      <li>Apliphatic Index</li>
      <li>Average Hydropathy</li>
      <li>Isoelectric Point</li>
      <li>Net Charge</li>
      <li>Amino Acid Composition</li>
    </ul>
  </li>
</ul>

<h3 id="may-28---june-2-week-5">May 28 - June 2 (Week 5)</h3>

<ul>
  <li>Added several additional adaptor methods to make usage more user
friendly.
    <ul>
      <li>public static final double getEnrichment(String sequence, char
aminoAcidCode)</li>
      <li>public static final double getEnrichment(String sequence, String
aminoAcidCode)</li>
      <li>public static final Map&lt;String, Double&gt;
getAACompositionString(String sequence)</li>
      <li>public static final Map&lt;Character, Double&gt;
getAACompositionChar(String sequence)</li>
    </ul>
  </li>
  <li>Implemented an additional properties, absorbance.</li>
  <li>Also implemented adaptor methods for making computation of
absorbance more user friendly.</li>
  <li>Implemented JUnit test cases for all properties.
    <ul>
      <li>Based on <a href="http://au.expasy.org/cgi-bin/protparam">http://au.expasy.org/cgi-bin/protparam</a> and
<a href="http://www.innovagen.se/custom-peptide-synthesis/peptide-property-calculator/peptide-property-calculator-notes.asp">http://www.innovagen.se/custom-peptide-synthesis/peptide-property-calculator/peptide-property-calculator-notes.asp</a></li>
      <li>However, many does not pass assertEquals due to precision
problem.</li>
    </ul>
  </li>
  <li>Added Javadoc for Constraints.java and PeptideProperties.java</li>
</ul>

<h3 id="june-3---june-10-week-6">June 3 - June 10 (Week 6)</h3>

<ul>
  <li>Removed Java docs from repository.</li>
  <li>Slightly modified current adaptor methods to allow user to decide on
the number of decimal places required for their usage.</li>
  <li>Added adaptor methods with default decimal places.</li>
  <li>All JUnit test cases passes now.</li>
  <li>Added a new class, Utils under aaproperties with main objective to
provide some utility methods that does not belong to any other
available classes.</li>
  <li>Added the following methods into Utils.
    <ul>
      <li>public final static double roundToDecimals(double, int) - this
method enable the rounding of value to desired decimal places.</li>
      <li>public final static boolean
doesSequenceContainInvalidChar(String) - this method checks if
the sequence contains invalid character. Return true if invalid
character exists, false otherwise.
        <ul>
          <li>Any character outside of the standard 20 amino acid codes
are considered to be invalid.</li>
        </ul>
      </li>
      <li>public final static String cleanSequence(String) - this method
replaces any invalid character in sequence with ‘-‘.</li>
    </ul>
  </li>
  <li>Added more test cases when invalid input are given to properties
generating methods.
    <ul>
      <li>Adjusted methods to handle such cases.</li>
      <li>If input protein sequence is null, a null exception will be
thrown.</li>
    </ul>
  </li>
</ul>

<h3 id="june-11---june-18-week-7">June 11 - June 18 (Week 7)</h3>

<ul>
  <li>Remove all catch exception in test cases and replace them with
expected = NullPointerException.class.</li>
  <li>Added several test methods to accommodate this.</li>
  <li>Change all adaptor methods to return full precision by default.</li>
  <li>Only fix the precision for the test cases to ensure they will not
fail due to precision problem.</li>
  <li>Created a new package - org.biojava3.aaproperties.xml
    <ul>
      <li>Created two classes under this package - IsotopeType.java and
IsotopeTable.java.</li>
      <li>The purpose of these package and classes are initial effort to
learn and utilize JAXB for properties’ value definition in
particular for molecular weight.</li>
    </ul>
  </li>
  <li>Populated two elements based on Peter’s Element class.</li>
  <li>Create a ElementTester to test the generated xml and its computed
standard atomic weight.</li>
  <li>Also created several classes to prepare for the generation of
PROFEAT properties.</li>
  <li>Finished implementing APIs for generating PROFEAT properties and
also added adaptor method for the ease of utilizing them</li>
</ul>

<h3 id="june-19---june-23-week-8">June 19 - June 23 (Week 8)</h3>

<ul>
  <li>Removed Utils.roundToDecimals from PeptidePropertiesImpl.java</li>
  <li>Embedded all data from the element information website</li>
  <li>Create Test Cases to ensure data are properly embedded</li>
  <li>Generated XML files for Element Mass and Amino Acid Composition.</li>
  <li>Able to read in XML files for Element Mass and Amino Acid
Composition into Java via JAXB.</li>
  <li>Create new methods to enable the usage of user specified xml file to
compute molecular mass</li>
  <li>Wrote test cases for these newly created methods</li>
</ul>

<h3 id="june-24---june-29-week-9">June 24 - June 29 (Week 9)</h3>

<ul>
  <li>Created ValidationEventHandler which allows the xml file to be
validated</li>
  <li>Validates xml files before they are unmarshal</li>
  <li>Added an additional method to auto find a elementMass file in
default location</li>
  <li>Generated the schema for aminoacidcomposition.xsd</li>
  <li>Change the XML file structure such that the element weight is
directly recorded instead of computed</li>
  <li>Remove all the test cases for the computation of the abundance.</li>
  <li>Successfully generated an XML schema for ElementMass.xsd and
AminoAcidComposition.xsd</li>
  <li>Remove abundance from ElementMass.xml</li>
  <li>Created in Cookbook with short examples showing how to utilize the
APIs</li>
  <li>How each properties are computed are also given (excluding
Instability Index and Net Charge due to syntax technicalities)</li>
  <li>Also added a test class in Maven to ensure that the short examples
given are error-free</li>
</ul>

<h3 id="june-30---july-7-week-10">June 30 - July 7 (Week 10)</h3>

<ul>
  <li>Added several methods in different classes to enable the usable of
methods that does not need to specify location of ElementMass.xml</li>
  <li>Improve the CookbookTester.java to combine shortExample 4 and 5 plus
utilizing methods that do not need to explicitly specify
ElementMass.xml in shortExamples 2 and 3</li>
  <li>Added annotations for AminoAcidComposition.java and Element.java.</li>
  <li>Simplified MolecularWeight.xml and ElementMass.xml.</li>
  <li>Work around JAXB inability to handle Map&lt;K, V&gt; and used List<e>
instead so as to made the XML files annotated to requirements.</e></li>
  <li>Adjust the XML requirements and made attributes as optional if
possible</li>
  <li>Checked that all Elements and Isotopes used in
AminoAcidComposition.xml can be found in ElementMass.xml</li>
  <li>Add CookBook pages on how Net Charge and Instability Index are
computed.</li>
  <li>Defined the standard and minimum XML files in the CookBook</li>
</ul>

<h3 id="july-8---july-13-week-11">July 8 - July 13 (Week 11)</h3>

<ul>
  <li>Clean up the codes on Computing Molecular Weights</li>
  <li>Change the Minimum XML files to more compact</li>
  <li>Change the naming of several XML attributes to more intuitive and
generic terms</li>
  <li>Created two advanced XML files to demonstrate how to define modified
amino acids such as radioactive labelled and phosphorylation.</li>
  <li>Did a simple test case for these advanced XML files as well.</li>
  <li>Check the attribute values to ensure that symbols are single
character and do not repeat themselves.</li>
  <li>Also check to ensure that element/isotope names define are also
found in ElementMass XML file.</li>
  <li>Mass and counts are also checked to make sure that they are &gt; 0.</li>
  <li>Emailed the BioJava community reporting the current status of the
project and also looking for feedback and suggestions on the
project.</li>
  <li>Implemented the first draft of the command prompt executable.</li>
</ul>

<h3 id="july-14---july-21-week-12">July 14 - July 21 (Week 12)</h3>

<ul>
  <li>Created a new class ModifiedAminoAcidCompoundSet which enables any
symbols to be specified in XML file to represent amino acids and
modified amino acids.</li>
  <li>Added three examples on how to run the tool.</li>
  <li>Change output file to optional and default to standard output.</li>
  <li>Set default decimal place to 4. Possible to change with -d argument.</li>
  <li>Change to use BioJava FASTA reader.</li>
  <li>Created CommandPromptTester.java to do testing of CommandPrompt.java</li>
  <li>Improved Help - Follows unix “man” style.</li>
</ul>

<h3 id="july-22---july-29-week-13">July 22 - July 29 (Week 13)</h3>

<ul>
  <li>Updated the cookbook on using extended XML file and
ModifiedAACompoundSet</li>
  <li>Split up the CommandPrompt run to be more “unit” style</li>
  <li>Had always assumed that cases are ignored, hence had to change many
method to consider case difference</li>
  <li>Introduced another class CaseFreeAminoAcidCompoundSet to ignore
cases</li>
  <li>By default, set ignore case. If they provide the XML file, then
cases will be differentiated</li>
  <li>Update the cookbook on Command Prompt</li>
</ul>

<h3 id="july-30---august-4-week-14">July 30 - August 4 (Week 14)</h3>

<ul>
  <li>Only consider cases difference for Molecular Weight. Revert all
cases consideration for other properties.</li>
  <li>Managed to link up with JABAWS via SVN. Could only checkout but not
able to commit.</li>
  <li>Able to commit after some configuration efforts.</li>
  <li>Committed initial attempt to provide AAProperties as a service in
JABAWS.</li>
  <li>Added AAProperties.jar into jabaws/binaries/windows.</li>
  <li>Added AAProperties description to conf/Executable.properties.</li>
  <li>Created AAPropertiesLimits.xml and AAPropertiesParameters.xml.</li>
  <li>Created AAProperties.java under compbio.runner.sequence and extends
SkeletalExecutable<aaproperties></aaproperties></li>
</ul>

<h3 id="august-5---august-12-week-15">August 5 - August 12 (Week 15)</h3>

<ul>
  <li>Wrote the parser for AAProperties.jar under JABAWS.</li>
  <li>Prepared test case for the parser.</li>
  <li>Uploaded a sample output of AAProperties.jar for testing purposes.</li>
  <li>Change the return type of parseAAProp to ScoreManager</li>
  <li>Adjust the configuration of pom.xml to generate a jar file with
org.biojava3.aaproperties.CommandPrompt as the main class. However,
need to rename it to AAProperties.jar</li>
  <li>Added SequenceName in the output of AAProperties.jar (in the first
column)</li>
  <li>Modify the SequenceUtil.AAprop result parser to cater for the change
above in the AAProperties.jar output</li>
  <li>Managed to rename in the configuration of pom.xml to generate a jar
file with name that includes AAProperties.</li>
</ul>

<h3 id="august-13---august-19-week-16">August 13 - August 19 (Week 16)</h3>

<ul>
  <li>Completed the integration of AAProperties.jar with JABAWS</li>
  <li>Checked all links on
<a href="http://biojava.org/wiki/GSoC:AAPropertiesComputation">http://biojava.org/wiki/GSoC:AAPropertiesComputation</a></li>
  <li>Rectified two links <a href="http://web.expasy.org/protparam/">http://web.expasy.org/protparam/</a> and
<a href="http://web.expasy.org/compute_pi/">http://web.expasy.org/compute_pi/</a></li>
  <li>Checked all links on <a href="http://biojava.org/wiki/BioJava:CookBook">http://biojava.org/wiki/BioJava:CookBook</a>
under Physicochemical Properties Computation</li>
  <li>Linked up the executable jar file on maven to Cookbook</li>
  <li>Simplified the section under Physicochemical Properties Computation
on <a href="http://biojava.org/wiki/BioJava:CookBook">http://biojava.org/wiki/BioJava:CookBook</a></li>
  <li>Added dummy id to be allowed in the xml file of
AminoAcidComposition.
    <ul>
      <li>It will be used in future as an option to compute the mass of
amino acid instead of based only on elements.</li>
    </ul>
  </li>
</ul>

<h3 id="august-20---current-week-17">August 20 - Current (Week 17)</h3>

<ul>
  <li>Shifted Profeat related classes to new packages called
aaproperties.profeat and aaproperties.profeat.convertor</li>
  <li>Added package-info.java for all packages</li>
  <li>Create a test class to test PROFEAT properties</li>
  <li>Added documentation in Cookbook on how to use PROFEAT properties API</li>
  <li>Update the BioJava community on what is achieved for this Google
Summer of Code project</li>
</ul>

<h2 id="skype-call-notes">Skype call notes</h2>

<p>Weekly skype calls are on Thursday 08 AM PST</p>

<p><code class="highlighter-rouge"> </code><a href="aapropsSkype_20110505" title="wikilink"><code class="highlighter-rouge">May</code> <code class="highlighter-rouge">5th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110512" title="wikilink"><code class="highlighter-rouge">May</code> <code class="highlighter-rouge">12th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110519" title="wikilink"><code class="highlighter-rouge">May</code> <code class="highlighter-rouge">19th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110526" title="wikilink"><code class="highlighter-rouge">May</code> <code class="highlighter-rouge">26th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110602" title="wikilink"><code class="highlighter-rouge">June</code> <code class="highlighter-rouge">2nd</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110609" title="wikilink"><code class="highlighter-rouge">June</code> <code class="highlighter-rouge">9th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110616" title="wikilink"><code class="highlighter-rouge">June</code> <code class="highlighter-rouge">16th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110623" title="wikilink"><code class="highlighter-rouge">June</code> <code class="highlighter-rouge">23th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110629" title="wikilink"><code class="highlighter-rouge">June</code> <code class="highlighter-rouge">29th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110706" title="wikilink"><code class="highlighter-rouge">July</code> <code class="highlighter-rouge">6th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110713" title="wikilink"><code class="highlighter-rouge">July</code> <code class="highlighter-rouge">13th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110721" title="wikilink"><code class="highlighter-rouge">July</code> <code class="highlighter-rouge">21st</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110728" title="wikilink"><code class="highlighter-rouge">July</code> <code class="highlighter-rouge">28th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110804" title="wikilink"><code class="highlighter-rouge">August</code> <code class="highlighter-rouge">4th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110812" title="wikilink"><code class="highlighter-rouge">August</code> <code class="highlighter-rouge">12th</code></a><br />
<code class="highlighter-rouge"> </code><a href="aapropsSkype_20110817" title="wikilink"><code class="highlighter-rouge">August</code> <code class="highlighter-rouge">17th</code></a></p>

<h2 id="references">References</h2>

<ul>
  <li><a href="http://bioinformatics.oxfordjournals.org/content/24/18/2096.abstract">BioJava</a> -
BioJava: an open-source framework for bioinformatics</li>
  <li><a href="http://www.worldscinet.com/jbcb/07/0706/S0219720009004436.html">SiriusPSB</a> -
Sirius PSB: A Generic System for Analysis of Biological Sequences</li>
  <li><a href="http://nar.oxfordjournals.org/content/34/suppl_2/W32.abstract">PROFEAT</a> -
PROFEAT: a web server for computing structural and physicochemical
features of proteins and peptides from amino acid sequence</li>
</ul>

<h2 id="comments">Comments</h2>

<p><em>Please add comments here…</em></p>
