<p><strong>BioJava Packages for Identification, Classification, and Visualization
of Posttranslational Modification of Proteins</strong></p>

<ul>
  <li>Student: <a href="Jianjiong Gao" title="wikilink">Jianjiong Gao</a></li>
  <li>Mentor: <a href="http://www.linkedin.com/in/peterrose">Peter Rose</a></li>
  <li>Co-menter: <a href="Andreas Prlic" title="wikilink">Andreas Prlic</a></li>
</ul>

<p>The original proposal is
<a href="http://docs.google.com/fileview?id=0B6z8AlO4d-qMMzAzZWFmYzQtMWRlNy00Y2Y2LThmZjktNjFhYzNlOTg3ZDEz&amp;hl=en">here</a></p>

<h2 id="background">Background</h2>

<p><a href="http://en.wikipedia.org/wiki/Posttranslational_modification">Posttranslational modifications
(PTM)</a> are
modifications to proteins after protein biosynthesis, which play a key
role in many cellular processes such as cellular differentiation,
protein degradation, signaling and regulatory processes, regulation of
gene expression, and protein-protein interactions. PTMs are present in
the 3D structures in the <a href="http://www.pdb.org/">Protein Data Bank</a>. It is
of vast interst to query and classify proteins by their PTMs. PTMs can
be classified in multiple ways. From an implementation perspective we
need to distinguish 3 major cases:</p>

<ul>
  <li>Case 1: Attachment of a chemical group (i.e. glycan)</li>
  <li>Case 2: Chemical modification of an amino acid (i.e. hydroxyproline)</li>
  <li>Case 3: Cross-linking (i.e. disulfide bonds, iso-peptide bonds)</li>
</ul>

<h2 id="major-project-contributions">Major Project Contributions</h2>

<p>The goal of this project is to develop BioJava packages that identify,
classify and visualize PTMs. Major deliverables of this project include:</p>

<ul>
  <li>A BioJava package to identify PTMs in a 3D protein structure (.pdb
or .cif file).</li>
  <li>A BioJava package to generate sequence diagrams with an option to
add PTM annotations.</li>
  <li>A BioJava package to generate 2D tree images of carbohydrate
(glycan) structures.</li>
</ul>

<p>Immediate applications of this project would be on Protein Data Bank
website:</p>

<ul>
  <li>Making PTMs searchable on PDB.</li>
  <li>Updating the sequence diagram on PDB with an option to display PTM
annotations.</li>
  <li>Listing PTMs in PDB ProteinWorkshop.</li>
</ul>

<p>This project will be beneficial to the research community by
facilitating structural analysis on PTMs and hence reinforcing our
understanding on the mechanisms of various PTMs.</p>

<h2 id="tasks-and-implementation">Tasks and Implementation</h2>

<h3 id="task1-making-a-list-of-ptm-types">Task1. Making a list of PTM types</h3>

<h4 id="resources">Resources</h4>

<ul>
  <li>UniProt
    <ul>
      <li><a href="http://www.uniprot.org/docs/ptmlist.txt">http://www.uniprot.org/docs/ptmlist.txt</a></li>
      <li><a href="http://www.uniprot.org/manual/mod_res">http://www.uniprot.org/manual/mod_res</a></li>
      <li>Paper: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15174124">Annotation of post-translational modifications in the
Swiss-Prot knowledge
base</a></li>
      <li>Paper: <a href="http://www.ncbi.nlm.nih.gov/pubmed/11680872">Annotation of glycoproteins in the SWISS-PROT
database</a></li>
      <li>Three types of PTM (FT): MOD_RES, CROSSLNK, LIPID</li>
      <li>Contains DR records that point to other PTM databases (i.e.
RESID)</li>
    </ul>
  </li>
  <li>RESID database
    <ul>
      <li><a href="http://www.ebi.ac.uk/RESID/">http://www.ebi.ac.uk/RESID/</a></li>
      <li><a href="http://srs.ebi.ac.uk/srsbin/cgi-bin/wgetz">http://srs.ebi.ac.uk/srsbin/cgi-bin/wgetz</a>?-page+LibInfo+-id+2F9eD1VPLvp+-lib+RESID</li>
      <li>Paper: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15174122">The RESID Database of Protein Modifications as a
resource and annotation
tool</a></li>
      <li>Has a link to a PDB file with PTM structure: Xref PDBHET</li>
    </ul>
  </li>
  <li>PSI-MOD
    <ul>
      <li><a href="http://psidev.sourceforge.net/mod/">http://psidev.sourceforge.net/mod/</a></li>
      <li>Paper: <a href="http://www.nature.com/nbt/journal/v26/n8/full/nbt0808-864.html">The PSI-MOD community standard for representation of
protein modification
data</a></li>
    </ul>
  </li>
  <li>ExPASy
    <ul>
      <li><a href="http://ca.expasy.org/tools/findmod/findmod_masses.html">http://ca.expasy.org/tools/findmod/findmod_masses.html</a></li>
    </ul>
  </li>
  <li>Delta Mass
    <ul>
      <li><a href="http://www.abrf.org/index.cfm/dm.home">http://www.abrf.org/index.cfm/dm.home</a></li>
      <li>Paper: <a href="http://www.ncbi.nlm.nih.gov/pubmed/12610572">Proteomic analysis of post-translational
modifications</a></li>
    </ul>
  </li>
  <li>GlycoSciences
    <ul>
      <li><a href="http://www.glycosciences.de/">http://www.glycosciences.de/</a></li>
      <li><a href="http://www.glycosciences.de/sweetdb/start.php?action=explore_linucsid&amp;linucsid=22560&amp;show=65#mass">http://www.glycosciences.de/sweetdb/start.php?action=explore_linucsid&amp;linucsid=22560&amp;show=65#mass</a></li>
      <li><a href="http://www.glycosciences.de/sweetdb/start.php?action=explore_linucsid&amp;linucsid=297&amp;show=4129#crosslinks">http://www.glycosciences.de/sweetdb/start.php?action=explore_linucsid&amp;linucsid=297&amp;show=4129#crosslinks</a></li>
      <li>Paper: <a href="http://bib.oxfordjournals.org/cgi/content/short/5/2/164">Bioinformatics for glycomics: Status, methods,
requirements and
perspectives</a></li>
      <li>Paper: <a href="http://www.biomedcentral.com/1471-2105/5/69">pdb-care (PDB carbohydrate residue check): a program to
support annotation of complex carbohydrate structures in PDB
files</a></li>
      <li>Paper: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15010309">Data mining the protein data bank: automatic detection
and assignment of carbohydrate
structures</a></li>
    </ul>
  </li>
  <li><a href="http://glycores.ncifcrf.gov/intr/index.html">Glycobiology resources listed at
NCI-Frederick</a></li>
  <li><a href="http://erice2008.docking.org/vcourse/04wed/1130-Baker/Ted%20Baker.pdf">Ted Baker: Erice 2008 talk on Isopeptide
bonds</a></li>
</ul>

<h4 id="procedure-to-get-a-list-of-ptms">Procedure to get a list of PTMs</h4>

<ul>
  <li>Retrieve the <a href="http://www.wwpdb.org/ccd.html">Chemical Component
Dictrionary</a></li>
  <li>Remove all obsolete or ambiguous chemical components by only keeping
entries with
    <ul>
      <li>_chem_comp.pdbx_release_status REL</li>
      <li>_chem_comp.pdbx_ambiguous_flag N</li>
    </ul>
  </li>
  <li>Create two subsets of the CCD
    <ul>
      <li>A. All chemical components that are
        <ul>
          <li>not the 20 standard amino acids and</li>
          <li>do not have a _chem_comp.mon_nstd_parent_comp_id field
corresponding to one of the 20 standard amino acids.</li>
        </ul>
      </li>
      <li>B. All chemical components that have a
_chem_comp.mon_nstd_parent_comp_id field corresponding to
one of the 20 standard amino acids.</li>
    </ul>
  </li>
  <li>For case 1 (attachments), match the RESID NameXref PDBHET ID with
chemical component set A to get <a href="http://spreadsheets1.google.com/ccc?key=tCBmM83dURWTKEwM34MwRsg&amp;hl=en">a list of attachment
groups</a>.</li>
  <li>For case 2 (modified amino acids), match the RESID NameXref PDBHET
ID with set B to get <a href="http://spreadsheets1.google.com/ccc?key=tmS1y7VhfTIIrapme6kSUug&amp;hl=en">a list of modified amino acids that are
PTMs</a>.</li>
  <li><a href="http://spreadsheets0.google.com/ccc?key=tCC48SYpb9v0ehoel6PlnZg&amp;hl=en">A initial list of
PTMs</a>
were then manually selected.</li>
</ul>

<h3 id="task2-defining-data-representation-of-ptms">Task2. Defining data representation of PTMs</h3>

<ul>
  <li><a href="http://code.open-bio.org/svnweb/index.cgi/biojava/browse/biojava-live/trunk/biojava3-protmod/src/main/resources/org/biojava3/protmod">An XML
file</a>
is used to store the information of PTMs.</li>
</ul>

<!-- -->

<ul>
  <li>A Java class <em>ProteinModification</em> to store different types of PTMs.</li>
</ul>

<!-- -->

<ul>
  <li>Three interfaces representing three cases.
    <ul>
      <li>ModifiedResidue</li>
      <li>Attachment</li>
      <li>CrossLink</li>
    </ul>
  </li>
</ul>

<h3 id="task3-reading-and-parsing-3d-protein-structure-files-pdb-or-cif">Task3. Reading and parsing 3D protein structure files (.pdb or .cif)</h3>

<ul>
  <li>Utilizing org.biojava.bio.structure
(http://www.biojava.org/wiki/BioJava:CookBook:PDB:read).</li>
  <li>The code will be based on BioJava 3.</li>
</ul>

<h3 id="task4-identifying-ptms-in-3d-protein-structures">Task4. Identifying PTMs in 3D protein structures</h3>

<ul>
  <li>For case 1, finding the attachment points for PTMs.
    <ul>
      <li>Reading modifications in HETATM.</li>
      <li>Scanning possible atoms on possible amino acids to locating
attachment point of each PTM by checking if the distance between
an amino acid atom and the PTM is less than a threshold.
        <ul>
          <li>The threshold will be the sum of the covalent bond length
(i.e. the sum of <a href="http://en.wikipedia.org/wiki/Covalent_radius">covalent
radiuses</a> of
both atoms) plus a tolerance of error, say 0.4 Angstrom
(this need to be decided later after analyzing the data).</li>
          <li>For different types of PTMs, only limited types of amino
acids and atoms can be attached to. Thus, only those atoms
need to be scanned.</li>
          <li>If multiple atoms in multiple amino acids are within the
distance threshold to the PTM, choose the one with the
shortest distance.</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>For case 2, finding the modified amino acids.
    <ul>
      <li>Finding the corresponding 3-letter code of each PTM type in
<a href="http://www.wwpdb.org/ccd.html">Chemical Component Dictrionary</a>.
        <ul>
          <li>If _chem_comp.mon_nstd_parent_comp_id field
corresponds one of the 20 standard amino acids, then it is a
modified amino acid.</li>
          <li>3-letter code is contained in the _chem_comp.id field.</li>
        </ul>
      </li>
      <li>Identifying PTMs by parsing through the residues in the protein
chain using the 3-letter code.</li>
    </ul>
  </li>
  <li>For case 3, finding cross-linked amino acids.
    <ul>
      <li>Identifying cross-linked amino acids by looking for short
distances between amino acid pairs in a similar way as case 1.
        <ul>
          <li><a href="http://en.wikipedia.org/wiki/Disulfide_bond">Disulfide
bonds</a></li>
          <li><a href="http://en.wikipedia.org/wiki/Isopeptide_bond">Iso-peptide
bonds</a></li>
        </ul>
      </li>
    </ul>
  </li>
  <li>For both case 1 and 3, filtering out the close contacts that have
nothing to do with PTMs.
    <ul>
      <li>The filtering strategy will be developed after analyzing the
data.</li>
    </ul>
  </li>
</ul>

<h3 id="task5-representing-ptms-in-text">Task5. Representing PTMs in text</h3>

<ul>
  <li>For case 1: Attached PTM
    <ul>
      <li>Two atoms that link between the amino acid and the attachment
        <ul>
          <li>3-letter code</li>
          <li>Chain id</li>
          <li>Residue number</li>
          <li>Atom name</li>
          <li>Distance between the two atoms</li>
          <li>Example:
            <ul>
              <li>ASN A 173 ND2</li>
              <li>NAG A 651 C1</li>
              <li>1.45 A</li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Two atoms that link between additional chemical components
        <ul>
          <li>3-letter code</li>
          <li>Chain id</li>
          <li>Residue number</li>
          <li>Atom name</li>
          <li>Distance between the two atoms</li>
          <li>Example:
            <ul>
              <li>NAG A 651 O4</li>
              <li>NAG A 652 C1</li>
              <li>1.43 A</li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li>For case 2: modified amino acid
    <ul>
      <li>3-letter code</li>
      <li>Chain id</li>
      <li>Residue number</li>
      <li>Example:
        <ul>
          <li>HYP A 123</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>For case 3: cross-link
    <ul>
      <li>3-letter code</li>
      <li>Chain id</li>
      <li>Residue number</li>
      <li>Insertion code</li>
      <li>Atom name</li>
      <li>Distance</li>
      <li>Example
        <ul>
          <li>LYS A 36 NZ</li>
          <li>ASN A 168 CG</li>
          <li>1.5 A</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<h3 id="task6-annotating-ptms-on-sequence-diagram-view">Task6. Annotating PTMs on sequence diagram view</h3>

<ul>
  <li>Refactoring the Java classes that are being used in PDB to diaplay
the sequence diagram into stand-alone Java classes and make it
available in BioJava.
    <ul>
      <li>In the sequence diagram of an entry in PDB (e.g. 3M6S), a user
can select information/annotations (e.g. Pfam domain, InterPro
domain, etc) of one’s interest.</li>
      <li>When refactoring, a key issue is to keep the flexibility to add
new annotations, such as PTM annotation.</li>
    </ul>
  </li>
  <li>Extending the diagram with an option to display PTMs in the
structure.
    <ul>
      <li>For attached PTMs (case 1) and modified amino acids (case 2),
place triangles and abbreviations over the residue.
        <ul>
          <li>Whether to display the PTM annotation is an option to the
users, just like other annotations.</li>
          <li>Different colors of the triangles and texts could be used to
distinguish case 1 and case 2 PTMs.</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>For cross-link, linking the two residues with a dotted line.
    <ul>
      <li>Currently in PDB, disulfide bonds is displayed without an
option.</li>
      <li>A better way will be leaving the cross-links as an optional
annotation to users.
        <ul>
          <li>Different types of cross-links (say disulfide bond and
iso-peptide bond) will have separate options.</li>
          <li>Different line color or style (dotted, dashed, etc) could be
used to represent different cross-link types.</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>Adding another option on the sequence diagram to display PTMs
annotations from UniProt (or other sources).
    <ul>
      <li>UniProt/Swiss-Prot contains tens of thousands of PTM
annotations, which can be displayed in the sequence diagram.</li>
      <li>The UniPort XML file (e.g.,
<a href="http://www.uniprot.org/uniprot/P56524.xml">http://www.uniprot.org/uniprot/P56524.xml</a>) can be utilized.</li>
      <li>Some code I have written in Musite could be refactored and used
here.</li>
    </ul>
  </li>
</ul>

<h3 id="task7-generating-2d-tree-images-of-glycan-structures">Task7. Generating 2D tree images of glycan structures</h3>

<ul>
  <li>Generating images to present the linkage patterns of glycan using
the guideline from the <a href="http://www.functionalglycomics.org/static/consortium/Nomenclature.shtml">Consortium for Functional
Glyconomics</a>.</li>
</ul>

<h2 id="timeline">Timeline</h2>

<ul>
  <li>05/24-05/25: Task 1-Making a list of PTM types.</li>
  <li>05/26-06/02: Task 2-Defining data representation of PTMs.</li>
  <li>06/03-06/09: Task 3-Reading and parsing 3D protein structure files
(.pdb or .cif).</li>
  <li>06/10-06/23: Task 4-Identifying PTMs in 3D protein structures.</li>
  <li>06/24-06/25: Task 5-Representing PTMs in text.</li>
  <li>06/26-07/30: Task 6.1-Refactor sequence diagram code from PDB.</li>
  <li>07/31-08/06: Task 6.2-Annotating PTMs on sequence diagram view.</li>
  <li>08/07-08/16: Task 7-Generating 2D tree images of glycan structures.</li>
</ul>

<h2 id="weekly-report">Weekly Report</h2>

<ul>
  <li>05/31/2010
    <ul>
      <li>A initial list of case 1 and case 2 PTMs has be identified.</li>
      <li>An XML format to store the information of PTMs has been defined.</li>
      <li>Java classes representing the PTMs have been committed to the
code repository.</li>
    </ul>
  </li>
  <li>06/07/2010
    <ul>
      <li>Case 3 PTMs were added into the XML file.</li>
      <li>XML parser.</li>
      <li>Identification of case 1 PTMs.</li>
      <li>Unit tests.</li>
    </ul>
  </li>
  <li>06/14/2010
    <ul>
      <li>Identification of PTMs of all three cases.</li>
      <li>More modifications were added into the XML file.</li>
      <li>Representing PTMs in text in test code.</li>
    </ul>
  </li>
  <li>06/22/2010
    <ul>
      <li>More modifications were added into the XML file, such as
isopeptide bonds.</li>
      <li>Improving the current code according to</li>
      <li>PDB sequence diagram code was refactored from PDB.org by
Andreas.</li>
    </ul>
  </li>
  <li>06/28/2010
    <ul>
      <li>Finished going over all RESID entries.
        <ul>
          <li>212 entries were added in the XML file.</li>
          <li>Special treatment should be applied to all ACE entries.</li>
        </ul>
      </li>
      <li>Identify additional attached ligands that are not directly
attached to amino acids.</li>
      <li>List unidentifiable atom linkages.</li>
    </ul>
  </li>
  <li>07/08/2010
    <ul>
      <li>In progress of scanning whole PDB.</li>
      <li>Hacking the PDB sequence diagram code.</li>
    </ul>
  </li>
  <li>07/16/2010
    <ul>
      <li>Attending ISMB from July 09-13</li>
      <li>Finished scanning whole PDB. Need to analyze the result.</li>
      <li>Started to refactor the sequence diagram code.</li>
    </ul>
  </li>
  <li>07/26/2010
    <ul>
      <li>Keywords added in the XML file.</li>
      <li>Analyzed results of PDB scanning:
<a href="http://spreadsheets.google.com/ccc?key=0Aqz8AlO4d-qMdGVtNWM5Zk5mWXlBQUIwMnZESjc2SkE&amp;hl=en&amp;authkey=CI6ggKsN">summary</a>.</li>
      <li>Still working on sequence diagram code.</li>
    </ul>
  </li>
  <li>08/02/2010
    <ul>
      <li>Refactor ModifiedCompound and ModifiedCompoundImpl.</li>
      <li>Serialization.</li>
    </ul>
  </li>
  <li>08/09/2010
    <ul>
      <li>Refactor package structures</li>
      <li>Integrate crosslinks in sequence diagram.</li>
    </ul>
  </li>
  <li>08/16/2010
    <ul>
      <li>Better representation of PTMs in sequence diagram.</li>
      <li>Support metal coordination modifications.</li>
    </ul>
  </li>
</ul>

<h2 id="skype-call-notes">Skype call notes</h2>

<ul>
  <li>07/29/2010
    <ul>
      <li>Participants: Jianjiong, Peter.</li>
      <li>Top priority: add PTM annotations in sequence diagram.</li>
      <li>Serialization: using Strings to represents identified PTMs.</li>
      <li>Metal Coordination
        <ul>
          <li>classify according to number of residues involved</li>
          <li>some metal ions links to multiple residues forming cross
links, e.g. AA0136</li>
        </ul>
      </li>
      <li>Unidentified modified residues: contact John Garavelli</li>
      <li>Do we need to scan different models in a structure? No, use the
first model.</li>
      <li>Improve Java doc</li>
      <li>Add a chapter in cookbook:
<a href="http://biojava.org/wiki/BioJava:CookBook3.0">http://biojava.org/wiki/BioJava:CookBook3.0</a></li>
    </ul>
  </li>
</ul>

<!-- -->

<ul>
  <li>06/15/2010
    <ul>
      <li>Participants: Jianjiong, Peter, Andreas.</li>
      <li>Find more test cases for cross-links (at least one for each type
of cross-link)</li>
      <li>Scan whole PDB to find more modifications
        <ul>
          <li>Classify non-natural modifications into the same classes as
the natural modifications (attachments, modified residues,
cross-links)</li>
          <li>Alternative confirmation (altloc, see <a href="ftp://ftp.wwpdb.org/pub/pdb/doc/format_descriptions/Format_v32_letter.pdf">PDB Content
Guide</a>
) could be a reason for close contacts.</li>
          <li>Send to Andreas if errors occur when reading structures.</li>
        </ul>
      </li>
      <li>How to deal with multiple modifications (in Green fluorescent
proteins), e.g. CRO in 3MIQ?</li>
      <li>About code
        <ul>
          <li>Break long functions (e.g. in
DefaultProteinModificationParser) into short ones.</li>
          <li>Collections.singletonList for 1-element list.</li>
          <li>Return empty collection instead of null.</li>
          <li>Use FindBugs plugin to detect potential bugs</li>
          <li>Remove any Eclipse/Java warnings if present</li>
          <li>Follow Sun JavaDoc conventions
<a href="http://java.sun.com/j2se/javadoc/writingdoccomments/">1</a></li>
          <li>Any Checkstyle template for BioJava?</li>
        </ul>
      </li>
      <li>Cookbook page after API is stable.</li>
      <li>Andreas is refactoring sequence diagram code from PDB web.</li>
      <li>In sequence diagram, dashed line is only for cross-link2. How to
represent cross-links that link more than 2 residues?</li>
    </ul>
  </li>
</ul>

<h2 id="comments">Comments</h2>

<p><em>Please add comments here…</em></p>

<ul>
  <li>Peter Rose (04/30): It’s an ever increasing list of PTMs. So instead
of hardcoding PTMs, it would be better to load them from a file,
i.e. xml.</li>
</ul>

