<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                  BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on Jan 4, 2006<a name="line.20"></a>
<span class="sourceLineNo">021</span> *<a name="line.21"></a>
<span class="sourceLineNo">022</span> */<a name="line.22"></a>
<span class="sourceLineNo">023</span>package org.biojava.nbio.structure;<a name="line.23"></a>
<span class="sourceLineNo">024</span><a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.io.File;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.io.FileInputStream;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.IOException;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.io.InputStream;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.util.ArrayList;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.HashMap;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.HashSet;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.Iterator;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.LinkedHashSet;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.util.List;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.util.Map;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.util.Set;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.structure.align.util.AtomCache;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.nbio.structure.contact.AtomContactSet;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.nbio.structure.contact.Grid;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.nbio.structure.io.FileParsingParameters;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.nbio.structure.io.PDBFileParser;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.nbio.structure.io.mmcif.chem.PolymerType;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.nbio.structure.io.mmcif.chem.ResidueType;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.nbio.structure.io.mmcif.model.ChemComp;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.nbio.structure.io.util.FileDownloadUtils;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.slf4j.Logger;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.slf4j.LoggerFactory;<a name="line.47"></a>
<span class="sourceLineNo">048</span><a name="line.48"></a>
<span class="sourceLineNo">049</span>/**<a name="line.49"></a>
<span class="sourceLineNo">050</span> * A class that provides some tool methods.<a name="line.50"></a>
<span class="sourceLineNo">051</span> *<a name="line.51"></a>
<span class="sourceLineNo">052</span> * @author Andreas Prlic, Jules Jacobsen<a name="line.52"></a>
<span class="sourceLineNo">053</span> * @since 1.0<a name="line.53"></a>
<span class="sourceLineNo">054</span> */<a name="line.54"></a>
<span class="sourceLineNo">055</span>public class StructureTools {<a name="line.55"></a>
<span class="sourceLineNo">056</span><a name="line.56"></a>
<span class="sourceLineNo">057</span>        private static final Logger logger = LoggerFactory<a name="line.57"></a>
<span class="sourceLineNo">058</span>                        .getLogger(StructureTools.class);<a name="line.58"></a>
<span class="sourceLineNo">059</span><a name="line.59"></a>
<span class="sourceLineNo">060</span>        // Amino Acid backbone<a name="line.60"></a>
<span class="sourceLineNo">061</span>        /**<a name="line.61"></a>
<span class="sourceLineNo">062</span>         * The atom name of the backbone C-alpha atom. Note that this can be<a name="line.62"></a>
<span class="sourceLineNo">063</span>         * ambiguous depending on the context since Calcium atoms use the same name<a name="line.63"></a>
<span class="sourceLineNo">064</span>         * in PDB.<a name="line.64"></a>
<span class="sourceLineNo">065</span>         */<a name="line.65"></a>
<span class="sourceLineNo">066</span>        public static final String CA_ATOM_NAME = "CA";<a name="line.66"></a>
<span class="sourceLineNo">067</span><a name="line.67"></a>
<span class="sourceLineNo">068</span>        /**<a name="line.68"></a>
<span class="sourceLineNo">069</span>         * The atom name for the backbone amide nitrogen<a name="line.69"></a>
<span class="sourceLineNo">070</span>         */<a name="line.70"></a>
<span class="sourceLineNo">071</span>        public static final String N_ATOM_NAME = "N";<a name="line.71"></a>
<span class="sourceLineNo">072</span><a name="line.72"></a>
<span class="sourceLineNo">073</span>        /**<a name="line.73"></a>
<span class="sourceLineNo">074</span>         * The atom name for the backbone carbonyl<a name="line.74"></a>
<span class="sourceLineNo">075</span>         */<a name="line.75"></a>
<span class="sourceLineNo">076</span>        public static final String C_ATOM_NAME = "C";<a name="line.76"></a>
<span class="sourceLineNo">077</span><a name="line.77"></a>
<span class="sourceLineNo">078</span>        /**<a name="line.78"></a>
<span class="sourceLineNo">079</span>         * The atom name for the backbone carbonyl oxygen<a name="line.79"></a>
<span class="sourceLineNo">080</span>         */<a name="line.80"></a>
<span class="sourceLineNo">081</span>        public static final String O_ATOM_NAME = "O";<a name="line.81"></a>
<span class="sourceLineNo">082</span><a name="line.82"></a>
<span class="sourceLineNo">083</span>        /**<a name="line.83"></a>
<span class="sourceLineNo">084</span>         * The atom name of the side-chain C-beta atom<a name="line.84"></a>
<span class="sourceLineNo">085</span>         */<a name="line.85"></a>
<span class="sourceLineNo">086</span>        public static final String CB_ATOM_NAME = "CB";<a name="line.86"></a>
<span class="sourceLineNo">087</span><a name="line.87"></a>
<span class="sourceLineNo">088</span>        // Nucleotide backbone<a name="line.88"></a>
<span class="sourceLineNo">089</span>        /**<a name="line.89"></a>
<span class="sourceLineNo">090</span>         * The atom name of the backbone C1' in RNA<a name="line.90"></a>
<span class="sourceLineNo">091</span>         */<a name="line.91"></a>
<span class="sourceLineNo">092</span>        public static final String C1_ATOM_NAME = "C1'";<a name="line.92"></a>
<span class="sourceLineNo">093</span>        /**<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * The atom name of the backbone C2' in RNA<a name="line.94"></a>
<span class="sourceLineNo">095</span>         */<a name="line.95"></a>
<span class="sourceLineNo">096</span>        public static final String C2_ATOM_NAME = "C2'";<a name="line.96"></a>
<span class="sourceLineNo">097</span>        /**<a name="line.97"></a>
<span class="sourceLineNo">098</span>         * The atom name of the backbone C3' in RNA<a name="line.98"></a>
<span class="sourceLineNo">099</span>         */<a name="line.99"></a>
<span class="sourceLineNo">100</span>        public static final String C3_ATOM_NAME = "C3'";<a name="line.100"></a>
<span class="sourceLineNo">101</span>        /**<a name="line.101"></a>
<span class="sourceLineNo">102</span>         * The atom name of the backbone C4' in RNA<a name="line.102"></a>
<span class="sourceLineNo">103</span>         */<a name="line.103"></a>
<span class="sourceLineNo">104</span>        public static final String C4_ATOM_NAME = "C4'";<a name="line.104"></a>
<span class="sourceLineNo">105</span>        /**<a name="line.105"></a>
<span class="sourceLineNo">106</span>         * The atom name of the backbone O2' in RNA<a name="line.106"></a>
<span class="sourceLineNo">107</span>         */<a name="line.107"></a>
<span class="sourceLineNo">108</span>        public static final String O2_ATOM_NAME = "O2'";<a name="line.108"></a>
<span class="sourceLineNo">109</span>        /**<a name="line.109"></a>
<span class="sourceLineNo">110</span>         * The atom name of the backbone O3' in RNA<a name="line.110"></a>
<span class="sourceLineNo">111</span>         */<a name="line.111"></a>
<span class="sourceLineNo">112</span>        public static final String O3_ATOM_NAME = "O3'";<a name="line.112"></a>
<span class="sourceLineNo">113</span>        /**<a name="line.113"></a>
<span class="sourceLineNo">114</span>         * The atom name of the backbone O4' in RNA<a name="line.114"></a>
<span class="sourceLineNo">115</span>         */<a name="line.115"></a>
<span class="sourceLineNo">116</span>        public static final String O4_ATOM_NAME = "O4'";<a name="line.116"></a>
<span class="sourceLineNo">117</span>        /**<a name="line.117"></a>
<span class="sourceLineNo">118</span>         * The atom name of the backbone O4' in RNA<a name="line.118"></a>
<span class="sourceLineNo">119</span>         */<a name="line.119"></a>
<span class="sourceLineNo">120</span>        public static final String O5_ATOM_NAME = "O5'";<a name="line.120"></a>
<span class="sourceLineNo">121</span>        /**<a name="line.121"></a>
<span class="sourceLineNo">122</span>         * The atom name of the backbone O4' in RNA<a name="line.122"></a>
<span class="sourceLineNo">123</span>         */<a name="line.123"></a>
<span class="sourceLineNo">124</span>        public static final String OP1_ATOM_NAME = "OP1";<a name="line.124"></a>
<span class="sourceLineNo">125</span>        /**<a name="line.125"></a>
<span class="sourceLineNo">126</span>         * The atom name of the backbone O4' in RNA<a name="line.126"></a>
<span class="sourceLineNo">127</span>         */<a name="line.127"></a>
<span class="sourceLineNo">128</span>        public static final String OP2_ATOM_NAME = "OP2";<a name="line.128"></a>
<span class="sourceLineNo">129</span>        /**<a name="line.129"></a>
<span class="sourceLineNo">130</span>         * The atom name of the backbone phosphate in RNA<a name="line.130"></a>
<span class="sourceLineNo">131</span>         */<a name="line.131"></a>
<span class="sourceLineNo">132</span>        public static final String P_ATOM_NAME = "P";<a name="line.132"></a>
<span class="sourceLineNo">133</span><a name="line.133"></a>
<span class="sourceLineNo">134</span>        /**<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * The atom used as representative for nucleotides, equivalent to<a name="line.135"></a>
<span class="sourceLineNo">136</span>         * {@link #CA_ATOM_NAME} for proteins<a name="line.136"></a>
<span class="sourceLineNo">137</span>         */<a name="line.137"></a>
<span class="sourceLineNo">138</span>        public static final String NUCLEOTIDE_REPRESENTATIVE = P_ATOM_NAME;<a name="line.138"></a>
<span class="sourceLineNo">139</span><a name="line.139"></a>
<span class="sourceLineNo">140</span>        /**<a name="line.140"></a>
<span class="sourceLineNo">141</span>         * The character to use for unknown compounds in sequence strings<a name="line.141"></a>
<span class="sourceLineNo">142</span>         */<a name="line.142"></a>
<span class="sourceLineNo">143</span>        public static final char UNKNOWN_GROUP_LABEL = 'X';<a name="line.143"></a>
<span class="sourceLineNo">144</span><a name="line.144"></a>
<span class="sourceLineNo">145</span>        /**<a name="line.145"></a>
<span class="sourceLineNo">146</span>         * Below this ratio of aminoacid/nucleotide residues to the sequence total,<a name="line.146"></a>
<span class="sourceLineNo">147</span>         * we use simple majority of aminoacid/nucleotide residues to decide the<a name="line.147"></a>
<span class="sourceLineNo">148</span>         * character of the chain (protein/nucleotide)<a name="line.148"></a>
<span class="sourceLineNo">149</span>         */<a name="line.149"></a>
<span class="sourceLineNo">150</span>        public static final double RATIO_RESIDUES_TO_TOTAL = 0.95;<a name="line.150"></a>
<span class="sourceLineNo">151</span><a name="line.151"></a>
<span class="sourceLineNo">152</span>        // there is a file format change in PDB 3.0 and nucleotides are being<a name="line.152"></a>
<span class="sourceLineNo">153</span>        // renamed<a name="line.153"></a>
<span class="sourceLineNo">154</span>        private static final Map&lt;String, Character&gt; nucleotides30;<a name="line.154"></a>
<span class="sourceLineNo">155</span>        private static final Map&lt;String, Character&gt; nucleotides23;<a name="line.155"></a>
<span class="sourceLineNo">156</span><a name="line.156"></a>
<span class="sourceLineNo">157</span>        // amino acid 3 and 1 letter code definitions<a name="line.157"></a>
<span class="sourceLineNo">158</span>        private static final Map&lt;String, Character&gt; aminoAcids;<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>        private static final Set&lt;Element&gt; hBondDonorAcceptors;<a name="line.160"></a>
<span class="sourceLineNo">161</span><a name="line.161"></a>
<span class="sourceLineNo">162</span>        static {<a name="line.162"></a>
<span class="sourceLineNo">163</span>                nucleotides30 = new HashMap&lt;String, Character&gt;();<a name="line.163"></a>
<span class="sourceLineNo">164</span>                nucleotides30.put("DA", 'A');<a name="line.164"></a>
<span class="sourceLineNo">165</span>                nucleotides30.put("DC", 'C');<a name="line.165"></a>
<span class="sourceLineNo">166</span>                nucleotides30.put("DG", 'G');<a name="line.166"></a>
<span class="sourceLineNo">167</span>                nucleotides30.put("DT", 'T');<a name="line.167"></a>
<span class="sourceLineNo">168</span>                nucleotides30.put("DI", 'I');<a name="line.168"></a>
<span class="sourceLineNo">169</span>                nucleotides30.put("A", 'A');<a name="line.169"></a>
<span class="sourceLineNo">170</span>                nucleotides30.put("G", 'G');<a name="line.170"></a>
<span class="sourceLineNo">171</span>                nucleotides30.put("C", 'C');<a name="line.171"></a>
<span class="sourceLineNo">172</span>                nucleotides30.put("U", 'U');<a name="line.172"></a>
<span class="sourceLineNo">173</span>                nucleotides30.put("I", 'I');<a name="line.173"></a>
<span class="sourceLineNo">174</span><a name="line.174"></a>
<span class="sourceLineNo">175</span>                // the DNA linkers - the +C , +G, +A +T +U and +I have been replaced<a name="line.175"></a>
<span class="sourceLineNo">176</span>                // with these:<a name="line.176"></a>
<span class="sourceLineNo">177</span>                nucleotides30.put("TAF", UNKNOWN_GROUP_LABEL); // Fluorinated Thymine<a name="line.177"></a>
<span class="sourceLineNo">178</span>                nucleotides30.put("TC1", UNKNOWN_GROUP_LABEL); // Furanosyl<a name="line.178"></a>
<span class="sourceLineNo">179</span>                nucleotides30.put("TFE", UNKNOWN_GROUP_LABEL); // Fluorinated Thymine<a name="line.179"></a>
<span class="sourceLineNo">180</span>                nucleotides30.put("TFO", UNKNOWN_GROUP_LABEL); // Tenofovir (3'<a name="line.180"></a>
<span class="sourceLineNo">181</span>                                                                                                                // terminator)<a name="line.181"></a>
<span class="sourceLineNo">182</span>                nucleotides30.put("TGP", UNKNOWN_GROUP_LABEL); // Guanine variant<a name="line.182"></a>
<span class="sourceLineNo">183</span>                nucleotides30.put("THX", UNKNOWN_GROUP_LABEL); // 5' terminator<a name="line.183"></a>
<span class="sourceLineNo">184</span>                nucleotides30.put("TLC", UNKNOWN_GROUP_LABEL); // Thymine with dicyclic<a name="line.184"></a>
<span class="sourceLineNo">185</span>                                                                                                                // sugar<a name="line.185"></a>
<span class="sourceLineNo">186</span>                nucleotides30.put("TLN", UNKNOWN_GROUP_LABEL); // locked Thymine<a name="line.186"></a>
<span class="sourceLineNo">187</span>                nucleotides30.put("LCG", UNKNOWN_GROUP_LABEL); // locked Guanine<a name="line.187"></a>
<span class="sourceLineNo">188</span>                nucleotides30.put("TP1", UNKNOWN_GROUP_LABEL); // Thymine peptide<a name="line.188"></a>
<span class="sourceLineNo">189</span>                                                                                                                // nucleic acid, with<a name="line.189"></a>
<span class="sourceLineNo">190</span>                                                                                                                // added methyl<a name="line.190"></a>
<span class="sourceLineNo">191</span>                nucleotides30.put("CP1", UNKNOWN_GROUP_LABEL); // Cytidine peptide<a name="line.191"></a>
<span class="sourceLineNo">192</span>                                                                                                                // nucleic acid, with<a name="line.192"></a>
<span class="sourceLineNo">193</span>                                                                                                                // added methyl<a name="line.193"></a>
<span class="sourceLineNo">194</span>                nucleotides30.put("TPN", UNKNOWN_GROUP_LABEL); // Thymine peptide<a name="line.194"></a>
<span class="sourceLineNo">195</span>                                                                                                                // nucleic acid<a name="line.195"></a>
<span class="sourceLineNo">196</span>                nucleotides30.put("CPN", UNKNOWN_GROUP_LABEL); // Cytidine peptide<a name="line.196"></a>
<span class="sourceLineNo">197</span>                                                                                                                // nucleic acid<a name="line.197"></a>
<span class="sourceLineNo">198</span>                nucleotides30.put("GPN", UNKNOWN_GROUP_LABEL); // Guanine peptide<a name="line.198"></a>
<span class="sourceLineNo">199</span>                                                                                                                // nucleic acid<a name="line.199"></a>
<span class="sourceLineNo">200</span>                nucleotides30.put("APN", UNKNOWN_GROUP_LABEL); // Adenosine peptide<a name="line.200"></a>
<span class="sourceLineNo">201</span>                                                                                                                // nucleic acid<a name="line.201"></a>
<span class="sourceLineNo">202</span>                nucleotides30.put("TPC", UNKNOWN_GROUP_LABEL); // Thymine variant<a name="line.202"></a>
<span class="sourceLineNo">203</span><a name="line.203"></a>
<span class="sourceLineNo">204</span>                // store nucleic acids (C, G, A, T, U, and I), and<a name="line.204"></a>
<span class="sourceLineNo">205</span>                // the modified versions of nucleic acids (+C, +G, +A, +T, +U, and +I),<a name="line.205"></a>
<span class="sourceLineNo">206</span>                // and<a name="line.206"></a>
<span class="sourceLineNo">207</span>                nucleotides23 = new HashMap&lt;String, Character&gt;();<a name="line.207"></a>
<span class="sourceLineNo">208</span>                String[] names = { "C", "G", "A", "T", "U", "I", "+C", "+G", "+A",<a name="line.208"></a>
<span class="sourceLineNo">209</span>                                "+T", "+U", "+I" };<a name="line.209"></a>
<span class="sourceLineNo">210</span>                for (String n : names) {<a name="line.210"></a>
<span class="sourceLineNo">211</span>                        nucleotides23.put(n, n.charAt(n.length() - 1));<a name="line.211"></a>
<span class="sourceLineNo">212</span>                }<a name="line.212"></a>
<span class="sourceLineNo">213</span><a name="line.213"></a>
<span class="sourceLineNo">214</span>                aminoAcids = new HashMap&lt;String, Character&gt;();<a name="line.214"></a>
<span class="sourceLineNo">215</span>                aminoAcids.put("GLY", 'G');<a name="line.215"></a>
<span class="sourceLineNo">216</span>                aminoAcids.put("ALA", 'A');<a name="line.216"></a>
<span class="sourceLineNo">217</span>                aminoAcids.put("VAL", 'V');<a name="line.217"></a>
<span class="sourceLineNo">218</span>                aminoAcids.put("LEU", 'L');<a name="line.218"></a>
<span class="sourceLineNo">219</span>                aminoAcids.put("ILE", 'I');<a name="line.219"></a>
<span class="sourceLineNo">220</span>                aminoAcids.put("PHE", 'F');<a name="line.220"></a>
<span class="sourceLineNo">221</span>                aminoAcids.put("TYR", 'Y');<a name="line.221"></a>
<span class="sourceLineNo">222</span>                aminoAcids.put("TRP", 'W');<a name="line.222"></a>
<span class="sourceLineNo">223</span>                aminoAcids.put("PRO", 'P');<a name="line.223"></a>
<span class="sourceLineNo">224</span>                aminoAcids.put("HIS", 'H');<a name="line.224"></a>
<span class="sourceLineNo">225</span>                aminoAcids.put("LYS", 'K');<a name="line.225"></a>
<span class="sourceLineNo">226</span>                aminoAcids.put("ARG", 'R');<a name="line.226"></a>
<span class="sourceLineNo">227</span>                aminoAcids.put("SER", 'S');<a name="line.227"></a>
<span class="sourceLineNo">228</span>                aminoAcids.put("THR", 'T');<a name="line.228"></a>
<span class="sourceLineNo">229</span>                aminoAcids.put("GLU", 'E');<a name="line.229"></a>
<span class="sourceLineNo">230</span>                aminoAcids.put("GLN", 'Q');<a name="line.230"></a>
<span class="sourceLineNo">231</span>                aminoAcids.put("ASP", 'D');<a name="line.231"></a>
<span class="sourceLineNo">232</span>                aminoAcids.put("ASN", 'N');<a name="line.232"></a>
<span class="sourceLineNo">233</span>                aminoAcids.put("CYS", 'C');<a name="line.233"></a>
<span class="sourceLineNo">234</span>                aminoAcids.put("MET", 'M');<a name="line.234"></a>
<span class="sourceLineNo">235</span>                // MSE is only found as a molecular replacement for MET<a name="line.235"></a>
<span class="sourceLineNo">236</span>                aminoAcids.put("MSE", 'M');<a name="line.236"></a>
<span class="sourceLineNo">237</span>                // 'non-standard', genetically encoded<a name="line.237"></a>
<span class="sourceLineNo">238</span>                // http://www.chem.qmul.ac.uk/iubmb/newsletter/1999/item3.html<a name="line.238"></a>
<span class="sourceLineNo">239</span>                // IUBMB recommended name is 'SEC' but the wwPDB currently use 'CSE'<a name="line.239"></a>
<span class="sourceLineNo">240</span>                // likewise 'PYL' (IUBMB) and 'PYH' (PDB)<a name="line.240"></a>
<span class="sourceLineNo">241</span>                aminoAcids.put("CSE", 'U');<a name="line.241"></a>
<span class="sourceLineNo">242</span>                aminoAcids.put("SEC", 'U');<a name="line.242"></a>
<span class="sourceLineNo">243</span>                aminoAcids.put("PYH", 'O');<a name="line.243"></a>
<span class="sourceLineNo">244</span>                aminoAcids.put("PYL", 'O');<a name="line.244"></a>
<span class="sourceLineNo">245</span><a name="line.245"></a>
<span class="sourceLineNo">246</span>                hBondDonorAcceptors = new HashSet&lt;Element&gt;();<a name="line.246"></a>
<span class="sourceLineNo">247</span>                hBondDonorAcceptors.add(Element.N);<a name="line.247"></a>
<span class="sourceLineNo">248</span>                hBondDonorAcceptors.add(Element.O);<a name="line.248"></a>
<span class="sourceLineNo">249</span>                hBondDonorAcceptors.add(Element.S);<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>        }<a name="line.251"></a>
<span class="sourceLineNo">252</span><a name="line.252"></a>
<span class="sourceLineNo">253</span>        /**<a name="line.253"></a>
<span class="sourceLineNo">254</span>         * Count how many Atoms are contained within a Structure object.<a name="line.254"></a>
<span class="sourceLineNo">255</span>         *<a name="line.255"></a>
<span class="sourceLineNo">256</span>         * @param s<a name="line.256"></a>
<span class="sourceLineNo">257</span>         *            the structure object<a name="line.257"></a>
<span class="sourceLineNo">258</span>         * @return the number of Atoms in this Structure<a name="line.258"></a>
<span class="sourceLineNo">259</span>         */<a name="line.259"></a>
<span class="sourceLineNo">260</span>        public static final int getNrAtoms(Structure s) {<a name="line.260"></a>
<span class="sourceLineNo">261</span><a name="line.261"></a>
<span class="sourceLineNo">262</span>                int nrAtoms = 0;<a name="line.262"></a>
<span class="sourceLineNo">263</span><a name="line.263"></a>
<span class="sourceLineNo">264</span>                Iterator&lt;Group&gt; iter = new GroupIterator(s);<a name="line.264"></a>
<span class="sourceLineNo">265</span><a name="line.265"></a>
<span class="sourceLineNo">266</span>                while (iter.hasNext()) {<a name="line.266"></a>
<span class="sourceLineNo">267</span>                        Group g = iter.next();<a name="line.267"></a>
<span class="sourceLineNo">268</span>                        nrAtoms += g.size();<a name="line.268"></a>
<span class="sourceLineNo">269</span>                }<a name="line.269"></a>
<span class="sourceLineNo">270</span><a name="line.270"></a>
<span class="sourceLineNo">271</span>                return nrAtoms;<a name="line.271"></a>
<span class="sourceLineNo">272</span>        }<a name="line.272"></a>
<span class="sourceLineNo">273</span><a name="line.273"></a>
<span class="sourceLineNo">274</span>        /**<a name="line.274"></a>
<span class="sourceLineNo">275</span>         * Count how many groups are contained within a structure object.<a name="line.275"></a>
<span class="sourceLineNo">276</span>         *<a name="line.276"></a>
<span class="sourceLineNo">277</span>         * @param s<a name="line.277"></a>
<span class="sourceLineNo">278</span>         *            the structure object<a name="line.278"></a>
<span class="sourceLineNo">279</span>         * @return the number of groups in the structure<a name="line.279"></a>
<span class="sourceLineNo">280</span>         */<a name="line.280"></a>
<span class="sourceLineNo">281</span>        public static final int getNrGroups(Structure s) {<a name="line.281"></a>
<span class="sourceLineNo">282</span>                int nrGroups = 0;<a name="line.282"></a>
<span class="sourceLineNo">283</span><a name="line.283"></a>
<span class="sourceLineNo">284</span>                List&lt;Chain&gt; chains = s.getChains(0);<a name="line.284"></a>
<span class="sourceLineNo">285</span>                for (Chain c : chains) {<a name="line.285"></a>
<span class="sourceLineNo">286</span>                        nrGroups += c.getAtomLength();<a name="line.286"></a>
<span class="sourceLineNo">287</span>                }<a name="line.287"></a>
<span class="sourceLineNo">288</span>                return nrGroups;<a name="line.288"></a>
<span class="sourceLineNo">289</span>        }<a name="line.289"></a>
<span class="sourceLineNo">290</span><a name="line.290"></a>
<span class="sourceLineNo">291</span>        /**<a name="line.291"></a>
<span class="sourceLineNo">292</span>         * Returns an array of the requested Atoms from the Structure object.<a name="line.292"></a>
<span class="sourceLineNo">293</span>         * Iterates over all groups and checks if the requested atoms are in this<a name="line.293"></a>
<span class="sourceLineNo">294</span>         * group, no matter if this is a {@link AminoAcid} or {@link HetatomImpl}<a name="line.294"></a>
<span class="sourceLineNo">295</span>         * group. If the group does not contain all requested atoms then no atoms<a name="line.295"></a>
<span class="sourceLineNo">296</span>         * are added for that group. For structures with more than one model, only<a name="line.296"></a>
<span class="sourceLineNo">297</span>         * model 0 will be used.<a name="line.297"></a>
<span class="sourceLineNo">298</span>         *<a name="line.298"></a>
<span class="sourceLineNo">299</span>         * @param s<a name="line.299"></a>
<span class="sourceLineNo">300</span>         *            the structure to get the atoms from<a name="line.300"></a>
<span class="sourceLineNo">301</span>         *<a name="line.301"></a>
<span class="sourceLineNo">302</span>         * @param atomNames<a name="line.302"></a>
<span class="sourceLineNo">303</span>         *            contains the atom names to be used.<a name="line.303"></a>
<span class="sourceLineNo">304</span>         * @return an Atom[] array<a name="line.304"></a>
<span class="sourceLineNo">305</span>         */<a name="line.305"></a>
<span class="sourceLineNo">306</span>        public static final Atom[] getAtomArray(Structure s, String[] atomNames) {<a name="line.306"></a>
<span class="sourceLineNo">307</span>                List&lt;Chain&gt; chains = s.getModel(0);<a name="line.307"></a>
<span class="sourceLineNo">308</span><a name="line.308"></a>
<span class="sourceLineNo">309</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.309"></a>
<span class="sourceLineNo">310</span><a name="line.310"></a>
<span class="sourceLineNo">311</span>                extractAtoms(atomNames, chains, atoms);<a name="line.311"></a>
<span class="sourceLineNo">312</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.312"></a>
<span class="sourceLineNo">313</span><a name="line.313"></a>
<span class="sourceLineNo">314</span>        }<a name="line.314"></a>
<span class="sourceLineNo">315</span><a name="line.315"></a>
<span class="sourceLineNo">316</span>        /**<a name="line.316"></a>
<span class="sourceLineNo">317</span>         * Returns an array of the requested Atoms from the Structure object. In<a name="line.317"></a>
<span class="sourceLineNo">318</span>         * contrast to {@link #getAtomArray(Structure, String[])} this method<a name="line.318"></a>
<span class="sourceLineNo">319</span>         * iterates over all chains. Iterates over all chains and groups and checks<a name="line.319"></a>
<span class="sourceLineNo">320</span>         * if the requested atoms are in this group, no matter if this is a<a name="line.320"></a>
<span class="sourceLineNo">321</span>         * {@link AminoAcid} or {@link HetatomImpl} group. If the group does not<a name="line.321"></a>
<span class="sourceLineNo">322</span>         * contain all requested atoms then no atoms are added for that group. For<a name="line.322"></a>
<span class="sourceLineNo">323</span>         * structures with more than one model, only model 0 will be used.<a name="line.323"></a>
<span class="sourceLineNo">324</span>         *<a name="line.324"></a>
<span class="sourceLineNo">325</span>         * @param s<a name="line.325"></a>
<span class="sourceLineNo">326</span>         *            the structure to get the atoms from<a name="line.326"></a>
<span class="sourceLineNo">327</span>         *<a name="line.327"></a>
<span class="sourceLineNo">328</span>         * @param atomNames<a name="line.328"></a>
<span class="sourceLineNo">329</span>         *            contains the atom names to be used.<a name="line.329"></a>
<span class="sourceLineNo">330</span>         * @return an Atom[] array<a name="line.330"></a>
<span class="sourceLineNo">331</span>         */<a name="line.331"></a>
<span class="sourceLineNo">332</span>        public static final Atom[] getAtomArrayAllModels(Structure s,<a name="line.332"></a>
<span class="sourceLineNo">333</span>                        String[] atomNames) {<a name="line.333"></a>
<span class="sourceLineNo">334</span><a name="line.334"></a>
<span class="sourceLineNo">335</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.335"></a>
<span class="sourceLineNo">336</span><a name="line.336"></a>
<span class="sourceLineNo">337</span>                for (int i = 0; i &lt; s.nrModels(); i++) {<a name="line.337"></a>
<span class="sourceLineNo">338</span>                        List&lt;Chain&gt; chains = s.getModel(i);<a name="line.338"></a>
<span class="sourceLineNo">339</span>                        extractAtoms(atomNames, chains, atoms);<a name="line.339"></a>
<span class="sourceLineNo">340</span>                }<a name="line.340"></a>
<span class="sourceLineNo">341</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.341"></a>
<span class="sourceLineNo">342</span><a name="line.342"></a>
<span class="sourceLineNo">343</span>        }<a name="line.343"></a>
<span class="sourceLineNo">344</span><a name="line.344"></a>
<span class="sourceLineNo">345</span>        /**<a name="line.345"></a>
<span class="sourceLineNo">346</span>         * Convert all atoms of the structure (first model) into an Atom array<a name="line.346"></a>
<span class="sourceLineNo">347</span>         *<a name="line.347"></a>
<span class="sourceLineNo">348</span>         * @param s<a name="line.348"></a>
<span class="sourceLineNo">349</span>         *            input structure<a name="line.349"></a>
<span class="sourceLineNo">350</span>         * @return all atom array<a name="line.350"></a>
<span class="sourceLineNo">351</span>         */<a name="line.351"></a>
<span class="sourceLineNo">352</span>        public static final Atom[] getAllAtomArray(Structure s) {<a name="line.352"></a>
<span class="sourceLineNo">353</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.353"></a>
<span class="sourceLineNo">354</span><a name="line.354"></a>
<span class="sourceLineNo">355</span>                AtomIterator iter = new AtomIterator(s);<a name="line.355"></a>
<span class="sourceLineNo">356</span>                while (iter.hasNext()) {<a name="line.356"></a>
<span class="sourceLineNo">357</span>                        Atom a = iter.next();<a name="line.357"></a>
<span class="sourceLineNo">358</span>                        atoms.add(a);<a name="line.358"></a>
<span class="sourceLineNo">359</span>                }<a name="line.359"></a>
<span class="sourceLineNo">360</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.360"></a>
<span class="sourceLineNo">361</span>        }<a name="line.361"></a>
<span class="sourceLineNo">362</span><a name="line.362"></a>
<span class="sourceLineNo">363</span>        /**<a name="line.363"></a>
<span class="sourceLineNo">364</span>         * Returns and array of all atoms of the chain (first model), including<a name="line.364"></a>
<span class="sourceLineNo">365</span>         * Hydrogens (if present) and all HETATOMs. Waters are not included.<a name="line.365"></a>
<span class="sourceLineNo">366</span>         *<a name="line.366"></a>
<span class="sourceLineNo">367</span>         * @param c<a name="line.367"></a>
<span class="sourceLineNo">368</span>         *            input chain<a name="line.368"></a>
<span class="sourceLineNo">369</span>         * @return all atom array<a name="line.369"></a>
<span class="sourceLineNo">370</span>         */<a name="line.370"></a>
<span class="sourceLineNo">371</span>        public static final Atom[] getAllAtomArray(Chain c) {<a name="line.371"></a>
<span class="sourceLineNo">372</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.372"></a>
<span class="sourceLineNo">373</span><a name="line.373"></a>
<span class="sourceLineNo">374</span>                for (Group g : c.getAtomGroups()) {<a name="line.374"></a>
<span class="sourceLineNo">375</span>                        if (g.isWater())<a name="line.375"></a>
<span class="sourceLineNo">376</span>                                continue;<a name="line.376"></a>
<span class="sourceLineNo">377</span>                        for (Atom a : g.getAtoms()) {<a name="line.377"></a>
<span class="sourceLineNo">378</span>                                atoms.add(a);<a name="line.378"></a>
<span class="sourceLineNo">379</span>                        }<a name="line.379"></a>
<span class="sourceLineNo">380</span>                }<a name="line.380"></a>
<span class="sourceLineNo">381</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.381"></a>
<span class="sourceLineNo">382</span>        }<a name="line.382"></a>
<span class="sourceLineNo">383</span><a name="line.383"></a>
<span class="sourceLineNo">384</span>        /**<a name="line.384"></a>
<span class="sourceLineNo">385</span>         * List of groups from the structure not included in ca (e.g. ligands).<a name="line.385"></a>
<span class="sourceLineNo">386</span>         *<a name="line.386"></a>
<span class="sourceLineNo">387</span>         * Unaligned groups are searched from all chains referenced in ca, as well<a name="line.387"></a>
<span class="sourceLineNo">388</span>         * as any chains in the first model of the structure from ca[0], if any.<a name="line.388"></a>
<span class="sourceLineNo">389</span>         *<a name="line.389"></a>
<span class="sourceLineNo">390</span>         * @param ca an array of atoms<a name="line.390"></a>
<span class="sourceLineNo">391</span>         * @return<a name="line.391"></a>
<span class="sourceLineNo">392</span>         */<a name="line.392"></a>
<span class="sourceLineNo">393</span>        public static List&lt;Group&gt; getUnalignedGroups(Atom[] ca) {<a name="line.393"></a>
<span class="sourceLineNo">394</span>                Set&lt;Chain&gt; chains = new HashSet&lt;Chain&gt;();<a name="line.394"></a>
<span class="sourceLineNo">395</span>                Set&lt;Group&gt; caGroups = new HashSet&lt;Group&gt;();<a name="line.395"></a>
<span class="sourceLineNo">396</span><a name="line.396"></a>
<span class="sourceLineNo">397</span>                // Create list of all chains in this structure<a name="line.397"></a>
<span class="sourceLineNo">398</span>                Structure s = null;<a name="line.398"></a>
<span class="sourceLineNo">399</span>                if (ca.length &gt; 0) {<a name="line.399"></a>
<span class="sourceLineNo">400</span>                        Group g = ca[0].getGroup();<a name="line.400"></a>
<span class="sourceLineNo">401</span>                        if (g != null) {<a name="line.401"></a>
<span class="sourceLineNo">402</span>                                Chain c = g.getChain();<a name="line.402"></a>
<span class="sourceLineNo">403</span>                                if (c != null) {<a name="line.403"></a>
<span class="sourceLineNo">404</span>                                        s = c.getStructure();<a name="line.404"></a>
<span class="sourceLineNo">405</span>                                }<a name="line.405"></a>
<span class="sourceLineNo">406</span>                        }<a name="line.406"></a>
<span class="sourceLineNo">407</span>                }<a name="line.407"></a>
<span class="sourceLineNo">408</span>                if (s != null) {<a name="line.408"></a>
<span class="sourceLineNo">409</span>                        // Add all chains from the structure<a name="line.409"></a>
<span class="sourceLineNo">410</span>                        for (Chain c : s.getChains(0)) {<a name="line.410"></a>
<span class="sourceLineNo">411</span>                                chains.add(c);<a name="line.411"></a>
<span class="sourceLineNo">412</span>                        }<a name="line.412"></a>
<span class="sourceLineNo">413</span>                }<a name="line.413"></a>
<span class="sourceLineNo">414</span><a name="line.414"></a>
<span class="sourceLineNo">415</span>                // Add groups and chains from ca<a name="line.415"></a>
<span class="sourceLineNo">416</span>                for (Atom a : ca) {<a name="line.416"></a>
<span class="sourceLineNo">417</span>                        Group g = a.getGroup();<a name="line.417"></a>
<span class="sourceLineNo">418</span>                        if (g != null) {<a name="line.418"></a>
<span class="sourceLineNo">419</span>                                caGroups.add(g);<a name="line.419"></a>
<span class="sourceLineNo">420</span><a name="line.420"></a>
<span class="sourceLineNo">421</span>                                Chain c = g.getChain();<a name="line.421"></a>
<span class="sourceLineNo">422</span>                                if (c != null) {<a name="line.422"></a>
<span class="sourceLineNo">423</span>                                        chains.add(c);<a name="line.423"></a>
<span class="sourceLineNo">424</span>                                }<a name="line.424"></a>
<span class="sourceLineNo">425</span>                        }<a name="line.425"></a>
<span class="sourceLineNo">426</span>                }<a name="line.426"></a>
<span class="sourceLineNo">427</span><a name="line.427"></a>
<span class="sourceLineNo">428</span>                // Iterate through all chains, finding groups not in ca<a name="line.428"></a>
<span class="sourceLineNo">429</span>                List&lt;Group&gt; unadded = new ArrayList&lt;Group&gt;();<a name="line.429"></a>
<span class="sourceLineNo">430</span>                for (Chain c : chains) {<a name="line.430"></a>
<span class="sourceLineNo">431</span>                        for (Group g : c.getAtomGroups()) {<a name="line.431"></a>
<span class="sourceLineNo">432</span>                                if (!caGroups.contains(g)) {<a name="line.432"></a>
<span class="sourceLineNo">433</span>                                        unadded.add(g);<a name="line.433"></a>
<span class="sourceLineNo">434</span>                                }<a name="line.434"></a>
<span class="sourceLineNo">435</span>                        }<a name="line.435"></a>
<span class="sourceLineNo">436</span>                }<a name="line.436"></a>
<span class="sourceLineNo">437</span>                return unadded;<a name="line.437"></a>
<span class="sourceLineNo">438</span>        }<a name="line.438"></a>
<span class="sourceLineNo">439</span><a name="line.439"></a>
<span class="sourceLineNo">440</span>        /**<a name="line.440"></a>
<span class="sourceLineNo">441</span>         * Returns and array of all non-Hydrogen atoms in the given Structure,<a name="line.441"></a>
<span class="sourceLineNo">442</span>         * optionally including HET atoms or not. Waters are not included.<a name="line.442"></a>
<span class="sourceLineNo">443</span>         *<a name="line.443"></a>
<span class="sourceLineNo">444</span>         * @param s<a name="line.444"></a>
<span class="sourceLineNo">445</span>         * @param hetAtoms<a name="line.445"></a>
<span class="sourceLineNo">446</span>         *            if true HET atoms are included in array, if false they are not<a name="line.446"></a>
<span class="sourceLineNo">447</span>         * @return<a name="line.447"></a>
<span class="sourceLineNo">448</span>         */<a name="line.448"></a>
<span class="sourceLineNo">449</span>        public static final Atom[] getAllNonHAtomArray(Structure s, boolean hetAtoms) {<a name="line.449"></a>
<span class="sourceLineNo">450</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.450"></a>
<span class="sourceLineNo">451</span><a name="line.451"></a>
<span class="sourceLineNo">452</span>                AtomIterator iter = new AtomIterator(s);<a name="line.452"></a>
<span class="sourceLineNo">453</span>                while (iter.hasNext()) {<a name="line.453"></a>
<span class="sourceLineNo">454</span>                        Atom a = iter.next();<a name="line.454"></a>
<span class="sourceLineNo">455</span>                        if (a.getElement() == Element.H)<a name="line.455"></a>
<span class="sourceLineNo">456</span>                                continue;<a name="line.456"></a>
<span class="sourceLineNo">457</span><a name="line.457"></a>
<span class="sourceLineNo">458</span>                        Group g = a.getGroup();<a name="line.458"></a>
<span class="sourceLineNo">459</span><a name="line.459"></a>
<span class="sourceLineNo">460</span>                        if (g.isWater())<a name="line.460"></a>
<span class="sourceLineNo">461</span>                                continue;<a name="line.461"></a>
<span class="sourceLineNo">462</span><a name="line.462"></a>
<span class="sourceLineNo">463</span>                        if (!hetAtoms &amp;&amp; g.getType().equals(GroupType.HETATM))<a name="line.463"></a>
<span class="sourceLineNo">464</span>                                continue;<a name="line.464"></a>
<span class="sourceLineNo">465</span><a name="line.465"></a>
<span class="sourceLineNo">466</span>                        atoms.add(a);<a name="line.466"></a>
<span class="sourceLineNo">467</span>                }<a name="line.467"></a>
<span class="sourceLineNo">468</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.468"></a>
<span class="sourceLineNo">469</span>        }<a name="line.469"></a>
<span class="sourceLineNo">470</span><a name="line.470"></a>
<span class="sourceLineNo">471</span>        /**<a name="line.471"></a>
<span class="sourceLineNo">472</span>         * Returns and array of all non-Hydrogen atoms in the given Chain,<a name="line.472"></a>
<span class="sourceLineNo">473</span>         * optionally including HET atoms or not Waters are not included.<a name="line.473"></a>
<span class="sourceLineNo">474</span>         *<a name="line.474"></a>
<span class="sourceLineNo">475</span>         * @param c<a name="line.475"></a>
<span class="sourceLineNo">476</span>         * @param hetAtoms<a name="line.476"></a>
<span class="sourceLineNo">477</span>         *            if true HET atoms are included in array, if false they are not<a name="line.477"></a>
<span class="sourceLineNo">478</span>         * @return<a name="line.478"></a>
<span class="sourceLineNo">479</span>         */<a name="line.479"></a>
<span class="sourceLineNo">480</span>        public static final Atom[] getAllNonHAtomArray(Chain c, boolean hetAtoms) {<a name="line.480"></a>
<span class="sourceLineNo">481</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.481"></a>
<span class="sourceLineNo">482</span><a name="line.482"></a>
<span class="sourceLineNo">483</span>                for (Group g : c.getAtomGroups()) {<a name="line.483"></a>
<span class="sourceLineNo">484</span>                        if (g.isWater())<a name="line.484"></a>
<span class="sourceLineNo">485</span>                                continue;<a name="line.485"></a>
<span class="sourceLineNo">486</span>                        for (Atom a : g.getAtoms()) {<a name="line.486"></a>
<span class="sourceLineNo">487</span><a name="line.487"></a>
<span class="sourceLineNo">488</span>                                if (a.getElement() == Element.H)<a name="line.488"></a>
<span class="sourceLineNo">489</span>                                        continue;<a name="line.489"></a>
<span class="sourceLineNo">490</span><a name="line.490"></a>
<span class="sourceLineNo">491</span>                                if (!hetAtoms &amp;&amp; g.getType().equals(GroupType.HETATM))<a name="line.491"></a>
<span class="sourceLineNo">492</span>                                        continue;<a name="line.492"></a>
<span class="sourceLineNo">493</span><a name="line.493"></a>
<span class="sourceLineNo">494</span>                                atoms.add(a);<a name="line.494"></a>
<span class="sourceLineNo">495</span>                        }<a name="line.495"></a>
<span class="sourceLineNo">496</span>                }<a name="line.496"></a>
<span class="sourceLineNo">497</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.497"></a>
<span class="sourceLineNo">498</span>        }<a name="line.498"></a>
<span class="sourceLineNo">499</span><a name="line.499"></a>
<span class="sourceLineNo">500</span>        /**<a name="line.500"></a>
<span class="sourceLineNo">501</span>         * Adds to the given atoms list, all atoms of groups that contained all<a name="line.501"></a>
<span class="sourceLineNo">502</span>         * requested atomNames, i.e. if a group does not contain all of the<a name="line.502"></a>
<span class="sourceLineNo">503</span>         * requested atom names, its atoms won't be added.<a name="line.503"></a>
<span class="sourceLineNo">504</span>         *<a name="line.504"></a>
<span class="sourceLineNo">505</span>         * @param atomNames<a name="line.505"></a>
<span class="sourceLineNo">506</span>         * @param chains<a name="line.506"></a>
<span class="sourceLineNo">507</span>         * @param atoms<a name="line.507"></a>
<span class="sourceLineNo">508</span>         */<a name="line.508"></a>
<span class="sourceLineNo">509</span>        private static void extractAtoms(String[] atomNames, List&lt;Chain&gt; chains,<a name="line.509"></a>
<span class="sourceLineNo">510</span>                        List&lt;Atom&gt; atoms) {<a name="line.510"></a>
<span class="sourceLineNo">511</span><a name="line.511"></a>
<span class="sourceLineNo">512</span>                for (Chain c : chains) {<a name="line.512"></a>
<span class="sourceLineNo">513</span><a name="line.513"></a>
<span class="sourceLineNo">514</span>                        for (Group g : c.getAtomGroups()) {<a name="line.514"></a>
<span class="sourceLineNo">515</span><a name="line.515"></a>
<span class="sourceLineNo">516</span>                                // a temp container for the atoms of this group<a name="line.516"></a>
<span class="sourceLineNo">517</span>                                List&lt;Atom&gt; thisGroupAtoms = new ArrayList&lt;Atom&gt;();<a name="line.517"></a>
<span class="sourceLineNo">518</span>                                // flag to check if this group contains all the requested atoms.<a name="line.518"></a>
<span class="sourceLineNo">519</span>                                boolean thisGroupAllAtoms = true;<a name="line.519"></a>
<span class="sourceLineNo">520</span>                                for (String atomName : atomNames) {<a name="line.520"></a>
<span class="sourceLineNo">521</span>                                        Atom a = g.getAtom(atomName);<a name="line.521"></a>
<span class="sourceLineNo">522</span><a name="line.522"></a>
<span class="sourceLineNo">523</span>                                        if (a == null) {<a name="line.523"></a>
<span class="sourceLineNo">524</span>                                                // this group does not have a required atom, skip it...<a name="line.524"></a>
<span class="sourceLineNo">525</span>                                                thisGroupAllAtoms = false;<a name="line.525"></a>
<span class="sourceLineNo">526</span>                                                break;<a name="line.526"></a>
<span class="sourceLineNo">527</span>                                        }<a name="line.527"></a>
<span class="sourceLineNo">528</span>                                        thisGroupAtoms.add(a);<a name="line.528"></a>
<span class="sourceLineNo">529</span>                                }<a name="line.529"></a>
<span class="sourceLineNo">530</span>                                if (thisGroupAllAtoms) {<a name="line.530"></a>
<span class="sourceLineNo">531</span>                                        // add the atoms of this group to the array.<a name="line.531"></a>
<span class="sourceLineNo">532</span>                                        for (Atom a : thisGroupAtoms) {<a name="line.532"></a>
<span class="sourceLineNo">533</span>                                                atoms.add(a);<a name="line.533"></a>
<span class="sourceLineNo">534</span>                                        }<a name="line.534"></a>
<span class="sourceLineNo">535</span>                                }<a name="line.535"></a>
<span class="sourceLineNo">536</span><a name="line.536"></a>
<span class="sourceLineNo">537</span>                        }<a name="line.537"></a>
<span class="sourceLineNo">538</span>                }<a name="line.538"></a>
<span class="sourceLineNo">539</span>        }<a name="line.539"></a>
<span class="sourceLineNo">540</span><a name="line.540"></a>
<span class="sourceLineNo">541</span>        /**<a name="line.541"></a>
<span class="sourceLineNo">542</span>         * Returns an array of the requested Atoms from the Chain object. Iterates<a name="line.542"></a>
<span class="sourceLineNo">543</span>         * over all groups and checks if the requested atoms are in this group, no<a name="line.543"></a>
<span class="sourceLineNo">544</span>         * matter if this is a AminoAcid or Hetatom group. If the group does not<a name="line.544"></a>
<span class="sourceLineNo">545</span>         * contain all requested atoms then no atoms are added for that group.<a name="line.545"></a>
<span class="sourceLineNo">546</span>         *<a name="line.546"></a>
<span class="sourceLineNo">547</span>         * @param c<a name="line.547"></a>
<span class="sourceLineNo">548</span>         *            the Chain to get the atoms from<a name="line.548"></a>
<span class="sourceLineNo">549</span>         *<a name="line.549"></a>
<span class="sourceLineNo">550</span>         * @param atomNames<a name="line.550"></a>
<span class="sourceLineNo">551</span>         *            contains the atom names to be used.<a name="line.551"></a>
<span class="sourceLineNo">552</span>         * @return an Atom[] array<a name="line.552"></a>
<span class="sourceLineNo">553</span>         */<a name="line.553"></a>
<span class="sourceLineNo">554</span>        public static final Atom[] getAtomArray(Chain c, String[] atomNames) {<a name="line.554"></a>
<span class="sourceLineNo">555</span><a name="line.555"></a>
<span class="sourceLineNo">556</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.556"></a>
<span class="sourceLineNo">557</span><a name="line.557"></a>
<span class="sourceLineNo">558</span>                for (Group g : c.getAtomGroups()) {<a name="line.558"></a>
<span class="sourceLineNo">559</span><a name="line.559"></a>
<span class="sourceLineNo">560</span>                        // a temp container for the atoms of this group<a name="line.560"></a>
<span class="sourceLineNo">561</span>                        List&lt;Atom&gt; thisGroupAtoms = new ArrayList&lt;Atom&gt;();<a name="line.561"></a>
<span class="sourceLineNo">562</span>                        // flag to check if this group contains all the requested atoms.<a name="line.562"></a>
<span class="sourceLineNo">563</span>                        boolean thisGroupAllAtoms = true;<a name="line.563"></a>
<span class="sourceLineNo">564</span>                        for (String atomName : atomNames) {<a name="line.564"></a>
<span class="sourceLineNo">565</span>                                Atom a = g.getAtom(atomName);<a name="line.565"></a>
<span class="sourceLineNo">566</span>                                if (a == null) {<a name="line.566"></a>
<span class="sourceLineNo">567</span>                                        logger.debug("Group " + g.getResidueNumber() + " ("<a name="line.567"></a>
<span class="sourceLineNo">568</span>                                                        + g.getPDBName()<a name="line.568"></a>
<span class="sourceLineNo">569</span>                                                        + ") does not have the required atom '" + atomName<a name="line.569"></a>
<span class="sourceLineNo">570</span>                                                        + "'");<a name="line.570"></a>
<span class="sourceLineNo">571</span>                                        // this group does not have a required atom, skip it...<a name="line.571"></a>
<span class="sourceLineNo">572</span>                                        thisGroupAllAtoms = false;<a name="line.572"></a>
<span class="sourceLineNo">573</span>                                        break;<a name="line.573"></a>
<span class="sourceLineNo">574</span>                                }<a name="line.574"></a>
<span class="sourceLineNo">575</span>                                thisGroupAtoms.add(a);<a name="line.575"></a>
<span class="sourceLineNo">576</span>                        }<a name="line.576"></a>
<span class="sourceLineNo">577</span><a name="line.577"></a>
<span class="sourceLineNo">578</span>                        if (thisGroupAllAtoms) {<a name="line.578"></a>
<span class="sourceLineNo">579</span>                                // add the atoms of this group to the array.<a name="line.579"></a>
<span class="sourceLineNo">580</span>                                for (Atom a : thisGroupAtoms) {<a name="line.580"></a>
<span class="sourceLineNo">581</span>                                        atoms.add(a);<a name="line.581"></a>
<span class="sourceLineNo">582</span>                                }<a name="line.582"></a>
<span class="sourceLineNo">583</span>                        }<a name="line.583"></a>
<span class="sourceLineNo">584</span><a name="line.584"></a>
<span class="sourceLineNo">585</span>                }<a name="line.585"></a>
<span class="sourceLineNo">586</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.586"></a>
<span class="sourceLineNo">587</span><a name="line.587"></a>
<span class="sourceLineNo">588</span>        }<a name="line.588"></a>
<span class="sourceLineNo">589</span><a name="line.589"></a>
<span class="sourceLineNo">590</span>        /**<a name="line.590"></a>
<span class="sourceLineNo">591</span>         * Returns an Atom array of the C-alpha atoms. Any atom that is a carbon and<a name="line.591"></a>
<span class="sourceLineNo">592</span>         * has CA name will be returned.<a name="line.592"></a>
<span class="sourceLineNo">593</span>         *<a name="line.593"></a>
<span class="sourceLineNo">594</span>         * @param c<a name="line.594"></a>
<span class="sourceLineNo">595</span>         *            the structure object<a name="line.595"></a>
<span class="sourceLineNo">596</span>         * @return an Atom[] array<a name="line.596"></a>
<span class="sourceLineNo">597</span>         * @see #getRepresentativeAtomArray(Chain)<a name="line.597"></a>
<span class="sourceLineNo">598</span>         */<a name="line.598"></a>
<span class="sourceLineNo">599</span>        public static final Atom[] getAtomCAArray(Chain c) {<a name="line.599"></a>
<span class="sourceLineNo">600</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.600"></a>
<span class="sourceLineNo">601</span><a name="line.601"></a>
<span class="sourceLineNo">602</span>                for (Group g : c.getAtomGroups()) {<a name="line.602"></a>
<span class="sourceLineNo">603</span>                        if (g.hasAtom(CA_ATOM_NAME)<a name="line.603"></a>
<span class="sourceLineNo">604</span>                                        &amp;&amp; g.getAtom(CA_ATOM_NAME).getElement() == Element.C) {<a name="line.604"></a>
<span class="sourceLineNo">605</span>                                atoms.add(g.getAtom(CA_ATOM_NAME));<a name="line.605"></a>
<span class="sourceLineNo">606</span>                        }<a name="line.606"></a>
<span class="sourceLineNo">607</span>                }<a name="line.607"></a>
<span class="sourceLineNo">608</span><a name="line.608"></a>
<span class="sourceLineNo">609</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.609"></a>
<span class="sourceLineNo">610</span>        }<a name="line.610"></a>
<span class="sourceLineNo">611</span><a name="line.611"></a>
<span class="sourceLineNo">612</span>        /**<a name="line.612"></a>
<span class="sourceLineNo">613</span>         * Gets a representative atom for each group that is part of the chain<a name="line.613"></a>
<span class="sourceLineNo">614</span>         * backbone. Note that modified aminoacids won't be returned as part of the<a name="line.614"></a>
<span class="sourceLineNo">615</span>         * backbone if the {@link org.biojava.nbio.structure.io.mmcif.ReducedChemCompProvider} was used to load the<a name="line.615"></a>
<span class="sourceLineNo">616</span>         * structure.<a name="line.616"></a>
<span class="sourceLineNo">617</span>         *<a name="line.617"></a>
<span class="sourceLineNo">618</span>         * For amino acids, the representative is a CA carbon. For nucleotides, the<a name="line.618"></a>
<span class="sourceLineNo">619</span>         * representative is the {@value #NUCLEOTIDE_REPRESENTATIVE}. Other group<a name="line.619"></a>
<span class="sourceLineNo">620</span>         * types will be ignored.<a name="line.620"></a>
<span class="sourceLineNo">621</span>         *<a name="line.621"></a>
<span class="sourceLineNo">622</span>         * @param c<a name="line.622"></a>
<span class="sourceLineNo">623</span>         * @return representative Atoms of the chain backbone<a name="line.623"></a>
<span class="sourceLineNo">624</span>         * @since Biojava 4.1.0<a name="line.624"></a>
<span class="sourceLineNo">625</span>         */<a name="line.625"></a>
<span class="sourceLineNo">626</span>        public static final Atom[] getRepresentativeAtomArray(Chain c) {<a name="line.626"></a>
<span class="sourceLineNo">627</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.627"></a>
<span class="sourceLineNo">628</span><a name="line.628"></a>
<span class="sourceLineNo">629</span>                for (Group g : c.getAtomGroups()) {<a name="line.629"></a>
<span class="sourceLineNo">630</span><a name="line.630"></a>
<span class="sourceLineNo">631</span>                        switch (g.getType()) {<a name="line.631"></a>
<span class="sourceLineNo">632</span>                        case AMINOACID:<a name="line.632"></a>
<span class="sourceLineNo">633</span>                                if (g.hasAtom(CA_ATOM_NAME)<a name="line.633"></a>
<span class="sourceLineNo">634</span>                                                &amp;&amp; g.getAtom(CA_ATOM_NAME).getElement() == Element.C) {<a name="line.634"></a>
<span class="sourceLineNo">635</span>                                        atoms.add(g.getAtom(CA_ATOM_NAME));<a name="line.635"></a>
<span class="sourceLineNo">636</span>                                }<a name="line.636"></a>
<span class="sourceLineNo">637</span>                                break;<a name="line.637"></a>
<span class="sourceLineNo">638</span>                        case NUCLEOTIDE:<a name="line.638"></a>
<span class="sourceLineNo">639</span>                                if (g.hasAtom(NUCLEOTIDE_REPRESENTATIVE)) {<a name="line.639"></a>
<span class="sourceLineNo">640</span>                                        atoms.add(g.getAtom(NUCLEOTIDE_REPRESENTATIVE));<a name="line.640"></a>
<span class="sourceLineNo">641</span>                                }<a name="line.641"></a>
<span class="sourceLineNo">642</span>                                break;<a name="line.642"></a>
<span class="sourceLineNo">643</span>                        default:<a name="line.643"></a>
<span class="sourceLineNo">644</span>                                // don't add<a name="line.644"></a>
<span class="sourceLineNo">645</span>                        }<a name="line.645"></a>
<span class="sourceLineNo">646</span>                }<a name="line.646"></a>
<span class="sourceLineNo">647</span><a name="line.647"></a>
<span class="sourceLineNo">648</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.648"></a>
<span class="sourceLineNo">649</span><a name="line.649"></a>
<span class="sourceLineNo">650</span>        }<a name="line.650"></a>
<span class="sourceLineNo">651</span><a name="line.651"></a>
<span class="sourceLineNo">652</span>        /**<a name="line.652"></a>
<span class="sourceLineNo">653</span>         * Provides an equivalent copy of Atoms in a new array. Clones everything,<a name="line.653"></a>
<span class="sourceLineNo">654</span>         * starting with parent groups and chains. The chain will only contain<a name="line.654"></a>
<span class="sourceLineNo">655</span>         * groups that are part of the input array.<a name="line.655"></a>
<span class="sourceLineNo">656</span>         *<a name="line.656"></a>
<span class="sourceLineNo">657</span>         * @param ca<a name="line.657"></a>
<span class="sourceLineNo">658</span>         *            array of representative atoms, e.g. CA atoms<a name="line.658"></a>
<span class="sourceLineNo">659</span>         * @return Atom array<a name="line.659"></a>
<span class="sourceLineNo">660</span>         * @deprecated Use the better-named {@link #cloneAtomArray(Atom[])} instead<a name="line.660"></a>
<span class="sourceLineNo">661</span>         */<a name="line.661"></a>
<span class="sourceLineNo">662</span>        @Deprecated<a name="line.662"></a>
<span class="sourceLineNo">663</span>        public static final Atom[] cloneCAArray(Atom[] ca) {<a name="line.663"></a>
<span class="sourceLineNo">664</span>                return cloneAtomArray(ca);<a name="line.664"></a>
<span class="sourceLineNo">665</span>        }<a name="line.665"></a>
<span class="sourceLineNo">666</span><a name="line.666"></a>
<span class="sourceLineNo">667</span>        /**<a name="line.667"></a>
<span class="sourceLineNo">668</span>         * Provides an equivalent copy of Atoms in a new array. Clones everything,<a name="line.668"></a>
<span class="sourceLineNo">669</span>         * starting with parent groups and chains. The chain will only contain<a name="line.669"></a>
<span class="sourceLineNo">670</span>         * groups that are part of the input array.<a name="line.670"></a>
<span class="sourceLineNo">671</span>         *<a name="line.671"></a>
<span class="sourceLineNo">672</span>         * @param ca<a name="line.672"></a>
<span class="sourceLineNo">673</span>         *            array of representative atoms, e.g. CA atoms<a name="line.673"></a>
<span class="sourceLineNo">674</span>         * @return Atom array<a name="line.674"></a>
<span class="sourceLineNo">675</span>         * @since Biojava 4.1.0<a name="line.675"></a>
<span class="sourceLineNo">676</span>         */<a name="line.676"></a>
<span class="sourceLineNo">677</span>        public static final Atom[] cloneAtomArray(Atom[] ca) {<a name="line.677"></a>
<span class="sourceLineNo">678</span>                Atom[] newCA = new Atom[ca.length];<a name="line.678"></a>
<span class="sourceLineNo">679</span><a name="line.679"></a>
<span class="sourceLineNo">680</span>                List&lt;Chain&gt; model = new ArrayList&lt;Chain&gt;();<a name="line.680"></a>
<span class="sourceLineNo">681</span>                int apos = -1;<a name="line.681"></a>
<span class="sourceLineNo">682</span>                for (Atom a : ca) {<a name="line.682"></a>
<span class="sourceLineNo">683</span>                        apos++;<a name="line.683"></a>
<span class="sourceLineNo">684</span>                        Group parentG = a.getGroup();<a name="line.684"></a>
<span class="sourceLineNo">685</span>                        Chain parentC = parentG.getChain();<a name="line.685"></a>
<span class="sourceLineNo">686</span><a name="line.686"></a>
<span class="sourceLineNo">687</span>                        Chain newChain = null;<a name="line.687"></a>
<span class="sourceLineNo">688</span>                        for (Chain c : model) {<a name="line.688"></a>
<span class="sourceLineNo">689</span>                                if (c.getChainID().equals(parentC.getChainID())) {<a name="line.689"></a>
<span class="sourceLineNo">690</span>                                        newChain = c;<a name="line.690"></a>
<span class="sourceLineNo">691</span>                                        break;<a name="line.691"></a>
<span class="sourceLineNo">692</span>                                }<a name="line.692"></a>
<span class="sourceLineNo">693</span>                        }<a name="line.693"></a>
<span class="sourceLineNo">694</span>                        if (newChain == null) {<a name="line.694"></a>
<span class="sourceLineNo">695</span>                                newChain = new ChainImpl();<a name="line.695"></a>
<span class="sourceLineNo">696</span>                                newChain.setChainID(parentC.getChainID());<a name="line.696"></a>
<span class="sourceLineNo">697</span>                                model.add(newChain);<a name="line.697"></a>
<span class="sourceLineNo">698</span>                        }<a name="line.698"></a>
<span class="sourceLineNo">699</span><a name="line.699"></a>
<span class="sourceLineNo">700</span>                        Group parentN = (Group) parentG.clone();<a name="line.700"></a>
<span class="sourceLineNo">701</span><a name="line.701"></a>
<span class="sourceLineNo">702</span>                        newCA[apos] = parentN.getAtom(a.getName());<a name="line.702"></a>
<span class="sourceLineNo">703</span>                        try {<a name="line.703"></a>
<span class="sourceLineNo">704</span>                                // if the group doesn't exist yet, this produces a StructureException<a name="line.704"></a>
<span class="sourceLineNo">705</span>                                newChain.getGroupByPDB(parentN.getResidueNumber());<a name="line.705"></a>
<span class="sourceLineNo">706</span>                        } catch (StructureException e) {<a name="line.706"></a>
<span class="sourceLineNo">707</span>                                // the group doesn't exist yet in the newChain, let's add it<a name="line.707"></a>
<span class="sourceLineNo">708</span>                                newChain.addGroup(parentN);<a name="line.708"></a>
<span class="sourceLineNo">709</span>                        }<a name="line.709"></a>
<span class="sourceLineNo">710</span><a name="line.710"></a>
<span class="sourceLineNo">711</span>                }<a name="line.711"></a>
<span class="sourceLineNo">712</span>                return newCA;<a name="line.712"></a>
<span class="sourceLineNo">713</span>        }<a name="line.713"></a>
<span class="sourceLineNo">714</span><a name="line.714"></a>
<span class="sourceLineNo">715</span>        /**<a name="line.715"></a>
<span class="sourceLineNo">716</span>         * Clone a set of representative Atoms, but returns the parent groups<a name="line.716"></a>
<span class="sourceLineNo">717</span>         *<a name="line.717"></a>
<span class="sourceLineNo">718</span>         * @param ca<a name="line.718"></a>
<span class="sourceLineNo">719</span>         *            Atom array<a name="line.719"></a>
<span class="sourceLineNo">720</span>         * @return Group array<a name="line.720"></a>
<span class="sourceLineNo">721</span>         */<a name="line.721"></a>
<span class="sourceLineNo">722</span>        public static Group[] cloneGroups(Atom[] ca) {<a name="line.722"></a>
<span class="sourceLineNo">723</span>                Group[] newGroup = new Group[ca.length];<a name="line.723"></a>
<span class="sourceLineNo">724</span><a name="line.724"></a>
<span class="sourceLineNo">725</span>                List&lt;Chain&gt; model = new ArrayList&lt;Chain&gt;();<a name="line.725"></a>
<span class="sourceLineNo">726</span>                int apos = -1;<a name="line.726"></a>
<span class="sourceLineNo">727</span>                for (Atom a : ca) {<a name="line.727"></a>
<span class="sourceLineNo">728</span>                        apos++;<a name="line.728"></a>
<span class="sourceLineNo">729</span>                        Group parentG = a.getGroup();<a name="line.729"></a>
<span class="sourceLineNo">730</span>                        Chain parentC = parentG.getChain();<a name="line.730"></a>
<span class="sourceLineNo">731</span><a name="line.731"></a>
<span class="sourceLineNo">732</span>                        Chain newChain = null;<a name="line.732"></a>
<span class="sourceLineNo">733</span>                        for (Chain c : model) {<a name="line.733"></a>
<span class="sourceLineNo">734</span>                                if (c.getChainID().equals(parentC.getChainID())) {<a name="line.734"></a>
<span class="sourceLineNo">735</span>                                        newChain = c;<a name="line.735"></a>
<span class="sourceLineNo">736</span>                                        break;<a name="line.736"></a>
<span class="sourceLineNo">737</span>                                }<a name="line.737"></a>
<span class="sourceLineNo">738</span>                        }<a name="line.738"></a>
<span class="sourceLineNo">739</span>                        if (newChain == null) {<a name="line.739"></a>
<span class="sourceLineNo">740</span>                                newChain = new ChainImpl();<a name="line.740"></a>
<span class="sourceLineNo">741</span>                                newChain.setChainID(parentC.getChainID());<a name="line.741"></a>
<span class="sourceLineNo">742</span>                                model.add(newChain);<a name="line.742"></a>
<span class="sourceLineNo">743</span>                        }<a name="line.743"></a>
<span class="sourceLineNo">744</span><a name="line.744"></a>
<span class="sourceLineNo">745</span>                        Group ng = (Group) parentG.clone();<a name="line.745"></a>
<span class="sourceLineNo">746</span>                        newGroup[apos] = ng;<a name="line.746"></a>
<span class="sourceLineNo">747</span>                        newChain.addGroup(ng);<a name="line.747"></a>
<span class="sourceLineNo">748</span>                }<a name="line.748"></a>
<span class="sourceLineNo">749</span>                return newGroup;<a name="line.749"></a>
<span class="sourceLineNo">750</span>        }<a name="line.750"></a>
<span class="sourceLineNo">751</span><a name="line.751"></a>
<span class="sourceLineNo">752</span>        /**<a name="line.752"></a>
<span class="sourceLineNo">753</span>         * Utility method for working with circular permutations. Creates a<a name="line.753"></a>
<span class="sourceLineNo">754</span>         * duplicated and cloned set of Calpha atoms from the input array.<a name="line.754"></a>
<span class="sourceLineNo">755</span>         *<a name="line.755"></a>
<span class="sourceLineNo">756</span>         * @param ca2<a name="line.756"></a>
<span class="sourceLineNo">757</span>         *            atom array<a name="line.757"></a>
<span class="sourceLineNo">758</span>         * @return cloned and duplicated set of input array<a name="line.758"></a>
<span class="sourceLineNo">759</span>         */<a name="line.759"></a>
<span class="sourceLineNo">760</span>        public static Atom[] duplicateCA2(Atom[] ca2) {<a name="line.760"></a>
<span class="sourceLineNo">761</span>                // we don't want to rotate input atoms, do we?<a name="line.761"></a>
<span class="sourceLineNo">762</span>                Atom[] ca2clone = new Atom[ca2.length * 2];<a name="line.762"></a>
<span class="sourceLineNo">763</span><a name="line.763"></a>
<span class="sourceLineNo">764</span>                int pos = 0;<a name="line.764"></a>
<span class="sourceLineNo">765</span><a name="line.765"></a>
<span class="sourceLineNo">766</span>                Chain c = null;<a name="line.766"></a>
<span class="sourceLineNo">767</span>                String prevChainId = "";<a name="line.767"></a>
<span class="sourceLineNo">768</span>                for (Atom a : ca2) {<a name="line.768"></a>
<span class="sourceLineNo">769</span>                        Group g = (Group) a.getGroup().clone(); // works because each group<a name="line.769"></a>
<span class="sourceLineNo">770</span>                                                                                                        // has only a single atom<a name="line.770"></a>
<span class="sourceLineNo">771</span><a name="line.771"></a>
<span class="sourceLineNo">772</span>                        if (c == null) {<a name="line.772"></a>
<span class="sourceLineNo">773</span>                                c = new ChainImpl();<a name="line.773"></a>
<span class="sourceLineNo">774</span>                                Chain orig = a.getGroup().getChain();<a name="line.774"></a>
<span class="sourceLineNo">775</span>                                c.setChainID(orig.getChainID());<a name="line.775"></a>
<span class="sourceLineNo">776</span>                        } else {<a name="line.776"></a>
<span class="sourceLineNo">777</span>                                Chain orig = a.getGroup().getChain();<a name="line.777"></a>
<span class="sourceLineNo">778</span>                                if (!orig.getChainID().equals(prevChainId)) {<a name="line.778"></a>
<span class="sourceLineNo">779</span>                                        c = new ChainImpl();<a name="line.779"></a>
<span class="sourceLineNo">780</span>                                        c.setChainID(orig.getChainID());<a name="line.780"></a>
<span class="sourceLineNo">781</span>                                }<a name="line.781"></a>
<span class="sourceLineNo">782</span>                        }<a name="line.782"></a>
<span class="sourceLineNo">783</span><a name="line.783"></a>
<span class="sourceLineNo">784</span>                        c.addGroup(g);<a name="line.784"></a>
<span class="sourceLineNo">785</span>                        ca2clone[pos] = g.getAtom(a.getName());<a name="line.785"></a>
<span class="sourceLineNo">786</span><a name="line.786"></a>
<span class="sourceLineNo">787</span>                        pos++;<a name="line.787"></a>
<span class="sourceLineNo">788</span>                }<a name="line.788"></a>
<span class="sourceLineNo">789</span><a name="line.789"></a>
<span class="sourceLineNo">790</span>                // Duplicate ca2!<a name="line.790"></a>
<span class="sourceLineNo">791</span>                c = null;<a name="line.791"></a>
<span class="sourceLineNo">792</span>                prevChainId = "";<a name="line.792"></a>
<span class="sourceLineNo">793</span>                for (Atom a : ca2) {<a name="line.793"></a>
<span class="sourceLineNo">794</span>                        Group g = (Group) a.getGroup().clone();<a name="line.794"></a>
<span class="sourceLineNo">795</span><a name="line.795"></a>
<span class="sourceLineNo">796</span>                        if (c == null) {<a name="line.796"></a>
<span class="sourceLineNo">797</span>                                c = new ChainImpl();<a name="line.797"></a>
<span class="sourceLineNo">798</span>                                Chain orig = a.getGroup().getChain();<a name="line.798"></a>
<span class="sourceLineNo">799</span>                                c.setChainID(orig.getChainID());<a name="line.799"></a>
<span class="sourceLineNo">800</span>                        } else {<a name="line.800"></a>
<span class="sourceLineNo">801</span>                                Chain orig = a.getGroup().getChain();<a name="line.801"></a>
<span class="sourceLineNo">802</span>                                if (!orig.getChainID().equals(prevChainId)) {<a name="line.802"></a>
<span class="sourceLineNo">803</span>                                        c = new ChainImpl();<a name="line.803"></a>
<span class="sourceLineNo">804</span>                                        c.setChainID(orig.getChainID());<a name="line.804"></a>
<span class="sourceLineNo">805</span>                                }<a name="line.805"></a>
<span class="sourceLineNo">806</span>                        }<a name="line.806"></a>
<span class="sourceLineNo">807</span><a name="line.807"></a>
<span class="sourceLineNo">808</span>                        c.addGroup(g);<a name="line.808"></a>
<span class="sourceLineNo">809</span>                        ca2clone[pos] = g.getAtom(a.getName());<a name="line.809"></a>
<span class="sourceLineNo">810</span><a name="line.810"></a>
<span class="sourceLineNo">811</span>                        pos++;<a name="line.811"></a>
<span class="sourceLineNo">812</span>                }<a name="line.812"></a>
<span class="sourceLineNo">813</span><a name="line.813"></a>
<span class="sourceLineNo">814</span>                return ca2clone;<a name="line.814"></a>
<span class="sourceLineNo">815</span><a name="line.815"></a>
<span class="sourceLineNo">816</span>        }<a name="line.816"></a>
<span class="sourceLineNo">817</span><a name="line.817"></a>
<span class="sourceLineNo">818</span>        /**<a name="line.818"></a>
<span class="sourceLineNo">819</span>         * Return an Atom array of the C-alpha atoms. Any atom that is a carbon and<a name="line.819"></a>
<span class="sourceLineNo">820</span>         * has CA name will be returned.<a name="line.820"></a>
<span class="sourceLineNo">821</span>         *<a name="line.821"></a>
<span class="sourceLineNo">822</span>         * @param s<a name="line.822"></a>
<span class="sourceLineNo">823</span>         *            the structure object<a name="line.823"></a>
<span class="sourceLineNo">824</span>         * @return an Atom[] array<a name="line.824"></a>
<span class="sourceLineNo">825</span>         * @see #getRepresentativeAtomArray(Structure)<a name="line.825"></a>
<span class="sourceLineNo">826</span>         */<a name="line.826"></a>
<span class="sourceLineNo">827</span>        public static Atom[] getAtomCAArray(Structure s) {<a name="line.827"></a>
<span class="sourceLineNo">828</span><a name="line.828"></a>
<span class="sourceLineNo">829</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.829"></a>
<span class="sourceLineNo">830</span><a name="line.830"></a>
<span class="sourceLineNo">831</span>                for (Chain c : s.getChains()) {<a name="line.831"></a>
<span class="sourceLineNo">832</span>                        for (Group g : c.getAtomGroups()) {<a name="line.832"></a>
<span class="sourceLineNo">833</span>                                if (g.hasAtom(CA_ATOM_NAME)<a name="line.833"></a>
<span class="sourceLineNo">834</span>                                                &amp;&amp; g.getAtom(CA_ATOM_NAME).getElement() == Element.C) {<a name="line.834"></a>
<span class="sourceLineNo">835</span>                                        atoms.add(g.getAtom(CA_ATOM_NAME));<a name="line.835"></a>
<span class="sourceLineNo">836</span>                                }<a name="line.836"></a>
<span class="sourceLineNo">837</span>                        }<a name="line.837"></a>
<span class="sourceLineNo">838</span>                }<a name="line.838"></a>
<span class="sourceLineNo">839</span><a name="line.839"></a>
<span class="sourceLineNo">840</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.840"></a>
<span class="sourceLineNo">841</span>        }<a name="line.841"></a>
<span class="sourceLineNo">842</span><a name="line.842"></a>
<span class="sourceLineNo">843</span>        /**<a name="line.843"></a>
<span class="sourceLineNo">844</span>         * Gets a representative atom for each group that is part of the chain<a name="line.844"></a>
<span class="sourceLineNo">845</span>         * backbone. Note that modified aminoacids won't be returned as part of the<a name="line.845"></a>
<span class="sourceLineNo">846</span>         * backbone if the {@link org.biojava.nbio.structure.io.mmcif.ReducedChemCompProvider} was used to load the<a name="line.846"></a>
<span class="sourceLineNo">847</span>         * structure.<a name="line.847"></a>
<span class="sourceLineNo">848</span>         *<a name="line.848"></a>
<span class="sourceLineNo">849</span>         * For amino acids, the representative is a CA carbon. For nucleotides, the<a name="line.849"></a>
<span class="sourceLineNo">850</span>         * representative is the {@value #NUCLEOTIDE_REPRESENTATIVE}. Other group<a name="line.850"></a>
<span class="sourceLineNo">851</span>         * types will be ignored.<a name="line.851"></a>
<span class="sourceLineNo">852</span>         *<a name="line.852"></a>
<span class="sourceLineNo">853</span>         * @param s<a name="line.853"></a>
<span class="sourceLineNo">854</span>         *            Input structure<a name="line.854"></a>
<span class="sourceLineNo">855</span>         * @return representative Atoms of the structure backbone<a name="line.855"></a>
<span class="sourceLineNo">856</span>         * @since Biojava 4.1.0<a name="line.856"></a>
<span class="sourceLineNo">857</span>         */<a name="line.857"></a>
<span class="sourceLineNo">858</span>        public static Atom[] getRepresentativeAtomArray(Structure s) {<a name="line.858"></a>
<span class="sourceLineNo">859</span><a name="line.859"></a>
<span class="sourceLineNo">860</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.860"></a>
<span class="sourceLineNo">861</span><a name="line.861"></a>
<span class="sourceLineNo">862</span>                for (Chain c : s.getChains()) {<a name="line.862"></a>
<span class="sourceLineNo">863</span>                        Atom[] chainAtoms = getRepresentativeAtomArray(c);<a name="line.863"></a>
<span class="sourceLineNo">864</span>                        for (Atom a : chainAtoms) {<a name="line.864"></a>
<span class="sourceLineNo">865</span>                                atoms.add(a);<a name="line.865"></a>
<span class="sourceLineNo">866</span>                        }<a name="line.866"></a>
<span class="sourceLineNo">867</span>                }<a name="line.867"></a>
<span class="sourceLineNo">868</span><a name="line.868"></a>
<span class="sourceLineNo">869</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.869"></a>
<span class="sourceLineNo">870</span>        }<a name="line.870"></a>
<span class="sourceLineNo">871</span><a name="line.871"></a>
<span class="sourceLineNo">872</span>        /**<a name="line.872"></a>
<span class="sourceLineNo">873</span>         * Return an Atom array of the main chain atoms: CA, C, N, O Any group that<a name="line.873"></a>
<span class="sourceLineNo">874</span>         * contains those atoms will be included, be it a standard aminoacid or not<a name="line.874"></a>
<span class="sourceLineNo">875</span>         *<a name="line.875"></a>
<span class="sourceLineNo">876</span>         * @param s<a name="line.876"></a>
<span class="sourceLineNo">877</span>         *            the structure object<a name="line.877"></a>
<span class="sourceLineNo">878</span>         * @return an Atom[] array<a name="line.878"></a>
<span class="sourceLineNo">879</span>         */<a name="line.879"></a>
<span class="sourceLineNo">880</span>        public static Atom[] getBackboneAtomArray(Structure s) {<a name="line.880"></a>
<span class="sourceLineNo">881</span><a name="line.881"></a>
<span class="sourceLineNo">882</span>                List&lt;Atom&gt; atoms = new ArrayList&lt;Atom&gt;();<a name="line.882"></a>
<span class="sourceLineNo">883</span><a name="line.883"></a>
<span class="sourceLineNo">884</span>                for (Chain c : s.getChains()) {<a name="line.884"></a>
<span class="sourceLineNo">885</span>                        for (Group g : c.getAtomGroups()) {<a name="line.885"></a>
<span class="sourceLineNo">886</span>                                if (g.hasAminoAtoms()) {<a name="line.886"></a>
<span class="sourceLineNo">887</span>                                        // this means we will only take atoms grom groups that have<a name="line.887"></a>
<span class="sourceLineNo">888</span>                                        // complete backbones<a name="line.888"></a>
<span class="sourceLineNo">889</span>                                        for (Atom a : g.getAtoms()) {<a name="line.889"></a>
<span class="sourceLineNo">890</span>                                                switch (g.getType()) {<a name="line.890"></a>
<span class="sourceLineNo">891</span>                                                case NUCLEOTIDE:<a name="line.891"></a>
<span class="sourceLineNo">892</span>                                                        // Nucleotide backbone<a name="line.892"></a>
<span class="sourceLineNo">893</span>                                                        if (a.getName().equals(C1_ATOM_NAME))<a name="line.893"></a>
<span class="sourceLineNo">894</span>                                                                atoms.add(a);<a name="line.894"></a>
<span class="sourceLineNo">895</span>                                                        if (a.getName().equals(C2_ATOM_NAME))<a name="line.895"></a>
<span class="sourceLineNo">896</span>                                                                atoms.add(a);<a name="line.896"></a>
<span class="sourceLineNo">897</span>                                                        if (a.getName().equals(C3_ATOM_NAME))<a name="line.897"></a>
<span class="sourceLineNo">898</span>                                                                atoms.add(a);<a name="line.898"></a>
<span class="sourceLineNo">899</span>                                                        if (a.getName().equals(C4_ATOM_NAME))<a name="line.899"></a>
<span class="sourceLineNo">900</span>                                                                atoms.add(a);<a name="line.900"></a>
<span class="sourceLineNo">901</span>                                                        if (a.getName().equals(O2_ATOM_NAME))<a name="line.901"></a>
<span class="sourceLineNo">902</span>                                                                atoms.add(a);<a name="line.902"></a>
<span class="sourceLineNo">903</span>                                                        if (a.getName().equals(O3_ATOM_NAME))<a name="line.903"></a>
<span class="sourceLineNo">904</span>                                                                atoms.add(a);<a name="line.904"></a>
<span class="sourceLineNo">905</span>                                                        if (a.getName().equals(O4_ATOM_NAME))<a name="line.905"></a>
<span class="sourceLineNo">906</span>                                                                atoms.add(a);<a name="line.906"></a>
<span class="sourceLineNo">907</span>                                                        if (a.getName().equals(O5_ATOM_NAME))<a name="line.907"></a>
<span class="sourceLineNo">908</span>                                                                atoms.add(a);<a name="line.908"></a>
<span class="sourceLineNo">909</span>                                                        if (a.getName().equals(OP1_ATOM_NAME))<a name="line.909"></a>
<span class="sourceLineNo">910</span>                                                                atoms.add(a);<a name="line.910"></a>
<span class="sourceLineNo">911</span>                                                        if (a.getName().equals(OP2_ATOM_NAME))<a name="line.911"></a>
<span class="sourceLineNo">912</span>                                                                atoms.add(a);<a name="line.912"></a>
<span class="sourceLineNo">913</span>                                                        if (a.getName().equals(P_ATOM_NAME))<a name="line.913"></a>
<span class="sourceLineNo">914</span>                                                                atoms.add(a);<a name="line.914"></a>
<span class="sourceLineNo">915</span>                                                        // TODO Allow C4* names as well as C4'? -SB 3/2015<a name="line.915"></a>
<span class="sourceLineNo">916</span>                                                        break;<a name="line.916"></a>
<span class="sourceLineNo">917</span>                                                case AMINOACID:<a name="line.917"></a>
<span class="sourceLineNo">918</span>                                                default:<a name="line.918"></a>
<span class="sourceLineNo">919</span>                                                        // we do it this way instead of with g.getAtom() to<a name="line.919"></a>
<span class="sourceLineNo">920</span>                                                        // be sure we always use the same order as original<a name="line.920"></a>
<span class="sourceLineNo">921</span>                                                        if (a.getName().equals(CA_ATOM_NAME))<a name="line.921"></a>
<span class="sourceLineNo">922</span>                                                                atoms.add(a);<a name="line.922"></a>
<span class="sourceLineNo">923</span>                                                        if (a.getName().equals(C_ATOM_NAME))<a name="line.923"></a>
<span class="sourceLineNo">924</span>                                                                atoms.add(a);<a name="line.924"></a>
<span class="sourceLineNo">925</span>                                                        if (a.getName().equals(N_ATOM_NAME))<a name="line.925"></a>
<span class="sourceLineNo">926</span>                                                                atoms.add(a);<a name="line.926"></a>
<span class="sourceLineNo">927</span>                                                        if (a.getName().equals(O_ATOM_NAME))<a name="line.927"></a>
<span class="sourceLineNo">928</span>                                                                atoms.add(a);<a name="line.928"></a>
<span class="sourceLineNo">929</span>                                                        break;<a name="line.929"></a>
<span class="sourceLineNo">930</span>                                                }<a name="line.930"></a>
<span class="sourceLineNo">931</span>                                        }<a name="line.931"></a>
<span class="sourceLineNo">932</span>                                }<a name="line.932"></a>
<span class="sourceLineNo">933</span>                        }<a name="line.933"></a>
<span class="sourceLineNo">934</span><a name="line.934"></a>
<span class="sourceLineNo">935</span>                }<a name="line.935"></a>
<span class="sourceLineNo">936</span><a name="line.936"></a>
<span class="sourceLineNo">937</span>                return atoms.toArray(new Atom[atoms.size()]);<a name="line.937"></a>
<span class="sourceLineNo">938</span>        }<a name="line.938"></a>
<span class="sourceLineNo">939</span><a name="line.939"></a>
<span class="sourceLineNo">940</span>        /**<a name="line.940"></a>
<span class="sourceLineNo">941</span>         * Convert three character amino acid codes into single character e.g.<a name="line.941"></a>
<span class="sourceLineNo">942</span>         * convert CYS to C. Valid 3-letter codes will be those of the standard 20<a name="line.942"></a>
<span class="sourceLineNo">943</span>         * amino acids plus MSE, CSE, SEC, PYH, PYL (see the {@link #aminoAcids}<a name="line.943"></a>
<span class="sourceLineNo">944</span>         * map)<a name="line.944"></a>
<span class="sourceLineNo">945</span>         *<a name="line.945"></a>
<span class="sourceLineNo">946</span>         * @return the 1 letter code, or null if the given 3 letter code does not<a name="line.946"></a>
<span class="sourceLineNo">947</span>         *         correspond to an amino acid code<a name="line.947"></a>
<span class="sourceLineNo">948</span>         * @param groupCode3<a name="line.948"></a>
<span class="sourceLineNo">949</span>         *            a three character amino acid representation String<a name="line.949"></a>
<span class="sourceLineNo">950</span>         * @see {@link #get1LetterCode(String)}<a name="line.950"></a>
<span class="sourceLineNo">951</span>         */<a name="line.951"></a>
<span class="sourceLineNo">952</span>        public static final Character get1LetterCodeAmino(String groupCode3) {<a name="line.952"></a>
<span class="sourceLineNo">953</span>                return aminoAcids.get(groupCode3);<a name="line.953"></a>
<span class="sourceLineNo">954</span>        }<a name="line.954"></a>
<span class="sourceLineNo">955</span><a name="line.955"></a>
<span class="sourceLineNo">956</span>        /**<a name="line.956"></a>
<span class="sourceLineNo">957</span>         *<a name="line.957"></a>
<span class="sourceLineNo">958</span>         * @param code3<a name="line.958"></a>
<span class="sourceLineNo">959</span>         * @return<a name="line.959"></a>
<span class="sourceLineNo">960</span>         * @deprecated Use {@link #get1LetterCodeAmino(String)} instead<a name="line.960"></a>
<span class="sourceLineNo">961</span>         */<a name="line.961"></a>
<span class="sourceLineNo">962</span>        @Deprecated<a name="line.962"></a>
<span class="sourceLineNo">963</span>        public static final Character convert_3code_1code(String code3) {<a name="line.963"></a>
<span class="sourceLineNo">964</span>                return get1LetterCodeAmino(code3);<a name="line.964"></a>
<span class="sourceLineNo">965</span>        }<a name="line.965"></a>
<span class="sourceLineNo">966</span><a name="line.966"></a>
<span class="sourceLineNo">967</span>        /**<a name="line.967"></a>
<span class="sourceLineNo">968</span>         * Convert a three letter amino acid or nucleotide code into a single<a name="line.968"></a>
<span class="sourceLineNo">969</span>         * character code. If the code does not correspond to an amino acid or<a name="line.969"></a>
<span class="sourceLineNo">970</span>         * nucleotide, returns {@link #UNKNOWN_GROUP_LABEL}.<a name="line.970"></a>
<span class="sourceLineNo">971</span>         *<a name="line.971"></a>
<span class="sourceLineNo">972</span>         * Returned null for nucleotides prior to version 4.0.1.<a name="line.972"></a>
<span class="sourceLineNo">973</span>         *<a name="line.973"></a>
<span class="sourceLineNo">974</span>         * @param groupCode3<a name="line.974"></a>
<span class="sourceLineNo">975</span>         *            three letter representation<a name="line.975"></a>
<span class="sourceLineNo">976</span>         * @return The 1-letter abbreviation<a name="line.976"></a>
<span class="sourceLineNo">977</span>         */<a name="line.977"></a>
<span class="sourceLineNo">978</span>        public static final Character get1LetterCode(String groupCode3) {<a name="line.978"></a>
<span class="sourceLineNo">979</span><a name="line.979"></a>
<span class="sourceLineNo">980</span>                Character code1;<a name="line.980"></a>
<span class="sourceLineNo">981</span><a name="line.981"></a>
<span class="sourceLineNo">982</span>                // is it a standard amino acid ?<a name="line.982"></a>
<span class="sourceLineNo">983</span>                code1 = get1LetterCodeAmino(groupCode3);<a name="line.983"></a>
<span class="sourceLineNo">984</span><a name="line.984"></a>
<span class="sourceLineNo">985</span>                if (code1 == null) {<a name="line.985"></a>
<span class="sourceLineNo">986</span>                        // hm groupCode3 is not standard<a name="line.986"></a>
<span class="sourceLineNo">987</span>                        // perhaps it is a nucleotide?<a name="line.987"></a>
<span class="sourceLineNo">988</span>                        groupCode3 = groupCode3.trim();<a name="line.988"></a>
<span class="sourceLineNo">989</span>                        if (isNucleotide(groupCode3)) {<a name="line.989"></a>
<span class="sourceLineNo">990</span>                                code1 = nucleotides30.get(groupCode3);<a name="line.990"></a>
<span class="sourceLineNo">991</span>                                if (code1 == null) {<a name="line.991"></a>
<span class="sourceLineNo">992</span>                                        code1 = nucleotides23.get(groupCode3);<a name="line.992"></a>
<span class="sourceLineNo">993</span>                                }<a name="line.993"></a>
<span class="sourceLineNo">994</span>                                if (code1 == null) {<a name="line.994"></a>
<span class="sourceLineNo">995</span>                                        code1 = UNKNOWN_GROUP_LABEL;<a name="line.995"></a>
<span class="sourceLineNo">996</span>                                }<a name="line.996"></a>
<span class="sourceLineNo">997</span>                        } else {<a name="line.997"></a>
<span class="sourceLineNo">998</span>                                // does not seem to be so let's assume it is<a name="line.998"></a>
<span class="sourceLineNo">999</span>                                // nonstandard aminoacid and label it "X"<a name="line.999"></a>
<span class="sourceLineNo">1000</span>                                // logger.warning("unknown group name "+groupCode3 );<a name="line.1000"></a>
<span class="sourceLineNo">1001</span>                                code1 = UNKNOWN_GROUP_LABEL;<a name="line.1001"></a>
<span class="sourceLineNo">1002</span>                        }<a name="line.1002"></a>
<span class="sourceLineNo">1003</span>                }<a name="line.1003"></a>
<span class="sourceLineNo">1004</span><a name="line.1004"></a>
<span class="sourceLineNo">1005</span>                return code1;<a name="line.1005"></a>
<span class="sourceLineNo">1006</span><a name="line.1006"></a>
<span class="sourceLineNo">1007</span>        }<a name="line.1007"></a>
<span class="sourceLineNo">1008</span><a name="line.1008"></a>
<span class="sourceLineNo">1009</span>        /**<a name="line.1009"></a>
<span class="sourceLineNo">1010</span>         * Test if the three-letter code of an ATOM entry corresponds to a<a name="line.1010"></a>
<span class="sourceLineNo">1011</span>         * nucleotide or to an aminoacid.<a name="line.1011"></a>
<span class="sourceLineNo">1012</span>         *<a name="line.1012"></a>
<span class="sourceLineNo">1013</span>         * @param groupCode3<a name="line.1013"></a>
<span class="sourceLineNo">1014</span>         *            3-character code for a group.<a name="line.1014"></a>
<span class="sourceLineNo">1015</span>         *<a name="line.1015"></a>
<span class="sourceLineNo">1016</span>         */<a name="line.1016"></a>
<span class="sourceLineNo">1017</span>        public static final boolean isNucleotide(String groupCode3) {<a name="line.1017"></a>
<span class="sourceLineNo">1018</span>                String code = groupCode3.trim();<a name="line.1018"></a>
<span class="sourceLineNo">1019</span>                return nucleotides30.containsKey(code)<a name="line.1019"></a>
<span class="sourceLineNo">1020</span>                                || nucleotides23.containsKey(code);<a name="line.1020"></a>
<span class="sourceLineNo">1021</span>        }<a name="line.1021"></a>
<span class="sourceLineNo">1022</span><a name="line.1022"></a>
<span class="sourceLineNo">1023</span>        /**<a name="line.1023"></a>
<span class="sourceLineNo">1024</span>         * Reduce a structure to provide a smaller representation . Only takes the<a name="line.1024"></a>
<span class="sourceLineNo">1025</span>         * first model of the structure. If chainId is provided only return a<a name="line.1025"></a>
<span class="sourceLineNo">1026</span>         * structure containing that Chain ID. Converts lower case chain IDs to<a name="line.1026"></a>
<span class="sourceLineNo">1027</span>         * upper case if structure does not contain a chain with that ID.<a name="line.1027"></a>
<span class="sourceLineNo">1028</span>         *<a name="line.1028"></a>
<span class="sourceLineNo">1029</span>         * @param s<a name="line.1029"></a>
<span class="sourceLineNo">1030</span>         * @param chainId<a name="line.1030"></a>
<span class="sourceLineNo">1031</span>         * @return Structure<a name="line.1031"></a>
<span class="sourceLineNo">1032</span>         * @since 3.0<a name="line.1032"></a>
<span class="sourceLineNo">1033</span>         * @deprecated Use {@link StructureIdentifier#reduce(Structure)} instead (v. 4.2.0)<a name="line.1033"></a>
<span class="sourceLineNo">1034</span>         */<a name="line.1034"></a>
<span class="sourceLineNo">1035</span>        @Deprecated<a name="line.1035"></a>
<span class="sourceLineNo">1036</span>        public static final Structure getReducedStructure(Structure s,<a name="line.1036"></a>
<span class="sourceLineNo">1037</span>                        String chainId) throws StructureException {<a name="line.1037"></a>
<span class="sourceLineNo">1038</span>                // since we deal here with structure alignments,<a name="line.1038"></a>
<span class="sourceLineNo">1039</span>                // only use Model 1...<a name="line.1039"></a>
<span class="sourceLineNo">1040</span><a name="line.1040"></a>
<span class="sourceLineNo">1041</span>                Structure newS = new StructureImpl();<a name="line.1041"></a>
<span class="sourceLineNo">1042</span>                newS.setPDBCode(s.getPDBCode());<a name="line.1042"></a>
<span class="sourceLineNo">1043</span>                newS.setPDBHeader(s.getPDBHeader());<a name="line.1043"></a>
<span class="sourceLineNo">1044</span>                newS.setName(s.getName());<a name="line.1044"></a>
<span class="sourceLineNo">1045</span>                newS.setSSBonds(s.getSSBonds());<a name="line.1045"></a>
<span class="sourceLineNo">1046</span>                newS.setDBRefs(s.getDBRefs());<a name="line.1046"></a>
<span class="sourceLineNo">1047</span>                newS.setSites(s.getSites());<a name="line.1047"></a>
<span class="sourceLineNo">1048</span>                newS.setBiologicalAssembly(s.isBiologicalAssembly());<a name="line.1048"></a>
<span class="sourceLineNo">1049</span>                newS.setCompounds(s.getCompounds());<a name="line.1049"></a>
<span class="sourceLineNo">1050</span>                newS.setConnections(s.getConnections());<a name="line.1050"></a>
<span class="sourceLineNo">1051</span>                newS.setSSBonds(s.getSSBonds());<a name="line.1051"></a>
<span class="sourceLineNo">1052</span>                newS.setSites(s.getSites());<a name="line.1052"></a>
<span class="sourceLineNo">1053</span><a name="line.1053"></a>
<span class="sourceLineNo">1054</span>                if (chainId != null)<a name="line.1054"></a>
<span class="sourceLineNo">1055</span>                        chainId = chainId.trim();<a name="line.1055"></a>
<span class="sourceLineNo">1056</span><a name="line.1056"></a>
<span class="sourceLineNo">1057</span>                if (chainId == null || chainId.equals("")) {<a name="line.1057"></a>
<span class="sourceLineNo">1058</span>                        // only get model 0<a name="line.1058"></a>
<span class="sourceLineNo">1059</span>                        List&lt;Chain&gt; model0 = s.getModel(0);<a name="line.1059"></a>
<span class="sourceLineNo">1060</span>                        for (Chain c : model0) {<a name="line.1060"></a>
<span class="sourceLineNo">1061</span>                                newS.addChain(c);<a name="line.1061"></a>
<span class="sourceLineNo">1062</span>                        }<a name="line.1062"></a>
<span class="sourceLineNo">1063</span>                        return newS;<a name="line.1063"></a>
<span class="sourceLineNo">1064</span><a name="line.1064"></a>
<span class="sourceLineNo">1065</span>                }<a name="line.1065"></a>
<span class="sourceLineNo">1066</span><a name="line.1066"></a>
<span class="sourceLineNo">1067</span>                Chain c = null;<a name="line.1067"></a>
<span class="sourceLineNo">1068</span>                try {<a name="line.1068"></a>
<span class="sourceLineNo">1069</span>                        c = s.getChainByPDB(chainId);<a name="line.1069"></a>
<span class="sourceLineNo">1070</span>                } catch (StructureException e) {<a name="line.1070"></a>
<span class="sourceLineNo">1071</span>                        logger.warn(e.getMessage() + ". Chain id " + chainId<a name="line.1071"></a>
<span class="sourceLineNo">1072</span>                                        + " did not match, trying upper case Chain id.");<a name="line.1072"></a>
<span class="sourceLineNo">1073</span>                        c = s.getChainByPDB(chainId.toUpperCase());<a name="line.1073"></a>
<span class="sourceLineNo">1074</span><a name="line.1074"></a>
<span class="sourceLineNo">1075</span>                }<a name="line.1075"></a>
<span class="sourceLineNo">1076</span>                if (c != null) {<a name="line.1076"></a>
<span class="sourceLineNo">1077</span>                        newS.addChain(c);<a name="line.1077"></a>
<span class="sourceLineNo">1078</span>                        for (Compound comp : s.getCompounds()) {<a name="line.1078"></a>
<span class="sourceLineNo">1079</span>                                if (comp.getChainIds() != null<a name="line.1079"></a>
<span class="sourceLineNo">1080</span>                                                &amp;&amp; comp.getChainIds().contains(c.getChainID())) {<a name="line.1080"></a>
<span class="sourceLineNo">1081</span>                                        // found matching compound. set description...<a name="line.1081"></a>
<span class="sourceLineNo">1082</span>                                        newS.getPDBHeader().setDescription(<a name="line.1082"></a>
<span class="sourceLineNo">1083</span>                                                        "Chain " + c.getChainID() + " of " + s.getPDBCode()<a name="line.1083"></a>
<span class="sourceLineNo">1084</span>                                                                        + " " + comp.getMolName());<a name="line.1084"></a>
<span class="sourceLineNo">1085</span>                                }<a name="line.1085"></a>
<span class="sourceLineNo">1086</span>                        }<a name="line.1086"></a>
<span class="sourceLineNo">1087</span>                }<a name="line.1087"></a>
<span class="sourceLineNo">1088</span><a name="line.1088"></a>
<span class="sourceLineNo">1089</span>                return newS;<a name="line.1089"></a>
<span class="sourceLineNo">1090</span>        }<a name="line.1090"></a>
<span class="sourceLineNo">1091</span><a name="line.1091"></a>
<span class="sourceLineNo">1092</span>        /**<a name="line.1092"></a>
<span class="sourceLineNo">1093</span>         * Reduce a structure to provide a smaller representation. Only takes the<a name="line.1093"></a>
<span class="sourceLineNo">1094</span>         * first model of the structure. If chainNr &gt;=0 only takes the chain at that<a name="line.1094"></a>
<span class="sourceLineNo">1095</span>         * position into account.<a name="line.1095"></a>
<span class="sourceLineNo">1096</span>         *<a name="line.1096"></a>
<span class="sourceLineNo">1097</span>         * @param s<a name="line.1097"></a>
<span class="sourceLineNo">1098</span>         * @param chainNr<a name="line.1098"></a>
<span class="sourceLineNo">1099</span>         *            can be -1 to request all chains of model 0, otherwise will<a name="line.1099"></a>
<span class="sourceLineNo">1100</span>         *            only add chain at this position<a name="line.1100"></a>
<span class="sourceLineNo">1101</span>         * @return Structure object<a name="line.1101"></a>
<span class="sourceLineNo">1102</span>         * @since 3.0<a name="line.1102"></a>
<span class="sourceLineNo">1103</span>         * @deprecated Use {@link StructureIdentifier#reduce(Structure)} instead (v. 4.2.0)<a name="line.1103"></a>
<span class="sourceLineNo">1104</span>         */<a name="line.1104"></a>
<span class="sourceLineNo">1105</span>        @Deprecated<a name="line.1105"></a>
<span class="sourceLineNo">1106</span>        public static final Structure getReducedStructure(Structure s, int chainNr)<a name="line.1106"></a>
<span class="sourceLineNo">1107</span>                        throws StructureException {<a name="line.1107"></a>
<span class="sourceLineNo">1108</span>                // since we deal here with structure alignments,<a name="line.1108"></a>
<span class="sourceLineNo">1109</span>                // only use Model 1...<a name="line.1109"></a>
<span class="sourceLineNo">1110</span><a name="line.1110"></a>
<span class="sourceLineNo">1111</span>                Structure newS = new StructureImpl();<a name="line.1111"></a>
<span class="sourceLineNo">1112</span>                newS.setPDBCode(s.getPDBCode());<a name="line.1112"></a>
<span class="sourceLineNo">1113</span>                newS.setPDBHeader(s.getPDBHeader());<a name="line.1113"></a>
<span class="sourceLineNo">1114</span>                newS.setName(s.getName());<a name="line.1114"></a>
<span class="sourceLineNo">1115</span>                newS.setSSBonds(s.getSSBonds());<a name="line.1115"></a>
<span class="sourceLineNo">1116</span>                newS.setDBRefs(s.getDBRefs());<a name="line.1116"></a>
<span class="sourceLineNo">1117</span>                newS.setSites(s.getSites());<a name="line.1117"></a>
<span class="sourceLineNo">1118</span>                newS.setBiologicalAssembly(s.isBiologicalAssembly());<a name="line.1118"></a>
<span class="sourceLineNo">1119</span>                newS.setCompounds(s.getCompounds());<a name="line.1119"></a>
<span class="sourceLineNo">1120</span>                newS.setConnections(s.getConnections());<a name="line.1120"></a>
<span class="sourceLineNo">1121</span>                newS.setSSBonds(s.getSSBonds());<a name="line.1121"></a>
<span class="sourceLineNo">1122</span>                newS.setSites(s.getSites());<a name="line.1122"></a>
<span class="sourceLineNo">1123</span>                newS.setCrystallographicInfo(s.getCrystallographicInfo());<a name="line.1123"></a>
<span class="sourceLineNo">1124</span>                newS.getPDBHeader().setDescription(<a name="line.1124"></a>
<span class="sourceLineNo">1125</span>                                "subset of " + s.getPDBCode() + " "<a name="line.1125"></a>
<span class="sourceLineNo">1126</span>                                                + s.getPDBHeader().getDescription());<a name="line.1126"></a>
<span class="sourceLineNo">1127</span><a name="line.1127"></a>
<span class="sourceLineNo">1128</span>                if (chainNr &lt; 0) {<a name="line.1128"></a>
<span class="sourceLineNo">1129</span><a name="line.1129"></a>
<span class="sourceLineNo">1130</span>                        // only get model 0<a name="line.1130"></a>
<span class="sourceLineNo">1131</span>                        List&lt;Chain&gt; model0 = s.getModel(0);<a name="line.1131"></a>
<span class="sourceLineNo">1132</span>                        for (Chain c : model0) {<a name="line.1132"></a>
<span class="sourceLineNo">1133</span>                                newS.addChain(c);<a name="line.1133"></a>
<span class="sourceLineNo">1134</span>                        }<a name="line.1134"></a>
<span class="sourceLineNo">1135</span>                        return newS;<a name="line.1135"></a>
<span class="sourceLineNo">1136</span>                }<a name="line.1136"></a>
<span class="sourceLineNo">1137</span>                Chain c = null;<a name="line.1137"></a>
<span class="sourceLineNo">1138</span><a name="line.1138"></a>
<span class="sourceLineNo">1139</span>                c = s.getChain(0, chainNr);<a name="line.1139"></a>
<span class="sourceLineNo">1140</span><a name="line.1140"></a>
<span class="sourceLineNo">1141</span>                newS.addChain(c);<a name="line.1141"></a>
<span class="sourceLineNo">1142</span><a name="line.1142"></a>
<span class="sourceLineNo">1143</span>                return newS;<a name="line.1143"></a>
<span class="sourceLineNo">1144</span>        }<a name="line.1144"></a>
<span class="sourceLineNo">1145</span><a name="line.1145"></a>
<span class="sourceLineNo">1146</span>        /**<a name="line.1146"></a>
<span class="sourceLineNo">1147</span>         * In addition to the functionality provided by<a name="line.1147"></a>
<span class="sourceLineNo">1148</span>         * {@link #getReducedStructure(Structure, int)} and<a name="line.1148"></a>
<span class="sourceLineNo">1149</span>         * {@link #getReducedStructure(Structure, String)}, also provides a way to<a name="line.1149"></a>
<span class="sourceLineNo">1150</span>         * specify sub-regions of a structure with the following specification:<a name="line.1150"></a>
<span class="sourceLineNo">1151</span>         *<a name="line.1151"></a>
<span class="sourceLineNo">1152</span>         * &lt;p&gt;<a name="line.1152"></a>
<span class="sourceLineNo">1153</span>         * &lt;li&gt;ranges can be surrounded by ( and ). (but will be removed).&lt;/li&gt;<a name="line.1153"></a>
<span class="sourceLineNo">1154</span>         * &lt;li&gt;ranges are specified as PDBresnum1 : PDBresnum2&lt;/li&gt;<a name="line.1154"></a>
<span class="sourceLineNo">1155</span>         *<a name="line.1155"></a>
<span class="sourceLineNo">1156</span>         * &lt;li&gt;a list of ranges is separated by ,&lt;/li&gt;<a name="line.1156"></a>
<span class="sourceLineNo">1157</span>         * &lt;/p&gt;<a name="line.1157"></a>
<span class="sourceLineNo">1158</span>         * Example<a name="line.1158"></a>
<span class="sourceLineNo">1159</span>         *<a name="line.1159"></a>
<span class="sourceLineNo">1160</span>         * &lt;pre&gt;<a name="line.1160"></a>
<span class="sourceLineNo">1161</span>         *  4GCR (A:1-83)<a name="line.1161"></a>
<span class="sourceLineNo">1162</span>         *  1CDG (A:407-495,A:582-686)<a name="line.1162"></a>
<span class="sourceLineNo">1163</span>         *  1CDG (A_407-495,A_582-686)<a name="line.1163"></a>
<span class="sourceLineNo">1164</span>         * &lt;/pre&gt;<a name="line.1164"></a>
<span class="sourceLineNo">1165</span>         *<a name="line.1165"></a>
<span class="sourceLineNo">1166</span>         * @param s<a name="line.1166"></a>
<span class="sourceLineNo">1167</span>         *            The full structure<a name="line.1167"></a>
<span class="sourceLineNo">1168</span>         * @param ranges<a name="line.1168"></a>
<span class="sourceLineNo">1169</span>         *            A comma-separated list of ranges, optionally surrounded by<a name="line.1169"></a>
<span class="sourceLineNo">1170</span>         *            parentheses<a name="line.1170"></a>
<span class="sourceLineNo">1171</span>         * @return Substructure of s specified by ranges<a name="line.1171"></a>
<span class="sourceLineNo">1172</span>         * @throws IllegalArgumentException for malformed range strings<a name="line.1172"></a>
<span class="sourceLineNo">1173</span>         * @throws StructureException for errors when reducing the Structure<a name="line.1173"></a>
<span class="sourceLineNo">1174</span>         * @deprecated Use {@link StructureIdentifier} instead (4.2.0)<a name="line.1174"></a>
<span class="sourceLineNo">1175</span>         */<a name="line.1175"></a>
<span class="sourceLineNo">1176</span>        @Deprecated<a name="line.1176"></a>
<span class="sourceLineNo">1177</span>        public static final Structure getSubRanges(Structure s, String ranges )<a name="line.1177"></a>
<span class="sourceLineNo">1178</span>                        throws StructureException<a name="line.1178"></a>
<span class="sourceLineNo">1179</span>        {<a name="line.1179"></a>
<span class="sourceLineNo">1180</span>                if (ranges == null || ranges.equals(""))<a name="line.1180"></a>
<span class="sourceLineNo">1181</span>                        throw new IllegalArgumentException("ranges can't be null or empty");<a name="line.1181"></a>
<span class="sourceLineNo">1182</span><a name="line.1182"></a>
<span class="sourceLineNo">1183</span>                ranges = ranges.trim();<a name="line.1183"></a>
<span class="sourceLineNo">1184</span><a name="line.1184"></a>
<span class="sourceLineNo">1185</span>                if (ranges.startsWith("("))<a name="line.1185"></a>
<span class="sourceLineNo">1186</span>                        ranges = ranges.substring(1);<a name="line.1186"></a>
<span class="sourceLineNo">1187</span>                if (ranges.endsWith(")")) {<a name="line.1187"></a>
<span class="sourceLineNo">1188</span>                        ranges = ranges.substring(0, ranges.length() - 1);<a name="line.1188"></a>
<span class="sourceLineNo">1189</span>                }<a name="line.1189"></a>
<span class="sourceLineNo">1190</span><a name="line.1190"></a>
<span class="sourceLineNo">1191</span>                // special case: '-' means 'everything'<a name="line.1191"></a>
<span class="sourceLineNo">1192</span>                if (ranges.equals("-")) {<a name="line.1192"></a>
<span class="sourceLineNo">1193</span>                        return s;<a name="line.1193"></a>
<span class="sourceLineNo">1194</span>                }<a name="line.1194"></a>
<span class="sourceLineNo">1195</span><a name="line.1195"></a>
<span class="sourceLineNo">1196</span>                List&lt;ResidueRange&gt; resRanges = ResidueRange.parseMultiple(ranges);<a name="line.1196"></a>
<span class="sourceLineNo">1197</span>                SubstructureIdentifier structId = new SubstructureIdentifier(null,resRanges);<a name="line.1197"></a>
<span class="sourceLineNo">1198</span>                return structId.reduce(s);<a name="line.1198"></a>
<span class="sourceLineNo">1199</span>        }<a name="line.1199"></a>
<span class="sourceLineNo">1200</span><a name="line.1200"></a>
<span class="sourceLineNo">1201</span>        public static final String convertAtomsToSeq(Atom[] atoms) {<a name="line.1201"></a>
<span class="sourceLineNo">1202</span><a name="line.1202"></a>
<span class="sourceLineNo">1203</span>                StringBuilder buf = new StringBuilder();<a name="line.1203"></a>
<span class="sourceLineNo">1204</span>                Group prevGroup = null;<a name="line.1204"></a>
<span class="sourceLineNo">1205</span>                for (Atom a : atoms) {<a name="line.1205"></a>
<span class="sourceLineNo">1206</span>                        Group g = a.getGroup();<a name="line.1206"></a>
<span class="sourceLineNo">1207</span>                        if (prevGroup != null) {<a name="line.1207"></a>
<span class="sourceLineNo">1208</span>                                if (prevGroup.equals(g)) {<a name="line.1208"></a>
<span class="sourceLineNo">1209</span>                                        // we add each group only once.<a name="line.1209"></a>
<span class="sourceLineNo">1210</span>                                        continue;<a name="line.1210"></a>
<span class="sourceLineNo">1211</span>                                }<a name="line.1211"></a>
<span class="sourceLineNo">1212</span>                        }<a name="line.1212"></a>
<span class="sourceLineNo">1213</span>                        String code3 = g.getPDBName();<a name="line.1213"></a>
<span class="sourceLineNo">1214</span>                        Character code1 = get1LetterCodeAmino(code3);<a name="line.1214"></a>
<span class="sourceLineNo">1215</span>                        if (code1 == null)<a name="line.1215"></a>
<span class="sourceLineNo">1216</span>                                code1 = UNKNOWN_GROUP_LABEL;<a name="line.1216"></a>
<span class="sourceLineNo">1217</span><a name="line.1217"></a>
<span class="sourceLineNo">1218</span>                        buf.append(code1);<a name="line.1218"></a>
<span class="sourceLineNo">1219</span><a name="line.1219"></a>
<span class="sourceLineNo">1220</span>                        prevGroup = g;<a name="line.1220"></a>
<span class="sourceLineNo">1221</span><a name="line.1221"></a>
<span class="sourceLineNo">1222</span>                }<a name="line.1222"></a>
<span class="sourceLineNo">1223</span>                return buf.toString();<a name="line.1223"></a>
<span class="sourceLineNo">1224</span>        }<a name="line.1224"></a>
<span class="sourceLineNo">1225</span><a name="line.1225"></a>
<span class="sourceLineNo">1226</span>        /**<a name="line.1226"></a>
<span class="sourceLineNo">1227</span>         * Get a group represented by a ResidueNumber.<a name="line.1227"></a>
<span class="sourceLineNo">1228</span>         *<a name="line.1228"></a>
<span class="sourceLineNo">1229</span>         * @param struc<a name="line.1229"></a>
<span class="sourceLineNo">1230</span>         *            a {@link Structure}<a name="line.1230"></a>
<span class="sourceLineNo">1231</span>         * @param pdbResNum<a name="line.1231"></a>
<span class="sourceLineNo">1232</span>         *            a {@link ResidueNumber}<a name="line.1232"></a>
<span class="sourceLineNo">1233</span>         * @return a group in the structure that is represented by the pdbResNum.<a name="line.1233"></a>
<span class="sourceLineNo">1234</span>         * @throws StructureException<a name="line.1234"></a>
<span class="sourceLineNo">1235</span>         *             if the group cannot be found.<a name="line.1235"></a>
<span class="sourceLineNo">1236</span>         */<a name="line.1236"></a>
<span class="sourceLineNo">1237</span>        public static final Group getGroupByPDBResidueNumber(Structure struc,<a name="line.1237"></a>
<span class="sourceLineNo">1238</span>                        ResidueNumber pdbResNum) throws StructureException {<a name="line.1238"></a>
<span class="sourceLineNo">1239</span>                if (struc == null || pdbResNum == null) {<a name="line.1239"></a>
<span class="sourceLineNo">1240</span>                        throw new IllegalArgumentException("Null argument(s).");<a name="line.1240"></a>
<span class="sourceLineNo">1241</span>                }<a name="line.1241"></a>
<span class="sourceLineNo">1242</span><a name="line.1242"></a>
<span class="sourceLineNo">1243</span>                Chain chain = struc.findChain(pdbResNum.getChainId());<a name="line.1243"></a>
<span class="sourceLineNo">1244</span><a name="line.1244"></a>
<span class="sourceLineNo">1245</span>                return chain.getGroupByPDB(pdbResNum);<a name="line.1245"></a>
<span class="sourceLineNo">1246</span>        }<a name="line.1246"></a>
<span class="sourceLineNo">1247</span><a name="line.1247"></a>
<span class="sourceLineNo">1248</span>        /**<a name="line.1248"></a>
<span class="sourceLineNo">1249</span>         * Returns the set of intra-chain contacts for the given chain for given<a name="line.1249"></a>
<span class="sourceLineNo">1250</span>         * atom names, i.e. the contact map. Uses a geometric hashing algorithm that<a name="line.1250"></a>
<span class="sourceLineNo">1251</span>         * speeds up the calculation without need of full distance matrix. The<a name="line.1251"></a>
<span class="sourceLineNo">1252</span>         * parsing mode {@link FileParsingParameters#setAlignSeqRes(boolean)} needs<a name="line.1252"></a>
<span class="sourceLineNo">1253</span>         * to be set to true for this to work.<a name="line.1253"></a>
<span class="sourceLineNo">1254</span>         *<a name="line.1254"></a>
<span class="sourceLineNo">1255</span>         * @param chain<a name="line.1255"></a>
<span class="sourceLineNo">1256</span>         * @param atomNames<a name="line.1256"></a>
<span class="sourceLineNo">1257</span>         *            the array with atom names to be used. Beware: CA will do both<a name="line.1257"></a>
<span class="sourceLineNo">1258</span>         *            C-alphas an Calciums if null all non-H atoms of non-hetatoms<a name="line.1258"></a>
<span class="sourceLineNo">1259</span>         *            will be used<a name="line.1259"></a>
<span class="sourceLineNo">1260</span>         * @param cutoff<a name="line.1260"></a>
<span class="sourceLineNo">1261</span>         * @return<a name="line.1261"></a>
<span class="sourceLineNo">1262</span>         */<a name="line.1262"></a>
<span class="sourceLineNo">1263</span>        public static AtomContactSet getAtomsInContact(Chain chain,<a name="line.1263"></a>
<span class="sourceLineNo">1264</span>                        String[] atomNames, double cutoff) {<a name="line.1264"></a>
<span class="sourceLineNo">1265</span>                Grid grid = new Grid(cutoff);<a name="line.1265"></a>
<span class="sourceLineNo">1266</span><a name="line.1266"></a>
<span class="sourceLineNo">1267</span>                Atom[] atoms = null;<a name="line.1267"></a>
<span class="sourceLineNo">1268</span>                if (atomNames == null) {<a name="line.1268"></a>
<span class="sourceLineNo">1269</span>                        atoms = getAllNonHAtomArray(chain, false);<a name="line.1269"></a>
<span class="sourceLineNo">1270</span>                } else {<a name="line.1270"></a>
<span class="sourceLineNo">1271</span>                        atoms = getAtomArray(chain, atomNames);<a name="line.1271"></a>
<span class="sourceLineNo">1272</span>                }<a name="line.1272"></a>
<span class="sourceLineNo">1273</span><a name="line.1273"></a>
<span class="sourceLineNo">1274</span>                grid.addAtoms(atoms);<a name="line.1274"></a>
<span class="sourceLineNo">1275</span><a name="line.1275"></a>
<span class="sourceLineNo">1276</span>                return grid.getContacts();<a name="line.1276"></a>
<span class="sourceLineNo">1277</span>        }<a name="line.1277"></a>
<span class="sourceLineNo">1278</span><a name="line.1278"></a>
<span class="sourceLineNo">1279</span>        /**<a name="line.1279"></a>
<span class="sourceLineNo">1280</span>         * Returns the set of intra-chain contacts for the given chain for all non-H<a name="line.1280"></a>
<span class="sourceLineNo">1281</span>         * atoms of non-hetatoms, i.e. the contact map. Uses a geometric hashing<a name="line.1281"></a>
<span class="sourceLineNo">1282</span>         * algorithm that speeds up the calculation without need of full distance<a name="line.1282"></a>
<span class="sourceLineNo">1283</span>         * matrix. The parsing mode<a name="line.1283"></a>
<span class="sourceLineNo">1284</span>         * {@link FileParsingParameters#setAlignSeqRes(boolean)} needs to be set to<a name="line.1284"></a>
<span class="sourceLineNo">1285</span>         * true for this to work.<a name="line.1285"></a>
<span class="sourceLineNo">1286</span>         *<a name="line.1286"></a>
<span class="sourceLineNo">1287</span>         * @param chain<a name="line.1287"></a>
<span class="sourceLineNo">1288</span>         * @param cutoff<a name="line.1288"></a>
<span class="sourceLineNo">1289</span>         * @return<a name="line.1289"></a>
<span class="sourceLineNo">1290</span>         */<a name="line.1290"></a>
<span class="sourceLineNo">1291</span>        public static AtomContactSet getAtomsInContact(Chain chain, double cutoff) {<a name="line.1291"></a>
<span class="sourceLineNo">1292</span>                return getAtomsInContact(chain, (String[]) null, cutoff);<a name="line.1292"></a>
<span class="sourceLineNo">1293</span>        }<a name="line.1293"></a>
<span class="sourceLineNo">1294</span><a name="line.1294"></a>
<span class="sourceLineNo">1295</span>        /**<a name="line.1295"></a>
<span class="sourceLineNo">1296</span>         * Returns the set of intra-chain contacts for the given chain for C-alpha<a name="line.1296"></a>
<span class="sourceLineNo">1297</span>         * atoms (including non-standard aminoacids appearing as HETATM groups),<a name="line.1297"></a>
<span class="sourceLineNo">1298</span>         * i.e. the contact map. Uses a geometric hashing algorithm that speeds up<a name="line.1298"></a>
<span class="sourceLineNo">1299</span>         * the calculation without need of full distance matrix. The parsing mode<a name="line.1299"></a>
<span class="sourceLineNo">1300</span>         * {@link FileParsingParameters#setAlignSeqRes(boolean)} needs to be set to<a name="line.1300"></a>
<span class="sourceLineNo">1301</span>         * true for this to work.<a name="line.1301"></a>
<span class="sourceLineNo">1302</span>         *<a name="line.1302"></a>
<span class="sourceLineNo">1303</span>         * @param chain<a name="line.1303"></a>
<span class="sourceLineNo">1304</span>         * @param cutoff<a name="line.1304"></a>
<span class="sourceLineNo">1305</span>         * @return<a name="line.1305"></a>
<span class="sourceLineNo">1306</span>         * @see {@link #getRepresentativeAtomsInContact(Chain, double)}<a name="line.1306"></a>
<span class="sourceLineNo">1307</span>         */<a name="line.1307"></a>
<span class="sourceLineNo">1308</span>        public static AtomContactSet getAtomsCAInContact(Chain chain, double cutoff) {<a name="line.1308"></a>
<span class="sourceLineNo">1309</span>                Grid grid = new Grid(cutoff);<a name="line.1309"></a>
<span class="sourceLineNo">1310</span><a name="line.1310"></a>
<span class="sourceLineNo">1311</span>                Atom[] atoms = getAtomCAArray(chain);<a name="line.1311"></a>
<span class="sourceLineNo">1312</span><a name="line.1312"></a>
<span class="sourceLineNo">1313</span>                grid.addAtoms(atoms);<a name="line.1313"></a>
<span class="sourceLineNo">1314</span><a name="line.1314"></a>
<span class="sourceLineNo">1315</span>                return grid.getContacts();<a name="line.1315"></a>
<span class="sourceLineNo">1316</span>        }<a name="line.1316"></a>
<span class="sourceLineNo">1317</span><a name="line.1317"></a>
<span class="sourceLineNo">1318</span>        /**<a name="line.1318"></a>
<span class="sourceLineNo">1319</span>         * Returns the set of intra-chain contacts for the given chain for C-alpha<a name="line.1319"></a>
<span class="sourceLineNo">1320</span>         * or C3' atoms (including non-standard aminoacids appearing as HETATM<a name="line.1320"></a>
<span class="sourceLineNo">1321</span>         * groups), i.e. the contact map. Uses a geometric hashing algorithm that<a name="line.1321"></a>
<span class="sourceLineNo">1322</span>         * speeds up the calculation without need of full distance matrix.<a name="line.1322"></a>
<span class="sourceLineNo">1323</span>         *<a name="line.1323"></a>
<span class="sourceLineNo">1324</span>         * @param chain<a name="line.1324"></a>
<span class="sourceLineNo">1325</span>         * @param cutoff<a name="line.1325"></a>
<span class="sourceLineNo">1326</span>         * @return<a name="line.1326"></a>
<span class="sourceLineNo">1327</span>         * @since Biojava 4.1.0<a name="line.1327"></a>
<span class="sourceLineNo">1328</span>         */<a name="line.1328"></a>
<span class="sourceLineNo">1329</span>        public static AtomContactSet getRepresentativeAtomsInContact(Chain chain,<a name="line.1329"></a>
<span class="sourceLineNo">1330</span>                        double cutoff) {<a name="line.1330"></a>
<span class="sourceLineNo">1331</span>                Grid grid = new Grid(cutoff);<a name="line.1331"></a>
<span class="sourceLineNo">1332</span><a name="line.1332"></a>
<span class="sourceLineNo">1333</span>                Atom[] atoms = getRepresentativeAtomArray(chain);<a name="line.1333"></a>
<span class="sourceLineNo">1334</span><a name="line.1334"></a>
<span class="sourceLineNo">1335</span>                grid.addAtoms(atoms);<a name="line.1335"></a>
<span class="sourceLineNo">1336</span><a name="line.1336"></a>
<span class="sourceLineNo">1337</span>                return grid.getContacts();<a name="line.1337"></a>
<span class="sourceLineNo">1338</span>        }<a name="line.1338"></a>
<span class="sourceLineNo">1339</span><a name="line.1339"></a>
<span class="sourceLineNo">1340</span>        /**<a name="line.1340"></a>
<span class="sourceLineNo">1341</span>         * Returns the set of inter-chain contacts between the two given chains for<a name="line.1341"></a>
<span class="sourceLineNo">1342</span>         * the given atom names. Uses a geometric hashing algorithm that speeds up<a name="line.1342"></a>
<span class="sourceLineNo">1343</span>         * the calculation without need of full distance matrix. The parsing mode<a name="line.1343"></a>
<span class="sourceLineNo">1344</span>         * {@link FileParsingParameters#setAlignSeqRes(boolean)} needs to be set to<a name="line.1344"></a>
<span class="sourceLineNo">1345</span>         * true for this to work.<a name="line.1345"></a>
<span class="sourceLineNo">1346</span>         *<a name="line.1346"></a>
<span class="sourceLineNo">1347</span>         * @param chain1<a name="line.1347"></a>
<span class="sourceLineNo">1348</span>         * @param chain2<a name="line.1348"></a>
<span class="sourceLineNo">1349</span>         * @param atomNames<a name="line.1349"></a>
<span class="sourceLineNo">1350</span>         *            the array with atom names to be used. For Calphas use {"CA"},<a name="line.1350"></a>
<span class="sourceLineNo">1351</span>         *            if null all non-H atoms will be used. Note HET atoms are<a name="line.1351"></a>
<span class="sourceLineNo">1352</span>         *            ignored unless this parameter is null.<a name="line.1352"></a>
<span class="sourceLineNo">1353</span>         * @param cutoff<a name="line.1353"></a>
<span class="sourceLineNo">1354</span>         * @param hetAtoms<a name="line.1354"></a>
<span class="sourceLineNo">1355</span>         *            if true HET atoms are included, if false they are not<a name="line.1355"></a>
<span class="sourceLineNo">1356</span>         * @return<a name="line.1356"></a>
<span class="sourceLineNo">1357</span>         */<a name="line.1357"></a>
<span class="sourceLineNo">1358</span>        public static AtomContactSet getAtomsInContact(Chain chain1, Chain chain2,<a name="line.1358"></a>
<span class="sourceLineNo">1359</span>                        String[] atomNames, double cutoff, boolean hetAtoms) {<a name="line.1359"></a>
<span class="sourceLineNo">1360</span>                Grid grid = new Grid(cutoff);<a name="line.1360"></a>
<span class="sourceLineNo">1361</span>                Atom[] atoms1 = null;<a name="line.1361"></a>
<span class="sourceLineNo">1362</span>                Atom[] atoms2 = null;<a name="line.1362"></a>
<span class="sourceLineNo">1363</span>                if (atomNames == null) {<a name="line.1363"></a>
<span class="sourceLineNo">1364</span>                        atoms1 = getAllNonHAtomArray(chain1, hetAtoms);<a name="line.1364"></a>
<span class="sourceLineNo">1365</span>                        atoms2 = getAllNonHAtomArray(chain2, hetAtoms);<a name="line.1365"></a>
<span class="sourceLineNo">1366</span>                } else {<a name="line.1366"></a>
<span class="sourceLineNo">1367</span>                        atoms1 = getAtomArray(chain1, atomNames);<a name="line.1367"></a>
<span class="sourceLineNo">1368</span>                        atoms2 = getAtomArray(chain2, atomNames);<a name="line.1368"></a>
<span class="sourceLineNo">1369</span>                }<a name="line.1369"></a>
<span class="sourceLineNo">1370</span>                grid.addAtoms(atoms1, atoms2);<a name="line.1370"></a>
<span class="sourceLineNo">1371</span><a name="line.1371"></a>
<span class="sourceLineNo">1372</span>                return grid.getContacts();<a name="line.1372"></a>
<span class="sourceLineNo">1373</span>        }<a name="line.1373"></a>
<span class="sourceLineNo">1374</span><a name="line.1374"></a>
<span class="sourceLineNo">1375</span>        /**<a name="line.1375"></a>
<span class="sourceLineNo">1376</span>         * Returns the set of inter-chain contacts between the two given chains for<a name="line.1376"></a>
<span class="sourceLineNo">1377</span>         * all non-H atoms. Uses a geometric hashing algorithm that speeds up the<a name="line.1377"></a>
<span class="sourceLineNo">1378</span>         * calculation without need of full distance matrix. The parsing mode<a name="line.1378"></a>
<span class="sourceLineNo">1379</span>         * {@link FileParsingParameters#setAlignSeqRes(boolean)} needs to be set to<a name="line.1379"></a>
<span class="sourceLineNo">1380</span>         * true for this to work.<a name="line.1380"></a>
<span class="sourceLineNo">1381</span>         *<a name="line.1381"></a>
<span class="sourceLineNo">1382</span>         * @param chain1<a name="line.1382"></a>
<span class="sourceLineNo">1383</span>         * @param chain2<a name="line.1383"></a>
<span class="sourceLineNo">1384</span>         * @param cutoff<a name="line.1384"></a>
<span class="sourceLineNo">1385</span>         * @param hetAtoms<a name="line.1385"></a>
<span class="sourceLineNo">1386</span>         *            if true HET atoms are included, if false they are not<a name="line.1386"></a>
<span class="sourceLineNo">1387</span>         * @return<a name="line.1387"></a>
<span class="sourceLineNo">1388</span>         */<a name="line.1388"></a>
<span class="sourceLineNo">1389</span>        public static AtomContactSet getAtomsInContact(Chain chain1, Chain chain2,<a name="line.1389"></a>
<span class="sourceLineNo">1390</span>                        double cutoff, boolean hetAtoms) {<a name="line.1390"></a>
<span class="sourceLineNo">1391</span>                return getAtomsInContact(chain1, chain2, null, cutoff, hetAtoms);<a name="line.1391"></a>
<span class="sourceLineNo">1392</span>        }<a name="line.1392"></a>
<span class="sourceLineNo">1393</span><a name="line.1393"></a>
<span class="sourceLineNo">1394</span>        /**<a name="line.1394"></a>
<span class="sourceLineNo">1395</span>         * Finds Groups in {@code structure} that contain at least one Atom that is<a name="line.1395"></a>
<span class="sourceLineNo">1396</span>         * within {@code radius} Angstroms of {@code centroid}.<a name="line.1396"></a>
<span class="sourceLineNo">1397</span>         *<a name="line.1397"></a>
<span class="sourceLineNo">1398</span>         * @param structure<a name="line.1398"></a>
<span class="sourceLineNo">1399</span>         *            The structure from which to find Groups<a name="line.1399"></a>
<span class="sourceLineNo">1400</span>         * @param centroid<a name="line.1400"></a>
<span class="sourceLineNo">1401</span>         *            The centroid of the shell<a name="line.1401"></a>
<span class="sourceLineNo">1402</span>         * @param excludeResidues<a name="line.1402"></a>
<span class="sourceLineNo">1403</span>         *            A list of ResidueNumbers to exclude<a name="line.1403"></a>
<span class="sourceLineNo">1404</span>         * @param radius<a name="line.1404"></a>
<span class="sourceLineNo">1405</span>         *            The radius from {@code centroid}, in Angstroms<a name="line.1405"></a>
<span class="sourceLineNo">1406</span>         * @param includeWater<a name="line.1406"></a>
<span class="sourceLineNo">1407</span>         *            Whether to include Groups whose &lt;em&gt;only&lt;/em&gt; atoms are water<a name="line.1407"></a>
<span class="sourceLineNo">1408</span>         * @param useAverageDistance<a name="line.1408"></a>
<span class="sourceLineNo">1409</span>         *            When set to true, distances are the arithmetic mean (1-norm)<a name="line.1409"></a>
<span class="sourceLineNo">1410</span>         *            of the distances of atoms that belong to the group and that<a name="line.1410"></a>
<span class="sourceLineNo">1411</span>         *            are within the shell; otherwise, distances are the minimum of<a name="line.1411"></a>
<span class="sourceLineNo">1412</span>         *            these values<a name="line.1412"></a>
<span class="sourceLineNo">1413</span>         * @return A map of Groups within (or partially within) the shell, to their<a name="line.1413"></a>
<span class="sourceLineNo">1414</span>         *         distances in Angstroms<a name="line.1414"></a>
<span class="sourceLineNo">1415</span>         */<a name="line.1415"></a>
<span class="sourceLineNo">1416</span>        public static Map&lt;Group, Double&gt; getGroupDistancesWithinShell(<a name="line.1416"></a>
<span class="sourceLineNo">1417</span>                        Structure structure, Atom centroid,<a name="line.1417"></a>
<span class="sourceLineNo">1418</span>                        Set&lt;ResidueNumber&gt; excludeResidues, double radius,<a name="line.1418"></a>
<span class="sourceLineNo">1419</span>                        boolean includeWater, boolean useAverageDistance) {<a name="line.1419"></a>
<span class="sourceLineNo">1420</span><a name="line.1420"></a>
<span class="sourceLineNo">1421</span>                // for speed, we avoid calculating square roots<a name="line.1421"></a>
<span class="sourceLineNo">1422</span>                radius = radius * radius;<a name="line.1422"></a>
<span class="sourceLineNo">1423</span><a name="line.1423"></a>
<span class="sourceLineNo">1424</span>                Map&lt;Group, Double&gt; distances = new HashMap&lt;Group, Double&gt;();<a name="line.1424"></a>
<span class="sourceLineNo">1425</span><a name="line.1425"></a>
<span class="sourceLineNo">1426</span>                // we only need this if we're averaging distances<a name="line.1426"></a>
<span class="sourceLineNo">1427</span>                // note that we can't use group.getAtoms().size() because some the<a name="line.1427"></a>
<span class="sourceLineNo">1428</span>                // group's atoms be outside the shell<a name="line.1428"></a>
<span class="sourceLineNo">1429</span>                Map&lt;Group, Integer&gt; atomCounts = new HashMap&lt;Group, Integer&gt;();<a name="line.1429"></a>
<span class="sourceLineNo">1430</span><a name="line.1430"></a>
<span class="sourceLineNo">1431</span>                for (Chain chain : structure.getChains()) {<a name="line.1431"></a>
<span class="sourceLineNo">1432</span>                        groupLoop: for (Group chainGroup : chain.getAtomGroups()) {<a name="line.1432"></a>
<span class="sourceLineNo">1433</span><a name="line.1433"></a>
<span class="sourceLineNo">1434</span>                                // exclude water<a name="line.1434"></a>
<span class="sourceLineNo">1435</span>                                if (!includeWater &amp;&amp; chainGroup.isWater())<a name="line.1435"></a>
<span class="sourceLineNo">1436</span>                                        continue;<a name="line.1436"></a>
<span class="sourceLineNo">1437</span><a name="line.1437"></a>
<span class="sourceLineNo">1438</span>                                // check blacklist of residue numbers<a name="line.1438"></a>
<span class="sourceLineNo">1439</span>                                for (ResidueNumber rn : excludeResidues) {<a name="line.1439"></a>
<span class="sourceLineNo">1440</span>                                        if (rn.equals(chainGroup.getResidueNumber()))<a name="line.1440"></a>
<span class="sourceLineNo">1441</span>                                                continue groupLoop;<a name="line.1441"></a>
<span class="sourceLineNo">1442</span>                                }<a name="line.1442"></a>
<span class="sourceLineNo">1443</span><a name="line.1443"></a>
<span class="sourceLineNo">1444</span>                                for (Atom testAtom : chainGroup.getAtoms()) {<a name="line.1444"></a>
<span class="sourceLineNo">1445</span><a name="line.1445"></a>
<span class="sourceLineNo">1446</span>                                        // use getDistanceFast as we are doing a lot of comparisons<a name="line.1446"></a>
<span class="sourceLineNo">1447</span>                                        double dist = Calc.getDistanceFast(centroid, testAtom);<a name="line.1447"></a>
<span class="sourceLineNo">1448</span><a name="line.1448"></a>
<span class="sourceLineNo">1449</span>                                        // if we're the shell<a name="line.1449"></a>
<span class="sourceLineNo">1450</span>                                        if (dist &lt;= radius) {<a name="line.1450"></a>
<span class="sourceLineNo">1451</span>                                                if (!distances.containsKey(chainGroup))<a name="line.1451"></a>
<span class="sourceLineNo">1452</span>                                                        distances.put(chainGroup, Double.POSITIVE_INFINITY);<a name="line.1452"></a>
<span class="sourceLineNo">1453</span>                                                if (useAverageDistance) {<a name="line.1453"></a>
<span class="sourceLineNo">1454</span>                                                        // sum the distance; we'll divide by the total<a name="line.1454"></a>
<span class="sourceLineNo">1455</span>                                                        // number later<a name="line.1455"></a>
<span class="sourceLineNo">1456</span>                                                        // here, we CANNOT use fastDistance (distance<a name="line.1456"></a>
<span class="sourceLineNo">1457</span>                                                        // squared) because we want the arithmetic mean<a name="line.1457"></a>
<span class="sourceLineNo">1458</span>                                                        distances.put(chainGroup, distances.get(chainGroup)<a name="line.1458"></a>
<span class="sourceLineNo">1459</span>                                                                        + Math.sqrt(dist));<a name="line.1459"></a>
<span class="sourceLineNo">1460</span>                                                        if (!atomCounts.containsKey(chainGroup))<a name="line.1460"></a>
<span class="sourceLineNo">1461</span>                                                                atomCounts.put(chainGroup, 0);<a name="line.1461"></a>
<span class="sourceLineNo">1462</span>                                                        atomCounts.put(chainGroup,<a name="line.1462"></a>
<span class="sourceLineNo">1463</span>                                                                        atomCounts.get(chainGroup) + 1);<a name="line.1463"></a>
<span class="sourceLineNo">1464</span>                                                } else {<a name="line.1464"></a>
<span class="sourceLineNo">1465</span>                                                        // take the minimum distance among all atoms of<a name="line.1465"></a>
<span class="sourceLineNo">1466</span>                                                        // chainGroup<a name="line.1466"></a>
<span class="sourceLineNo">1467</span>                                                        // note that we can't break here because we might<a name="line.1467"></a>
<span class="sourceLineNo">1468</span>                                                        // find a smaller distance<a name="line.1468"></a>
<span class="sourceLineNo">1469</span>                                                        if (dist &lt; distances.get(chainGroup)) {<a name="line.1469"></a>
<span class="sourceLineNo">1470</span>                                                                distances.put(chainGroup, dist);<a name="line.1470"></a>
<span class="sourceLineNo">1471</span>                                                        }<a name="line.1471"></a>
<span class="sourceLineNo">1472</span>                                                }<a name="line.1472"></a>
<span class="sourceLineNo">1473</span>                                        }<a name="line.1473"></a>
<span class="sourceLineNo">1474</span><a name="line.1474"></a>
<span class="sourceLineNo">1475</span>                                }<a name="line.1475"></a>
<span class="sourceLineNo">1476</span>                        }<a name="line.1476"></a>
<span class="sourceLineNo">1477</span>                }<a name="line.1477"></a>
<span class="sourceLineNo">1478</span><a name="line.1478"></a>
<span class="sourceLineNo">1479</span>                if (useAverageDistance) {<a name="line.1479"></a>
<span class="sourceLineNo">1480</span>                        for (Map.Entry&lt;Group, Double&gt; entry : distances.entrySet()) {<a name="line.1480"></a>
<span class="sourceLineNo">1481</span>                                int count = atomCounts.get(entry.getKey());<a name="line.1481"></a>
<span class="sourceLineNo">1482</span>                                distances.put(entry.getKey(), entry.getValue() / count);<a name="line.1482"></a>
<span class="sourceLineNo">1483</span>                        }<a name="line.1483"></a>
<span class="sourceLineNo">1484</span>                } else {<a name="line.1484"></a>
<span class="sourceLineNo">1485</span>                        // in this case we used getDistanceFast<a name="line.1485"></a>
<span class="sourceLineNo">1486</span>                        for (Map.Entry&lt;Group, Double&gt; entry : distances.entrySet()) {<a name="line.1486"></a>
<span class="sourceLineNo">1487</span>                                distances.put(entry.getKey(), Math.sqrt(entry.getValue()));<a name="line.1487"></a>
<span class="sourceLineNo">1488</span>                        }<a name="line.1488"></a>
<span class="sourceLineNo">1489</span>                }<a name="line.1489"></a>
<span class="sourceLineNo">1490</span><a name="line.1490"></a>
<span class="sourceLineNo">1491</span>                return distances;<a name="line.1491"></a>
<span class="sourceLineNo">1492</span><a name="line.1492"></a>
<span class="sourceLineNo">1493</span>        }<a name="line.1493"></a>
<span class="sourceLineNo">1494</span><a name="line.1494"></a>
<span class="sourceLineNo">1495</span>        public static Set&lt;Group&gt; getGroupsWithinShell(Structure structure,<a name="line.1495"></a>
<span class="sourceLineNo">1496</span>                        Atom atom, Set&lt;ResidueNumber&gt; excludeResidues, double distance,<a name="line.1496"></a>
<span class="sourceLineNo">1497</span>                        boolean includeWater) {<a name="line.1497"></a>
<span class="sourceLineNo">1498</span><a name="line.1498"></a>
<span class="sourceLineNo">1499</span>                // square the distance to use as a comparison against getDistanceFast<a name="line.1499"></a>
<span class="sourceLineNo">1500</span>                // which returns the square of a distance.<a name="line.1500"></a>
<span class="sourceLineNo">1501</span>                distance = distance * distance;<a name="line.1501"></a>
<span class="sourceLineNo">1502</span><a name="line.1502"></a>
<span class="sourceLineNo">1503</span>                Set&lt;Group&gt; returnSet = new LinkedHashSet&lt;Group&gt;();<a name="line.1503"></a>
<span class="sourceLineNo">1504</span>                for (Chain chain : structure.getChains()) {<a name="line.1504"></a>
<span class="sourceLineNo">1505</span>                        groupLoop: for (Group chainGroup : chain.getAtomGroups()) {<a name="line.1505"></a>
<span class="sourceLineNo">1506</span>                                if (!includeWater &amp;&amp; chainGroup.isWater())<a name="line.1506"></a>
<span class="sourceLineNo">1507</span>                                        continue;<a name="line.1507"></a>
<span class="sourceLineNo">1508</span>                                for (ResidueNumber rn : excludeResidues) {<a name="line.1508"></a>
<span class="sourceLineNo">1509</span>                                        if (rn.equals(chainGroup.getResidueNumber()))<a name="line.1509"></a>
<span class="sourceLineNo">1510</span>                                                continue groupLoop;<a name="line.1510"></a>
<span class="sourceLineNo">1511</span>                                }<a name="line.1511"></a>
<span class="sourceLineNo">1512</span>                                for (Atom atomB : chainGroup.getAtoms()) {<a name="line.1512"></a>
<span class="sourceLineNo">1513</span><a name="line.1513"></a>
<span class="sourceLineNo">1514</span>                                        // use getDistanceFast as we are doing a lot of comparisons<a name="line.1514"></a>
<span class="sourceLineNo">1515</span>                                        double dist = Calc.getDistanceFast(atom, atomB);<a name="line.1515"></a>
<span class="sourceLineNo">1516</span>                                        if (dist &lt;= distance) {<a name="line.1516"></a>
<span class="sourceLineNo">1517</span>                                                returnSet.add(chainGroup);<a name="line.1517"></a>
<span class="sourceLineNo">1518</span>                                                break;<a name="line.1518"></a>
<span class="sourceLineNo">1519</span>                                        }<a name="line.1519"></a>
<span class="sourceLineNo">1520</span><a name="line.1520"></a>
<span class="sourceLineNo">1521</span>                                }<a name="line.1521"></a>
<span class="sourceLineNo">1522</span>                        }<a name="line.1522"></a>
<span class="sourceLineNo">1523</span>                }<a name="line.1523"></a>
<span class="sourceLineNo">1524</span>                return returnSet;<a name="line.1524"></a>
<span class="sourceLineNo">1525</span>        }<a name="line.1525"></a>
<span class="sourceLineNo">1526</span><a name="line.1526"></a>
<span class="sourceLineNo">1527</span>        /**<a name="line.1527"></a>
<span class="sourceLineNo">1528</span>         * &lt;p&gt;<a name="line.1528"></a>
<span class="sourceLineNo">1529</span>         * Returns a Set of Groups in a structure within the distance specified of a<a name="line.1529"></a>
<span class="sourceLineNo">1530</span>         * given group.<a name="line.1530"></a>
<span class="sourceLineNo">1531</span>         * &lt;/p&gt;<a name="line.1531"></a>
<span class="sourceLineNo">1532</span>         * &lt;p&gt;<a name="line.1532"></a>
<span class="sourceLineNo">1533</span>         * Updated 18-Sep-2015 sroughley to return a Set so only a unique set of<a name="line.1533"></a>
<span class="sourceLineNo">1534</span>         * Groups returned<a name="line.1534"></a>
<span class="sourceLineNo">1535</span>         *<a name="line.1535"></a>
<span class="sourceLineNo">1536</span>         * @param structure<a name="line.1536"></a>
<span class="sourceLineNo">1537</span>         *            The structure to work with<a name="line.1537"></a>
<span class="sourceLineNo">1538</span>         * @param group<a name="line.1538"></a>
<span class="sourceLineNo">1539</span>         *            The 'query' group<a name="line.1539"></a>
<span class="sourceLineNo">1540</span>         * @param distance<a name="line.1540"></a>
<span class="sourceLineNo">1541</span>         *            The cutoff distance<a name="line.1541"></a>
<span class="sourceLineNo">1542</span>         * @param includeWater<a name="line.1542"></a>
<span class="sourceLineNo">1543</span>         *            Should water residues be included in the output?<a name="line.1543"></a>
<span class="sourceLineNo">1544</span>         * @return {@link LinkedHashSet} of {@link Group}s within at least one atom<a name="line.1544"></a>
<span class="sourceLineNo">1545</span>         *         with {@code distance} of at least one atom in {@code group}<a name="line.1545"></a>
<span class="sourceLineNo">1546</span>         */<a name="line.1546"></a>
<span class="sourceLineNo">1547</span>        public static Set&lt;Group&gt; getGroupsWithinShell(Structure structure,<a name="line.1547"></a>
<span class="sourceLineNo">1548</span>                        Group group, double distance, boolean includeWater) {<a name="line.1548"></a>
<span class="sourceLineNo">1549</span><a name="line.1549"></a>
<span class="sourceLineNo">1550</span>                Set&lt;Group&gt; returnList = new LinkedHashSet&lt;Group&gt;();<a name="line.1550"></a>
<span class="sourceLineNo">1551</span><a name="line.1551"></a>
<span class="sourceLineNo">1552</span>                Set&lt;ResidueNumber&gt; excludeGroups = new HashSet&lt;ResidueNumber&gt;();<a name="line.1552"></a>
<span class="sourceLineNo">1553</span>                excludeGroups.add(group.getResidueNumber());<a name="line.1553"></a>
<span class="sourceLineNo">1554</span>                for (Atom atom : group.getAtoms()) {<a name="line.1554"></a>
<span class="sourceLineNo">1555</span>                        Set&lt;Group&gt; set = getGroupsWithinShell(structure, atom,<a name="line.1555"></a>
<span class="sourceLineNo">1556</span>                                        excludeGroups, distance, includeWater);<a name="line.1556"></a>
<span class="sourceLineNo">1557</span>                        returnList.addAll(set);<a name="line.1557"></a>
<span class="sourceLineNo">1558</span>                }<a name="line.1558"></a>
<span class="sourceLineNo">1559</span><a name="line.1559"></a>
<span class="sourceLineNo">1560</span>                return returnList;<a name="line.1560"></a>
<span class="sourceLineNo">1561</span>        }<a name="line.1561"></a>
<span class="sourceLineNo">1562</span><a name="line.1562"></a>
<span class="sourceLineNo">1563</span>        /**<a name="line.1563"></a>
<span class="sourceLineNo">1564</span>         * Remove all models from a Structure and keep only the first<a name="line.1564"></a>
<span class="sourceLineNo">1565</span>         *<a name="line.1565"></a>
<span class="sourceLineNo">1566</span>         * @param s<a name="line.1566"></a>
<span class="sourceLineNo">1567</span>         *            original Structure<a name="line.1567"></a>
<span class="sourceLineNo">1568</span>         * @return a structure that contains only the first model<a name="line.1568"></a>
<span class="sourceLineNo">1569</span>         * @since 3.0.5<a name="line.1569"></a>
<span class="sourceLineNo">1570</span>         */<a name="line.1570"></a>
<span class="sourceLineNo">1571</span>        public static Structure removeModels(Structure s) {<a name="line.1571"></a>
<span class="sourceLineNo">1572</span>                if (s.nrModels() == 1)<a name="line.1572"></a>
<span class="sourceLineNo">1573</span>                        return s;<a name="line.1573"></a>
<span class="sourceLineNo">1574</span><a name="line.1574"></a>
<span class="sourceLineNo">1575</span>                Structure n = new StructureImpl();<a name="line.1575"></a>
<span class="sourceLineNo">1576</span>                // go through whole substructure and clone ...<a name="line.1576"></a>
<span class="sourceLineNo">1577</span><a name="line.1577"></a>
<span class="sourceLineNo">1578</span>                // copy structure data<a name="line.1578"></a>
<span class="sourceLineNo">1579</span><a name="line.1579"></a>
<span class="sourceLineNo">1580</span>                n.setPDBCode(s.getPDBCode());<a name="line.1580"></a>
<span class="sourceLineNo">1581</span>                n.setName(s.getName());<a name="line.1581"></a>
<span class="sourceLineNo">1582</span><a name="line.1582"></a>
<span class="sourceLineNo">1583</span>                // TODO: do deep copying of data!<a name="line.1583"></a>
<span class="sourceLineNo">1584</span>                n.setPDBHeader(s.getPDBHeader());<a name="line.1584"></a>
<span class="sourceLineNo">1585</span>                n.setDBRefs(s.getDBRefs());<a name="line.1585"></a>
<span class="sourceLineNo">1586</span><a name="line.1586"></a>
<span class="sourceLineNo">1587</span>                n.setSites(s.getSites());<a name="line.1587"></a>
<span class="sourceLineNo">1588</span><a name="line.1588"></a>
<span class="sourceLineNo">1589</span>                n.setChains(s.getModel(0));<a name="line.1589"></a>
<span class="sourceLineNo">1590</span><a name="line.1590"></a>
<span class="sourceLineNo">1591</span>                return n;<a name="line.1591"></a>
<span class="sourceLineNo">1592</span><a name="line.1592"></a>
<span class="sourceLineNo">1593</span>        }<a name="line.1593"></a>
<span class="sourceLineNo">1594</span><a name="line.1594"></a>
<span class="sourceLineNo">1595</span>        /**<a name="line.1595"></a>
<span class="sourceLineNo">1596</span>         * Removes all polymeric and solvent groups from a list of groups<a name="line.1596"></a>
<span class="sourceLineNo">1597</span>         *<a name="line.1597"></a>
<span class="sourceLineNo">1598</span>         */<a name="line.1598"></a>
<span class="sourceLineNo">1599</span>        public static List&lt;Group&gt; filterLigands(List&lt;Group&gt; allGroups) {<a name="line.1599"></a>
<span class="sourceLineNo">1600</span><a name="line.1600"></a>
<span class="sourceLineNo">1601</span>                List&lt;Group&gt; groups = new ArrayList&lt;Group&gt;();<a name="line.1601"></a>
<span class="sourceLineNo">1602</span>                for (Group g : allGroups) {<a name="line.1602"></a>
<span class="sourceLineNo">1603</span><a name="line.1603"></a>
<span class="sourceLineNo">1604</span>                        ChemComp cc = g.getChemComp();<a name="line.1604"></a>
<span class="sourceLineNo">1605</span><a name="line.1605"></a>
<span class="sourceLineNo">1606</span>                        if (ResidueType.lPeptideLinking.equals(cc.getResidueType())<a name="line.1606"></a>
<span class="sourceLineNo">1607</span>                                        || PolymerType.PROTEIN_ONLY.contains(cc.getPolymerType())<a name="line.1607"></a>
<span class="sourceLineNo">1608</span>                                        || PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc<a name="line.1608"></a>
<span class="sourceLineNo">1609</span>                                                        .getPolymerType())) {<a name="line.1609"></a>
<span class="sourceLineNo">1610</span>                                continue;<a name="line.1610"></a>
<span class="sourceLineNo">1611</span>                        }<a name="line.1611"></a>
<span class="sourceLineNo">1612</span>                        if (!g.isWater()) {<a name="line.1612"></a>
<span class="sourceLineNo">1613</span>                                groups.add(g);<a name="line.1613"></a>
<span class="sourceLineNo">1614</span>                        }<a name="line.1614"></a>
<span class="sourceLineNo">1615</span>                }<a name="line.1615"></a>
<span class="sourceLineNo">1616</span><a name="line.1616"></a>
<span class="sourceLineNo">1617</span>                return groups;<a name="line.1617"></a>
<span class="sourceLineNo">1618</span>        }<a name="line.1618"></a>
<span class="sourceLineNo">1619</span><a name="line.1619"></a>
<span class="sourceLineNo">1620</span>        /**<a name="line.1620"></a>
<span class="sourceLineNo">1621</span>         * Short version of {@link #getStructure(String, PDBFileParser, AtomCache)}<a name="line.1621"></a>
<span class="sourceLineNo">1622</span>         * which creates new parsers when needed<a name="line.1622"></a>
<span class="sourceLineNo">1623</span>         *<a name="line.1623"></a>
<span class="sourceLineNo">1624</span>         * @param name<a name="line.1624"></a>
<span class="sourceLineNo">1625</span>         * @return<a name="line.1625"></a>
<span class="sourceLineNo">1626</span>         * @throws IOException<a name="line.1626"></a>
<span class="sourceLineNo">1627</span>         * @throws StructureException<a name="line.1627"></a>
<span class="sourceLineNo">1628</span>         */<a name="line.1628"></a>
<span class="sourceLineNo">1629</span>        public static Structure getStructure(String name) throws IOException,<a name="line.1629"></a>
<span class="sourceLineNo">1630</span>                        StructureException {<a name="line.1630"></a>
<span class="sourceLineNo">1631</span>                return StructureTools.getStructure(name, null, null);<a name="line.1631"></a>
<span class="sourceLineNo">1632</span>        }<a name="line.1632"></a>
<span class="sourceLineNo">1633</span><a name="line.1633"></a>
<span class="sourceLineNo">1634</span>        /**<a name="line.1634"></a>
<span class="sourceLineNo">1635</span>         * Flexibly get a structure from an input String. The intent of this method<a name="line.1635"></a>
<span class="sourceLineNo">1636</span>         * is to allow any reasonable string which could refer to a structure to be<a name="line.1636"></a>
<span class="sourceLineNo">1637</span>         * correctly parsed. The following are currently supported:<a name="line.1637"></a>
<span class="sourceLineNo">1638</span>         * &lt;ol&gt;<a name="line.1638"></a>
<span class="sourceLineNo">1639</span>         * &lt;li&gt;Filename (if name refers to an existing file)<a name="line.1639"></a>
<span class="sourceLineNo">1640</span>         * &lt;li&gt;PDB ID<a name="line.1640"></a>
<span class="sourceLineNo">1641</span>         * &lt;li&gt;SCOP domains<a name="line.1641"></a>
<span class="sourceLineNo">1642</span>         * &lt;li&gt;PDP domains<a name="line.1642"></a>
<span class="sourceLineNo">1643</span>         * &lt;li&gt;Residue ranges<a name="line.1643"></a>
<span class="sourceLineNo">1644</span>         * &lt;li&gt;Other formats supported by AtomCache<a name="line.1644"></a>
<span class="sourceLineNo">1645</span>         * &lt;/ol&gt;<a name="line.1645"></a>
<span class="sourceLineNo">1646</span>         *<a name="line.1646"></a>
<span class="sourceLineNo">1647</span>         * @param name<a name="line.1647"></a>
<span class="sourceLineNo">1648</span>         *            Some reference to the protein structure<a name="line.1648"></a>
<span class="sourceLineNo">1649</span>         * @param parser<a name="line.1649"></a>
<span class="sourceLineNo">1650</span>         *            A clean PDBFileParser to use if it is a file. If null, a<a name="line.1650"></a>
<span class="sourceLineNo">1651</span>         *            PDBFileParser will be instantiated if needed.<a name="line.1651"></a>
<span class="sourceLineNo">1652</span>         * @param cache<a name="line.1652"></a>
<span class="sourceLineNo">1653</span>         *            An AtomCache to use if the structure can be fetched from the<a name="line.1653"></a>
<span class="sourceLineNo">1654</span>         *            PDB. If null, a AtomCache will be instantiated if needed.<a name="line.1654"></a>
<span class="sourceLineNo">1655</span>         * @return A Structure object<a name="line.1655"></a>
<span class="sourceLineNo">1656</span>         * @throws IOException<a name="line.1656"></a>
<span class="sourceLineNo">1657</span>         *             if name is an existing file, but doesn't parse correctly<a name="line.1657"></a>
<span class="sourceLineNo">1658</span>         * @throws StructureException<a name="line.1658"></a>
<span class="sourceLineNo">1659</span>         *             if the format is unknown, or if AtomCache throws an<a name="line.1659"></a>
<span class="sourceLineNo">1660</span>         *             exception.<a name="line.1660"></a>
<span class="sourceLineNo">1661</span>         */<a name="line.1661"></a>
<span class="sourceLineNo">1662</span>        public static Structure getStructure(String name, PDBFileParser parser,<a name="line.1662"></a>
<span class="sourceLineNo">1663</span>                        AtomCache cache) throws IOException, StructureException {<a name="line.1663"></a>
<span class="sourceLineNo">1664</span>                File f = new File(FileDownloadUtils.expandUserHome(name));<a name="line.1664"></a>
<span class="sourceLineNo">1665</span>                if (f.exists()) {<a name="line.1665"></a>
<span class="sourceLineNo">1666</span>                        if (parser == null) {<a name="line.1666"></a>
<span class="sourceLineNo">1667</span>                                parser = new PDBFileParser();<a name="line.1667"></a>
<span class="sourceLineNo">1668</span>                        }<a name="line.1668"></a>
<span class="sourceLineNo">1669</span>                        InputStream inStream = new FileInputStream(f);<a name="line.1669"></a>
<span class="sourceLineNo">1670</span>                        return parser.parsePDBFile(inStream);<a name="line.1670"></a>
<span class="sourceLineNo">1671</span>                } else {<a name="line.1671"></a>
<span class="sourceLineNo">1672</span>                        if (cache == null) {<a name="line.1672"></a>
<span class="sourceLineNo">1673</span>                                cache = new AtomCache();<a name="line.1673"></a>
<span class="sourceLineNo">1674</span>                        }<a name="line.1674"></a>
<span class="sourceLineNo">1675</span>                        return cache.getStructure(name);<a name="line.1675"></a>
<span class="sourceLineNo">1676</span>                }<a name="line.1676"></a>
<span class="sourceLineNo">1677</span>        }<a name="line.1677"></a>
<span class="sourceLineNo">1678</span><a name="line.1678"></a>
<span class="sourceLineNo">1679</span>        /**<a name="line.1679"></a>
<span class="sourceLineNo">1680</span>         * Tell whether given chain is a protein chain<a name="line.1680"></a>
<span class="sourceLineNo">1681</span>         *<a name="line.1681"></a>
<span class="sourceLineNo">1682</span>         * @param c<a name="line.1682"></a>
<span class="sourceLineNo">1683</span>         * @return true if protein, false if nucleotide or ligand<a name="line.1683"></a>
<span class="sourceLineNo">1684</span>         * @see #getPredominantGroupType(Chain)<a name="line.1684"></a>
<span class="sourceLineNo">1685</span>         */<a name="line.1685"></a>
<span class="sourceLineNo">1686</span>        public static boolean isProtein(Chain c) {<a name="line.1686"></a>
<span class="sourceLineNo">1687</span>                return getPredominantGroupType(c) == GroupType.AMINOACID;<a name="line.1687"></a>
<span class="sourceLineNo">1688</span>        }<a name="line.1688"></a>
<span class="sourceLineNo">1689</span><a name="line.1689"></a>
<span class="sourceLineNo">1690</span>        /**<a name="line.1690"></a>
<span class="sourceLineNo">1691</span>         * Tell whether given chain is DNA or RNA<a name="line.1691"></a>
<span class="sourceLineNo">1692</span>         *<a name="line.1692"></a>
<span class="sourceLineNo">1693</span>         * @param c<a name="line.1693"></a>
<span class="sourceLineNo">1694</span>         * @return true if nucleic acid, false if protein or ligand<a name="line.1694"></a>
<span class="sourceLineNo">1695</span>         * @see #getPredominantGroupType(Chain)<a name="line.1695"></a>
<span class="sourceLineNo">1696</span>         */<a name="line.1696"></a>
<span class="sourceLineNo">1697</span>        public static boolean isNucleicAcid(Chain c) {<a name="line.1697"></a>
<span class="sourceLineNo">1698</span>                return getPredominantGroupType(c) == GroupType.NUCLEOTIDE;<a name="line.1698"></a>
<span class="sourceLineNo">1699</span>        }<a name="line.1699"></a>
<span class="sourceLineNo">1700</span><a name="line.1700"></a>
<span class="sourceLineNo">1701</span>        /**<a name="line.1701"></a>
<span class="sourceLineNo">1702</span>         * Get the predominant {@link GroupType} for a given Chain, following these<a name="line.1702"></a>
<span class="sourceLineNo">1703</span>         * rules: &lt;li&gt;if the ratio of number of residues of a certain<a name="line.1703"></a>
<span class="sourceLineNo">1704</span>         * {@link GroupType} to total non-water residues is above the threshold<a name="line.1704"></a>
<span class="sourceLineNo">1705</span>         * {@value #RATIO_RESIDUES_TO_TOTAL}, then that {@link GroupType} is<a name="line.1705"></a>
<span class="sourceLineNo">1706</span>         * returned&lt;/li&gt; &lt;li&gt;if there is no {@link GroupType} that is above the<a name="line.1706"></a>
<span class="sourceLineNo">1707</span>         * threshold then the {@link GroupType} with most members is chosen, logging<a name="line.1707"></a>
<span class="sourceLineNo">1708</span>         * it&lt;/li&gt;<a name="line.1708"></a>
<span class="sourceLineNo">1709</span>         * &lt;p&gt;<a name="line.1709"></a>
<span class="sourceLineNo">1710</span>         * See also {@link ChemComp#getPolymerType()} and<a name="line.1710"></a>
<span class="sourceLineNo">1711</span>         * {@link ChemComp#getResidueType()} which follow the PDB chemical component<a name="line.1711"></a>
<span class="sourceLineNo">1712</span>         * dictionary and provide a much more accurate description of groups and<a name="line.1712"></a>
<span class="sourceLineNo">1713</span>         * their linking.<a name="line.1713"></a>
<span class="sourceLineNo">1714</span>         * &lt;/p&gt;<a name="line.1714"></a>
<span class="sourceLineNo">1715</span>         *<a name="line.1715"></a>
<span class="sourceLineNo">1716</span>         * @param c<a name="line.1716"></a>
<span class="sourceLineNo">1717</span>         * @return<a name="line.1717"></a>
<span class="sourceLineNo">1718</span>         */<a name="line.1718"></a>
<span class="sourceLineNo">1719</span>        public static GroupType getPredominantGroupType(Chain c) {<a name="line.1719"></a>
<span class="sourceLineNo">1720</span>                int sizeAminos = c.getAtomGroups(GroupType.AMINOACID).size();<a name="line.1720"></a>
<span class="sourceLineNo">1721</span>                int sizeNucleotides = c.getAtomGroups(GroupType.NUCLEOTIDE).size();<a name="line.1721"></a>
<span class="sourceLineNo">1722</span>                List&lt;Group&gt; hetAtoms = c.getAtomGroups(GroupType.HETATM);<a name="line.1722"></a>
<span class="sourceLineNo">1723</span>                int sizeHetatoms = hetAtoms.size();<a name="line.1723"></a>
<span class="sourceLineNo">1724</span>                int sizeWaters = 0;<a name="line.1724"></a>
<span class="sourceLineNo">1725</span>                for (Group g : hetAtoms) {<a name="line.1725"></a>
<span class="sourceLineNo">1726</span>                        if (g.isWater())<a name="line.1726"></a>
<span class="sourceLineNo">1727</span>                                sizeWaters++;<a name="line.1727"></a>
<span class="sourceLineNo">1728</span>                }<a name="line.1728"></a>
<span class="sourceLineNo">1729</span>                int sizeHetatomsWithoutWater = sizeHetatoms - sizeWaters;<a name="line.1729"></a>
<span class="sourceLineNo">1730</span><a name="line.1730"></a>
<span class="sourceLineNo">1731</span>                int fullSize = sizeAminos + sizeNucleotides + sizeHetatomsWithoutWater;<a name="line.1731"></a>
<span class="sourceLineNo">1732</span><a name="line.1732"></a>
<span class="sourceLineNo">1733</span>                if ((double) sizeAminos / (double) fullSize &gt; RATIO_RESIDUES_TO_TOTAL)<a name="line.1733"></a>
<span class="sourceLineNo">1734</span>                        return GroupType.AMINOACID;<a name="line.1734"></a>
<span class="sourceLineNo">1735</span><a name="line.1735"></a>
<span class="sourceLineNo">1736</span>                if ((double) sizeNucleotides / (double) fullSize &gt; RATIO_RESIDUES_TO_TOTAL)<a name="line.1736"></a>
<span class="sourceLineNo">1737</span>                        return GroupType.NUCLEOTIDE;<a name="line.1737"></a>
<span class="sourceLineNo">1738</span><a name="line.1738"></a>
<span class="sourceLineNo">1739</span>                if ((double) (sizeHetatomsWithoutWater) / (double) fullSize &gt; RATIO_RESIDUES_TO_TOTAL)<a name="line.1739"></a>
<span class="sourceLineNo">1740</span>                        return GroupType.HETATM;<a name="line.1740"></a>
<span class="sourceLineNo">1741</span><a name="line.1741"></a>
<span class="sourceLineNo">1742</span>                // finally if neither condition works, we try based on majority, but log<a name="line.1742"></a>
<span class="sourceLineNo">1743</span>                // it<a name="line.1743"></a>
<span class="sourceLineNo">1744</span>                GroupType max;<a name="line.1744"></a>
<span class="sourceLineNo">1745</span>                if (sizeNucleotides &gt; sizeAminos) {<a name="line.1745"></a>
<span class="sourceLineNo">1746</span>                        if (sizeNucleotides &gt; sizeHetatomsWithoutWater) {<a name="line.1746"></a>
<span class="sourceLineNo">1747</span>                                max = GroupType.NUCLEOTIDE;<a name="line.1747"></a>
<span class="sourceLineNo">1748</span>                        } else {<a name="line.1748"></a>
<span class="sourceLineNo">1749</span>                                max = GroupType.HETATM;<a name="line.1749"></a>
<span class="sourceLineNo">1750</span>                        }<a name="line.1750"></a>
<span class="sourceLineNo">1751</span>                } else {<a name="line.1751"></a>
<span class="sourceLineNo">1752</span>                        if (sizeAminos &gt; sizeHetatomsWithoutWater) {<a name="line.1752"></a>
<span class="sourceLineNo">1753</span>                                max = GroupType.AMINOACID;<a name="line.1753"></a>
<span class="sourceLineNo">1754</span>                        } else {<a name="line.1754"></a>
<span class="sourceLineNo">1755</span>                                max = GroupType.HETATM;<a name="line.1755"></a>
<span class="sourceLineNo">1756</span>                        }<a name="line.1756"></a>
<span class="sourceLineNo">1757</span>                }<a name="line.1757"></a>
<span class="sourceLineNo">1758</span>                logger.debug(<a name="line.1758"></a>
<span class="sourceLineNo">1759</span>                                "Ratio of residues to total for chain {} is below {}. Assuming it is a {} chain. "<a name="line.1759"></a>
<span class="sourceLineNo">1760</span>                                                + "Counts: # aa residues: {}, # nuc residues: {}, # non-water het residues: {}, # waters: {}, "<a name="line.1760"></a>
<span class="sourceLineNo">1761</span>                                                + "ratio aa/total: {}, ratio nuc/total: {}",<a name="line.1761"></a>
<span class="sourceLineNo">1762</span>                                c.getChainID(), RATIO_RESIDUES_TO_TOTAL, max, sizeAminos,<a name="line.1762"></a>
<span class="sourceLineNo">1763</span>                                sizeNucleotides, sizeHetatomsWithoutWater, sizeWaters,<a name="line.1763"></a>
<span class="sourceLineNo">1764</span>                                (double) sizeAminos / (double) fullSize,<a name="line.1764"></a>
<span class="sourceLineNo">1765</span>                                (double) sizeNucleotides / (double) fullSize);<a name="line.1765"></a>
<span class="sourceLineNo">1766</span><a name="line.1766"></a>
<span class="sourceLineNo">1767</span>                return max;<a name="line.1767"></a>
<span class="sourceLineNo">1768</span>        }<a name="line.1768"></a>
<span class="sourceLineNo">1769</span><a name="line.1769"></a>
<span class="sourceLineNo">1770</span>        /**<a name="line.1770"></a>
<span class="sourceLineNo">1771</span>         * Returns true if the given chain is composed of water molecules only<a name="line.1771"></a>
<span class="sourceLineNo">1772</span>         *<a name="line.1772"></a>
<span class="sourceLineNo">1773</span>         * @param c<a name="line.1773"></a>
<span class="sourceLineNo">1774</span>         * @return<a name="line.1774"></a>
<span class="sourceLineNo">1775</span>         */<a name="line.1775"></a>
<span class="sourceLineNo">1776</span>        public static boolean isChainWaterOnly(Chain c) {<a name="line.1776"></a>
<span class="sourceLineNo">1777</span>                boolean waterOnly = true;<a name="line.1777"></a>
<span class="sourceLineNo">1778</span>                for (Group g : c.getAtomGroups()) {<a name="line.1778"></a>
<span class="sourceLineNo">1779</span>                        if (!g.isWater())<a name="line.1779"></a>
<span class="sourceLineNo">1780</span>                                waterOnly = false;<a name="line.1780"></a>
<span class="sourceLineNo">1781</span>                        break;<a name="line.1781"></a>
<span class="sourceLineNo">1782</span>                }<a name="line.1782"></a>
<span class="sourceLineNo">1783</span>                return waterOnly;<a name="line.1783"></a>
<span class="sourceLineNo">1784</span>        }<a name="line.1784"></a>
<span class="sourceLineNo">1785</span><a name="line.1785"></a>
<span class="sourceLineNo">1786</span>        /**<a name="line.1786"></a>
<span class="sourceLineNo">1787</span>         * Returns true if the given chain is composed of non-polymeric groups only<a name="line.1787"></a>
<span class="sourceLineNo">1788</span>         *<a name="line.1788"></a>
<span class="sourceLineNo">1789</span>         * @param c<a name="line.1789"></a>
<span class="sourceLineNo">1790</span>         * @return<a name="line.1790"></a>
<span class="sourceLineNo">1791</span>         */<a name="line.1791"></a>
<span class="sourceLineNo">1792</span>        public static boolean isChainPureNonPolymer(Chain c) {<a name="line.1792"></a>
<span class="sourceLineNo">1793</span><a name="line.1793"></a>
<span class="sourceLineNo">1794</span>                for (Group g : c.getAtomGroups()) {<a name="line.1794"></a>
<span class="sourceLineNo">1795</span>                        if (g.getType() == GroupType.AMINOACID<a name="line.1795"></a>
<span class="sourceLineNo">1796</span>                                        || g.getType() == GroupType.NUCLEOTIDE)<a name="line.1796"></a>
<span class="sourceLineNo">1797</span>                                return false;<a name="line.1797"></a>
<span class="sourceLineNo">1798</span><a name="line.1798"></a>
<span class="sourceLineNo">1799</span>                }<a name="line.1799"></a>
<span class="sourceLineNo">1800</span>                return true;<a name="line.1800"></a>
<span class="sourceLineNo">1801</span>        }<a name="line.1801"></a>
<span class="sourceLineNo">1802</span>}<a name="line.1802"></a>




























































</pre>
</div>
</body>
</html>
