<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package org.biojava.nbio.structure.io;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>import org.biojava.nbio.alignment.Alignments;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.alignment.SimpleGapPenalty;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.alignment.template.GapPenalty;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.alignment.template.PairwiseSequenceAligner;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.alignment.template.SequencePair;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.DNASequence;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.nbio.core.sequence.RNASequence;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.structure.*;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.slf4j.Logger;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.slf4j.LoggerFactory;<a name="line.39"></a>
<span class="sourceLineNo">040</span><a name="line.40"></a>
<span class="sourceLineNo">041</span>import java.util.ArrayList;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import java.util.Collections;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import java.util.Comparator;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import java.util.HashMap;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import java.util.List;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import java.util.Map;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import java.util.Set;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import java.util.TreeMap;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import java.util.TreeSet;<a name="line.49"></a>
<span class="sourceLineNo">050</span><a name="line.50"></a>
<span class="sourceLineNo">051</span>/**<a name="line.51"></a>
<span class="sourceLineNo">052</span> * Heuristical finding of Compounds (called Entities in mmCIF dictionary)<a name="line.52"></a>
<span class="sourceLineNo">053</span> * in a given Structure. Compounds are the groups of sequence identical NCS-related polymer chains<a name="line.53"></a>
<span class="sourceLineNo">054</span> * in the Structure.<a name="line.54"></a>
<span class="sourceLineNo">055</span> *<a name="line.55"></a>
<span class="sourceLineNo">056</span> * This is related to {@link SeqRes2AtomAligner} but it is intended for raw PDB files where<a name="line.56"></a>
<span class="sourceLineNo">057</span> * possibly no SEQRES is given.<a name="line.57"></a>
<span class="sourceLineNo">058</span> *<a name="line.58"></a>
<span class="sourceLineNo">059</span> * @author duarte_j<a name="line.59"></a>
<span class="sourceLineNo">060</span> */<a name="line.60"></a>
<span class="sourceLineNo">061</span>public class CompoundFinder {<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span>        private Structure s;<a name="line.63"></a>
<span class="sourceLineNo">064</span><a name="line.64"></a>
<span class="sourceLineNo">065</span>        private static final Logger logger = LoggerFactory.getLogger(CompoundFinder.class);<a name="line.65"></a>
<span class="sourceLineNo">066</span><a name="line.66"></a>
<span class="sourceLineNo">067</span>        /**<a name="line.67"></a>
<span class="sourceLineNo">068</span>         * Above this ratio of mismatching residue types for same residue numbers we<a name="line.68"></a>
<span class="sourceLineNo">069</span>         * consider the 2 chains to have mismatching residue numbers and warn about it<a name="line.69"></a>
<span class="sourceLineNo">070</span>         */<a name="line.70"></a>
<span class="sourceLineNo">071</span>        public static final double RATIO_GAPS_FOR_MISMATCH = 0.50;<a name="line.71"></a>
<span class="sourceLineNo">072</span><a name="line.72"></a>
<span class="sourceLineNo">073</span>        /**<a name="line.73"></a>
<span class="sourceLineNo">074</span>         * Identity value for 2 chains to be considered part of same entity<a name="line.74"></a>
<span class="sourceLineNo">075</span>         */<a name="line.75"></a>
<span class="sourceLineNo">076</span>        public static final double IDENTITY_THRESHOLD = 0.99999;<a name="line.76"></a>
<span class="sourceLineNo">077</span><a name="line.77"></a>
<span class="sourceLineNo">078</span>        /**<a name="line.78"></a>
<span class="sourceLineNo">079</span>         * Gap coverage value (num gaps over length of sequence) for each chain of the match:<a name="line.79"></a>
<span class="sourceLineNo">080</span>         * 2 chains with more gap coverage than this value will not be considered part of the same entity<a name="line.80"></a>
<span class="sourceLineNo">081</span>         */<a name="line.81"></a>
<span class="sourceLineNo">082</span>        public static final double GAP_COVERAGE_THRESHOLD = 0.3;<a name="line.82"></a>
<span class="sourceLineNo">083</span><a name="line.83"></a>
<span class="sourceLineNo">084</span><a name="line.84"></a>
<span class="sourceLineNo">085</span>        public CompoundFinder(Structure s) {<a name="line.85"></a>
<span class="sourceLineNo">086</span>                this.s = s;<a name="line.86"></a>
<span class="sourceLineNo">087</span>        }<a name="line.87"></a>
<span class="sourceLineNo">088</span><a name="line.88"></a>
<span class="sourceLineNo">089</span>        /**<a name="line.89"></a>
<span class="sourceLineNo">090</span>         * Utility method that employs some heuristics to find the Compounds<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * for this Structure in case the information is missing in PDB/mmCIF file<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * @return<a name="line.92"></a>
<span class="sourceLineNo">093</span>         */<a name="line.93"></a>
<span class="sourceLineNo">094</span>        public List&lt;Compound&gt; findCompounds() {<a name="line.94"></a>
<span class="sourceLineNo">095</span><a name="line.95"></a>
<span class="sourceLineNo">096</span>                TreeMap&lt;String,Compound&gt; chainIds2entities = findCompoundsFromAlignment();<a name="line.96"></a>
<span class="sourceLineNo">097</span><a name="line.97"></a>
<span class="sourceLineNo">098</span>                List&lt;Compound&gt; compounds = findUniqueCompounds(chainIds2entities);<a name="line.98"></a>
<span class="sourceLineNo">099</span><a name="line.99"></a>
<span class="sourceLineNo">100</span>                createPurelyNonPolyCompounds(compounds);<a name="line.100"></a>
<span class="sourceLineNo">101</span><a name="line.101"></a>
<span class="sourceLineNo">102</span>                return compounds;<a name="line.102"></a>
<span class="sourceLineNo">103</span>        }<a name="line.103"></a>
<span class="sourceLineNo">104</span><a name="line.104"></a>
<span class="sourceLineNo">105</span>        /**<a name="line.105"></a>
<span class="sourceLineNo">106</span>         * Utility method to obtain a list of unique entities from the chainIds2entities map<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * @return<a name="line.107"></a>
<span class="sourceLineNo">108</span>         */<a name="line.108"></a>
<span class="sourceLineNo">109</span>        private static List&lt;Compound&gt; findUniqueCompounds(TreeMap&lt;String,Compound&gt; chainIds2entities) {<a name="line.109"></a>
<span class="sourceLineNo">110</span><a name="line.110"></a>
<span class="sourceLineNo">111</span>                List&lt;Compound&gt; list = new ArrayList&lt;Compound&gt;();<a name="line.111"></a>
<span class="sourceLineNo">112</span><a name="line.112"></a>
<span class="sourceLineNo">113</span>                for (Compound cluster:chainIds2entities.values()) {<a name="line.113"></a>
<span class="sourceLineNo">114</span>                        boolean present = false;<a name="line.114"></a>
<span class="sourceLineNo">115</span>                        for (Compound cl:list) {<a name="line.115"></a>
<span class="sourceLineNo">116</span>                                if (cl==cluster) {<a name="line.116"></a>
<span class="sourceLineNo">117</span>                                        present = true;<a name="line.117"></a>
<span class="sourceLineNo">118</span>                                        break;<a name="line.118"></a>
<span class="sourceLineNo">119</span>                                }<a name="line.119"></a>
<span class="sourceLineNo">120</span>                        }<a name="line.120"></a>
<span class="sourceLineNo">121</span>                        if (!present) list.add(cluster);<a name="line.121"></a>
<span class="sourceLineNo">122</span>                }<a name="line.122"></a>
<span class="sourceLineNo">123</span>                return list;<a name="line.123"></a>
<span class="sourceLineNo">124</span>        }<a name="line.124"></a>
<span class="sourceLineNo">125</span><a name="line.125"></a>
<span class="sourceLineNo">126</span>        private void createPurelyNonPolyCompounds(List&lt;Compound&gt; compounds) {<a name="line.126"></a>
<span class="sourceLineNo">127</span><a name="line.127"></a>
<span class="sourceLineNo">128</span>                List&lt;Chain&gt; pureNonPolymerChains = new ArrayList&lt;Chain&gt;();<a name="line.128"></a>
<span class="sourceLineNo">129</span>                for (int i=0;i&lt;s.getChains().size();i++) {<a name="line.129"></a>
<span class="sourceLineNo">130</span>                        if (StructureTools.isChainPureNonPolymer(s.getChain(i))) {<a name="line.130"></a>
<span class="sourceLineNo">131</span>                                pureNonPolymerChains.add(s.getChain(i));<a name="line.131"></a>
<span class="sourceLineNo">132</span>                        }<a name="line.132"></a>
<span class="sourceLineNo">133</span>                }<a name="line.133"></a>
<span class="sourceLineNo">134</span><a name="line.134"></a>
<span class="sourceLineNo">135</span>                if (!pureNonPolymerChains.isEmpty()) {<a name="line.135"></a>
<span class="sourceLineNo">136</span><a name="line.136"></a>
<span class="sourceLineNo">137</span>                        int maxMolId = 0; // if we have no compounds at all we just use 0, so that first assignment is 1<a name="line.137"></a>
<span class="sourceLineNo">138</span>                        if (!compounds.isEmpty()) {<a name="line.138"></a>
<span class="sourceLineNo">139</span>                                Collections.max(compounds, new Comparator&lt;Compound&gt;() {<a name="line.139"></a>
<span class="sourceLineNo">140</span>                                        @Override<a name="line.140"></a>
<span class="sourceLineNo">141</span>                                        public int compare(Compound o1, Compound o2) {<a name="line.141"></a>
<span class="sourceLineNo">142</span>                                                return new Integer(o1.getMolId()).compareTo(o2.getMolId());<a name="line.142"></a>
<span class="sourceLineNo">143</span>                                        }<a name="line.143"></a>
<span class="sourceLineNo">144</span>                                }).getMolId();<a name="line.144"></a>
<span class="sourceLineNo">145</span>                        }<a name="line.145"></a>
<span class="sourceLineNo">146</span><a name="line.146"></a>
<span class="sourceLineNo">147</span>                        int molId = maxMolId + 1;<a name="line.147"></a>
<span class="sourceLineNo">148</span>                        // for the purely non-polymeric chains we assign additional compounds<a name="line.148"></a>
<span class="sourceLineNo">149</span>                        for (Chain c: pureNonPolymerChains) {<a name="line.149"></a>
<span class="sourceLineNo">150</span>                                Compound comp = new Compound();<a name="line.150"></a>
<span class="sourceLineNo">151</span>                                comp.addChain(c);<a name="line.151"></a>
<span class="sourceLineNo">152</span>                                comp.setMolId(molId);<a name="line.152"></a>
<span class="sourceLineNo">153</span>                                logger.warn("Chain {} is purely non-polymeric, will assign a new Compound (entity) to it (entity id {})", c.getChainID(), molId);<a name="line.153"></a>
<span class="sourceLineNo">154</span>                                molId++;<a name="line.154"></a>
<span class="sourceLineNo">155</span><a name="line.155"></a>
<span class="sourceLineNo">156</span>                                compounds.add(comp);<a name="line.156"></a>
<span class="sourceLineNo">157</span>                        }<a name="line.157"></a>
<span class="sourceLineNo">158</span><a name="line.158"></a>
<span class="sourceLineNo">159</span>                }<a name="line.159"></a>
<span class="sourceLineNo">160</span>        }<a name="line.160"></a>
<span class="sourceLineNo">161</span><a name="line.161"></a>
<span class="sourceLineNo">162</span><a name="line.162"></a>
<span class="sourceLineNo">163</span>        private static boolean areResNumbersAligned(Chain c1, Chain c2) {<a name="line.163"></a>
<span class="sourceLineNo">164</span><a name="line.164"></a>
<span class="sourceLineNo">165</span>                boolean isC1prot = StructureTools.isProtein(c1);<a name="line.165"></a>
<span class="sourceLineNo">166</span>                boolean isC2prot = StructureTools.isProtein(c2);<a name="line.166"></a>
<span class="sourceLineNo">167</span><a name="line.167"></a>
<span class="sourceLineNo">168</span>                // different kind of chain: we won't try to align them<a name="line.168"></a>
<span class="sourceLineNo">169</span>                if (isC1prot != isC2prot ) return false;<a name="line.169"></a>
<span class="sourceLineNo">170</span><a name="line.170"></a>
<span class="sourceLineNo">171</span>                List&lt;Group&gt; c1AtomGroups = null;<a name="line.171"></a>
<span class="sourceLineNo">172</span>                if (isC1prot) {<a name="line.172"></a>
<span class="sourceLineNo">173</span>                        c1AtomGroups = c1.getAtomGroups(GroupType.AMINOACID);<a name="line.173"></a>
<span class="sourceLineNo">174</span>                }<a name="line.174"></a>
<span class="sourceLineNo">175</span>                else {<a name="line.175"></a>
<span class="sourceLineNo">176</span>                        c1AtomGroups = c1.getAtomGroups(GroupType.NUCLEOTIDE);<a name="line.176"></a>
<span class="sourceLineNo">177</span>                }<a name="line.177"></a>
<span class="sourceLineNo">178</span><a name="line.178"></a>
<span class="sourceLineNo">179</span>                int countNonExisting = 0;<a name="line.179"></a>
<span class="sourceLineNo">180</span><a name="line.180"></a>
<span class="sourceLineNo">181</span>                for (Group g1:c1AtomGroups) {<a name="line.181"></a>
<span class="sourceLineNo">182</span>                        try {<a name="line.182"></a>
<span class="sourceLineNo">183</span>                                Group g2 = c2.getGroupByPDB(g1.getResidueNumber());<a name="line.183"></a>
<span class="sourceLineNo">184</span>                                if (!g2.getPDBName().equals(g1.getPDBName())) {<a name="line.184"></a>
<span class="sourceLineNo">185</span>                                        logger.debug("Mismatch of residues between chains {},{} for residue number {}: {} {}",<a name="line.185"></a>
<span class="sourceLineNo">186</span>                                                        c1.getChainID(),c2.getChainID(),g1.getResidueNumber(), g1.getPDBName(), g2.getPDBName());<a name="line.186"></a>
<span class="sourceLineNo">187</span>                                        return false;<a name="line.187"></a>
<span class="sourceLineNo">188</span>                                }<a name="line.188"></a>
<span class="sourceLineNo">189</span>                        } catch (StructureException e) {<a name="line.189"></a>
<span class="sourceLineNo">190</span>                                // the group doesn't exist (no density) in the chain, go on<a name="line.190"></a>
<span class="sourceLineNo">191</span>                                countNonExisting++;<a name="line.191"></a>
<span class="sourceLineNo">192</span>                                continue;<a name="line.192"></a>
<span class="sourceLineNo">193</span>                        }<a name="line.193"></a>
<span class="sourceLineNo">194</span>                }<a name="line.194"></a>
<span class="sourceLineNo">195</span><a name="line.195"></a>
<span class="sourceLineNo">196</span>                if ((double)countNonExisting/(double)c1AtomGroups.size() &gt; RATIO_GAPS_FOR_MISMATCH) {<a name="line.196"></a>
<span class="sourceLineNo">197</span>                        logger.debug("More than {} of the residues ({} out of {}) are not present in chain {} when comparing by residue numbers to chain {}.",<a name="line.197"></a>
<span class="sourceLineNo">198</span>                                        RATIO_GAPS_FOR_MISMATCH, countNonExisting, c1AtomGroups.size(), c2.getChainID(), c1.getChainID());<a name="line.198"></a>
<span class="sourceLineNo">199</span>                        return false;<a name="line.199"></a>
<span class="sourceLineNo">200</span>                }<a name="line.200"></a>
<span class="sourceLineNo">201</span><a name="line.201"></a>
<span class="sourceLineNo">202</span>                return true;<a name="line.202"></a>
<span class="sourceLineNo">203</span>        }<a name="line.203"></a>
<span class="sourceLineNo">204</span><a name="line.204"></a>
<span class="sourceLineNo">205</span><a name="line.205"></a>
<span class="sourceLineNo">206</span><a name="line.206"></a>
<span class="sourceLineNo">207</span>        private TreeMap&lt;String,Compound&gt; findCompoundsFromAlignment() {<a name="line.207"></a>
<span class="sourceLineNo">208</span><a name="line.208"></a>
<span class="sourceLineNo">209</span>                // first we determine which chains to consider: anything not looking<a name="line.209"></a>
<span class="sourceLineNo">210</span>                // polymeric (protein or nucleotide chain) should be discarded<a name="line.210"></a>
<span class="sourceLineNo">211</span>                Set&lt;Integer&gt; polyChainIndices = new TreeSet&lt;Integer&gt;();<a name="line.211"></a>
<span class="sourceLineNo">212</span>                List&lt;Chain&gt; pureNonPolymerChains = new ArrayList&lt;Chain&gt;();<a name="line.212"></a>
<span class="sourceLineNo">213</span>                for (int i=0;i&lt;s.getChains().size();i++) {<a name="line.213"></a>
<span class="sourceLineNo">214</span>                        if (StructureTools.isChainPureNonPolymer(s.getChain(i))) {<a name="line.214"></a>
<span class="sourceLineNo">215</span>                                pureNonPolymerChains.add(s.getChain(i));<a name="line.215"></a>
<span class="sourceLineNo">216</span>                        } else {<a name="line.216"></a>
<span class="sourceLineNo">217</span>                                polyChainIndices.add(i);<a name="line.217"></a>
<span class="sourceLineNo">218</span>                        }<a name="line.218"></a>
<span class="sourceLineNo">219</span>                }<a name="line.219"></a>
<span class="sourceLineNo">220</span><a name="line.220"></a>
<span class="sourceLineNo">221</span><a name="line.221"></a>
<span class="sourceLineNo">222</span>                TreeMap&lt;String, Compound&gt; chainIds2compounds = new TreeMap&lt;String,Compound&gt;();<a name="line.222"></a>
<span class="sourceLineNo">223</span><a name="line.223"></a>
<span class="sourceLineNo">224</span>                int molId = 1;<a name="line.224"></a>
<span class="sourceLineNo">225</span><a name="line.225"></a>
<span class="sourceLineNo">226</span>                outer:<a name="line.226"></a>
<span class="sourceLineNo">227</span>                        for (int i:polyChainIndices) {<a name="line.227"></a>
<span class="sourceLineNo">228</span>                                for (int j:polyChainIndices) {<a name="line.228"></a>
<span class="sourceLineNo">229</span><a name="line.229"></a>
<span class="sourceLineNo">230</span>                                        if (j&lt;=i) continue;<a name="line.230"></a>
<span class="sourceLineNo">231</span><a name="line.231"></a>
<span class="sourceLineNo">232</span>                                        Chain c1 = s.getChain(i);<a name="line.232"></a>
<span class="sourceLineNo">233</span>                                        Chain c2 = s.getChain(j);<a name="line.233"></a>
<span class="sourceLineNo">234</span><a name="line.234"></a>
<span class="sourceLineNo">235</span>                                        Map&lt;Integer,Integer&gt; positionIndex1 = new HashMap&lt;Integer, Integer&gt;();<a name="line.235"></a>
<span class="sourceLineNo">236</span>                                        Map&lt;Integer,Integer&gt; positionIndex2 = new HashMap&lt;Integer, Integer&gt;();<a name="line.236"></a>
<span class="sourceLineNo">237</span>                                        // here we use false, which means that X will be used for unknown compounds<a name="line.237"></a>
<span class="sourceLineNo">238</span>                                        String str1 = SeqRes2AtomAligner.getFullAtomSequence(c1.getAtomGroups(), positionIndex1, false);<a name="line.238"></a>
<span class="sourceLineNo">239</span>                                        String str2 = SeqRes2AtomAligner.getFullAtomSequence(c2.getAtomGroups(), positionIndex2, false);<a name="line.239"></a>
<span class="sourceLineNo">240</span><a name="line.240"></a>
<span class="sourceLineNo">241</span>                                        int seq1Length = 0;<a name="line.241"></a>
<span class="sourceLineNo">242</span>                                        int seq2Length = 0;<a name="line.242"></a>
<span class="sourceLineNo">243</span><a name="line.243"></a>
<span class="sourceLineNo">244</span>                                        SequencePair&lt;?,?&gt; pair = null;<a name="line.244"></a>
<span class="sourceLineNo">245</span>                                        if (isProteinSequence(str1) &amp;&amp; isProteinSequence(str2)) {<a name="line.245"></a>
<span class="sourceLineNo">246</span>                                                ProteinSequence s1 = getProteinSequence(str1);<a name="line.246"></a>
<span class="sourceLineNo">247</span>                                                ProteinSequence s2 = getProteinSequence(str2);<a name="line.247"></a>
<span class="sourceLineNo">248</span>                                                seq1Length = s1.getLength();<a name="line.248"></a>
<span class="sourceLineNo">249</span>                                                seq2Length = s2.getLength();<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>                                                pair = alignProtein(s1,s2);<a name="line.251"></a>
<span class="sourceLineNo">252</span><a name="line.252"></a>
<span class="sourceLineNo">253</span>                                        } else if (isDNASequence(str1) &amp;&amp; isDNASequence(str2)) {<a name="line.253"></a>
<span class="sourceLineNo">254</span>                                                DNASequence s1 = getDNASequence(str1);<a name="line.254"></a>
<span class="sourceLineNo">255</span>                                                DNASequence s2 = getDNASequence(str2);<a name="line.255"></a>
<span class="sourceLineNo">256</span>                                                seq1Length = s1.getLength();<a name="line.256"></a>
<span class="sourceLineNo">257</span>                                                seq2Length = s2.getLength();<a name="line.257"></a>
<span class="sourceLineNo">258</span><a name="line.258"></a>
<span class="sourceLineNo">259</span>                                                pair = alignDNA(s1,s2);<a name="line.259"></a>
<span class="sourceLineNo">260</span><a name="line.260"></a>
<span class="sourceLineNo">261</span>                                        } else if (isRNASequence(str1) &amp;&amp; isRNASequence(str2)) {<a name="line.261"></a>
<span class="sourceLineNo">262</span>                                                RNASequence s1 = getRNASequence(str1);<a name="line.262"></a>
<span class="sourceLineNo">263</span>                                                RNASequence s2 = getRNASequence(str2);<a name="line.263"></a>
<span class="sourceLineNo">264</span>                                                seq1Length = s1.getLength();<a name="line.264"></a>
<span class="sourceLineNo">265</span>                                                seq2Length = s2.getLength();<a name="line.265"></a>
<span class="sourceLineNo">266</span><a name="line.266"></a>
<span class="sourceLineNo">267</span>                                                pair = alignRNA(s1,s2);<a name="line.267"></a>
<span class="sourceLineNo">268</span><a name="line.268"></a>
<span class="sourceLineNo">269</span>                                        } else {<a name="line.269"></a>
<span class="sourceLineNo">270</span>                                                logger.debug("Chains {},{} are either different kind of polymers or could not be recognized as protein or nucleotide polymers");<a name="line.270"></a>
<span class="sourceLineNo">271</span>                                                continue;<a name="line.271"></a>
<span class="sourceLineNo">272</span>                                        }<a name="line.272"></a>
<span class="sourceLineNo">273</span><a name="line.273"></a>
<span class="sourceLineNo">274</span><a name="line.274"></a>
<span class="sourceLineNo">275</span>                                        int numGaps = getNumGaps(pair);<a name="line.275"></a>
<span class="sourceLineNo">276</span>                                        int numGaps1 = getNumGapsQuery(pair);<a name="line.276"></a>
<span class="sourceLineNo">277</span>                                        int numGaps2 = getNumGapsTarget(pair);<a name="line.277"></a>
<span class="sourceLineNo">278</span><a name="line.278"></a>
<span class="sourceLineNo">279</span>                                        int nonGaps = pair.getLength() - numGaps;<a name="line.279"></a>
<span class="sourceLineNo">280</span><a name="line.280"></a>
<span class="sourceLineNo">281</span>                                        double identity = (double)pair.getNumIdenticals()/(double)nonGaps;<a name="line.281"></a>
<span class="sourceLineNo">282</span>                                        double gapCov1 = (double) numGaps1 / (double) seq1Length;<a name="line.282"></a>
<span class="sourceLineNo">283</span>                                        double gapCov2 = (double) numGaps2 / (double) seq2Length;<a name="line.283"></a>
<span class="sourceLineNo">284</span><a name="line.284"></a>
<span class="sourceLineNo">285</span>                                        logger.debug("Alignment for chain pair {},{}: identity: {}, gap coverage 1: {}, gap coverage 2: {}",<a name="line.285"></a>
<span class="sourceLineNo">286</span>                                                        c1.getChainID(), c2.getChainID(), String.format("%4.2f",identity), String.format("%4.2f",gapCov1), String.format("%4.2f",gapCov2));<a name="line.286"></a>
<span class="sourceLineNo">287</span>                                        logger.debug("\n"+pair.toString(100));<a name="line.287"></a>
<span class="sourceLineNo">288</span><a name="line.288"></a>
<span class="sourceLineNo">289</span>                                        if (identity &gt; IDENTITY_THRESHOLD &amp;&amp; gapCov1&lt;GAP_COVERAGE_THRESHOLD &amp;&amp; gapCov2&lt;GAP_COVERAGE_THRESHOLD) {<a name="line.289"></a>
<span class="sourceLineNo">290</span>                                                if (    !chainIds2compounds.containsKey(c1.getChainID()) &amp;&amp;<a name="line.290"></a>
<span class="sourceLineNo">291</span>                                                                !chainIds2compounds.containsKey(c2.getChainID())) {<a name="line.291"></a>
<span class="sourceLineNo">292</span>                                                        logger.debug("Creating Compound with chains {},{}",c1.getChainID(),c2.getChainID());<a name="line.292"></a>
<span class="sourceLineNo">293</span><a name="line.293"></a>
<span class="sourceLineNo">294</span>                                                        Compound ent = new Compound();<a name="line.294"></a>
<span class="sourceLineNo">295</span>                                                        ent.addChain(c1);<a name="line.295"></a>
<span class="sourceLineNo">296</span>                                                        ent.addChain(c2);<a name="line.296"></a>
<span class="sourceLineNo">297</span>                                                        ent.setMolId(molId++);<a name="line.297"></a>
<span class="sourceLineNo">298</span>                                                        chainIds2compounds.put(c1.getChainID(), ent);<a name="line.298"></a>
<span class="sourceLineNo">299</span>                                                        chainIds2compounds.put(c2.getChainID(), ent);<a name="line.299"></a>
<span class="sourceLineNo">300</span><a name="line.300"></a>
<span class="sourceLineNo">301</span><a name="line.301"></a>
<span class="sourceLineNo">302</span>                                                } else {<a name="line.302"></a>
<span class="sourceLineNo">303</span>                                                        Compound ent = chainIds2compounds.get(c1.getChainID());<a name="line.303"></a>
<span class="sourceLineNo">304</span><a name="line.304"></a>
<span class="sourceLineNo">305</span>                                                        if (ent==null) {<a name="line.305"></a>
<span class="sourceLineNo">306</span>                                                                logger.debug("Adding chain {} to entity {}",c1.getChainID(),c2.getChainID());<a name="line.306"></a>
<span class="sourceLineNo">307</span>                                                                ent = chainIds2compounds.get(c2.getChainID());<a name="line.307"></a>
<span class="sourceLineNo">308</span>                                                                ent.addChain(c1);<a name="line.308"></a>
<span class="sourceLineNo">309</span>                                                                chainIds2compounds.put(c1.getChainID(), ent);<a name="line.309"></a>
<span class="sourceLineNo">310</span><a name="line.310"></a>
<span class="sourceLineNo">311</span>                                                        } else {<a name="line.311"></a>
<span class="sourceLineNo">312</span>                                                                logger.debug("Adding chain {} to entity {}",c2.getChainID(),c1.getChainID());<a name="line.312"></a>
<span class="sourceLineNo">313</span>                                                                ent.addChain(c2);<a name="line.313"></a>
<span class="sourceLineNo">314</span>                                                                chainIds2compounds.put(c2.getChainID(), ent);<a name="line.314"></a>
<span class="sourceLineNo">315</span><a name="line.315"></a>
<span class="sourceLineNo">316</span>                                                        }<a name="line.316"></a>
<span class="sourceLineNo">317</span>                                                }<a name="line.317"></a>
<span class="sourceLineNo">318</span>                                                if (!areResNumbersAligned(c1, c2)) {<a name="line.318"></a>
<span class="sourceLineNo">319</span>                                                        logger.warn("Including 100% identical chains {},{} in same Compound, although they have misaligned residue numbers",<a name="line.319"></a>
<span class="sourceLineNo">320</span>                                                                        c1.getChainID(),c2.getChainID());<a name="line.320"></a>
<span class="sourceLineNo">321</span>                                                }<a name="line.321"></a>
<span class="sourceLineNo">322</span>                                        }<a name="line.322"></a>
<span class="sourceLineNo">323</span><a name="line.323"></a>
<span class="sourceLineNo">324</span>                                        if (identity&gt;1) {<a name="line.324"></a>
<span class="sourceLineNo">325</span>                                                logger.warn("Identity for chains {},{} above 1. {} identicals out of {} non-gap-aligned residues (identity {})",<a name="line.325"></a>
<span class="sourceLineNo">326</span>                                                                c1.getChainID(),c2.getChainID(),pair.getNumIdenticals(),nonGaps,identity);<a name="line.326"></a>
<span class="sourceLineNo">327</span>                                                logger.warn("\n"+pair.toString(100));<a name="line.327"></a>
<span class="sourceLineNo">328</span>                                        }<a name="line.328"></a>
<span class="sourceLineNo">329</span><a name="line.329"></a>
<span class="sourceLineNo">330</span>                                        if (chainIds2compounds.size()==polyChainIndices.size()) // we've got all chains in entities<a name="line.330"></a>
<span class="sourceLineNo">331</span>                                                break outer;<a name="line.331"></a>
<span class="sourceLineNo">332</span>                                }<a name="line.332"></a>
<span class="sourceLineNo">333</span>                        }<a name="line.333"></a>
<span class="sourceLineNo">334</span><a name="line.334"></a>
<span class="sourceLineNo">335</span>                // anything not in a Compound will be its own Compound<a name="line.335"></a>
<span class="sourceLineNo">336</span>                for (int i:polyChainIndices) {<a name="line.336"></a>
<span class="sourceLineNo">337</span>                        Chain c = s.getChain(i);<a name="line.337"></a>
<span class="sourceLineNo">338</span>                        if (!chainIds2compounds.containsKey(c.getChainID())) {<a name="line.338"></a>
<span class="sourceLineNo">339</span>                                logger.debug("Creating a 1-member Compound for chain {}",c.getChainID());<a name="line.339"></a>
<span class="sourceLineNo">340</span><a name="line.340"></a>
<span class="sourceLineNo">341</span>                                Compound ent = new Compound();<a name="line.341"></a>
<span class="sourceLineNo">342</span>                                ent.addChain(c);<a name="line.342"></a>
<span class="sourceLineNo">343</span>                                ent.setMolId(molId++);<a name="line.343"></a>
<span class="sourceLineNo">344</span><a name="line.344"></a>
<span class="sourceLineNo">345</span>                                chainIds2compounds.put(c.getChainID(),ent);<a name="line.345"></a>
<span class="sourceLineNo">346</span>                        }<a name="line.346"></a>
<span class="sourceLineNo">347</span>                }<a name="line.347"></a>
<span class="sourceLineNo">348</span><a name="line.348"></a>
<span class="sourceLineNo">349</span><a name="line.349"></a>
<span class="sourceLineNo">350</span>                return chainIds2compounds;<a name="line.350"></a>
<span class="sourceLineNo">351</span>        }<a name="line.351"></a>
<span class="sourceLineNo">352</span><a name="line.352"></a>
<span class="sourceLineNo">353</span>        private SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; alignProtein(ProteinSequence s1, ProteinSequence s2) {<a name="line.353"></a>
<span class="sourceLineNo">354</span>                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix = SubstitutionMatrixHelper.getIdentity();<a name="line.354"></a>
<span class="sourceLineNo">355</span><a name="line.355"></a>
<span class="sourceLineNo">356</span>                GapPenalty penalty = new SimpleGapPenalty(8, 1);<a name="line.356"></a>
<span class="sourceLineNo">357</span><a name="line.357"></a>
<span class="sourceLineNo">358</span>                PairwiseSequenceAligner&lt;ProteinSequence, AminoAcidCompound&gt; nw =<a name="line.358"></a>
<span class="sourceLineNo">359</span>                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);<a name="line.359"></a>
<span class="sourceLineNo">360</span><a name="line.360"></a>
<span class="sourceLineNo">361</span>                return nw.getPair();<a name="line.361"></a>
<span class="sourceLineNo">362</span>        }<a name="line.362"></a>
<span class="sourceLineNo">363</span><a name="line.363"></a>
<span class="sourceLineNo">364</span>        private SequencePair&lt;DNASequence, NucleotideCompound&gt; alignDNA(DNASequence s1, DNASequence s2) {<a name="line.364"></a>
<span class="sourceLineNo">365</span>                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();<a name="line.365"></a>
<span class="sourceLineNo">366</span><a name="line.366"></a>
<span class="sourceLineNo">367</span>                GapPenalty penalty = new SimpleGapPenalty(8, 1);<a name="line.367"></a>
<span class="sourceLineNo">368</span><a name="line.368"></a>
<span class="sourceLineNo">369</span>                PairwiseSequenceAligner&lt;DNASequence, NucleotideCompound&gt; nw =<a name="line.369"></a>
<span class="sourceLineNo">370</span>                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);<a name="line.370"></a>
<span class="sourceLineNo">371</span><a name="line.371"></a>
<span class="sourceLineNo">372</span>                return nw.getPair();<a name="line.372"></a>
<span class="sourceLineNo">373</span>        }<a name="line.373"></a>
<span class="sourceLineNo">374</span><a name="line.374"></a>
<span class="sourceLineNo">375</span>        private SequencePair&lt;RNASequence, NucleotideCompound&gt; alignRNA(RNASequence s1, RNASequence s2) {<a name="line.375"></a>
<span class="sourceLineNo">376</span>                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();<a name="line.376"></a>
<span class="sourceLineNo">377</span><a name="line.377"></a>
<span class="sourceLineNo">378</span>                GapPenalty penalty = new SimpleGapPenalty(8, 1);<a name="line.378"></a>
<span class="sourceLineNo">379</span><a name="line.379"></a>
<span class="sourceLineNo">380</span>                PairwiseSequenceAligner&lt;RNASequence, NucleotideCompound&gt; nw =<a name="line.380"></a>
<span class="sourceLineNo">381</span>                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);<a name="line.381"></a>
<span class="sourceLineNo">382</span><a name="line.382"></a>
<span class="sourceLineNo">383</span>                return nw.getPair();<a name="line.383"></a>
<span class="sourceLineNo">384</span>        }<a name="line.384"></a>
<span class="sourceLineNo">385</span><a name="line.385"></a>
<span class="sourceLineNo">386</span><a name="line.386"></a>
<span class="sourceLineNo">387</span><a name="line.387"></a>
<span class="sourceLineNo">388</span>        private static int getNumGaps(SequencePair&lt;?, ?&gt; pair) {<a name="line.388"></a>
<span class="sourceLineNo">389</span>                int numGaps = 0;<a name="line.389"></a>
<span class="sourceLineNo">390</span>                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {<a name="line.390"></a>
<span class="sourceLineNo">391</span>                        if (pair.hasGap(alignmentIndex)) numGaps++;<a name="line.391"></a>
<span class="sourceLineNo">392</span>                }<a name="line.392"></a>
<span class="sourceLineNo">393</span>                return numGaps;<a name="line.393"></a>
<span class="sourceLineNo">394</span>        }<a name="line.394"></a>
<span class="sourceLineNo">395</span><a name="line.395"></a>
<span class="sourceLineNo">396</span>        private static int getNumGapsQuery(SequencePair&lt;?, ?&gt; pair) {<a name="line.396"></a>
<span class="sourceLineNo">397</span>                int numGaps = 0;<a name="line.397"></a>
<span class="sourceLineNo">398</span>                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {<a name="line.398"></a>
<span class="sourceLineNo">399</span>                        if (pair.getCompoundInQueryAt(alignmentIndex).getShortName().equals("-")) {<a name="line.399"></a>
<span class="sourceLineNo">400</span>                                numGaps++;<a name="line.400"></a>
<span class="sourceLineNo">401</span>                        }<a name="line.401"></a>
<span class="sourceLineNo">402</span>                }<a name="line.402"></a>
<span class="sourceLineNo">403</span>                return numGaps;<a name="line.403"></a>
<span class="sourceLineNo">404</span>        }<a name="line.404"></a>
<span class="sourceLineNo">405</span><a name="line.405"></a>
<span class="sourceLineNo">406</span>        private static int getNumGapsTarget(SequencePair&lt;?, ?&gt; pair) {<a name="line.406"></a>
<span class="sourceLineNo">407</span>                int numGaps = 0;<a name="line.407"></a>
<span class="sourceLineNo">408</span>                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {<a name="line.408"></a>
<span class="sourceLineNo">409</span>                        if (pair.getCompoundInTargetAt(alignmentIndex).getShortName().equals("-")) {<a name="line.409"></a>
<span class="sourceLineNo">410</span>                                numGaps++;<a name="line.410"></a>
<span class="sourceLineNo">411</span>                        }<a name="line.411"></a>
<span class="sourceLineNo">412</span>                }<a name="line.412"></a>
<span class="sourceLineNo">413</span>                return numGaps;<a name="line.413"></a>
<span class="sourceLineNo">414</span>        }<a name="line.414"></a>
<span class="sourceLineNo">415</span><a name="line.415"></a>
<span class="sourceLineNo">416</span>        private boolean isProteinSequence(String str) {<a name="line.416"></a>
<span class="sourceLineNo">417</span>                try {<a name="line.417"></a>
<span class="sourceLineNo">418</span>                        new ProteinSequence(str);<a name="line.418"></a>
<span class="sourceLineNo">419</span>                } catch (CompoundNotFoundException e) {<a name="line.419"></a>
<span class="sourceLineNo">420</span><a name="line.420"></a>
<span class="sourceLineNo">421</span>                        return false;<a name="line.421"></a>
<span class="sourceLineNo">422</span>                }<a name="line.422"></a>
<span class="sourceLineNo">423</span>                return true;<a name="line.423"></a>
<span class="sourceLineNo">424</span>        }<a name="line.424"></a>
<span class="sourceLineNo">425</span><a name="line.425"></a>
<span class="sourceLineNo">426</span>        private boolean isDNASequence(String str) {<a name="line.426"></a>
<span class="sourceLineNo">427</span>                try {<a name="line.427"></a>
<span class="sourceLineNo">428</span>                        new DNASequence(str);<a name="line.428"></a>
<span class="sourceLineNo">429</span>                } catch (CompoundNotFoundException e) {<a name="line.429"></a>
<span class="sourceLineNo">430</span><a name="line.430"></a>
<span class="sourceLineNo">431</span>                        return false;<a name="line.431"></a>
<span class="sourceLineNo">432</span>                }<a name="line.432"></a>
<span class="sourceLineNo">433</span>                return true;<a name="line.433"></a>
<span class="sourceLineNo">434</span>        }<a name="line.434"></a>
<span class="sourceLineNo">435</span><a name="line.435"></a>
<span class="sourceLineNo">436</span>        private boolean isRNASequence(String str) {<a name="line.436"></a>
<span class="sourceLineNo">437</span>                try {<a name="line.437"></a>
<span class="sourceLineNo">438</span>                        new RNASequence(str);<a name="line.438"></a>
<span class="sourceLineNo">439</span>                } catch (CompoundNotFoundException e) {<a name="line.439"></a>
<span class="sourceLineNo">440</span><a name="line.440"></a>
<span class="sourceLineNo">441</span>                        return false;<a name="line.441"></a>
<span class="sourceLineNo">442</span>                }<a name="line.442"></a>
<span class="sourceLineNo">443</span>                return true;<a name="line.443"></a>
<span class="sourceLineNo">444</span><a name="line.444"></a>
<span class="sourceLineNo">445</span>        }<a name="line.445"></a>
<span class="sourceLineNo">446</span><a name="line.446"></a>
<span class="sourceLineNo">447</span>        /**<a name="line.447"></a>
<span class="sourceLineNo">448</span>         * Returns the ProteinSequence or null if one can't be created<a name="line.448"></a>
<span class="sourceLineNo">449</span>         * @param str<a name="line.449"></a>
<span class="sourceLineNo">450</span>         * @return<a name="line.450"></a>
<span class="sourceLineNo">451</span>         */<a name="line.451"></a>
<span class="sourceLineNo">452</span>        private ProteinSequence getProteinSequence(String str) {<a name="line.452"></a>
<span class="sourceLineNo">453</span>                try {<a name="line.453"></a>
<span class="sourceLineNo">454</span>                        ProteinSequence s = new ProteinSequence(str);<a name="line.454"></a>
<span class="sourceLineNo">455</span>                        return s;<a name="line.455"></a>
<span class="sourceLineNo">456</span>                } catch (CompoundNotFoundException e) {<a name="line.456"></a>
<span class="sourceLineNo">457</span><a name="line.457"></a>
<span class="sourceLineNo">458</span>                        logger.error("Unexpected error when creating ProteinSequence",e);<a name="line.458"></a>
<span class="sourceLineNo">459</span>                }<a name="line.459"></a>
<span class="sourceLineNo">460</span>                return null;<a name="line.460"></a>
<span class="sourceLineNo">461</span>        }<a name="line.461"></a>
<span class="sourceLineNo">462</span><a name="line.462"></a>
<span class="sourceLineNo">463</span>        /**<a name="line.463"></a>
<span class="sourceLineNo">464</span>         * Returns the DNASequence or null if one can't be created<a name="line.464"></a>
<span class="sourceLineNo">465</span>         * @param str<a name="line.465"></a>
<span class="sourceLineNo">466</span>         * @return<a name="line.466"></a>
<span class="sourceLineNo">467</span>         */<a name="line.467"></a>
<span class="sourceLineNo">468</span>        private DNASequence getDNASequence(String str) {<a name="line.468"></a>
<span class="sourceLineNo">469</span>                try {<a name="line.469"></a>
<span class="sourceLineNo">470</span>                        DNASequence s = new DNASequence(str);<a name="line.470"></a>
<span class="sourceLineNo">471</span>                        return s;<a name="line.471"></a>
<span class="sourceLineNo">472</span>                } catch (CompoundNotFoundException e) {<a name="line.472"></a>
<span class="sourceLineNo">473</span><a name="line.473"></a>
<span class="sourceLineNo">474</span>                        logger.error("Unexpected error when creating DNASequence ",e);<a name="line.474"></a>
<span class="sourceLineNo">475</span>                }<a name="line.475"></a>
<span class="sourceLineNo">476</span>                return null;<a name="line.476"></a>
<span class="sourceLineNo">477</span>        }<a name="line.477"></a>
<span class="sourceLineNo">478</span><a name="line.478"></a>
<span class="sourceLineNo">479</span>        /**<a name="line.479"></a>
<span class="sourceLineNo">480</span>         * Returns the RNASequence or null if one can't be created<a name="line.480"></a>
<span class="sourceLineNo">481</span>         * @param str<a name="line.481"></a>
<span class="sourceLineNo">482</span>         * @return<a name="line.482"></a>
<span class="sourceLineNo">483</span>         */<a name="line.483"></a>
<span class="sourceLineNo">484</span>        private RNASequence getRNASequence(String str) {<a name="line.484"></a>
<span class="sourceLineNo">485</span>                try {<a name="line.485"></a>
<span class="sourceLineNo">486</span>                        RNASequence s = new RNASequence(str);<a name="line.486"></a>
<span class="sourceLineNo">487</span>                        return s;<a name="line.487"></a>
<span class="sourceLineNo">488</span>                } catch (CompoundNotFoundException e) {<a name="line.488"></a>
<span class="sourceLineNo">489</span><a name="line.489"></a>
<span class="sourceLineNo">490</span>                        logger.error("Unexpected error when creating RNASequence ",e);<a name="line.490"></a>
<span class="sourceLineNo">491</span>                }<a name="line.491"></a>
<span class="sourceLineNo">492</span>                return null;<a name="line.492"></a>
<span class="sourceLineNo">493</span>        }<a name="line.493"></a>
<span class="sourceLineNo">494</span><a name="line.494"></a>
<span class="sourceLineNo">495</span>}<a name="line.495"></a>




























































</pre>
</div>
</body>
</html>
