<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:53 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Deprecated List (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Deprecated List (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li class="navBarCell1Rev">Deprecated</li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?deprecated-list.html" target="_top">Frames</a></li>
<li><a href="deprecated-list.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Deprecated API" class="title">Deprecated API</h1>
<h2 title="Contents">Contents</h2>
<ul>
<li><a href="#class">Deprecated Classes</a></li>
<li><a href="#method">Deprecated Methods</a></li>
<li><a href="#constructor">Deprecated Constructors</a></li>
</ul>
</div>
<div class="contentContainer"><a name="class">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Deprecated Classes table, listing deprecated classes, and an explanation">
<caption><span>Deprecated Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Identifier.html" title="class in org.biojava.nbio.structure">org.biojava.nbio.structure.Identifier</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client"><code>StructureName</code></a> instead. Deprecated in v. 4.2.0</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/gui/RasmolCommandListener.html" title="class in org.biojava.nbio.structure.gui">org.biojava.nbio.structure.gui.RasmolCommandListener</a>
<div class="block"><i>use  org.biojava.nbio.structure.align.gui.jmol.RasmolCommandListener instead</i></div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
<a name="method">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Deprecated Methods table, listing deprecated methods, and an explanation">
<caption><span>Deprecated Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/alignment/io/StockholmStructure.html#addSequence(java.lang.String,%20java.lang.String)">org.biojava.nbio.alignment.io.StockholmStructure.addSequence(String, String)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/alignment/io/StockholmStructure.html#appendToSequence(java.lang.String,%20java.lang.String)"><code>StockholmStructure.appendToSequence(String,String)</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">org.biojava.nbio.structure.align.ce.CECalculator.calc_rmsd(Atom[], Atom[], int, boolean, boolean)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)"><code>CECalculator.calc_rmsd(Atom[],Atom[],int,boolean)</code></a> instead</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureTools.html#cloneCAArray(org.biojava.nbio.structure.Atom[])">org.biojava.nbio.structure.StructureTools.cloneCAArray(Atom[])</a>
<div class="block"><i>Use the better-named <a href="org/biojava/nbio/structure/StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom[])"><code>StructureTools.cloneAtomArray(Atom[])</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureTools.html#convert_3code_1code(java.lang.String)">org.biojava.nbio.structure.StructureTools.convert_3code_1code(String)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/StructureTools.html#get1LetterCodeAmino(java.lang.String)"><code>StructureTools.get1LetterCodeAmino(String)</code></a> instead</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/PDBFileReader.html#downloadPDB(java.lang.String)">org.biojava.nbio.structure.io.PDBFileReader.downloadPDB(String)</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/UserConfiguration.html#getAutoFetch()">org.biojava.nbio.structure.align.util.UserConfiguration.getAutoFetch()</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/util/UserConfiguration.html#getFetchBehavior()"><code>UserConfiguration.getFetchBehavior()</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#getAverageTraceRmsd()">org.biojava.nbio.structure.symmetry.core.QuatSymmetryResults.getAverageTraceRmsd()</a>
<div class="block"><i>use <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#getScores()"><code>QuatSymmetryResults.getScores()</code></a> instead.</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#getAverageTraceTmScoreMin()">org.biojava.nbio.structure.symmetry.core.QuatSymmetryResults.getAverageTraceTmScoreMin()</a>
<div class="block"><i>use <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html#getScores()"><code>QuatSymmetryResults.getScores()</code></a> instead.</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalUnit(java.lang.String)">org.biojava.nbio.structure.align.util.AtomCache.getBiologicalUnit(String)</a>
<div class="block"><i>Renamed to <a href="org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly(java.lang.String)"><code>AtomCache.getBiologicalAssembly(String)</code></a> in 4.2</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/gui/DisplayAFP.html#getBlockNrForAlignPos(org.biojava.nbio.structure.align.model.AFPChain,%20int)">org.biojava.nbio.structure.align.gui.DisplayAFP.getBlockNrForAlignPos(AFPChain, int)</a>
<div class="block"><i>use AFPAlignmentDisplay.getBlockNrForAlignPos instead...</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/CAConverter.html#getCAOnly(org.biojava.nbio.structure.Chain)">org.biojava.nbio.structure.io.CAConverter.getCAOnly(Chain)</a>
<div class="block"><i>Use the more generic <a href="org/biojava/nbio/structure/io/CAConverter.html#getRepresentativeAtomsOnly(org.biojava.nbio.structure.Chain)"><code>CAConverter.getRepresentativeAtomsOnly(Chain)</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/CAConverter.html#getCAOnly(java.util.List)">org.biojava.nbio.structure.io.CAConverter.getCAOnly(List<Chain>)</a>
<div class="block"><i>Use the more generic <a href="org/biojava/nbio/structure/io/CAConverter.html#getRepresentativeAtomsOnly(java.util.List)"><code>CAConverter.getRepresentativeAtomsOnly(List)</code></a> instead</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#getConnections()">org.biojava.nbio.structure.Structure.getConnections()</a>
<div class="block"><i>use <a href="org/biojava/nbio/structure/Atom.html#getBonds()"><code>Atom.getBonds()</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html#getConsituents()">org.biojava.nbio.core.sequence.compound.NucleotideCompound.getConsituents()</a>
<div class="block"><i>replaced with <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html#getConstituents()"><code>NucleotideCompound.getConstituents()</code></a> due to typographical error</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/ChainImpl.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber,%20boolean)">org.biojava.nbio.structure.ChainImpl.getGroupsByPDB(ResidueNumber, ResidueNumber, boolean)</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/client/StructureName.html#getName()">org.biojava.nbio.structure.align.client.StructureName.getName()</a>
<div class="block"><i>use <a href="org/biojava/nbio/structure/align/client/StructureName.html#getIdentifier()"><code>StructureName.getIdentifier()</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/protmod/structure/ModifiedCompoundImpl.html#getOriginalModificationId()">org.biojava.nbio.protmod.structure.ModifiedCompoundImpl.getOriginalModificationId()</a>
<div class="block"><i>use getModification().getId()</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/RotationAxis.html#getOtherTranslation()">org.biojava.nbio.structure.align.util.RotationAxis.getOtherTranslation()</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/ChainImpl.html#getParent()">org.biojava.nbio.structure.ChainImpl.getParent()</a>
<div class="block"><i>use getStructure instead.</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Chain.html#getParent()">org.biojava.nbio.structure.Chain.getParent()</a>
<div class="block"><i>use getStructure(Structure) instead.</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureImpl.html#getPdbId()">org.biojava.nbio.structure.StructureImpl.getPdbId()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#getPdbId()">org.biojava.nbio.structure.Structure.getPdbId()</a>
<div class="block"><i>From BioJava 4.2, use <a href="org/biojava/nbio/structure/Structure.html#getPDBCode()"><code>Structure.getPDBCode()</code></a> or
  <code>getStructureIdentifier().toCanonical().getPdbId()</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/cath/CathDomain.html#getPdbId()">org.biojava.nbio.structure.cath.CathDomain.getPdbId()</a>
<div class="block"><i>This method is poorly named; use <a href="org/biojava/nbio/structure/cath/CathDomain.html#getThePdbId()"><code>CathDomain.getThePdbId()</code></a> or <a href="org/biojava/nbio/structure/cath/CathDomain.html#getPdbIdAndChain()"><code>CathDomain.getPdbIdAndChain()</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureImpl.html#getRanges()">org.biojava.nbio.structure.StructureImpl.getRanges()</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#getRanges()">org.biojava.nbio.structure.Structure.getRanges()</a>
<div class="block"><i>From BioJava 4.2, use
  <code>getStructureIdentifier().toCanonical().getRanges()</code></i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getReducedCAStructure(org.biojava.nbio.structure.Structure)">org.biojava.nbio.structure.quaternary.BioAssemblyTools.getReducedCAStructure(Structure)</a>
<div class="block"><i>Use the more generic <a href="org/biojava/nbio/structure/quaternary/BioAssemblyTools.html#getReducedStructure(org.biojava.nbio.structure.Structure)"><code>BioAssemblyTools.getReducedStructure(Structure)</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureTools.html#getReducedStructure(org.biojava.nbio.structure.Structure,%20int)">org.biojava.nbio.structure.StructureTools.getReducedStructure(Structure, int)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureTools.html#getReducedStructure(org.biojava.nbio.structure.Structure,%20java.lang.String)">org.biojava.nbio.structure.StructureTools.getReducedStructure(Structure, String)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureImpl.html#getResidueRanges()">org.biojava.nbio.structure.StructureImpl.getResidueRanges()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#getResidueRanges()">org.biojava.nbio.structure.Structure.getResidueRanges()</a>
<div class="block"><i>From BioJava 4.2, use
  <code>getStructureIdentifier().toCanonical().getResidueRanges()</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/symmetry/analysis/CalcBioAssemblySymmetry.html#getRotationGroup()">org.biojava.nbio.structure.symmetry.analysis.CalcBioAssemblySymmetry.getRotationGroup()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html#getSaturation()">org.biojava.nbio.structure.gui.JMatrixPanel.getSaturation()</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html#getScalevalue()">org.biojava.nbio.structure.gui.JMatrixPanel.getScalevalue()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/scop/ScopInstallation.html#getScopDownloadURL()">org.biojava.nbio.structure.scop.ScopInstallation.getScopDownloadURL()</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/model/AFPChain.html#getSimilarity1()">org.biojava.nbio.structure.align.model.AFPChain.getSimilarity1()</a>
<div class="block"><i>use getCoverage1() instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/model/AFPChain.html#getSimilarity2()">org.biojava.nbio.structure.align.model.AFPChain.getSimilarity2()</a>
<div class="block"><i>use getCoverage2() instead</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureTools.html#getSubRanges(org.biojava.nbio.structure.Structure,%20java.lang.String)">org.biojava.nbio.structure.StructureTools.getSubRanges(Structure, String)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure"><code>StructureIdentifier</code></a> instead (4.2.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/PDBCrystallographicInfo.html#getZ()">org.biojava.nbio.structure.PDBCrystallographicInfo.getZ()</a>
<div class="block"><i>As of 4.0, use <a href="org/biojava/nbio/structure/xtal/SpaceGroup.html#getMultiplicity()"><code>SpaceGroup.getMultiplicity()</code></a> and <a href="org/biojava/nbio/structure/Structure.html#size()"><code>Structure.size()</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#isAutoFetch()">org.biojava.nbio.structure.align.util.AtomCache.isAutoFetch()</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/util/AtomCache.html#getFetchBehavior()"><code>AtomCache.getFetchBehavior()</code></a></i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html#isAutoFetch()">org.biojava.nbio.structure.io.LocalPDBDirectory.isAutoFetch()</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html#getFetchBehavior()"><code>LocalPDBDirectory.getFetchBehavior()</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/StructurePairAligner.html#isDebug()">org.biojava.nbio.structure.align.StructurePairAligner.isDebug()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#isFetchCurrent()">org.biojava.nbio.structure.align.util.AtomCache.isFetchCurrent()</a>
<div class="block"><i>Use <code>FileParsingParameters#getObsoleteBehavior()</code> instead (4.0.0)</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/PDBFileReader.html#isFetchCurrent()">org.biojava.nbio.structure.io.PDBFileReader.isFetchCurrent()</a>
<div class="block"><i>Use <code>FileParsingParameters#getObsoleteBehavior()</code></i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#isFetchFileEvenIfObsolete()">org.biojava.nbio.structure.align.util.AtomCache.isFetchFileEvenIfObsolete()</a>
<div class="block"><i>Use <code>FileParsingParameters#getObsoleteBehavior()</code> instead (4.0.0)</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/PDBFileReader.html#isFetchFileEvenIfObsolete()">org.biojava.nbio.structure.io.PDBFileReader.isFetchFileEvenIfObsolete()</a>
<div class="block"><i>Use <code>FileParsingParameters#getObsoleteBehavior()</code></i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/ce/StartupParameters.html#isPdbDirSplit()">org.biojava.nbio.structure.align.ce.StartupParameters.isPdbDirSplit()</a>
<div class="block"><i>Always returns true (4.0.0)</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#isStrictSCOP()">org.biojava.nbio.structure.align.util.AtomCache.isStrictSCOP()</a>
<div class="block"><i>since 4.2</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/FileParsingParameters.html#isUpdateRemediatedFiles()">org.biojava.nbio.structure.io.FileParsingParameters.isUpdateRemediatedFiles()</a>
<div class="block"><i>Properties which impact downloading and caching behavior
  have been moved to the <a href="org/biojava/nbio/structure/io/StructureIOFile.html" title="interface in org.biojava.nbio.structure.io"><code>StructureIOFile</code></a> implementations.
  See <code>LocalPDBDirectory#getFetchBehavior(LocalPDBDirectory.FetchBehavior)</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Identifier.html#loadIdentifier(java.lang.String,%20org.biojava.nbio.structure.align.util.AtomCache)">org.biojava.nbio.structure.Identifier.loadIdentifier(String, AtomCache)</a>
<div class="block"><i>Create a new <a href="org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client"><code>StructureName</code></a> instead.</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/genome/parsers/gff/LocIterator.html#main(java.lang.String[])">org.biojava.nbio.genome.parsers.gff.LocIterator.main(String[])</a>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/genome/parsers/gff/Location.html#main(java.lang.String[])">org.biojava.nbio.genome.parsers.gff.Location.main(String[])</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/genome/parsers/gff/Feature.html#main(java.lang.String[])">org.biojava.nbio.genome.parsers.gff.Feature.main(String[])</a>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/genome/parsers/gff/FeatureList.html#main(java.lang.String[])">org.biojava.nbio.genome.parsers.gff.FeatureList.main(String[])</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/UserConfiguration.html#setAutoFetch(boolean)">org.biojava.nbio.structure.align.util.UserConfiguration.setAutoFetch(boolean)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/util/UserConfiguration.html#getFetchBehavior()"><code>UserConfiguration.getFetchBehavior()</code></a></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#setAutoFetch(boolean)">org.biojava.nbio.structure.align.util.AtomCache.setAutoFetch(boolean)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/align/util/AtomCache.html#getFetchBehavior()"><code>AtomCache.getFetchBehavior()</code></a></i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html#setAutoFetch(boolean)">org.biojava.nbio.structure.io.LocalPDBDirectory.setAutoFetch(boolean)</a>
<div class="block"><i>Use <code>#setFetchBehavior()</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#setConnections(java.util.List)">org.biojava.nbio.structure.Structure.setConnections(List<Map<String, Integer>>)</a>
<div class="block"><i>use <a href="org/biojava/nbio/structure/Atom.html#addBond(org.biojava.nbio.structure.Bond)"><code>Atom.addBond(Bond)</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/StructurePairAligner.html#setDebug(boolean)">org.biojava.nbio.structure.align.StructurePairAligner.setDebug(boolean)</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#setFetchCurrent(boolean)">org.biojava.nbio.structure.align.util.AtomCache.setFetchCurrent(boolean)</a>
<div class="block"><i>Use <code>FileParsingParameters#setObsoleteBehavior()</code> instead (4.0.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/PDBFileReader.html#setFetchCurrent(boolean)">org.biojava.nbio.structure.io.PDBFileReader.setFetchCurrent(boolean)</a>
<div class="block"><i>Use <code>FileParsingParameters#setObsoleteBehavior(ObsoleteBehavior)</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#setFetchFileEvenIfObsolete(boolean)">org.biojava.nbio.structure.align.util.AtomCache.setFetchFileEvenIfObsolete(boolean)</a>
<div class="block"><i>Use <code>FileParsingParameters#setObsoleteBehavior()</code> instead (4.0.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/PDBFileReader.html#setFetchFileEvenIfObsolete(boolean)">org.biojava.nbio.structure.io.PDBFileReader.setFetchFileEvenIfObsolete(boolean)</a>
<div class="block"><i>Use <code>FileParsingParameters#setObsoleteBehavior(ObsoleteBehavior)</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/StructureImpl.html#setNmr(boolean)">org.biojava.nbio.structure.StructureImpl.setNmr(boolean)</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Structure.html#setNmr(boolean)">org.biojava.nbio.structure.Structure.setNmr(boolean)</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/ChainImpl.html#setParent(org.biojava.nbio.structure.Structure)">org.biojava.nbio.structure.ChainImpl.setParent(Structure)</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/Chain.html#setParent(org.biojava.nbio.structure.Structure)">org.biojava.nbio.structure.Chain.setParent(Structure)</a>
<div class="block"><i>use setStructure instead</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/ce/StartupParameters.html#setPdbDirSplit(boolean)">org.biojava.nbio.structure.align.ce.StartupParameters.setPdbDirSplit(boolean)</a>
<div class="block"><i>Ignored (4.0.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html#setRefineResult(boolean)">org.biojava.nbio.structure.symmetry.internal.CESymmParameters.setRefineResult(boolean)</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html#setSaturation(float)">org.biojava.nbio.structure.gui.JMatrixPanel.setSaturation(float)</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html#setScalevalue(float)">org.biojava.nbio.structure.gui.JMatrixPanel.setScalevalue(float)</a></td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/scop/ScopInstallation.html#setScopDownloadURL(java.lang.String)">org.biojava.nbio.structure.scop.ScopInstallation.setScopDownloadURL(String)</a>
<div class="block"><i>Use <a href="org/biojava/nbio/structure/scop/ScopInstallation.html#addMirror(java.lang.String)"><code>ScopInstallation.addMirror(java.lang.String)</code></a> instead</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#setStrictSCOP(boolean)">org.biojava.nbio.structure.align.util.AtomCache.setStrictSCOP(boolean)</a>
<div class="block"><i>Removed in 4.2.0</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/io/FileParsingParameters.html#setUpdateRemediatedFiles(boolean)">org.biojava.nbio.structure.io.FileParsingParameters.setUpdateRemediatedFiles(boolean)</a>
<div class="block"><i>Properties which impact downloading and caching behavior
  have been moved to the <a href="org/biojava/nbio/structure/io/StructureIOFile.html" title="interface in org.biojava.nbio.structure.io"><code>StructureIOFile</code></a> implementations.
  See <a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html#setFetchBehavior(org.biojava.nbio.structure.io.LocalPDBDirectory.FetchBehavior)"><code>LocalPDBDirectory.setFetchBehavior(LocalPDBDirectory.FetchBehavior)</code></a></i></div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
<a name="constructor">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Deprecated Constructors table, listing deprecated constructors, and an explanation">
<caption><span>Deprecated Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/model/AFPChain.html#AFPChain()">org.biojava.nbio.structure.align.model.AFPChain()</a></td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#AtomCache(java.lang.String,%20boolean)">org.biojava.nbio.structure.align.util.AtomCache(String, boolean)</a>
<div class="block"><i>isSplit parameter is ignored (4.0.0)</i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/structure/align/util/AtomCache.html#AtomCache(java.lang.String,%20java.lang.String,%20boolean)">org.biojava.nbio.structure.align.util.AtomCache(String, String, boolean)</a>
<div class="block"><i>isSplit parameter is ignored (4.0.0)</i></div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.transcription.Table,%20boolean,%20boolean,%20boolean)">org.biojava.nbio.core.sequence.transcription.RNAToAminoAcidTranslator(SequenceCreatorInterface<AminoAcidCompound>, CompoundSet<NucleotideCompound>, CompoundSet<Table.Codon>, CompoundSet<AminoAcidCompound>, Table, boolean, boolean, boolean)</a>
<div class="block"><i>Retained for backwards compatability, setting
             <a href="org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html#stopAtStopCodons"><code>RNAToAminoAcidTranslator.stopAtStopCodons</code></a> to <code>false</code></i></div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html#RNAToAminoAcidTranslator(org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.transcription.Table,%20boolean,%20boolean,%20boolean,%20boolean)">org.biojava.nbio.core.sequence.transcription.RNAToAminoAcidTranslator(SequenceCreatorInterface<AminoAcidCompound>, CompoundSet<NucleotideCompound>, CompoundSet<Table.Codon>, CompoundSet<AminoAcidCompound>, Table, boolean, boolean, boolean, boolean)</a></td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li class="navBarCell1Rev">Deprecated</li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="index.html?deprecated-list.html" target="_top">Frames</a></li>
<li><a href="deprecated-list.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
